{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.core.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))\n",
    "from matplotlib.pyplot import figure\n",
    "figure(num=None, figsize=(15, 10), dpi=100, facecolor='w', edgecolor='k')\n",
    "import pandas as pd\n",
    "from scipy.stats import zscore, norm\n",
    "import matplotlib.pyplot as plt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca_per_pathway = pd.read_csv('../data/pca_per_pathway.csv', index_col=0)\n",
    "\n",
    "new_clinical_patient = pd.read_csv('../data/data_clinical_patient.txt', sep='\\t', index_col=0).iloc[4:]\n",
    "new_clinical_sample = pd.read_csv('../data/data_clinical_sample.txt', sep='\\t', index_col=0).iloc[4:]\n",
    "new_clinical = pd.concat([new_clinical_patient, new_clinical_sample.reindex(new_clinical_patient.index)], axis=1)\n",
    "new_clinical['Triple Neg'] = new_clinical.apply(lambda row: True if ((row['ER Status'] == 'Negative') \n",
    "                                                                     and (row['PR Status'] == 'Negative') \n",
    "                                                                     and (row['HER2 Status'] == 'Negative')) else False, axis = 1)\n",
    "\n",
    "new_clinical['ER-/PR-/HER2+'] = new_clinical.apply(lambda row: True if ((row['ER Status'] == 'Negative') \n",
    "                                                                     and (row['PR Status'] == 'Negative') \n",
    "                                                                     and (row['HER2 Status'] == 'Positive')) else False, axis = 1)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_df = pd.concat([pca_per_pathway, new_clinical.reindex(pca_per_pathway.index)], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Interleukin-6 signaling</th>\n",
       "      <th>Apoptosis</th>\n",
       "      <th>Hemostasis</th>\n",
       "      <th>Intrinsic Pathway for Apoptosis</th>\n",
       "      <th>PKB-mediated events</th>\n",
       "      <th>PI3K Cascade</th>\n",
       "      <th>MAPK3 (ERK1) activation</th>\n",
       "      <th>Translesion synthesis by REV1</th>\n",
       "      <th>Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template</th>\n",
       "      <th>Recognition of DNA damage by PCNA-containing replication complex</th>\n",
       "      <th>...</th>\n",
       "      <th>Intclust 9 or not</th>\n",
       "      <th>Intclust 7 or not</th>\n",
       "      <th>Intclust 3 or not</th>\n",
       "      <th>Intclust 8 or not</th>\n",
       "      <th>Intclust 10 or not</th>\n",
       "      <th>Intclust 1 or not</th>\n",
       "      <th>Intclust 2 or not</th>\n",
       "      <th>Intclust 5 or not</th>\n",
       "      <th>Intclust 4ER- or not</th>\n",
       "      <th>Intclust 6 or not</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>-0.464915</td>\n",
       "      <td>1.150390</td>\n",
       "      <td>7.995550</td>\n",
       "      <td>-2.133505</td>\n",
       "      <td>-1.005971</td>\n",
       "      <td>-1.934175</td>\n",
       "      <td>0.682674</td>\n",
       "      <td>-1.740452</td>\n",
       "      <td>-1.758971</td>\n",
       "      <td>2.306896</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>1.135676</td>\n",
       "      <td>-5.398171</td>\n",
       "      <td>-1.468534</td>\n",
       "      <td>-2.783638</td>\n",
       "      <td>-0.042169</td>\n",
       "      <td>-1.152715</td>\n",
       "      <td>-0.440677</td>\n",
       "      <td>-0.738142</td>\n",
       "      <td>-0.950279</td>\n",
       "      <td>0.857385</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0010</th>\n",
       "      <td>-0.016407</td>\n",
       "      <td>-2.252856</td>\n",
       "      <td>1.688631</td>\n",
       "      <td>-1.840132</td>\n",
       "      <td>0.439943</td>\n",
       "      <td>0.907014</td>\n",
       "      <td>-0.946115</td>\n",
       "      <td>-1.195843</td>\n",
       "      <td>-2.019880</td>\n",
       "      <td>2.452047</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>7</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0035</th>\n",
       "      <td>-0.045984</td>\n",
       "      <td>3.050125</td>\n",
       "      <td>14.221321</td>\n",
       "      <td>-1.010664</td>\n",
       "      <td>-2.460802</td>\n",
       "      <td>0.033379</td>\n",
       "      <td>-0.040401</td>\n",
       "      <td>1.589719</td>\n",
       "      <td>0.338310</td>\n",
       "      <td>-0.025657</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0036</th>\n",
       "      <td>-1.107911</td>\n",
       "      <td>1.445771</td>\n",
       "      <td>5.790792</td>\n",
       "      <td>1.275586</td>\n",
       "      <td>-1.040463</td>\n",
       "      <td>-2.897256</td>\n",
       "      <td>0.357232</td>\n",
       "      <td>0.276760</td>\n",
       "      <td>1.075780</td>\n",
       "      <td>-1.226257</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0050</th>\n",
       "      <td>1.003819</td>\n",
       "      <td>1.441920</td>\n",
       "      <td>-1.964812</td>\n",
       "      <td>-0.102851</td>\n",
       "      <td>-0.049525</td>\n",
       "      <td>-3.192611</td>\n",
       "      <td>-1.181677</td>\n",
       "      <td>0.429309</td>\n",
       "      <td>1.425590</td>\n",
       "      <td>-1.375613</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0059</th>\n",
       "      <td>-0.473150</td>\n",
       "      <td>0.186910</td>\n",
       "      <td>6.189742</td>\n",
       "      <td>-1.018816</td>\n",
       "      <td>-1.353297</td>\n",
       "      <td>-2.260648</td>\n",
       "      <td>0.119735</td>\n",
       "      <td>0.142322</td>\n",
       "      <td>-0.111307</td>\n",
       "      <td>-0.685721</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0060</th>\n",
       "      <td>0.681486</td>\n",
       "      <td>-12.524534</td>\n",
       "      <td>-0.370151</td>\n",
       "      <td>-2.756785</td>\n",
       "      <td>0.077299</td>\n",
       "      <td>-0.448584</td>\n",
       "      <td>-1.396643</td>\n",
       "      <td>-3.632946</td>\n",
       "      <td>-5.847386</td>\n",
       "      <td>6.291344</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0066</th>\n",
       "      <td>0.539735</td>\n",
       "      <td>-6.459812</td>\n",
       "      <td>-0.231027</td>\n",
       "      <td>-1.086133</td>\n",
       "      <td>-0.017782</td>\n",
       "      <td>-0.218056</td>\n",
       "      <td>-0.731439</td>\n",
       "      <td>-0.210748</td>\n",
       "      <td>-0.525239</td>\n",
       "      <td>1.251225</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>7</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0101</th>\n",
       "      <td>1.391563</td>\n",
       "      <td>-2.378717</td>\n",
       "      <td>-4.880185</td>\n",
       "      <td>1.025275</td>\n",
       "      <td>-0.772772</td>\n",
       "      <td>-0.612536</td>\n",
       "      <td>-0.850303</td>\n",
       "      <td>1.193591</td>\n",
       "      <td>2.045964</td>\n",
       "      <td>-1.989934</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0102</th>\n",
       "      <td>0.975992</td>\n",
       "      <td>-4.692292</td>\n",
       "      <td>-2.159457</td>\n",
       "      <td>-1.916437</td>\n",
       "      <td>-0.424554</td>\n",
       "      <td>0.999638</td>\n",
       "      <td>-1.688466</td>\n",
       "      <td>0.167435</td>\n",
       "      <td>-1.977181</td>\n",
       "      <td>1.916599</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0106</th>\n",
       "      <td>-1.716131</td>\n",
       "      <td>3.055953</td>\n",
       "      <td>2.940126</td>\n",
       "      <td>0.884104</td>\n",
       "      <td>-0.823157</td>\n",
       "      <td>-0.688135</td>\n",
       "      <td>1.656749</td>\n",
       "      <td>-0.002856</td>\n",
       "      <td>2.292241</td>\n",
       "      <td>-1.652733</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0112</th>\n",
       "      <td>0.721624</td>\n",
       "      <td>-1.939072</td>\n",
       "      <td>-0.898322</td>\n",
       "      <td>0.265369</td>\n",
       "      <td>-0.368646</td>\n",
       "      <td>-1.915540</td>\n",
       "      <td>-1.256289</td>\n",
       "      <td>1.112242</td>\n",
       "      <td>1.566163</td>\n",
       "      <td>-1.291966</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0115</th>\n",
       "      <td>2.891250</td>\n",
       "      <td>-7.520513</td>\n",
       "      <td>-4.479024</td>\n",
       "      <td>-4.139860</td>\n",
       "      <td>-0.845741</td>\n",
       "      <td>-0.875985</td>\n",
       "      <td>-2.905533</td>\n",
       "      <td>-3.344632</td>\n",
       "      <td>-3.650589</td>\n",
       "      <td>3.451235</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0117</th>\n",
       "      <td>-1.218244</td>\n",
       "      <td>-0.139969</td>\n",
       "      <td>-2.140516</td>\n",
       "      <td>0.796431</td>\n",
       "      <td>1.913941</td>\n",
       "      <td>0.425881</td>\n",
       "      <td>0.909888</td>\n",
       "      <td>0.240269</td>\n",
       "      <td>1.165142</td>\n",
       "      <td>-1.165904</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0119</th>\n",
       "      <td>0.324922</td>\n",
       "      <td>-1.494166</td>\n",
       "      <td>2.681937</td>\n",
       "      <td>-0.732578</td>\n",
       "      <td>-1.160084</td>\n",
       "      <td>-0.281727</td>\n",
       "      <td>-0.094547</td>\n",
       "      <td>0.530474</td>\n",
       "      <td>0.249794</td>\n",
       "      <td>-0.944412</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>7</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0120</th>\n",
       "      <td>-1.282450</td>\n",
       "      <td>-1.312260</td>\n",
       "      <td>-5.283445</td>\n",
       "      <td>-1.769936</td>\n",
       "      <td>-0.042372</td>\n",
       "      <td>1.623986</td>\n",
       "      <td>0.562100</td>\n",
       "      <td>-0.550128</td>\n",
       "      <td>-1.018389</td>\n",
       "      <td>0.599823</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0121</th>\n",
       "      <td>0.066524</td>\n",
       "      <td>-3.182866</td>\n",
       "      <td>0.578324</td>\n",
       "      <td>-1.262635</td>\n",
       "      <td>-0.358424</td>\n",
       "      <td>-0.596381</td>\n",
       "      <td>-0.376231</td>\n",
       "      <td>-1.426032</td>\n",
       "      <td>0.031787</td>\n",
       "      <td>0.056474</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0122</th>\n",
       "      <td>0.241645</td>\n",
       "      <td>-0.896690</td>\n",
       "      <td>4.363559</td>\n",
       "      <td>-0.332297</td>\n",
       "      <td>-0.933940</td>\n",
       "      <td>-2.378229</td>\n",
       "      <td>0.431727</td>\n",
       "      <td>-0.993327</td>\n",
       "      <td>0.664990</td>\n",
       "      <td>-0.454606</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0123</th>\n",
       "      <td>0.632205</td>\n",
       "      <td>-4.939672</td>\n",
       "      <td>1.387583</td>\n",
       "      <td>-2.019270</td>\n",
       "      <td>1.906553</td>\n",
       "      <td>1.130355</td>\n",
       "      <td>-0.342736</td>\n",
       "      <td>-4.240944</td>\n",
       "      <td>-4.235351</td>\n",
       "      <td>4.132531</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0126</th>\n",
       "      <td>0.430240</td>\n",
       "      <td>0.585299</td>\n",
       "      <td>0.323267</td>\n",
       "      <td>-1.390457</td>\n",
       "      <td>0.169088</td>\n",
       "      <td>-4.192595</td>\n",
       "      <td>-1.174283</td>\n",
       "      <td>1.428986</td>\n",
       "      <td>1.529271</td>\n",
       "      <td>-2.216522</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0131</th>\n",
       "      <td>0.059170</td>\n",
       "      <td>0.794649</td>\n",
       "      <td>0.309136</td>\n",
       "      <td>0.093709</td>\n",
       "      <td>-0.878579</td>\n",
       "      <td>-0.526578</td>\n",
       "      <td>-0.248608</td>\n",
       "      <td>0.558131</td>\n",
       "      <td>0.547173</td>\n",
       "      <td>-0.530690</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0133</th>\n",
       "      <td>-0.570782</td>\n",
       "      <td>4.560938</td>\n",
       "      <td>3.690203</td>\n",
       "      <td>2.302302</td>\n",
       "      <td>0.705017</td>\n",
       "      <td>-0.455561</td>\n",
       "      <td>0.440143</td>\n",
       "      <td>1.551490</td>\n",
       "      <td>2.748361</td>\n",
       "      <td>-2.443490</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0135</th>\n",
       "      <td>-0.117803</td>\n",
       "      <td>-0.869882</td>\n",
       "      <td>0.971675</td>\n",
       "      <td>-1.517378</td>\n",
       "      <td>-1.827313</td>\n",
       "      <td>-2.082254</td>\n",
       "      <td>-0.976155</td>\n",
       "      <td>-1.920829</td>\n",
       "      <td>-1.999828</td>\n",
       "      <td>2.118295</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>2</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0136</th>\n",
       "      <td>0.874496</td>\n",
       "      <td>-3.379140</td>\n",
       "      <td>-9.386417</td>\n",
       "      <td>-0.044186</td>\n",
       "      <td>0.481281</td>\n",
       "      <td>-0.811673</td>\n",
       "      <td>-0.483516</td>\n",
       "      <td>-0.405053</td>\n",
       "      <td>0.319846</td>\n",
       "      <td>-0.551310</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0138</th>\n",
       "      <td>1.158647</td>\n",
       "      <td>4.847099</td>\n",
       "      <td>0.673716</td>\n",
       "      <td>1.441657</td>\n",
       "      <td>-0.541410</td>\n",
       "      <td>-0.216848</td>\n",
       "      <td>-0.574181</td>\n",
       "      <td>1.703794</td>\n",
       "      <td>2.426379</td>\n",
       "      <td>-2.162349</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0139</th>\n",
       "      <td>2.100204</td>\n",
       "      <td>3.857392</td>\n",
       "      <td>-2.110865</td>\n",
       "      <td>1.680220</td>\n",
       "      <td>0.543285</td>\n",
       "      <td>-2.839074</td>\n",
       "      <td>-1.455348</td>\n",
       "      <td>2.729110</td>\n",
       "      <td>4.206067</td>\n",
       "      <td>-4.028101</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0140</th>\n",
       "      <td>-0.166482</td>\n",
       "      <td>1.533517</td>\n",
       "      <td>2.621224</td>\n",
       "      <td>-0.157213</td>\n",
       "      <td>-0.250768</td>\n",
       "      <td>-0.001997</td>\n",
       "      <td>-0.033955</td>\n",
       "      <td>-1.053808</td>\n",
       "      <td>-1.000072</td>\n",
       "      <td>1.821616</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0142</th>\n",
       "      <td>-0.881733</td>\n",
       "      <td>2.119339</td>\n",
       "      <td>11.769112</td>\n",
       "      <td>1.807500</td>\n",
       "      <td>-1.036924</td>\n",
       "      <td>-1.392334</td>\n",
       "      <td>0.138771</td>\n",
       "      <td>3.207029</td>\n",
       "      <td>3.105644</td>\n",
       "      <td>-3.524379</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0143</th>\n",
       "      <td>0.237474</td>\n",
       "      <td>1.306009</td>\n",
       "      <td>4.406096</td>\n",
       "      <td>0.412958</td>\n",
       "      <td>-2.160744</td>\n",
       "      <td>-1.154757</td>\n",
       "      <td>-0.658322</td>\n",
       "      <td>1.296105</td>\n",
       "      <td>0.730830</td>\n",
       "      <td>-0.765246</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7264</th>\n",
       "      <td>0.052616</td>\n",
       "      <td>0.475062</td>\n",
       "      <td>-8.357461</td>\n",
       "      <td>2.719353</td>\n",
       "      <td>1.340783</td>\n",
       "      <td>1.720229</td>\n",
       "      <td>0.486650</td>\n",
       "      <td>1.187940</td>\n",
       "      <td>1.369706</td>\n",
       "      <td>-0.953181</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7266</th>\n",
       "      <td>-2.427949</td>\n",
       "      <td>2.504505</td>\n",
       "      <td>12.534635</td>\n",
       "      <td>0.578593</td>\n",
       "      <td>-1.317527</td>\n",
       "      <td>-0.400218</td>\n",
       "      <td>0.768747</td>\n",
       "      <td>-1.814783</td>\n",
       "      <td>-0.804663</td>\n",
       "      <td>1.092236</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7267</th>\n",
       "      <td>1.027192</td>\n",
       "      <td>-3.108903</td>\n",
       "      <td>-3.791368</td>\n",
       "      <td>-4.825164</td>\n",
       "      <td>-0.267898</td>\n",
       "      <td>0.911133</td>\n",
       "      <td>-1.830152</td>\n",
       "      <td>-2.247315</td>\n",
       "      <td>-4.519671</td>\n",
       "      <td>5.070132</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7268</th>\n",
       "      <td>0.899954</td>\n",
       "      <td>0.782333</td>\n",
       "      <td>6.789504</td>\n",
       "      <td>-0.559499</td>\n",
       "      <td>-1.123089</td>\n",
       "      <td>-0.000125</td>\n",
       "      <td>-1.000879</td>\n",
       "      <td>-0.244545</td>\n",
       "      <td>-1.339817</td>\n",
       "      <td>0.690283</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7269</th>\n",
       "      <td>-1.542249</td>\n",
       "      <td>-6.540713</td>\n",
       "      <td>-6.588955</td>\n",
       "      <td>-0.905963</td>\n",
       "      <td>-0.975963</td>\n",
       "      <td>1.026272</td>\n",
       "      <td>0.386235</td>\n",
       "      <td>-1.722726</td>\n",
       "      <td>-2.492191</td>\n",
       "      <td>2.197838</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7270</th>\n",
       "      <td>-0.818144</td>\n",
       "      <td>-2.313261</td>\n",
       "      <td>0.407206</td>\n",
       "      <td>-1.231119</td>\n",
       "      <td>1.065474</td>\n",
       "      <td>-0.039110</td>\n",
       "      <td>-0.105982</td>\n",
       "      <td>-5.473630</td>\n",
       "      <td>-5.901622</td>\n",
       "      <td>6.593430</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7273</th>\n",
       "      <td>0.133795</td>\n",
       "      <td>-7.936353</td>\n",
       "      <td>-6.036484</td>\n",
       "      <td>-1.853518</td>\n",
       "      <td>4.553318</td>\n",
       "      <td>2.124215</td>\n",
       "      <td>-0.638000</td>\n",
       "      <td>-0.010831</td>\n",
       "      <td>-1.326545</td>\n",
       "      <td>0.197701</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>5</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7275</th>\n",
       "      <td>0.119508</td>\n",
       "      <td>-3.830375</td>\n",
       "      <td>-7.073864</td>\n",
       "      <td>-3.164525</td>\n",
       "      <td>0.449460</td>\n",
       "      <td>-2.310980</td>\n",
       "      <td>-0.657651</td>\n",
       "      <td>0.304326</td>\n",
       "      <td>-0.530877</td>\n",
       "      <td>0.338423</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>5</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7276</th>\n",
       "      <td>1.452774</td>\n",
       "      <td>-0.532131</td>\n",
       "      <td>0.833160</td>\n",
       "      <td>0.619618</td>\n",
       "      <td>0.006122</td>\n",
       "      <td>-0.870304</td>\n",
       "      <td>-0.551065</td>\n",
       "      <td>-0.493778</td>\n",
       "      <td>1.031235</td>\n",
       "      <td>-0.607397</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7277</th>\n",
       "      <td>-0.836269</td>\n",
       "      <td>-2.383064</td>\n",
       "      <td>3.939454</td>\n",
       "      <td>0.441016</td>\n",
       "      <td>-0.337889</td>\n",
       "      <td>-1.065002</td>\n",
       "      <td>0.148280</td>\n",
       "      <td>-0.753005</td>\n",
       "      <td>0.553203</td>\n",
       "      <td>0.260013</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7278</th>\n",
       "      <td>-0.750752</td>\n",
       "      <td>5.208024</td>\n",
       "      <td>4.780788</td>\n",
       "      <td>1.853149</td>\n",
       "      <td>-0.159819</td>\n",
       "      <td>-1.297136</td>\n",
       "      <td>0.650400</td>\n",
       "      <td>0.951615</td>\n",
       "      <td>2.215585</td>\n",
       "      <td>-1.703582</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7279</th>\n",
       "      <td>-0.591865</td>\n",
       "      <td>3.546974</td>\n",
       "      <td>6.984630</td>\n",
       "      <td>3.625156</td>\n",
       "      <td>-0.001635</td>\n",
       "      <td>3.217996</td>\n",
       "      <td>1.287618</td>\n",
       "      <td>2.715368</td>\n",
       "      <td>1.277015</td>\n",
       "      <td>-1.698399</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7280</th>\n",
       "      <td>3.168238</td>\n",
       "      <td>3.348863</td>\n",
       "      <td>2.440740</td>\n",
       "      <td>0.188479</td>\n",
       "      <td>0.550790</td>\n",
       "      <td>0.139212</td>\n",
       "      <td>-1.515452</td>\n",
       "      <td>-0.981956</td>\n",
       "      <td>-0.771772</td>\n",
       "      <td>0.736326</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>7</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7281</th>\n",
       "      <td>-0.708823</td>\n",
       "      <td>-1.609151</td>\n",
       "      <td>-5.090321</td>\n",
       "      <td>0.525543</td>\n",
       "      <td>0.303806</td>\n",
       "      <td>2.379776</td>\n",
       "      <td>0.551161</td>\n",
       "      <td>0.059091</td>\n",
       "      <td>0.492005</td>\n",
       "      <td>-0.799303</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>5</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7283</th>\n",
       "      <td>-1.524783</td>\n",
       "      <td>6.702166</td>\n",
       "      <td>12.168263</td>\n",
       "      <td>-0.127327</td>\n",
       "      <td>-2.139917</td>\n",
       "      <td>0.550919</td>\n",
       "      <td>0.340746</td>\n",
       "      <td>-0.509159</td>\n",
       "      <td>-1.216963</td>\n",
       "      <td>0.841771</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7284</th>\n",
       "      <td>-1.585521</td>\n",
       "      <td>2.486830</td>\n",
       "      <td>3.896734</td>\n",
       "      <td>0.832220</td>\n",
       "      <td>-0.452137</td>\n",
       "      <td>0.954621</td>\n",
       "      <td>2.116733</td>\n",
       "      <td>-0.226250</td>\n",
       "      <td>-0.314260</td>\n",
       "      <td>-0.059567</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7285</th>\n",
       "      <td>2.112241</td>\n",
       "      <td>-4.308329</td>\n",
       "      <td>-10.057469</td>\n",
       "      <td>-1.500446</td>\n",
       "      <td>0.268187</td>\n",
       "      <td>0.055063</td>\n",
       "      <td>-2.512579</td>\n",
       "      <td>-0.288444</td>\n",
       "      <td>-0.051126</td>\n",
       "      <td>-0.545414</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7286</th>\n",
       "      <td>0.464739</td>\n",
       "      <td>0.329545</td>\n",
       "      <td>0.350963</td>\n",
       "      <td>-1.888931</td>\n",
       "      <td>-0.192087</td>\n",
       "      <td>-3.063396</td>\n",
       "      <td>-1.545657</td>\n",
       "      <td>0.978685</td>\n",
       "      <td>0.061323</td>\n",
       "      <td>-0.506717</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7287</th>\n",
       "      <td>-0.461483</td>\n",
       "      <td>-6.788634</td>\n",
       "      <td>-4.532691</td>\n",
       "      <td>-0.648133</td>\n",
       "      <td>0.898894</td>\n",
       "      <td>-0.037595</td>\n",
       "      <td>-0.657459</td>\n",
       "      <td>-0.130143</td>\n",
       "      <td>-1.043855</td>\n",
       "      <td>0.465353</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>5</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7288</th>\n",
       "      <td>0.084161</td>\n",
       "      <td>-5.730170</td>\n",
       "      <td>-0.889219</td>\n",
       "      <td>-3.048048</td>\n",
       "      <td>-0.746198</td>\n",
       "      <td>-0.468645</td>\n",
       "      <td>-1.043475</td>\n",
       "      <td>-0.499714</td>\n",
       "      <td>-0.199662</td>\n",
       "      <td>0.688354</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>8</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7289</th>\n",
       "      <td>0.948163</td>\n",
       "      <td>-2.915446</td>\n",
       "      <td>-9.226002</td>\n",
       "      <td>-1.145563</td>\n",
       "      <td>-0.094301</td>\n",
       "      <td>-1.420274</td>\n",
       "      <td>-1.017675</td>\n",
       "      <td>-0.254609</td>\n",
       "      <td>1.379707</td>\n",
       "      <td>-1.184310</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7291</th>\n",
       "      <td>1.676027</td>\n",
       "      <td>-10.800482</td>\n",
       "      <td>-13.742885</td>\n",
       "      <td>-3.676589</td>\n",
       "      <td>0.619119</td>\n",
       "      <td>1.983350</td>\n",
       "      <td>-2.676246</td>\n",
       "      <td>-2.079944</td>\n",
       "      <td>-2.595874</td>\n",
       "      <td>1.998022</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7292</th>\n",
       "      <td>-0.991138</td>\n",
       "      <td>-5.250101</td>\n",
       "      <td>0.637368</td>\n",
       "      <td>-2.451837</td>\n",
       "      <td>-1.300904</td>\n",
       "      <td>-1.138077</td>\n",
       "      <td>-0.387298</td>\n",
       "      <td>-0.420057</td>\n",
       "      <td>-2.341863</td>\n",
       "      <td>1.763887</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7293</th>\n",
       "      <td>0.295280</td>\n",
       "      <td>-5.007402</td>\n",
       "      <td>-6.453299</td>\n",
       "      <td>-0.666104</td>\n",
       "      <td>-0.802562</td>\n",
       "      <td>-0.854217</td>\n",
       "      <td>-0.086637</td>\n",
       "      <td>-1.197647</td>\n",
       "      <td>0.210241</td>\n",
       "      <td>0.307119</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7294</th>\n",
       "      <td>-0.241594</td>\n",
       "      <td>-6.497156</td>\n",
       "      <td>7.094790</td>\n",
       "      <td>-0.526408</td>\n",
       "      <td>-0.312066</td>\n",
       "      <td>-1.249597</td>\n",
       "      <td>-0.619259</td>\n",
       "      <td>-1.585000</td>\n",
       "      <td>-0.730437</td>\n",
       "      <td>0.616089</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7295</th>\n",
       "      <td>-0.097627</td>\n",
       "      <td>-4.277540</td>\n",
       "      <td>-5.748505</td>\n",
       "      <td>-3.180437</td>\n",
       "      <td>-0.865192</td>\n",
       "      <td>-0.979926</td>\n",
       "      <td>-1.582625</td>\n",
       "      <td>-0.735149</td>\n",
       "      <td>-0.468519</td>\n",
       "      <td>0.321434</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>3</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7296</th>\n",
       "      <td>-0.616957</td>\n",
       "      <td>-4.852698</td>\n",
       "      <td>1.617394</td>\n",
       "      <td>-3.523299</td>\n",
       "      <td>-1.540831</td>\n",
       "      <td>1.360917</td>\n",
       "      <td>-0.889991</td>\n",
       "      <td>-1.775554</td>\n",
       "      <td>-2.728289</td>\n",
       "      <td>1.738995</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>5</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7297</th>\n",
       "      <td>-0.546331</td>\n",
       "      <td>-8.314218</td>\n",
       "      <td>-2.774328</td>\n",
       "      <td>-2.203446</td>\n",
       "      <td>-0.507499</td>\n",
       "      <td>-0.153017</td>\n",
       "      <td>-0.957462</td>\n",
       "      <td>-0.821364</td>\n",
       "      <td>-1.953621</td>\n",
       "      <td>2.779270</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7298</th>\n",
       "      <td>1.160625</td>\n",
       "      <td>-6.085674</td>\n",
       "      <td>-3.616640</td>\n",
       "      <td>-2.815058</td>\n",
       "      <td>-0.101404</td>\n",
       "      <td>-0.417562</td>\n",
       "      <td>-1.730290</td>\n",
       "      <td>0.260045</td>\n",
       "      <td>0.183894</td>\n",
       "      <td>-0.057154</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>1</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7299</th>\n",
       "      <td>0.110470</td>\n",
       "      <td>-7.775091</td>\n",
       "      <td>-12.366570</td>\n",
       "      <td>-3.473074</td>\n",
       "      <td>1.077668</td>\n",
       "      <td>0.441568</td>\n",
       "      <td>-1.243941</td>\n",
       "      <td>-0.409818</td>\n",
       "      <td>-1.179034</td>\n",
       "      <td>0.926047</td>\n",
       "      <td>...</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>10</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "      <td>Not</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1992 rows  2118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Interleukin-6 signaling  Apoptosis  Hemostasis  \\\n",
       "MB-0002                -0.464915   1.150390    7.995550   \n",
       "MB-0008                 1.135676  -5.398171   -1.468534   \n",
       "MB-0010                -0.016407  -2.252856    1.688631   \n",
       "MB-0035                -0.045984   3.050125   14.221321   \n",
       "MB-0036                -1.107911   1.445771    5.790792   \n",
       "MB-0050                 1.003819   1.441920   -1.964812   \n",
       "MB-0059                -0.473150   0.186910    6.189742   \n",
       "MB-0060                 0.681486 -12.524534   -0.370151   \n",
       "MB-0066                 0.539735  -6.459812   -0.231027   \n",
       "MB-0101                 1.391563  -2.378717   -4.880185   \n",
       "MB-0102                 0.975992  -4.692292   -2.159457   \n",
       "MB-0106                -1.716131   3.055953    2.940126   \n",
       "MB-0112                 0.721624  -1.939072   -0.898322   \n",
       "MB-0115                 2.891250  -7.520513   -4.479024   \n",
       "MB-0117                -1.218244  -0.139969   -2.140516   \n",
       "MB-0119                 0.324922  -1.494166    2.681937   \n",
       "MB-0120                -1.282450  -1.312260   -5.283445   \n",
       "MB-0121                 0.066524  -3.182866    0.578324   \n",
       "MB-0122                 0.241645  -0.896690    4.363559   \n",
       "MB-0123                 0.632205  -4.939672    1.387583   \n",
       "MB-0126                 0.430240   0.585299    0.323267   \n",
       "MB-0131                 0.059170   0.794649    0.309136   \n",
       "MB-0133                -0.570782   4.560938    3.690203   \n",
       "MB-0135                -0.117803  -0.869882    0.971675   \n",
       "MB-0136                 0.874496  -3.379140   -9.386417   \n",
       "MB-0138                 1.158647   4.847099    0.673716   \n",
       "MB-0139                 2.100204   3.857392   -2.110865   \n",
       "MB-0140                -0.166482   1.533517    2.621224   \n",
       "MB-0142                -0.881733   2.119339   11.769112   \n",
       "MB-0143                 0.237474   1.306009    4.406096   \n",
       "...                          ...        ...         ...   \n",
       "MB-7264                 0.052616   0.475062   -8.357461   \n",
       "MB-7266                -2.427949   2.504505   12.534635   \n",
       "MB-7267                 1.027192  -3.108903   -3.791368   \n",
       "MB-7268                 0.899954   0.782333    6.789504   \n",
       "MB-7269                -1.542249  -6.540713   -6.588955   \n",
       "MB-7270                -0.818144  -2.313261    0.407206   \n",
       "MB-7273                 0.133795  -7.936353   -6.036484   \n",
       "MB-7275                 0.119508  -3.830375   -7.073864   \n",
       "MB-7276                 1.452774  -0.532131    0.833160   \n",
       "MB-7277                -0.836269  -2.383064    3.939454   \n",
       "MB-7278                -0.750752   5.208024    4.780788   \n",
       "MB-7279                -0.591865   3.546974    6.984630   \n",
       "MB-7280                 3.168238   3.348863    2.440740   \n",
       "MB-7281                -0.708823  -1.609151   -5.090321   \n",
       "MB-7283                -1.524783   6.702166   12.168263   \n",
       "MB-7284                -1.585521   2.486830    3.896734   \n",
       "MB-7285                 2.112241  -4.308329  -10.057469   \n",
       "MB-7286                 0.464739   0.329545    0.350963   \n",
       "MB-7287                -0.461483  -6.788634   -4.532691   \n",
       "MB-7288                 0.084161  -5.730170   -0.889219   \n",
       "MB-7289                 0.948163  -2.915446   -9.226002   \n",
       "MB-7291                 1.676027 -10.800482  -13.742885   \n",
       "MB-7292                -0.991138  -5.250101    0.637368   \n",
       "MB-7293                 0.295280  -5.007402   -6.453299   \n",
       "MB-7294                -0.241594  -6.497156    7.094790   \n",
       "MB-7295                -0.097627  -4.277540   -5.748505   \n",
       "MB-7296                -0.616957  -4.852698    1.617394   \n",
       "MB-7297                -0.546331  -8.314218   -2.774328   \n",
       "MB-7298                 1.160625  -6.085674   -3.616640   \n",
       "MB-7299                 0.110470  -7.775091  -12.366570   \n",
       "\n",
       "         Intrinsic Pathway for Apoptosis  PKB-mediated events  PI3K Cascade  \\\n",
       "MB-0002                        -2.133505            -1.005971     -1.934175   \n",
       "MB-0008                        -2.783638            -0.042169     -1.152715   \n",
       "MB-0010                        -1.840132             0.439943      0.907014   \n",
       "MB-0035                        -1.010664            -2.460802      0.033379   \n",
       "MB-0036                         1.275586            -1.040463     -2.897256   \n",
       "MB-0050                        -0.102851            -0.049525     -3.192611   \n",
       "MB-0059                        -1.018816            -1.353297     -2.260648   \n",
       "MB-0060                        -2.756785             0.077299     -0.448584   \n",
       "MB-0066                        -1.086133            -0.017782     -0.218056   \n",
       "MB-0101                         1.025275            -0.772772     -0.612536   \n",
       "MB-0102                        -1.916437            -0.424554      0.999638   \n",
       "MB-0106                         0.884104            -0.823157     -0.688135   \n",
       "MB-0112                         0.265369            -0.368646     -1.915540   \n",
       "MB-0115                        -4.139860            -0.845741     -0.875985   \n",
       "MB-0117                         0.796431             1.913941      0.425881   \n",
       "MB-0119                        -0.732578            -1.160084     -0.281727   \n",
       "MB-0120                        -1.769936            -0.042372      1.623986   \n",
       "MB-0121                        -1.262635            -0.358424     -0.596381   \n",
       "MB-0122                        -0.332297            -0.933940     -2.378229   \n",
       "MB-0123                        -2.019270             1.906553      1.130355   \n",
       "MB-0126                        -1.390457             0.169088     -4.192595   \n",
       "MB-0131                         0.093709            -0.878579     -0.526578   \n",
       "MB-0133                         2.302302             0.705017     -0.455561   \n",
       "MB-0135                        -1.517378            -1.827313     -2.082254   \n",
       "MB-0136                        -0.044186             0.481281     -0.811673   \n",
       "MB-0138                         1.441657            -0.541410     -0.216848   \n",
       "MB-0139                         1.680220             0.543285     -2.839074   \n",
       "MB-0140                        -0.157213            -0.250768     -0.001997   \n",
       "MB-0142                         1.807500            -1.036924     -1.392334   \n",
       "MB-0143                         0.412958            -2.160744     -1.154757   \n",
       "...                                  ...                  ...           ...   \n",
       "MB-7264                         2.719353             1.340783      1.720229   \n",
       "MB-7266                         0.578593            -1.317527     -0.400218   \n",
       "MB-7267                        -4.825164            -0.267898      0.911133   \n",
       "MB-7268                        -0.559499            -1.123089     -0.000125   \n",
       "MB-7269                        -0.905963            -0.975963      1.026272   \n",
       "MB-7270                        -1.231119             1.065474     -0.039110   \n",
       "MB-7273                        -1.853518             4.553318      2.124215   \n",
       "MB-7275                        -3.164525             0.449460     -2.310980   \n",
       "MB-7276                         0.619618             0.006122     -0.870304   \n",
       "MB-7277                         0.441016            -0.337889     -1.065002   \n",
       "MB-7278                         1.853149            -0.159819     -1.297136   \n",
       "MB-7279                         3.625156            -0.001635      3.217996   \n",
       "MB-7280                         0.188479             0.550790      0.139212   \n",
       "MB-7281                         0.525543             0.303806      2.379776   \n",
       "MB-7283                        -0.127327            -2.139917      0.550919   \n",
       "MB-7284                         0.832220            -0.452137      0.954621   \n",
       "MB-7285                        -1.500446             0.268187      0.055063   \n",
       "MB-7286                        -1.888931            -0.192087     -3.063396   \n",
       "MB-7287                        -0.648133             0.898894     -0.037595   \n",
       "MB-7288                        -3.048048            -0.746198     -0.468645   \n",
       "MB-7289                        -1.145563            -0.094301     -1.420274   \n",
       "MB-7291                        -3.676589             0.619119      1.983350   \n",
       "MB-7292                        -2.451837            -1.300904     -1.138077   \n",
       "MB-7293                        -0.666104            -0.802562     -0.854217   \n",
       "MB-7294                        -0.526408            -0.312066     -1.249597   \n",
       "MB-7295                        -3.180437            -0.865192     -0.979926   \n",
       "MB-7296                        -3.523299            -1.540831      1.360917   \n",
       "MB-7297                        -2.203446            -0.507499     -0.153017   \n",
       "MB-7298                        -2.815058            -0.101404     -0.417562   \n",
       "MB-7299                        -3.473074             1.077668      0.441568   \n",
       "\n",
       "         MAPK3 (ERK1) activation  Translesion synthesis by REV1  \\\n",
       "MB-0002                 0.682674                      -1.740452   \n",
       "MB-0008                -0.440677                      -0.738142   \n",
       "MB-0010                -0.946115                      -1.195843   \n",
       "MB-0035                -0.040401                       1.589719   \n",
       "MB-0036                 0.357232                       0.276760   \n",
       "MB-0050                -1.181677                       0.429309   \n",
       "MB-0059                 0.119735                       0.142322   \n",
       "MB-0060                -1.396643                      -3.632946   \n",
       "MB-0066                -0.731439                      -0.210748   \n",
       "MB-0101                -0.850303                       1.193591   \n",
       "MB-0102                -1.688466                       0.167435   \n",
       "MB-0106                 1.656749                      -0.002856   \n",
       "MB-0112                -1.256289                       1.112242   \n",
       "MB-0115                -2.905533                      -3.344632   \n",
       "MB-0117                 0.909888                       0.240269   \n",
       "MB-0119                -0.094547                       0.530474   \n",
       "MB-0120                 0.562100                      -0.550128   \n",
       "MB-0121                -0.376231                      -1.426032   \n",
       "MB-0122                 0.431727                      -0.993327   \n",
       "MB-0123                -0.342736                      -4.240944   \n",
       "MB-0126                -1.174283                       1.428986   \n",
       "MB-0131                -0.248608                       0.558131   \n",
       "MB-0133                 0.440143                       1.551490   \n",
       "MB-0135                -0.976155                      -1.920829   \n",
       "MB-0136                -0.483516                      -0.405053   \n",
       "MB-0138                -0.574181                       1.703794   \n",
       "MB-0139                -1.455348                       2.729110   \n",
       "MB-0140                -0.033955                      -1.053808   \n",
       "MB-0142                 0.138771                       3.207029   \n",
       "MB-0143                -0.658322                       1.296105   \n",
       "...                          ...                            ...   \n",
       "MB-7264                 0.486650                       1.187940   \n",
       "MB-7266                 0.768747                      -1.814783   \n",
       "MB-7267                -1.830152                      -2.247315   \n",
       "MB-7268                -1.000879                      -0.244545   \n",
       "MB-7269                 0.386235                      -1.722726   \n",
       "MB-7270                -0.105982                      -5.473630   \n",
       "MB-7273                -0.638000                      -0.010831   \n",
       "MB-7275                -0.657651                       0.304326   \n",
       "MB-7276                -0.551065                      -0.493778   \n",
       "MB-7277                 0.148280                      -0.753005   \n",
       "MB-7278                 0.650400                       0.951615   \n",
       "MB-7279                 1.287618                       2.715368   \n",
       "MB-7280                -1.515452                      -0.981956   \n",
       "MB-7281                 0.551161                       0.059091   \n",
       "MB-7283                 0.340746                      -0.509159   \n",
       "MB-7284                 2.116733                      -0.226250   \n",
       "MB-7285                -2.512579                      -0.288444   \n",
       "MB-7286                -1.545657                       0.978685   \n",
       "MB-7287                -0.657459                      -0.130143   \n",
       "MB-7288                -1.043475                      -0.499714   \n",
       "MB-7289                -1.017675                      -0.254609   \n",
       "MB-7291                -2.676246                      -2.079944   \n",
       "MB-7292                -0.387298                      -0.420057   \n",
       "MB-7293                -0.086637                      -1.197647   \n",
       "MB-7294                -0.619259                      -1.585000   \n",
       "MB-7295                -1.582625                      -0.735149   \n",
       "MB-7296                -0.889991                      -1.775554   \n",
       "MB-7297                -0.957462                      -0.821364   \n",
       "MB-7298                -1.730290                       0.260045   \n",
       "MB-7299                -1.243941                      -0.409818   \n",
       "\n",
       "         Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template  \\\n",
       "MB-0002                                          -1.758971                                    \n",
       "MB-0008                                          -0.950279                                    \n",
       "MB-0010                                          -2.019880                                    \n",
       "MB-0035                                           0.338310                                    \n",
       "MB-0036                                           1.075780                                    \n",
       "MB-0050                                           1.425590                                    \n",
       "MB-0059                                          -0.111307                                    \n",
       "MB-0060                                          -5.847386                                    \n",
       "MB-0066                                          -0.525239                                    \n",
       "MB-0101                                           2.045964                                    \n",
       "MB-0102                                          -1.977181                                    \n",
       "MB-0106                                           2.292241                                    \n",
       "MB-0112                                           1.566163                                    \n",
       "MB-0115                                          -3.650589                                    \n",
       "MB-0117                                           1.165142                                    \n",
       "MB-0119                                           0.249794                                    \n",
       "MB-0120                                          -1.018389                                    \n",
       "MB-0121                                           0.031787                                    \n",
       "MB-0122                                           0.664990                                    \n",
       "MB-0123                                          -4.235351                                    \n",
       "MB-0126                                           1.529271                                    \n",
       "MB-0131                                           0.547173                                    \n",
       "MB-0133                                           2.748361                                    \n",
       "MB-0135                                          -1.999828                                    \n",
       "MB-0136                                           0.319846                                    \n",
       "MB-0138                                           2.426379                                    \n",
       "MB-0139                                           4.206067                                    \n",
       "MB-0140                                          -1.000072                                    \n",
       "MB-0142                                           3.105644                                    \n",
       "MB-0143                                           0.730830                                    \n",
       "...                                                    ...                                    \n",
       "MB-7264                                           1.369706                                    \n",
       "MB-7266                                          -0.804663                                    \n",
       "MB-7267                                          -4.519671                                    \n",
       "MB-7268                                          -1.339817                                    \n",
       "MB-7269                                          -2.492191                                    \n",
       "MB-7270                                          -5.901622                                    \n",
       "MB-7273                                          -1.326545                                    \n",
       "MB-7275                                          -0.530877                                    \n",
       "MB-7276                                           1.031235                                    \n",
       "MB-7277                                           0.553203                                    \n",
       "MB-7278                                           2.215585                                    \n",
       "MB-7279                                           1.277015                                    \n",
       "MB-7280                                          -0.771772                                    \n",
       "MB-7281                                           0.492005                                    \n",
       "MB-7283                                          -1.216963                                    \n",
       "MB-7284                                          -0.314260                                    \n",
       "MB-7285                                          -0.051126                                    \n",
       "MB-7286                                           0.061323                                    \n",
       "MB-7287                                          -1.043855                                    \n",
       "MB-7288                                          -0.199662                                    \n",
       "MB-7289                                           1.379707                                    \n",
       "MB-7291                                          -2.595874                                    \n",
       "MB-7292                                          -2.341863                                    \n",
       "MB-7293                                           0.210241                                    \n",
       "MB-7294                                          -0.730437                                    \n",
       "MB-7295                                          -0.468519                                    \n",
       "MB-7296                                          -2.728289                                    \n",
       "MB-7297                                          -1.953621                                    \n",
       "MB-7298                                           0.183894                                    \n",
       "MB-7299                                          -1.179034                                    \n",
       "\n",
       "         Recognition of DNA damage by PCNA-containing replication complex  \\\n",
       "MB-0002                                           2.306896                  \n",
       "MB-0008                                           0.857385                  \n",
       "MB-0010                                           2.452047                  \n",
       "MB-0035                                          -0.025657                  \n",
       "MB-0036                                          -1.226257                  \n",
       "MB-0050                                          -1.375613                  \n",
       "MB-0059                                          -0.685721                  \n",
       "MB-0060                                           6.291344                  \n",
       "MB-0066                                           1.251225                  \n",
       "MB-0101                                          -1.989934                  \n",
       "MB-0102                                           1.916599                  \n",
       "MB-0106                                          -1.652733                  \n",
       "MB-0112                                          -1.291966                  \n",
       "MB-0115                                           3.451235                  \n",
       "MB-0117                                          -1.165904                  \n",
       "MB-0119                                          -0.944412                  \n",
       "MB-0120                                           0.599823                  \n",
       "MB-0121                                           0.056474                  \n",
       "MB-0122                                          -0.454606                  \n",
       "MB-0123                                           4.132531                  \n",
       "MB-0126                                          -2.216522                  \n",
       "MB-0131                                          -0.530690                  \n",
       "MB-0133                                          -2.443490                  \n",
       "MB-0135                                           2.118295                  \n",
       "MB-0136                                          -0.551310                  \n",
       "MB-0138                                          -2.162349                  \n",
       "MB-0139                                          -4.028101                  \n",
       "MB-0140                                           1.821616                  \n",
       "MB-0142                                          -3.524379                  \n",
       "MB-0143                                          -0.765246                  \n",
       "...                                                    ...                  \n",
       "MB-7264                                          -0.953181                  \n",
       "MB-7266                                           1.092236                  \n",
       "MB-7267                                           5.070132                  \n",
       "MB-7268                                           0.690283                  \n",
       "MB-7269                                           2.197838                  \n",
       "MB-7270                                           6.593430                  \n",
       "MB-7273                                           0.197701                  \n",
       "MB-7275                                           0.338423                  \n",
       "MB-7276                                          -0.607397                  \n",
       "MB-7277                                           0.260013                  \n",
       "MB-7278                                          -1.703582                  \n",
       "MB-7279                                          -1.698399                  \n",
       "MB-7280                                           0.736326                  \n",
       "MB-7281                                          -0.799303                  \n",
       "MB-7283                                           0.841771                  \n",
       "MB-7284                                          -0.059567                  \n",
       "MB-7285                                          -0.545414                  \n",
       "MB-7286                                          -0.506717                  \n",
       "MB-7287                                           0.465353                  \n",
       "MB-7288                                           0.688354                  \n",
       "MB-7289                                          -1.184310                  \n",
       "MB-7291                                           1.998022                  \n",
       "MB-7292                                           1.763887                  \n",
       "MB-7293                                           0.307119                  \n",
       "MB-7294                                           0.616089                  \n",
       "MB-7295                                           0.321434                  \n",
       "MB-7296                                           1.738995                  \n",
       "MB-7297                                           2.779270                  \n",
       "MB-7298                                          -0.057154                  \n",
       "MB-7299                                           0.926047                  \n",
       "\n",
       "               ...          Intclust 9 or not  Intclust 7 or not  \\\n",
       "MB-0002        ...                        Not                Not   \n",
       "MB-0008        ...                          9                Not   \n",
       "MB-0010        ...                        Not                  7   \n",
       "MB-0035        ...                        Not                Not   \n",
       "MB-0036        ...                        Not                Not   \n",
       "MB-0050        ...                        Not                Not   \n",
       "MB-0059        ...                        Not                Not   \n",
       "MB-0060        ...                        Not                Not   \n",
       "MB-0066        ...                        Not                  7   \n",
       "MB-0101        ...                        Not                Not   \n",
       "MB-0102        ...                        Not                Not   \n",
       "MB-0106        ...                        Not                Not   \n",
       "MB-0112        ...                        Not                Not   \n",
       "MB-0115        ...                        Not                Not   \n",
       "MB-0117        ...                        Not                Not   \n",
       "MB-0119        ...                        Not                  7   \n",
       "MB-0120        ...                          9                Not   \n",
       "MB-0121        ...                        Not                Not   \n",
       "MB-0122        ...                        Not                Not   \n",
       "MB-0123        ...                        Not                Not   \n",
       "MB-0126        ...                        Not                Not   \n",
       "MB-0131        ...                        Not                Not   \n",
       "MB-0133        ...                        Not                Not   \n",
       "MB-0135        ...                        Not                Not   \n",
       "MB-0136        ...                        Not                Not   \n",
       "MB-0138        ...                        Not                Not   \n",
       "MB-0139        ...                        Not                Not   \n",
       "MB-0140        ...                        Not                Not   \n",
       "MB-0142        ...                        Not                Not   \n",
       "MB-0143        ...                          9                Not   \n",
       "...            ...                        ...                ...   \n",
       "MB-7264        ...                        Not                Not   \n",
       "MB-7266        ...                          9                Not   \n",
       "MB-7267        ...                        Not                Not   \n",
       "MB-7268        ...                        Not                Not   \n",
       "MB-7269        ...                        Not                Not   \n",
       "MB-7270        ...                        Not                Not   \n",
       "MB-7273        ...                        Not                Not   \n",
       "MB-7275        ...                        Not                Not   \n",
       "MB-7276        ...                        Not                Not   \n",
       "MB-7277        ...                        Not                Not   \n",
       "MB-7278        ...                        Not                Not   \n",
       "MB-7279        ...                          9                Not   \n",
       "MB-7280        ...                        Not                  7   \n",
       "MB-7281        ...                        Not                Not   \n",
       "MB-7283        ...                        Not                Not   \n",
       "MB-7284        ...                        Not                Not   \n",
       "MB-7285        ...                        Not                Not   \n",
       "MB-7286        ...                        Not                Not   \n",
       "MB-7287        ...                        Not                Not   \n",
       "MB-7288        ...                        Not                Not   \n",
       "MB-7289        ...                        Not                Not   \n",
       "MB-7291        ...                        Not                Not   \n",
       "MB-7292        ...                        Not                Not   \n",
       "MB-7293        ...                        Not                Not   \n",
       "MB-7294        ...                        Not                Not   \n",
       "MB-7295        ...                        Not                Not   \n",
       "MB-7296        ...                        Not                Not   \n",
       "MB-7297        ...                        Not                Not   \n",
       "MB-7298        ...                        Not                Not   \n",
       "MB-7299        ...                        Not                Not   \n",
       "\n",
       "         Intclust 3 or not  Intclust 8 or not  Intclust 10 or not  \\\n",
       "MB-0002                Not                Not                 Not   \n",
       "MB-0008                Not                Not                 Not   \n",
       "MB-0010                Not                Not                 Not   \n",
       "MB-0035                  3                Not                 Not   \n",
       "MB-0036                  3                Not                 Not   \n",
       "MB-0050                Not                  8                 Not   \n",
       "MB-0059                Not                  8                 Not   \n",
       "MB-0060                Not                Not                  10   \n",
       "MB-0066                Not                Not                 Not   \n",
       "MB-0101                Not                  8                 Not   \n",
       "MB-0102                  3                Not                 Not   \n",
       "MB-0106                Not                  8                 Not   \n",
       "MB-0112                  3                Not                 Not   \n",
       "MB-0115                Not                Not                  10   \n",
       "MB-0117                Not                Not                 Not   \n",
       "MB-0119                Not                Not                 Not   \n",
       "MB-0120                Not                Not                 Not   \n",
       "MB-0121                Not                  8                 Not   \n",
       "MB-0122                Not                  8                 Not   \n",
       "MB-0123                Not                Not                 Not   \n",
       "MB-0126                Not                  8                 Not   \n",
       "MB-0131                Not                  8                 Not   \n",
       "MB-0133                Not                  8                 Not   \n",
       "MB-0135                Not                Not                 Not   \n",
       "MB-0136                  3                Not                 Not   \n",
       "MB-0138                  3                Not                 Not   \n",
       "MB-0139                  3                Not                 Not   \n",
       "MB-0140                Not                  8                 Not   \n",
       "MB-0142                Not                Not                 Not   \n",
       "MB-0143                Not                Not                 Not   \n",
       "...                    ...                ...                 ...   \n",
       "MB-7264                Not                Not                 Not   \n",
       "MB-7266                Not                Not                 Not   \n",
       "MB-7267                Not                Not                  10   \n",
       "MB-7268                Not                  8                 Not   \n",
       "MB-7269                Not                Not                  10   \n",
       "MB-7270                Not                Not                  10   \n",
       "MB-7273                Not                Not                 Not   \n",
       "MB-7275                Not                Not                 Not   \n",
       "MB-7276                Not                  8                 Not   \n",
       "MB-7277                Not                  8                 Not   \n",
       "MB-7278                  3                Not                 Not   \n",
       "MB-7279                Not                Not                 Not   \n",
       "MB-7280                Not                Not                 Not   \n",
       "MB-7281                Not                Not                 Not   \n",
       "MB-7283                Not                Not                 Not   \n",
       "MB-7284                Not                Not                 Not   \n",
       "MB-7285                Not                Not                 Not   \n",
       "MB-7286                  3                Not                 Not   \n",
       "MB-7287                Not                Not                 Not   \n",
       "MB-7288                Not                  8                 Not   \n",
       "MB-7289                Not                Not                 Not   \n",
       "MB-7291                Not                Not                 Not   \n",
       "MB-7292                Not                Not                 Not   \n",
       "MB-7293                  3                Not                 Not   \n",
       "MB-7294                Not                Not                 Not   \n",
       "MB-7295                  3                Not                 Not   \n",
       "MB-7296                Not                Not                 Not   \n",
       "MB-7297                Not                Not                 Not   \n",
       "MB-7298                Not                Not                 Not   \n",
       "MB-7299                Not                Not                  10   \n",
       "\n",
       "         Intclust 1 or not  Intclust 2 or not  Intclust 5 or not  \\\n",
       "MB-0002                Not                Not                Not   \n",
       "MB-0008                Not                Not                Not   \n",
       "MB-0010                Not                Not                Not   \n",
       "MB-0035                Not                Not                Not   \n",
       "MB-0036                Not                Not                Not   \n",
       "MB-0050                Not                Not                Not   \n",
       "MB-0059                Not                Not                Not   \n",
       "MB-0060                Not                Not                Not   \n",
       "MB-0066                Not                Not                Not   \n",
       "MB-0101                Not                Not                Not   \n",
       "MB-0102                Not                Not                Not   \n",
       "MB-0106                Not                Not                Not   \n",
       "MB-0112                Not                Not                Not   \n",
       "MB-0115                Not                Not                Not   \n",
       "MB-0117                  1                Not                Not   \n",
       "MB-0119                Not                Not                Not   \n",
       "MB-0120                Not                Not                Not   \n",
       "MB-0121                Not                Not                Not   \n",
       "MB-0122                Not                Not                Not   \n",
       "MB-0123                  1                Not                Not   \n",
       "MB-0126                Not                Not                Not   \n",
       "MB-0131                Not                Not                Not   \n",
       "MB-0133                Not                Not                Not   \n",
       "MB-0135                Not                  2                Not   \n",
       "MB-0136                Not                Not                Not   \n",
       "MB-0138                Not                Not                Not   \n",
       "MB-0139                Not                Not                Not   \n",
       "MB-0140                Not                Not                Not   \n",
       "MB-0142                  1                Not                Not   \n",
       "MB-0143                Not                Not                Not   \n",
       "...                    ...                ...                ...   \n",
       "MB-7264                Not                Not                Not   \n",
       "MB-7266                Not                Not                Not   \n",
       "MB-7267                Not                Not                Not   \n",
       "MB-7268                Not                Not                Not   \n",
       "MB-7269                Not                Not                Not   \n",
       "MB-7270                Not                Not                Not   \n",
       "MB-7273                Not                Not                  5   \n",
       "MB-7275                Not                Not                  5   \n",
       "MB-7276                Not                Not                Not   \n",
       "MB-7277                Not                Not                Not   \n",
       "MB-7278                Not                Not                Not   \n",
       "MB-7279                Not                Not                Not   \n",
       "MB-7280                Not                Not                Not   \n",
       "MB-7281                Not                Not                  5   \n",
       "MB-7283                Not                Not                Not   \n",
       "MB-7284                Not                Not                Not   \n",
       "MB-7285                Not                Not                Not   \n",
       "MB-7286                Not                Not                Not   \n",
       "MB-7287                Not                Not                  5   \n",
       "MB-7288                Not                Not                Not   \n",
       "MB-7289                Not                Not                Not   \n",
       "MB-7291                  1                Not                Not   \n",
       "MB-7292                Not                Not                Not   \n",
       "MB-7293                Not                Not                Not   \n",
       "MB-7294                  1                Not                Not   \n",
       "MB-7295                Not                Not                Not   \n",
       "MB-7296                Not                Not                  5   \n",
       "MB-7297                  1                Not                Not   \n",
       "MB-7298                  1                Not                Not   \n",
       "MB-7299                Not                Not                Not   \n",
       "\n",
       "         Intclust 4ER- or not  Intclust 6 or not  \n",
       "MB-0002                   Not                Not  \n",
       "MB-0008                   Not                Not  \n",
       "MB-0010                   Not                Not  \n",
       "MB-0035                   Not                Not  \n",
       "MB-0036                   Not                Not  \n",
       "MB-0050                   Not                Not  \n",
       "MB-0059                   Not                Not  \n",
       "MB-0060                   Not                Not  \n",
       "MB-0066                   Not                Not  \n",
       "MB-0101                   Not                Not  \n",
       "MB-0102                   Not                Not  \n",
       "MB-0106                   Not                Not  \n",
       "MB-0112                   Not                Not  \n",
       "MB-0115                   Not                Not  \n",
       "MB-0117                   Not                Not  \n",
       "MB-0119                   Not                Not  \n",
       "MB-0120                   Not                Not  \n",
       "MB-0121                   Not                Not  \n",
       "MB-0122                   Not                Not  \n",
       "MB-0123                   Not                Not  \n",
       "MB-0126                   Not                Not  \n",
       "MB-0131                   Not                Not  \n",
       "MB-0133                   Not                Not  \n",
       "MB-0135                   Not                Not  \n",
       "MB-0136                   Not                Not  \n",
       "MB-0138                   Not                Not  \n",
       "MB-0139                   Not                Not  \n",
       "MB-0140                   Not                Not  \n",
       "MB-0142                   Not                Not  \n",
       "MB-0143                   Not                Not  \n",
       "...                       ...                ...  \n",
       "MB-7264                   Not                Not  \n",
       "MB-7266                   Not                Not  \n",
       "MB-7267                   Not                Not  \n",
       "MB-7268                   Not                Not  \n",
       "MB-7269                   Not                Not  \n",
       "MB-7270                   Not                Not  \n",
       "MB-7273                   Not                Not  \n",
       "MB-7275                   Not                Not  \n",
       "MB-7276                   Not                Not  \n",
       "MB-7277                   Not                Not  \n",
       "MB-7278                   Not                Not  \n",
       "MB-7279                   Not                Not  \n",
       "MB-7280                   Not                Not  \n",
       "MB-7281                   Not                Not  \n",
       "MB-7283                   Not                  6  \n",
       "MB-7284                   Not                  6  \n",
       "MB-7285                   Not                  6  \n",
       "MB-7286                   Not                Not  \n",
       "MB-7287                   Not                Not  \n",
       "MB-7288                   Not                Not  \n",
       "MB-7289                   Not                Not  \n",
       "MB-7291                   Not                Not  \n",
       "MB-7292                   Not                  6  \n",
       "MB-7293                   Not                Not  \n",
       "MB-7294                   Not                Not  \n",
       "MB-7295                   Not                Not  \n",
       "MB-7296                   Not                Not  \n",
       "MB-7297                   Not                Not  \n",
       "MB-7298                   Not                Not  \n",
       "MB-7299                   Not                Not  \n",
       "\n",
       "[1992 rows x 2118 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_df = pd.concat([pca_per_pathway, new_clinical.reindex(pca_per_pathway.index)], axis=1)\n",
    "for i in full_df['Integrative Cluster'].unique():\n",
    "    if type(i) == str:\n",
    "        full_df[f'Intclust {i} or not'] = full_df.apply(lambda row: i if (row['Integrative Cluster'] == i) else 'Not', axis = 1)\n",
    "        \n",
    "full_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Interleukin-6 signaling</th>\n",
       "      <th>Apoptosis</th>\n",
       "      <th>Hemostasis</th>\n",
       "      <th>Intrinsic Pathway for Apoptosis</th>\n",
       "      <th>PKB-mediated events</th>\n",
       "      <th>PI3K Cascade</th>\n",
       "      <th>MAPK3 (ERK1) activation</th>\n",
       "      <th>Translesion synthesis by REV1</th>\n",
       "      <th>Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template</th>\n",
       "      <th>Recognition of DNA damage by PCNA-containing replication complex</th>\n",
       "      <th>...</th>\n",
       "      <th>Antigen processing: Ubiquitination &amp; Proteasome degradation</th>\n",
       "      <th>Class I MHC mediated antigen processing &amp; presentation</th>\n",
       "      <th>Antigen Presentation: Folding, assembly and peptide loading of class I MHC</th>\n",
       "      <th>Kinesins</th>\n",
       "      <th>Factors involved in megakaryocyte development and platelet production</th>\n",
       "      <th>Antigen activates B Cell Receptor (BCR) leading to generation of second messengers</th>\n",
       "      <th>Signaling by the B Cell Receptor (BCR)</th>\n",
       "      <th>Ion channel transport</th>\n",
       "      <th>Activation of GABAB receptors</th>\n",
       "      <th>Inhibition  of voltage gated Ca2+ channels via Gbeta/gamma subunits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>-0.464915</td>\n",
       "      <td>1.150390</td>\n",
       "      <td>7.995550</td>\n",
       "      <td>-2.133505</td>\n",
       "      <td>-1.005971</td>\n",
       "      <td>-1.934175</td>\n",
       "      <td>0.682674</td>\n",
       "      <td>-1.740452</td>\n",
       "      <td>-1.758971</td>\n",
       "      <td>2.306896</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.181212</td>\n",
       "      <td>10.501022</td>\n",
       "      <td>-10.621735</td>\n",
       "      <td>-2.783924</td>\n",
       "      <td>-1.614593</td>\n",
       "      <td>-3.780976</td>\n",
       "      <td>-5.213696</td>\n",
       "      <td>2.340642</td>\n",
       "      <td>-2.081063</td>\n",
       "      <td>-1.718509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>1.135676</td>\n",
       "      <td>-5.398171</td>\n",
       "      <td>-1.468534</td>\n",
       "      <td>-2.783638</td>\n",
       "      <td>-0.042169</td>\n",
       "      <td>-1.152715</td>\n",
       "      <td>-0.440677</td>\n",
       "      <td>-0.738142</td>\n",
       "      <td>-0.950279</td>\n",
       "      <td>0.857385</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.103682</td>\n",
       "      <td>-4.968357</td>\n",
       "      <td>3.579570</td>\n",
       "      <td>2.017033</td>\n",
       "      <td>3.156719</td>\n",
       "      <td>3.019890</td>\n",
       "      <td>4.179122</td>\n",
       "      <td>2.678819</td>\n",
       "      <td>0.760122</td>\n",
       "      <td>0.911403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0010</th>\n",
       "      <td>-0.016407</td>\n",
       "      <td>-2.252856</td>\n",
       "      <td>1.688631</td>\n",
       "      <td>-1.840132</td>\n",
       "      <td>0.439943</td>\n",
       "      <td>0.907014</td>\n",
       "      <td>-0.946115</td>\n",
       "      <td>-1.195843</td>\n",
       "      <td>-2.019880</td>\n",
       "      <td>2.452047</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.429469</td>\n",
       "      <td>2.602108</td>\n",
       "      <td>-4.274843</td>\n",
       "      <td>5.476014</td>\n",
       "      <td>5.792390</td>\n",
       "      <td>-5.347735</td>\n",
       "      <td>-3.749546</td>\n",
       "      <td>-0.974028</td>\n",
       "      <td>-2.141831</td>\n",
       "      <td>-1.831642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0035</th>\n",
       "      <td>-0.045984</td>\n",
       "      <td>3.050125</td>\n",
       "      <td>14.221321</td>\n",
       "      <td>-1.010664</td>\n",
       "      <td>-2.460802</td>\n",
       "      <td>0.033379</td>\n",
       "      <td>-0.040401</td>\n",
       "      <td>1.589719</td>\n",
       "      <td>0.338310</td>\n",
       "      <td>-0.025657</td>\n",
       "      <td>...</td>\n",
       "      <td>2.307189</td>\n",
       "      <td>11.536765</td>\n",
       "      <td>-9.387837</td>\n",
       "      <td>-1.037284</td>\n",
       "      <td>-1.172208</td>\n",
       "      <td>-6.415124</td>\n",
       "      <td>-7.129510</td>\n",
       "      <td>0.346868</td>\n",
       "      <td>2.015445</td>\n",
       "      <td>2.369224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0036</th>\n",
       "      <td>-1.107911</td>\n",
       "      <td>1.445771</td>\n",
       "      <td>5.790792</td>\n",
       "      <td>1.275586</td>\n",
       "      <td>-1.040463</td>\n",
       "      <td>-2.897256</td>\n",
       "      <td>0.357232</td>\n",
       "      <td>0.276760</td>\n",
       "      <td>1.075780</td>\n",
       "      <td>-1.226257</td>\n",
       "      <td>...</td>\n",
       "      <td>0.652174</td>\n",
       "      <td>2.737150</td>\n",
       "      <td>-1.630176</td>\n",
       "      <td>-1.113311</td>\n",
       "      <td>-1.828239</td>\n",
       "      <td>-4.086297</td>\n",
       "      <td>-3.703407</td>\n",
       "      <td>3.655877</td>\n",
       "      <td>-0.986929</td>\n",
       "      <td>-1.141993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0050</th>\n",
       "      <td>1.003819</td>\n",
       "      <td>1.441920</td>\n",
       "      <td>-1.964812</td>\n",
       "      <td>-0.102851</td>\n",
       "      <td>-0.049525</td>\n",
       "      <td>-3.192611</td>\n",
       "      <td>-1.181677</td>\n",
       "      <td>0.429309</td>\n",
       "      <td>1.425590</td>\n",
       "      <td>-1.375613</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.234766</td>\n",
       "      <td>5.523601</td>\n",
       "      <td>-4.244558</td>\n",
       "      <td>-5.265519</td>\n",
       "      <td>-5.665442</td>\n",
       "      <td>2.556065</td>\n",
       "      <td>-0.543039</td>\n",
       "      <td>-0.737432</td>\n",
       "      <td>1.518420</td>\n",
       "      <td>1.390619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0059</th>\n",
       "      <td>-0.473150</td>\n",
       "      <td>0.186910</td>\n",
       "      <td>6.189742</td>\n",
       "      <td>-1.018816</td>\n",
       "      <td>-1.353297</td>\n",
       "      <td>-2.260648</td>\n",
       "      <td>0.119735</td>\n",
       "      <td>0.142322</td>\n",
       "      <td>-0.111307</td>\n",
       "      <td>-0.685721</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.292349</td>\n",
       "      <td>7.735078</td>\n",
       "      <td>-7.386242</td>\n",
       "      <td>-0.915784</td>\n",
       "      <td>-0.762069</td>\n",
       "      <td>-6.173665</td>\n",
       "      <td>-6.540617</td>\n",
       "      <td>0.229692</td>\n",
       "      <td>0.725576</td>\n",
       "      <td>1.175208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0060</th>\n",
       "      <td>0.681486</td>\n",
       "      <td>-12.524534</td>\n",
       "      <td>-0.370151</td>\n",
       "      <td>-2.756785</td>\n",
       "      <td>0.077299</td>\n",
       "      <td>-0.448584</td>\n",
       "      <td>-1.396643</td>\n",
       "      <td>-3.632946</td>\n",
       "      <td>-5.847386</td>\n",
       "      <td>6.291344</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.273609</td>\n",
       "      <td>-19.867712</td>\n",
       "      <td>14.711741</td>\n",
       "      <td>5.430391</td>\n",
       "      <td>6.268124</td>\n",
       "      <td>-1.857444</td>\n",
       "      <td>7.910823</td>\n",
       "      <td>-2.220703</td>\n",
       "      <td>-0.937799</td>\n",
       "      <td>0.211784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0066</th>\n",
       "      <td>0.539735</td>\n",
       "      <td>-6.459812</td>\n",
       "      <td>-0.231027</td>\n",
       "      <td>-1.086133</td>\n",
       "      <td>-0.017782</td>\n",
       "      <td>-0.218056</td>\n",
       "      <td>-0.731439</td>\n",
       "      <td>-0.210748</td>\n",
       "      <td>-0.525239</td>\n",
       "      <td>1.251225</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.044344</td>\n",
       "      <td>-10.432402</td>\n",
       "      <td>7.673403</td>\n",
       "      <td>-1.260229</td>\n",
       "      <td>-0.731123</td>\n",
       "      <td>0.353554</td>\n",
       "      <td>4.875847</td>\n",
       "      <td>1.274467</td>\n",
       "      <td>-0.689864</td>\n",
       "      <td>-0.615685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0101</th>\n",
       "      <td>1.391563</td>\n",
       "      <td>-2.378717</td>\n",
       "      <td>-4.880185</td>\n",
       "      <td>1.025275</td>\n",
       "      <td>-0.772772</td>\n",
       "      <td>-0.612536</td>\n",
       "      <td>-0.850303</td>\n",
       "      <td>1.193591</td>\n",
       "      <td>2.045964</td>\n",
       "      <td>-1.989934</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.349735</td>\n",
       "      <td>-5.605123</td>\n",
       "      <td>3.450237</td>\n",
       "      <td>-2.927546</td>\n",
       "      <td>-1.869052</td>\n",
       "      <td>5.157788</td>\n",
       "      <td>5.911775</td>\n",
       "      <td>0.584062</td>\n",
       "      <td>-1.963235</td>\n",
       "      <td>-2.617657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0102</th>\n",
       "      <td>0.975992</td>\n",
       "      <td>-4.692292</td>\n",
       "      <td>-2.159457</td>\n",
       "      <td>-1.916437</td>\n",
       "      <td>-0.424554</td>\n",
       "      <td>0.999638</td>\n",
       "      <td>-1.688466</td>\n",
       "      <td>0.167435</td>\n",
       "      <td>-1.977181</td>\n",
       "      <td>1.916599</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.961124</td>\n",
       "      <td>-7.064365</td>\n",
       "      <td>3.386175</td>\n",
       "      <td>1.580122</td>\n",
       "      <td>1.705319</td>\n",
       "      <td>2.999873</td>\n",
       "      <td>5.668929</td>\n",
       "      <td>-2.046671</td>\n",
       "      <td>-3.694572</td>\n",
       "      <td>-3.004808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0106</th>\n",
       "      <td>-1.716131</td>\n",
       "      <td>3.055953</td>\n",
       "      <td>2.940126</td>\n",
       "      <td>0.884104</td>\n",
       "      <td>-0.823157</td>\n",
       "      <td>-0.688135</td>\n",
       "      <td>1.656749</td>\n",
       "      <td>-0.002856</td>\n",
       "      <td>2.292241</td>\n",
       "      <td>-1.652733</td>\n",
       "      <td>...</td>\n",
       "      <td>2.509509</td>\n",
       "      <td>6.347529</td>\n",
       "      <td>-3.986429</td>\n",
       "      <td>-5.207336</td>\n",
       "      <td>-5.675126</td>\n",
       "      <td>-3.342922</td>\n",
       "      <td>-4.616443</td>\n",
       "      <td>1.865447</td>\n",
       "      <td>1.400919</td>\n",
       "      <td>1.725544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0112</th>\n",
       "      <td>0.721624</td>\n",
       "      <td>-1.939072</td>\n",
       "      <td>-0.898322</td>\n",
       "      <td>0.265369</td>\n",
       "      <td>-0.368646</td>\n",
       "      <td>-1.915540</td>\n",
       "      <td>-1.256289</td>\n",
       "      <td>1.112242</td>\n",
       "      <td>1.566163</td>\n",
       "      <td>-1.291966</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.463129</td>\n",
       "      <td>-2.173827</td>\n",
       "      <td>0.766776</td>\n",
       "      <td>-3.123365</td>\n",
       "      <td>-3.398603</td>\n",
       "      <td>0.281370</td>\n",
       "      <td>1.766547</td>\n",
       "      <td>-1.689046</td>\n",
       "      <td>-2.135168</td>\n",
       "      <td>-1.900310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0115</th>\n",
       "      <td>2.891250</td>\n",
       "      <td>-7.520513</td>\n",
       "      <td>-4.479024</td>\n",
       "      <td>-4.139860</td>\n",
       "      <td>-0.845741</td>\n",
       "      <td>-0.875985</td>\n",
       "      <td>-2.905533</td>\n",
       "      <td>-3.344632</td>\n",
       "      <td>-3.650589</td>\n",
       "      <td>3.451235</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.246096</td>\n",
       "      <td>-12.025943</td>\n",
       "      <td>9.564923</td>\n",
       "      <td>6.066910</td>\n",
       "      <td>6.558084</td>\n",
       "      <td>3.361716</td>\n",
       "      <td>6.288507</td>\n",
       "      <td>-8.519237</td>\n",
       "      <td>-0.397631</td>\n",
       "      <td>1.017734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0117</th>\n",
       "      <td>-1.218244</td>\n",
       "      <td>-0.139969</td>\n",
       "      <td>-2.140516</td>\n",
       "      <td>0.796431</td>\n",
       "      <td>1.913941</td>\n",
       "      <td>0.425881</td>\n",
       "      <td>0.909888</td>\n",
       "      <td>0.240269</td>\n",
       "      <td>1.165142</td>\n",
       "      <td>-1.165904</td>\n",
       "      <td>...</td>\n",
       "      <td>0.818814</td>\n",
       "      <td>0.280332</td>\n",
       "      <td>0.812773</td>\n",
       "      <td>-2.006572</td>\n",
       "      <td>-2.226116</td>\n",
       "      <td>3.127701</td>\n",
       "      <td>2.317191</td>\n",
       "      <td>0.827094</td>\n",
       "      <td>1.706968</td>\n",
       "      <td>1.396368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0119</th>\n",
       "      <td>0.324922</td>\n",
       "      <td>-1.494166</td>\n",
       "      <td>2.681937</td>\n",
       "      <td>-0.732578</td>\n",
       "      <td>-1.160084</td>\n",
       "      <td>-0.281727</td>\n",
       "      <td>-0.094547</td>\n",
       "      <td>0.530474</td>\n",
       "      <td>0.249794</td>\n",
       "      <td>-0.944412</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.911408</td>\n",
       "      <td>6.341346</td>\n",
       "      <td>-6.942924</td>\n",
       "      <td>-0.424924</td>\n",
       "      <td>0.076581</td>\n",
       "      <td>-0.182723</td>\n",
       "      <td>-1.025801</td>\n",
       "      <td>0.215425</td>\n",
       "      <td>-2.736801</td>\n",
       "      <td>-2.474468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0120</th>\n",
       "      <td>-1.282450</td>\n",
       "      <td>-1.312260</td>\n",
       "      <td>-5.283445</td>\n",
       "      <td>-1.769936</td>\n",
       "      <td>-0.042372</td>\n",
       "      <td>1.623986</td>\n",
       "      <td>0.562100</td>\n",
       "      <td>-0.550128</td>\n",
       "      <td>-1.018389</td>\n",
       "      <td>0.599823</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.535755</td>\n",
       "      <td>-3.482248</td>\n",
       "      <td>1.582937</td>\n",
       "      <td>0.281499</td>\n",
       "      <td>1.953857</td>\n",
       "      <td>5.230402</td>\n",
       "      <td>3.683432</td>\n",
       "      <td>0.937613</td>\n",
       "      <td>-2.077385</td>\n",
       "      <td>-1.412313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0121</th>\n",
       "      <td>0.066524</td>\n",
       "      <td>-3.182866</td>\n",
       "      <td>0.578324</td>\n",
       "      <td>-1.262635</td>\n",
       "      <td>-0.358424</td>\n",
       "      <td>-0.596381</td>\n",
       "      <td>-0.376231</td>\n",
       "      <td>-1.426032</td>\n",
       "      <td>0.031787</td>\n",
       "      <td>0.056474</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.407047</td>\n",
       "      <td>0.354208</td>\n",
       "      <td>-3.060088</td>\n",
       "      <td>-3.040142</td>\n",
       "      <td>-3.220359</td>\n",
       "      <td>-2.223592</td>\n",
       "      <td>-0.487749</td>\n",
       "      <td>0.361180</td>\n",
       "      <td>-2.343898</td>\n",
       "      <td>-2.393808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0122</th>\n",
       "      <td>0.241645</td>\n",
       "      <td>-0.896690</td>\n",
       "      <td>4.363559</td>\n",
       "      <td>-0.332297</td>\n",
       "      <td>-0.933940</td>\n",
       "      <td>-2.378229</td>\n",
       "      <td>0.431727</td>\n",
       "      <td>-0.993327</td>\n",
       "      <td>0.664990</td>\n",
       "      <td>-0.454606</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.043551</td>\n",
       "      <td>7.237955</td>\n",
       "      <td>-7.595239</td>\n",
       "      <td>-5.352230</td>\n",
       "      <td>-5.548123</td>\n",
       "      <td>-2.932324</td>\n",
       "      <td>-2.600730</td>\n",
       "      <td>1.526380</td>\n",
       "      <td>-0.434253</td>\n",
       "      <td>-0.132890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0123</th>\n",
       "      <td>0.632205</td>\n",
       "      <td>-4.939672</td>\n",
       "      <td>1.387583</td>\n",
       "      <td>-2.019270</td>\n",
       "      <td>1.906553</td>\n",
       "      <td>1.130355</td>\n",
       "      <td>-0.342736</td>\n",
       "      <td>-4.240944</td>\n",
       "      <td>-4.235351</td>\n",
       "      <td>4.132531</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.242506</td>\n",
       "      <td>4.165809</td>\n",
       "      <td>-7.355088</td>\n",
       "      <td>6.224326</td>\n",
       "      <td>6.990339</td>\n",
       "      <td>-1.275137</td>\n",
       "      <td>-0.749794</td>\n",
       "      <td>1.262404</td>\n",
       "      <td>-3.915522</td>\n",
       "      <td>-3.345704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0126</th>\n",
       "      <td>0.430240</td>\n",
       "      <td>0.585299</td>\n",
       "      <td>0.323267</td>\n",
       "      <td>-1.390457</td>\n",
       "      <td>0.169088</td>\n",
       "      <td>-4.192595</td>\n",
       "      <td>-1.174283</td>\n",
       "      <td>1.428986</td>\n",
       "      <td>1.529271</td>\n",
       "      <td>-2.216522</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.852170</td>\n",
       "      <td>3.772588</td>\n",
       "      <td>-3.266582</td>\n",
       "      <td>-3.031947</td>\n",
       "      <td>-3.440848</td>\n",
       "      <td>-3.510166</td>\n",
       "      <td>-4.230219</td>\n",
       "      <td>0.113761</td>\n",
       "      <td>-0.226449</td>\n",
       "      <td>0.015675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0131</th>\n",
       "      <td>0.059170</td>\n",
       "      <td>0.794649</td>\n",
       "      <td>0.309136</td>\n",
       "      <td>0.093709</td>\n",
       "      <td>-0.878579</td>\n",
       "      <td>-0.526578</td>\n",
       "      <td>-0.248608</td>\n",
       "      <td>0.558131</td>\n",
       "      <td>0.547173</td>\n",
       "      <td>-0.530690</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.802845</td>\n",
       "      <td>2.986812</td>\n",
       "      <td>-3.750671</td>\n",
       "      <td>-1.548971</td>\n",
       "      <td>-1.714889</td>\n",
       "      <td>-5.144326</td>\n",
       "      <td>-4.683748</td>\n",
       "      <td>1.047786</td>\n",
       "      <td>0.745869</td>\n",
       "      <td>-0.520462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0133</th>\n",
       "      <td>-0.570782</td>\n",
       "      <td>4.560938</td>\n",
       "      <td>3.690203</td>\n",
       "      <td>2.302302</td>\n",
       "      <td>0.705017</td>\n",
       "      <td>-0.455561</td>\n",
       "      <td>0.440143</td>\n",
       "      <td>1.551490</td>\n",
       "      <td>2.748361</td>\n",
       "      <td>-2.443490</td>\n",
       "      <td>...</td>\n",
       "      <td>2.981783</td>\n",
       "      <td>10.042189</td>\n",
       "      <td>-6.609117</td>\n",
       "      <td>-6.219803</td>\n",
       "      <td>-6.761085</td>\n",
       "      <td>-2.707678</td>\n",
       "      <td>-5.525304</td>\n",
       "      <td>0.238325</td>\n",
       "      <td>3.263350</td>\n",
       "      <td>2.628679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0135</th>\n",
       "      <td>-0.117803</td>\n",
       "      <td>-0.869882</td>\n",
       "      <td>0.971675</td>\n",
       "      <td>-1.517378</td>\n",
       "      <td>-1.827313</td>\n",
       "      <td>-2.082254</td>\n",
       "      <td>-0.976155</td>\n",
       "      <td>-1.920829</td>\n",
       "      <td>-1.999828</td>\n",
       "      <td>2.118295</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.796854</td>\n",
       "      <td>5.749571</td>\n",
       "      <td>-4.539532</td>\n",
       "      <td>3.203228</td>\n",
       "      <td>2.322894</td>\n",
       "      <td>0.924889</td>\n",
       "      <td>-0.893908</td>\n",
       "      <td>2.643863</td>\n",
       "      <td>-3.307961</td>\n",
       "      <td>-2.965665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0136</th>\n",
       "      <td>0.874496</td>\n",
       "      <td>-3.379140</td>\n",
       "      <td>-9.386417</td>\n",
       "      <td>-0.044186</td>\n",
       "      <td>0.481281</td>\n",
       "      <td>-0.811673</td>\n",
       "      <td>-0.483516</td>\n",
       "      <td>-0.405053</td>\n",
       "      <td>0.319846</td>\n",
       "      <td>-0.551310</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.329268</td>\n",
       "      <td>-6.115133</td>\n",
       "      <td>3.480316</td>\n",
       "      <td>-1.196467</td>\n",
       "      <td>-2.514433</td>\n",
       "      <td>2.689187</td>\n",
       "      <td>4.386436</td>\n",
       "      <td>-0.859905</td>\n",
       "      <td>-1.844360</td>\n",
       "      <td>-2.120232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0138</th>\n",
       "      <td>1.158647</td>\n",
       "      <td>4.847099</td>\n",
       "      <td>0.673716</td>\n",
       "      <td>1.441657</td>\n",
       "      <td>-0.541410</td>\n",
       "      <td>-0.216848</td>\n",
       "      <td>-0.574181</td>\n",
       "      <td>1.703794</td>\n",
       "      <td>2.426379</td>\n",
       "      <td>-2.162349</td>\n",
       "      <td>...</td>\n",
       "      <td>2.631016</td>\n",
       "      <td>6.741145</td>\n",
       "      <td>-4.268781</td>\n",
       "      <td>-2.985508</td>\n",
       "      <td>-3.976150</td>\n",
       "      <td>0.794619</td>\n",
       "      <td>-2.320587</td>\n",
       "      <td>2.239991</td>\n",
       "      <td>1.720395</td>\n",
       "      <td>1.111475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0139</th>\n",
       "      <td>2.100204</td>\n",
       "      <td>3.857392</td>\n",
       "      <td>-2.110865</td>\n",
       "      <td>1.680220</td>\n",
       "      <td>0.543285</td>\n",
       "      <td>-2.839074</td>\n",
       "      <td>-1.455348</td>\n",
       "      <td>2.729110</td>\n",
       "      <td>4.206067</td>\n",
       "      <td>-4.028101</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.110126</td>\n",
       "      <td>7.031245</td>\n",
       "      <td>-3.852609</td>\n",
       "      <td>-6.474811</td>\n",
       "      <td>-7.965247</td>\n",
       "      <td>-0.300409</td>\n",
       "      <td>-4.190813</td>\n",
       "      <td>0.154110</td>\n",
       "      <td>2.372650</td>\n",
       "      <td>1.756206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0140</th>\n",
       "      <td>-0.166482</td>\n",
       "      <td>1.533517</td>\n",
       "      <td>2.621224</td>\n",
       "      <td>-0.157213</td>\n",
       "      <td>-0.250768</td>\n",
       "      <td>-0.001997</td>\n",
       "      <td>-0.033955</td>\n",
       "      <td>-1.053808</td>\n",
       "      <td>-1.000072</td>\n",
       "      <td>1.821616</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.463628</td>\n",
       "      <td>9.266462</td>\n",
       "      <td>-8.249615</td>\n",
       "      <td>0.315797</td>\n",
       "      <td>-0.339202</td>\n",
       "      <td>-0.292043</td>\n",
       "      <td>-3.042215</td>\n",
       "      <td>3.212023</td>\n",
       "      <td>-4.014152</td>\n",
       "      <td>-4.762921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0142</th>\n",
       "      <td>-0.881733</td>\n",
       "      <td>2.119339</td>\n",
       "      <td>11.769112</td>\n",
       "      <td>1.807500</td>\n",
       "      <td>-1.036924</td>\n",
       "      <td>-1.392334</td>\n",
       "      <td>0.138771</td>\n",
       "      <td>3.207029</td>\n",
       "      <td>3.105644</td>\n",
       "      <td>-3.524379</td>\n",
       "      <td>...</td>\n",
       "      <td>1.139717</td>\n",
       "      <td>11.023046</td>\n",
       "      <td>-8.919237</td>\n",
       "      <td>-5.648750</td>\n",
       "      <td>-6.126320</td>\n",
       "      <td>-3.622275</td>\n",
       "      <td>-4.970809</td>\n",
       "      <td>4.395123</td>\n",
       "      <td>-0.061439</td>\n",
       "      <td>-0.582873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0143</th>\n",
       "      <td>0.237474</td>\n",
       "      <td>1.306009</td>\n",
       "      <td>4.406096</td>\n",
       "      <td>0.412958</td>\n",
       "      <td>-2.160744</td>\n",
       "      <td>-1.154757</td>\n",
       "      <td>-0.658322</td>\n",
       "      <td>1.296105</td>\n",
       "      <td>0.730830</td>\n",
       "      <td>-0.765246</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.103454</td>\n",
       "      <td>6.532816</td>\n",
       "      <td>-6.297841</td>\n",
       "      <td>-2.127102</td>\n",
       "      <td>-2.455823</td>\n",
       "      <td>-2.132147</td>\n",
       "      <td>-2.518095</td>\n",
       "      <td>2.648164</td>\n",
       "      <td>-0.377887</td>\n",
       "      <td>-0.254645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7264</th>\n",
       "      <td>0.052616</td>\n",
       "      <td>0.475062</td>\n",
       "      <td>-8.357461</td>\n",
       "      <td>2.719353</td>\n",
       "      <td>1.340783</td>\n",
       "      <td>1.720229</td>\n",
       "      <td>0.486650</td>\n",
       "      <td>1.187940</td>\n",
       "      <td>1.369706</td>\n",
       "      <td>-0.953181</td>\n",
       "      <td>...</td>\n",
       "      <td>2.978944</td>\n",
       "      <td>-8.202431</td>\n",
       "      <td>7.960864</td>\n",
       "      <td>-0.747237</td>\n",
       "      <td>-0.958069</td>\n",
       "      <td>8.858024</td>\n",
       "      <td>8.301377</td>\n",
       "      <td>-1.376696</td>\n",
       "      <td>-1.475901</td>\n",
       "      <td>-1.209309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7266</th>\n",
       "      <td>-2.427949</td>\n",
       "      <td>2.504505</td>\n",
       "      <td>12.534635</td>\n",
       "      <td>0.578593</td>\n",
       "      <td>-1.317527</td>\n",
       "      <td>-0.400218</td>\n",
       "      <td>0.768747</td>\n",
       "      <td>-1.814783</td>\n",
       "      <td>-0.804663</td>\n",
       "      <td>1.092236</td>\n",
       "      <td>...</td>\n",
       "      <td>0.236276</td>\n",
       "      <td>2.667299</td>\n",
       "      <td>1.511926</td>\n",
       "      <td>-1.543994</td>\n",
       "      <td>-1.962177</td>\n",
       "      <td>-6.226096</td>\n",
       "      <td>-5.982386</td>\n",
       "      <td>3.705963</td>\n",
       "      <td>-0.049891</td>\n",
       "      <td>0.316525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7267</th>\n",
       "      <td>1.027192</td>\n",
       "      <td>-3.108903</td>\n",
       "      <td>-3.791368</td>\n",
       "      <td>-4.825164</td>\n",
       "      <td>-0.267898</td>\n",
       "      <td>0.911133</td>\n",
       "      <td>-1.830152</td>\n",
       "      <td>-2.247315</td>\n",
       "      <td>-4.519671</td>\n",
       "      <td>5.070132</td>\n",
       "      <td>...</td>\n",
       "      <td>1.234845</td>\n",
       "      <td>-0.908210</td>\n",
       "      <td>-1.150582</td>\n",
       "      <td>8.357569</td>\n",
       "      <td>9.896810</td>\n",
       "      <td>6.050182</td>\n",
       "      <td>5.022543</td>\n",
       "      <td>-4.776596</td>\n",
       "      <td>-4.174924</td>\n",
       "      <td>-3.890250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7268</th>\n",
       "      <td>0.899954</td>\n",
       "      <td>0.782333</td>\n",
       "      <td>6.789504</td>\n",
       "      <td>-0.559499</td>\n",
       "      <td>-1.123089</td>\n",
       "      <td>-0.000125</td>\n",
       "      <td>-1.000879</td>\n",
       "      <td>-0.244545</td>\n",
       "      <td>-1.339817</td>\n",
       "      <td>0.690283</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.645635</td>\n",
       "      <td>8.161488</td>\n",
       "      <td>-7.698986</td>\n",
       "      <td>-0.410765</td>\n",
       "      <td>-0.940649</td>\n",
       "      <td>-3.436144</td>\n",
       "      <td>-4.130540</td>\n",
       "      <td>0.890925</td>\n",
       "      <td>0.940230</td>\n",
       "      <td>0.841507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7269</th>\n",
       "      <td>-1.542249</td>\n",
       "      <td>-6.540713</td>\n",
       "      <td>-6.588955</td>\n",
       "      <td>-0.905963</td>\n",
       "      <td>-0.975963</td>\n",
       "      <td>1.026272</td>\n",
       "      <td>0.386235</td>\n",
       "      <td>-1.722726</td>\n",
       "      <td>-2.492191</td>\n",
       "      <td>2.197838</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.246201</td>\n",
       "      <td>-20.507946</td>\n",
       "      <td>18.701986</td>\n",
       "      <td>5.781100</td>\n",
       "      <td>6.876094</td>\n",
       "      <td>5.948816</td>\n",
       "      <td>10.695359</td>\n",
       "      <td>-5.042941</td>\n",
       "      <td>-0.128779</td>\n",
       "      <td>0.578354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7270</th>\n",
       "      <td>-0.818144</td>\n",
       "      <td>-2.313261</td>\n",
       "      <td>0.407206</td>\n",
       "      <td>-1.231119</td>\n",
       "      <td>1.065474</td>\n",
       "      <td>-0.039110</td>\n",
       "      <td>-0.105982</td>\n",
       "      <td>-5.473630</td>\n",
       "      <td>-5.901622</td>\n",
       "      <td>6.593430</td>\n",
       "      <td>...</td>\n",
       "      <td>0.104617</td>\n",
       "      <td>-3.871092</td>\n",
       "      <td>0.700887</td>\n",
       "      <td>6.023480</td>\n",
       "      <td>9.270829</td>\n",
       "      <td>1.519696</td>\n",
       "      <td>3.669500</td>\n",
       "      <td>-5.082238</td>\n",
       "      <td>-2.763787</td>\n",
       "      <td>-2.605367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7273</th>\n",
       "      <td>0.133795</td>\n",
       "      <td>-7.936353</td>\n",
       "      <td>-6.036484</td>\n",
       "      <td>-1.853518</td>\n",
       "      <td>4.553318</td>\n",
       "      <td>2.124215</td>\n",
       "      <td>-0.638000</td>\n",
       "      <td>-0.010831</td>\n",
       "      <td>-1.326545</td>\n",
       "      <td>0.197701</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.583868</td>\n",
       "      <td>-9.537011</td>\n",
       "      <td>4.357210</td>\n",
       "      <td>1.776304</td>\n",
       "      <td>3.194313</td>\n",
       "      <td>5.475915</td>\n",
       "      <td>8.468404</td>\n",
       "      <td>-3.994619</td>\n",
       "      <td>-1.469257</td>\n",
       "      <td>-2.158524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7275</th>\n",
       "      <td>0.119508</td>\n",
       "      <td>-3.830375</td>\n",
       "      <td>-7.073864</td>\n",
       "      <td>-3.164525</td>\n",
       "      <td>0.449460</td>\n",
       "      <td>-2.310980</td>\n",
       "      <td>-0.657651</td>\n",
       "      <td>0.304326</td>\n",
       "      <td>-0.530877</td>\n",
       "      <td>0.338423</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.027479</td>\n",
       "      <td>-1.047748</td>\n",
       "      <td>-1.119716</td>\n",
       "      <td>-1.453237</td>\n",
       "      <td>-1.435250</td>\n",
       "      <td>-0.190809</td>\n",
       "      <td>0.521049</td>\n",
       "      <td>-1.946833</td>\n",
       "      <td>-2.628785</td>\n",
       "      <td>-2.593172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7276</th>\n",
       "      <td>1.452774</td>\n",
       "      <td>-0.532131</td>\n",
       "      <td>0.833160</td>\n",
       "      <td>0.619618</td>\n",
       "      <td>0.006122</td>\n",
       "      <td>-0.870304</td>\n",
       "      <td>-0.551065</td>\n",
       "      <td>-0.493778</td>\n",
       "      <td>1.031235</td>\n",
       "      <td>-0.607397</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.924477</td>\n",
       "      <td>-2.676211</td>\n",
       "      <td>3.357074</td>\n",
       "      <td>-2.800376</td>\n",
       "      <td>-3.255994</td>\n",
       "      <td>0.752929</td>\n",
       "      <td>2.364048</td>\n",
       "      <td>0.047615</td>\n",
       "      <td>-1.582996</td>\n",
       "      <td>-1.464334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7277</th>\n",
       "      <td>-0.836269</td>\n",
       "      <td>-2.383064</td>\n",
       "      <td>3.939454</td>\n",
       "      <td>0.441016</td>\n",
       "      <td>-0.337889</td>\n",
       "      <td>-1.065002</td>\n",
       "      <td>0.148280</td>\n",
       "      <td>-0.753005</td>\n",
       "      <td>0.553203</td>\n",
       "      <td>0.260013</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.903476</td>\n",
       "      <td>0.343151</td>\n",
       "      <td>-1.629507</td>\n",
       "      <td>-3.523731</td>\n",
       "      <td>-3.699770</td>\n",
       "      <td>-2.449184</td>\n",
       "      <td>0.086364</td>\n",
       "      <td>0.993515</td>\n",
       "      <td>-1.396922</td>\n",
       "      <td>-1.670098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7278</th>\n",
       "      <td>-0.750752</td>\n",
       "      <td>5.208024</td>\n",
       "      <td>4.780788</td>\n",
       "      <td>1.853149</td>\n",
       "      <td>-0.159819</td>\n",
       "      <td>-1.297136</td>\n",
       "      <td>0.650400</td>\n",
       "      <td>0.951615</td>\n",
       "      <td>2.215585</td>\n",
       "      <td>-1.703582</td>\n",
       "      <td>...</td>\n",
       "      <td>2.780067</td>\n",
       "      <td>9.351828</td>\n",
       "      <td>-7.465249</td>\n",
       "      <td>-5.574749</td>\n",
       "      <td>-5.874127</td>\n",
       "      <td>-0.789296</td>\n",
       "      <td>-4.013051</td>\n",
       "      <td>-0.210591</td>\n",
       "      <td>-1.246493</td>\n",
       "      <td>-1.164350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7279</th>\n",
       "      <td>-0.591865</td>\n",
       "      <td>3.546974</td>\n",
       "      <td>6.984630</td>\n",
       "      <td>3.625156</td>\n",
       "      <td>-0.001635</td>\n",
       "      <td>3.217996</td>\n",
       "      <td>1.287618</td>\n",
       "      <td>2.715368</td>\n",
       "      <td>1.277015</td>\n",
       "      <td>-1.698399</td>\n",
       "      <td>...</td>\n",
       "      <td>7.453155</td>\n",
       "      <td>-6.069090</td>\n",
       "      <td>8.397773</td>\n",
       "      <td>4.981525</td>\n",
       "      <td>5.700316</td>\n",
       "      <td>-1.411486</td>\n",
       "      <td>-0.457442</td>\n",
       "      <td>3.062864</td>\n",
       "      <td>2.922412</td>\n",
       "      <td>2.362305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7280</th>\n",
       "      <td>3.168238</td>\n",
       "      <td>3.348863</td>\n",
       "      <td>2.440740</td>\n",
       "      <td>0.188479</td>\n",
       "      <td>0.550790</td>\n",
       "      <td>0.139212</td>\n",
       "      <td>-1.515452</td>\n",
       "      <td>-0.981956</td>\n",
       "      <td>-0.771772</td>\n",
       "      <td>0.736326</td>\n",
       "      <td>...</td>\n",
       "      <td>0.306340</td>\n",
       "      <td>9.265659</td>\n",
       "      <td>-6.396199</td>\n",
       "      <td>-1.326755</td>\n",
       "      <td>-0.831227</td>\n",
       "      <td>-0.360274</td>\n",
       "      <td>-4.104418</td>\n",
       "      <td>0.139250</td>\n",
       "      <td>-2.458190</td>\n",
       "      <td>-2.486319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7281</th>\n",
       "      <td>-0.708823</td>\n",
       "      <td>-1.609151</td>\n",
       "      <td>-5.090321</td>\n",
       "      <td>0.525543</td>\n",
       "      <td>0.303806</td>\n",
       "      <td>2.379776</td>\n",
       "      <td>0.551161</td>\n",
       "      <td>0.059091</td>\n",
       "      <td>0.492005</td>\n",
       "      <td>-0.799303</td>\n",
       "      <td>...</td>\n",
       "      <td>0.093180</td>\n",
       "      <td>-4.191171</td>\n",
       "      <td>2.704735</td>\n",
       "      <td>0.990196</td>\n",
       "      <td>2.182252</td>\n",
       "      <td>2.212883</td>\n",
       "      <td>3.244051</td>\n",
       "      <td>-3.805825</td>\n",
       "      <td>0.245758</td>\n",
       "      <td>0.122075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7283</th>\n",
       "      <td>-1.524783</td>\n",
       "      <td>6.702166</td>\n",
       "      <td>12.168263</td>\n",
       "      <td>-0.127327</td>\n",
       "      <td>-2.139917</td>\n",
       "      <td>0.550919</td>\n",
       "      <td>0.340746</td>\n",
       "      <td>-0.509159</td>\n",
       "      <td>-1.216963</td>\n",
       "      <td>0.841771</td>\n",
       "      <td>...</td>\n",
       "      <td>5.662923</td>\n",
       "      <td>17.127012</td>\n",
       "      <td>-12.113644</td>\n",
       "      <td>2.423513</td>\n",
       "      <td>2.340146</td>\n",
       "      <td>-3.977006</td>\n",
       "      <td>-8.119009</td>\n",
       "      <td>3.242792</td>\n",
       "      <td>2.700290</td>\n",
       "      <td>2.586261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7284</th>\n",
       "      <td>-1.585521</td>\n",
       "      <td>2.486830</td>\n",
       "      <td>3.896734</td>\n",
       "      <td>0.832220</td>\n",
       "      <td>-0.452137</td>\n",
       "      <td>0.954621</td>\n",
       "      <td>2.116733</td>\n",
       "      <td>-0.226250</td>\n",
       "      <td>-0.314260</td>\n",
       "      <td>-0.059567</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.075522</td>\n",
       "      <td>9.258899</td>\n",
       "      <td>-6.996106</td>\n",
       "      <td>-2.956348</td>\n",
       "      <td>-3.568803</td>\n",
       "      <td>-1.943100</td>\n",
       "      <td>-4.104070</td>\n",
       "      <td>2.396170</td>\n",
       "      <td>3.613272</td>\n",
       "      <td>3.582648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7285</th>\n",
       "      <td>2.112241</td>\n",
       "      <td>-4.308329</td>\n",
       "      <td>-10.057469</td>\n",
       "      <td>-1.500446</td>\n",
       "      <td>0.268187</td>\n",
       "      <td>0.055063</td>\n",
       "      <td>-2.512579</td>\n",
       "      <td>-0.288444</td>\n",
       "      <td>-0.051126</td>\n",
       "      <td>-0.545414</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.234862</td>\n",
       "      <td>-1.385978</td>\n",
       "      <td>-1.994269</td>\n",
       "      <td>-0.925810</td>\n",
       "      <td>-0.647075</td>\n",
       "      <td>2.705253</td>\n",
       "      <td>3.287334</td>\n",
       "      <td>-2.418825</td>\n",
       "      <td>-3.060629</td>\n",
       "      <td>-3.616641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7286</th>\n",
       "      <td>0.464739</td>\n",
       "      <td>0.329545</td>\n",
       "      <td>0.350963</td>\n",
       "      <td>-1.888931</td>\n",
       "      <td>-0.192087</td>\n",
       "      <td>-3.063396</td>\n",
       "      <td>-1.545657</td>\n",
       "      <td>0.978685</td>\n",
       "      <td>0.061323</td>\n",
       "      <td>-0.506717</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.372422</td>\n",
       "      <td>2.396867</td>\n",
       "      <td>-1.212736</td>\n",
       "      <td>0.224308</td>\n",
       "      <td>-0.593532</td>\n",
       "      <td>-2.600160</td>\n",
       "      <td>-4.352315</td>\n",
       "      <td>2.006475</td>\n",
       "      <td>1.551420</td>\n",
       "      <td>1.824794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7287</th>\n",
       "      <td>-0.461483</td>\n",
       "      <td>-6.788634</td>\n",
       "      <td>-4.532691</td>\n",
       "      <td>-0.648133</td>\n",
       "      <td>0.898894</td>\n",
       "      <td>-0.037595</td>\n",
       "      <td>-0.657459</td>\n",
       "      <td>-0.130143</td>\n",
       "      <td>-1.043855</td>\n",
       "      <td>0.465353</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.005773</td>\n",
       "      <td>-10.989277</td>\n",
       "      <td>6.587463</td>\n",
       "      <td>1.458673</td>\n",
       "      <td>2.337528</td>\n",
       "      <td>5.110461</td>\n",
       "      <td>8.699111</td>\n",
       "      <td>-1.883775</td>\n",
       "      <td>-2.270477</td>\n",
       "      <td>-2.428680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7288</th>\n",
       "      <td>0.084161</td>\n",
       "      <td>-5.730170</td>\n",
       "      <td>-0.889219</td>\n",
       "      <td>-3.048048</td>\n",
       "      <td>-0.746198</td>\n",
       "      <td>-0.468645</td>\n",
       "      <td>-1.043475</td>\n",
       "      <td>-0.499714</td>\n",
       "      <td>-0.199662</td>\n",
       "      <td>0.688354</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.528711</td>\n",
       "      <td>-4.563002</td>\n",
       "      <td>1.488169</td>\n",
       "      <td>-1.655888</td>\n",
       "      <td>-1.267448</td>\n",
       "      <td>2.466028</td>\n",
       "      <td>3.618482</td>\n",
       "      <td>-0.904307</td>\n",
       "      <td>-3.733653</td>\n",
       "      <td>-3.187384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7289</th>\n",
       "      <td>0.948163</td>\n",
       "      <td>-2.915446</td>\n",
       "      <td>-9.226002</td>\n",
       "      <td>-1.145563</td>\n",
       "      <td>-0.094301</td>\n",
       "      <td>-1.420274</td>\n",
       "      <td>-1.017675</td>\n",
       "      <td>-0.254609</td>\n",
       "      <td>1.379707</td>\n",
       "      <td>-1.184310</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.259995</td>\n",
       "      <td>-1.504258</td>\n",
       "      <td>-0.598862</td>\n",
       "      <td>-4.488339</td>\n",
       "      <td>-4.522991</td>\n",
       "      <td>5.110595</td>\n",
       "      <td>4.190233</td>\n",
       "      <td>-2.456881</td>\n",
       "      <td>-2.388495</td>\n",
       "      <td>-2.264113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7291</th>\n",
       "      <td>1.676027</td>\n",
       "      <td>-10.800482</td>\n",
       "      <td>-13.742885</td>\n",
       "      <td>-3.676589</td>\n",
       "      <td>0.619119</td>\n",
       "      <td>1.983350</td>\n",
       "      <td>-2.676246</td>\n",
       "      <td>-2.079944</td>\n",
       "      <td>-2.595874</td>\n",
       "      <td>1.998022</td>\n",
       "      <td>...</td>\n",
       "      <td>-10.244129</td>\n",
       "      <td>-13.150474</td>\n",
       "      <td>6.470609</td>\n",
       "      <td>2.560807</td>\n",
       "      <td>3.581800</td>\n",
       "      <td>5.677145</td>\n",
       "      <td>10.225146</td>\n",
       "      <td>-5.255381</td>\n",
       "      <td>-4.037444</td>\n",
       "      <td>-3.730545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7292</th>\n",
       "      <td>-0.991138</td>\n",
       "      <td>-5.250101</td>\n",
       "      <td>0.637368</td>\n",
       "      <td>-2.451837</td>\n",
       "      <td>-1.300904</td>\n",
       "      <td>-1.138077</td>\n",
       "      <td>-0.387298</td>\n",
       "      <td>-0.420057</td>\n",
       "      <td>-2.341863</td>\n",
       "      <td>1.763887</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.732811</td>\n",
       "      <td>-2.052564</td>\n",
       "      <td>0.100131</td>\n",
       "      <td>1.853204</td>\n",
       "      <td>2.265941</td>\n",
       "      <td>-3.248151</td>\n",
       "      <td>-0.804346</td>\n",
       "      <td>-0.104385</td>\n",
       "      <td>-4.864537</td>\n",
       "      <td>-4.864987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7293</th>\n",
       "      <td>0.295280</td>\n",
       "      <td>-5.007402</td>\n",
       "      <td>-6.453299</td>\n",
       "      <td>-0.666104</td>\n",
       "      <td>-0.802562</td>\n",
       "      <td>-0.854217</td>\n",
       "      <td>-0.086637</td>\n",
       "      <td>-1.197647</td>\n",
       "      <td>0.210241</td>\n",
       "      <td>0.307119</td>\n",
       "      <td>...</td>\n",
       "      <td>-8.022697</td>\n",
       "      <td>-5.319050</td>\n",
       "      <td>1.760966</td>\n",
       "      <td>-4.034656</td>\n",
       "      <td>-4.194694</td>\n",
       "      <td>0.838636</td>\n",
       "      <td>2.630240</td>\n",
       "      <td>-3.092360</td>\n",
       "      <td>-4.634184</td>\n",
       "      <td>-4.191624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7294</th>\n",
       "      <td>-0.241594</td>\n",
       "      <td>-6.497156</td>\n",
       "      <td>7.094790</td>\n",
       "      <td>-0.526408</td>\n",
       "      <td>-0.312066</td>\n",
       "      <td>-1.249597</td>\n",
       "      <td>-0.619259</td>\n",
       "      <td>-1.585000</td>\n",
       "      <td>-0.730437</td>\n",
       "      <td>0.616089</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.888719</td>\n",
       "      <td>-3.931945</td>\n",
       "      <td>0.534228</td>\n",
       "      <td>-0.663199</td>\n",
       "      <td>-0.068830</td>\n",
       "      <td>-2.938476</td>\n",
       "      <td>1.727928</td>\n",
       "      <td>2.497594</td>\n",
       "      <td>-3.496975</td>\n",
       "      <td>-3.691912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7295</th>\n",
       "      <td>-0.097627</td>\n",
       "      <td>-4.277540</td>\n",
       "      <td>-5.748505</td>\n",
       "      <td>-3.180437</td>\n",
       "      <td>-0.865192</td>\n",
       "      <td>-0.979926</td>\n",
       "      <td>-1.582625</td>\n",
       "      <td>-0.735149</td>\n",
       "      <td>-0.468519</td>\n",
       "      <td>0.321434</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.578778</td>\n",
       "      <td>-0.410079</td>\n",
       "      <td>-2.211857</td>\n",
       "      <td>-2.745921</td>\n",
       "      <td>-2.813467</td>\n",
       "      <td>-1.638444</td>\n",
       "      <td>-0.490903</td>\n",
       "      <td>0.070499</td>\n",
       "      <td>-3.442825</td>\n",
       "      <td>-3.079476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7296</th>\n",
       "      <td>-0.616957</td>\n",
       "      <td>-4.852698</td>\n",
       "      <td>1.617394</td>\n",
       "      <td>-3.523299</td>\n",
       "      <td>-1.540831</td>\n",
       "      <td>1.360917</td>\n",
       "      <td>-0.889991</td>\n",
       "      <td>-1.775554</td>\n",
       "      <td>-2.728289</td>\n",
       "      <td>1.738995</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.110505</td>\n",
       "      <td>0.600571</td>\n",
       "      <td>-2.990283</td>\n",
       "      <td>5.587256</td>\n",
       "      <td>5.737139</td>\n",
       "      <td>-2.236260</td>\n",
       "      <td>-1.231161</td>\n",
       "      <td>0.606136</td>\n",
       "      <td>-0.894533</td>\n",
       "      <td>-0.717148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7297</th>\n",
       "      <td>-0.546331</td>\n",
       "      <td>-8.314218</td>\n",
       "      <td>-2.774328</td>\n",
       "      <td>-2.203446</td>\n",
       "      <td>-0.507499</td>\n",
       "      <td>-0.153017</td>\n",
       "      <td>-0.957462</td>\n",
       "      <td>-0.821364</td>\n",
       "      <td>-1.953621</td>\n",
       "      <td>2.779270</td>\n",
       "      <td>...</td>\n",
       "      <td>-10.178550</td>\n",
       "      <td>-6.728893</td>\n",
       "      <td>2.466814</td>\n",
       "      <td>2.097607</td>\n",
       "      <td>3.400695</td>\n",
       "      <td>6.833561</td>\n",
       "      <td>8.375035</td>\n",
       "      <td>-1.153599</td>\n",
       "      <td>-5.412340</td>\n",
       "      <td>-4.411184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7298</th>\n",
       "      <td>1.160625</td>\n",
       "      <td>-6.085674</td>\n",
       "      <td>-3.616640</td>\n",
       "      <td>-2.815058</td>\n",
       "      <td>-0.101404</td>\n",
       "      <td>-0.417562</td>\n",
       "      <td>-1.730290</td>\n",
       "      <td>0.260045</td>\n",
       "      <td>0.183894</td>\n",
       "      <td>-0.057154</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.366374</td>\n",
       "      <td>-5.495021</td>\n",
       "      <td>3.069672</td>\n",
       "      <td>0.566302</td>\n",
       "      <td>1.095581</td>\n",
       "      <td>3.134051</td>\n",
       "      <td>5.135721</td>\n",
       "      <td>1.004459</td>\n",
       "      <td>-2.584929</td>\n",
       "      <td>-2.664394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7299</th>\n",
       "      <td>0.110470</td>\n",
       "      <td>-7.775091</td>\n",
       "      <td>-12.366570</td>\n",
       "      <td>-3.473074</td>\n",
       "      <td>1.077668</td>\n",
       "      <td>0.441568</td>\n",
       "      <td>-1.243941</td>\n",
       "      <td>-0.409818</td>\n",
       "      <td>-1.179034</td>\n",
       "      <td>0.926047</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.908489</td>\n",
       "      <td>-11.564774</td>\n",
       "      <td>7.531423</td>\n",
       "      <td>-0.303380</td>\n",
       "      <td>-0.488631</td>\n",
       "      <td>10.309915</td>\n",
       "      <td>10.764680</td>\n",
       "      <td>-1.806220</td>\n",
       "      <td>-2.178345</td>\n",
       "      <td>-1.081335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1992 rows  2074 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Interleukin-6 signaling  Apoptosis  Hemostasis  \\\n",
       "MB-0002                -0.464915   1.150390    7.995550   \n",
       "MB-0008                 1.135676  -5.398171   -1.468534   \n",
       "MB-0010                -0.016407  -2.252856    1.688631   \n",
       "MB-0035                -0.045984   3.050125   14.221321   \n",
       "MB-0036                -1.107911   1.445771    5.790792   \n",
       "MB-0050                 1.003819   1.441920   -1.964812   \n",
       "MB-0059                -0.473150   0.186910    6.189742   \n",
       "MB-0060                 0.681486 -12.524534   -0.370151   \n",
       "MB-0066                 0.539735  -6.459812   -0.231027   \n",
       "MB-0101                 1.391563  -2.378717   -4.880185   \n",
       "MB-0102                 0.975992  -4.692292   -2.159457   \n",
       "MB-0106                -1.716131   3.055953    2.940126   \n",
       "MB-0112                 0.721624  -1.939072   -0.898322   \n",
       "MB-0115                 2.891250  -7.520513   -4.479024   \n",
       "MB-0117                -1.218244  -0.139969   -2.140516   \n",
       "MB-0119                 0.324922  -1.494166    2.681937   \n",
       "MB-0120                -1.282450  -1.312260   -5.283445   \n",
       "MB-0121                 0.066524  -3.182866    0.578324   \n",
       "MB-0122                 0.241645  -0.896690    4.363559   \n",
       "MB-0123                 0.632205  -4.939672    1.387583   \n",
       "MB-0126                 0.430240   0.585299    0.323267   \n",
       "MB-0131                 0.059170   0.794649    0.309136   \n",
       "MB-0133                -0.570782   4.560938    3.690203   \n",
       "MB-0135                -0.117803  -0.869882    0.971675   \n",
       "MB-0136                 0.874496  -3.379140   -9.386417   \n",
       "MB-0138                 1.158647   4.847099    0.673716   \n",
       "MB-0139                 2.100204   3.857392   -2.110865   \n",
       "MB-0140                -0.166482   1.533517    2.621224   \n",
       "MB-0142                -0.881733   2.119339   11.769112   \n",
       "MB-0143                 0.237474   1.306009    4.406096   \n",
       "...                          ...        ...         ...   \n",
       "MB-7264                 0.052616   0.475062   -8.357461   \n",
       "MB-7266                -2.427949   2.504505   12.534635   \n",
       "MB-7267                 1.027192  -3.108903   -3.791368   \n",
       "MB-7268                 0.899954   0.782333    6.789504   \n",
       "MB-7269                -1.542249  -6.540713   -6.588955   \n",
       "MB-7270                -0.818144  -2.313261    0.407206   \n",
       "MB-7273                 0.133795  -7.936353   -6.036484   \n",
       "MB-7275                 0.119508  -3.830375   -7.073864   \n",
       "MB-7276                 1.452774  -0.532131    0.833160   \n",
       "MB-7277                -0.836269  -2.383064    3.939454   \n",
       "MB-7278                -0.750752   5.208024    4.780788   \n",
       "MB-7279                -0.591865   3.546974    6.984630   \n",
       "MB-7280                 3.168238   3.348863    2.440740   \n",
       "MB-7281                -0.708823  -1.609151   -5.090321   \n",
       "MB-7283                -1.524783   6.702166   12.168263   \n",
       "MB-7284                -1.585521   2.486830    3.896734   \n",
       "MB-7285                 2.112241  -4.308329  -10.057469   \n",
       "MB-7286                 0.464739   0.329545    0.350963   \n",
       "MB-7287                -0.461483  -6.788634   -4.532691   \n",
       "MB-7288                 0.084161  -5.730170   -0.889219   \n",
       "MB-7289                 0.948163  -2.915446   -9.226002   \n",
       "MB-7291                 1.676027 -10.800482  -13.742885   \n",
       "MB-7292                -0.991138  -5.250101    0.637368   \n",
       "MB-7293                 0.295280  -5.007402   -6.453299   \n",
       "MB-7294                -0.241594  -6.497156    7.094790   \n",
       "MB-7295                -0.097627  -4.277540   -5.748505   \n",
       "MB-7296                -0.616957  -4.852698    1.617394   \n",
       "MB-7297                -0.546331  -8.314218   -2.774328   \n",
       "MB-7298                 1.160625  -6.085674   -3.616640   \n",
       "MB-7299                 0.110470  -7.775091  -12.366570   \n",
       "\n",
       "         Intrinsic Pathway for Apoptosis  PKB-mediated events  PI3K Cascade  \\\n",
       "MB-0002                        -2.133505            -1.005971     -1.934175   \n",
       "MB-0008                        -2.783638            -0.042169     -1.152715   \n",
       "MB-0010                        -1.840132             0.439943      0.907014   \n",
       "MB-0035                        -1.010664            -2.460802      0.033379   \n",
       "MB-0036                         1.275586            -1.040463     -2.897256   \n",
       "MB-0050                        -0.102851            -0.049525     -3.192611   \n",
       "MB-0059                        -1.018816            -1.353297     -2.260648   \n",
       "MB-0060                        -2.756785             0.077299     -0.448584   \n",
       "MB-0066                        -1.086133            -0.017782     -0.218056   \n",
       "MB-0101                         1.025275            -0.772772     -0.612536   \n",
       "MB-0102                        -1.916437            -0.424554      0.999638   \n",
       "MB-0106                         0.884104            -0.823157     -0.688135   \n",
       "MB-0112                         0.265369            -0.368646     -1.915540   \n",
       "MB-0115                        -4.139860            -0.845741     -0.875985   \n",
       "MB-0117                         0.796431             1.913941      0.425881   \n",
       "MB-0119                        -0.732578            -1.160084     -0.281727   \n",
       "MB-0120                        -1.769936            -0.042372      1.623986   \n",
       "MB-0121                        -1.262635            -0.358424     -0.596381   \n",
       "MB-0122                        -0.332297            -0.933940     -2.378229   \n",
       "MB-0123                        -2.019270             1.906553      1.130355   \n",
       "MB-0126                        -1.390457             0.169088     -4.192595   \n",
       "MB-0131                         0.093709            -0.878579     -0.526578   \n",
       "MB-0133                         2.302302             0.705017     -0.455561   \n",
       "MB-0135                        -1.517378            -1.827313     -2.082254   \n",
       "MB-0136                        -0.044186             0.481281     -0.811673   \n",
       "MB-0138                         1.441657            -0.541410     -0.216848   \n",
       "MB-0139                         1.680220             0.543285     -2.839074   \n",
       "MB-0140                        -0.157213            -0.250768     -0.001997   \n",
       "MB-0142                         1.807500            -1.036924     -1.392334   \n",
       "MB-0143                         0.412958            -2.160744     -1.154757   \n",
       "...                                  ...                  ...           ...   \n",
       "MB-7264                         2.719353             1.340783      1.720229   \n",
       "MB-7266                         0.578593            -1.317527     -0.400218   \n",
       "MB-7267                        -4.825164            -0.267898      0.911133   \n",
       "MB-7268                        -0.559499            -1.123089     -0.000125   \n",
       "MB-7269                        -0.905963            -0.975963      1.026272   \n",
       "MB-7270                        -1.231119             1.065474     -0.039110   \n",
       "MB-7273                        -1.853518             4.553318      2.124215   \n",
       "MB-7275                        -3.164525             0.449460     -2.310980   \n",
       "MB-7276                         0.619618             0.006122     -0.870304   \n",
       "MB-7277                         0.441016            -0.337889     -1.065002   \n",
       "MB-7278                         1.853149            -0.159819     -1.297136   \n",
       "MB-7279                         3.625156            -0.001635      3.217996   \n",
       "MB-7280                         0.188479             0.550790      0.139212   \n",
       "MB-7281                         0.525543             0.303806      2.379776   \n",
       "MB-7283                        -0.127327            -2.139917      0.550919   \n",
       "MB-7284                         0.832220            -0.452137      0.954621   \n",
       "MB-7285                        -1.500446             0.268187      0.055063   \n",
       "MB-7286                        -1.888931            -0.192087     -3.063396   \n",
       "MB-7287                        -0.648133             0.898894     -0.037595   \n",
       "MB-7288                        -3.048048            -0.746198     -0.468645   \n",
       "MB-7289                        -1.145563            -0.094301     -1.420274   \n",
       "MB-7291                        -3.676589             0.619119      1.983350   \n",
       "MB-7292                        -2.451837            -1.300904     -1.138077   \n",
       "MB-7293                        -0.666104            -0.802562     -0.854217   \n",
       "MB-7294                        -0.526408            -0.312066     -1.249597   \n",
       "MB-7295                        -3.180437            -0.865192     -0.979926   \n",
       "MB-7296                        -3.523299            -1.540831      1.360917   \n",
       "MB-7297                        -2.203446            -0.507499     -0.153017   \n",
       "MB-7298                        -2.815058            -0.101404     -0.417562   \n",
       "MB-7299                        -3.473074             1.077668      0.441568   \n",
       "\n",
       "         MAPK3 (ERK1) activation  Translesion synthesis by REV1  \\\n",
       "MB-0002                 0.682674                      -1.740452   \n",
       "MB-0008                -0.440677                      -0.738142   \n",
       "MB-0010                -0.946115                      -1.195843   \n",
       "MB-0035                -0.040401                       1.589719   \n",
       "MB-0036                 0.357232                       0.276760   \n",
       "MB-0050                -1.181677                       0.429309   \n",
       "MB-0059                 0.119735                       0.142322   \n",
       "MB-0060                -1.396643                      -3.632946   \n",
       "MB-0066                -0.731439                      -0.210748   \n",
       "MB-0101                -0.850303                       1.193591   \n",
       "MB-0102                -1.688466                       0.167435   \n",
       "MB-0106                 1.656749                      -0.002856   \n",
       "MB-0112                -1.256289                       1.112242   \n",
       "MB-0115                -2.905533                      -3.344632   \n",
       "MB-0117                 0.909888                       0.240269   \n",
       "MB-0119                -0.094547                       0.530474   \n",
       "MB-0120                 0.562100                      -0.550128   \n",
       "MB-0121                -0.376231                      -1.426032   \n",
       "MB-0122                 0.431727                      -0.993327   \n",
       "MB-0123                -0.342736                      -4.240944   \n",
       "MB-0126                -1.174283                       1.428986   \n",
       "MB-0131                -0.248608                       0.558131   \n",
       "MB-0133                 0.440143                       1.551490   \n",
       "MB-0135                -0.976155                      -1.920829   \n",
       "MB-0136                -0.483516                      -0.405053   \n",
       "MB-0138                -0.574181                       1.703794   \n",
       "MB-0139                -1.455348                       2.729110   \n",
       "MB-0140                -0.033955                      -1.053808   \n",
       "MB-0142                 0.138771                       3.207029   \n",
       "MB-0143                -0.658322                       1.296105   \n",
       "...                          ...                            ...   \n",
       "MB-7264                 0.486650                       1.187940   \n",
       "MB-7266                 0.768747                      -1.814783   \n",
       "MB-7267                -1.830152                      -2.247315   \n",
       "MB-7268                -1.000879                      -0.244545   \n",
       "MB-7269                 0.386235                      -1.722726   \n",
       "MB-7270                -0.105982                      -5.473630   \n",
       "MB-7273                -0.638000                      -0.010831   \n",
       "MB-7275                -0.657651                       0.304326   \n",
       "MB-7276                -0.551065                      -0.493778   \n",
       "MB-7277                 0.148280                      -0.753005   \n",
       "MB-7278                 0.650400                       0.951615   \n",
       "MB-7279                 1.287618                       2.715368   \n",
       "MB-7280                -1.515452                      -0.981956   \n",
       "MB-7281                 0.551161                       0.059091   \n",
       "MB-7283                 0.340746                      -0.509159   \n",
       "MB-7284                 2.116733                      -0.226250   \n",
       "MB-7285                -2.512579                      -0.288444   \n",
       "MB-7286                -1.545657                       0.978685   \n",
       "MB-7287                -0.657459                      -0.130143   \n",
       "MB-7288                -1.043475                      -0.499714   \n",
       "MB-7289                -1.017675                      -0.254609   \n",
       "MB-7291                -2.676246                      -2.079944   \n",
       "MB-7292                -0.387298                      -0.420057   \n",
       "MB-7293                -0.086637                      -1.197647   \n",
       "MB-7294                -0.619259                      -1.585000   \n",
       "MB-7295                -1.582625                      -0.735149   \n",
       "MB-7296                -0.889991                      -1.775554   \n",
       "MB-7297                -0.957462                      -0.821364   \n",
       "MB-7298                -1.730290                       0.260045   \n",
       "MB-7299                -1.243941                      -0.409818   \n",
       "\n",
       "         Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template  \\\n",
       "MB-0002                                          -1.758971                                    \n",
       "MB-0008                                          -0.950279                                    \n",
       "MB-0010                                          -2.019880                                    \n",
       "MB-0035                                           0.338310                                    \n",
       "MB-0036                                           1.075780                                    \n",
       "MB-0050                                           1.425590                                    \n",
       "MB-0059                                          -0.111307                                    \n",
       "MB-0060                                          -5.847386                                    \n",
       "MB-0066                                          -0.525239                                    \n",
       "MB-0101                                           2.045964                                    \n",
       "MB-0102                                          -1.977181                                    \n",
       "MB-0106                                           2.292241                                    \n",
       "MB-0112                                           1.566163                                    \n",
       "MB-0115                                          -3.650589                                    \n",
       "MB-0117                                           1.165142                                    \n",
       "MB-0119                                           0.249794                                    \n",
       "MB-0120                                          -1.018389                                    \n",
       "MB-0121                                           0.031787                                    \n",
       "MB-0122                                           0.664990                                    \n",
       "MB-0123                                          -4.235351                                    \n",
       "MB-0126                                           1.529271                                    \n",
       "MB-0131                                           0.547173                                    \n",
       "MB-0133                                           2.748361                                    \n",
       "MB-0135                                          -1.999828                                    \n",
       "MB-0136                                           0.319846                                    \n",
       "MB-0138                                           2.426379                                    \n",
       "MB-0139                                           4.206067                                    \n",
       "MB-0140                                          -1.000072                                    \n",
       "MB-0142                                           3.105644                                    \n",
       "MB-0143                                           0.730830                                    \n",
       "...                                                    ...                                    \n",
       "MB-7264                                           1.369706                                    \n",
       "MB-7266                                          -0.804663                                    \n",
       "MB-7267                                          -4.519671                                    \n",
       "MB-7268                                          -1.339817                                    \n",
       "MB-7269                                          -2.492191                                    \n",
       "MB-7270                                          -5.901622                                    \n",
       "MB-7273                                          -1.326545                                    \n",
       "MB-7275                                          -0.530877                                    \n",
       "MB-7276                                           1.031235                                    \n",
       "MB-7277                                           0.553203                                    \n",
       "MB-7278                                           2.215585                                    \n",
       "MB-7279                                           1.277015                                    \n",
       "MB-7280                                          -0.771772                                    \n",
       "MB-7281                                           0.492005                                    \n",
       "MB-7283                                          -1.216963                                    \n",
       "MB-7284                                          -0.314260                                    \n",
       "MB-7285                                          -0.051126                                    \n",
       "MB-7286                                           0.061323                                    \n",
       "MB-7287                                          -1.043855                                    \n",
       "MB-7288                                          -0.199662                                    \n",
       "MB-7289                                           1.379707                                    \n",
       "MB-7291                                          -2.595874                                    \n",
       "MB-7292                                          -2.341863                                    \n",
       "MB-7293                                           0.210241                                    \n",
       "MB-7294                                          -0.730437                                    \n",
       "MB-7295                                          -0.468519                                    \n",
       "MB-7296                                          -2.728289                                    \n",
       "MB-7297                                          -1.953621                                    \n",
       "MB-7298                                           0.183894                                    \n",
       "MB-7299                                          -1.179034                                    \n",
       "\n",
       "         Recognition of DNA damage by PCNA-containing replication complex  \\\n",
       "MB-0002                                           2.306896                  \n",
       "MB-0008                                           0.857385                  \n",
       "MB-0010                                           2.452047                  \n",
       "MB-0035                                          -0.025657                  \n",
       "MB-0036                                          -1.226257                  \n",
       "MB-0050                                          -1.375613                  \n",
       "MB-0059                                          -0.685721                  \n",
       "MB-0060                                           6.291344                  \n",
       "MB-0066                                           1.251225                  \n",
       "MB-0101                                          -1.989934                  \n",
       "MB-0102                                           1.916599                  \n",
       "MB-0106                                          -1.652733                  \n",
       "MB-0112                                          -1.291966                  \n",
       "MB-0115                                           3.451235                  \n",
       "MB-0117                                          -1.165904                  \n",
       "MB-0119                                          -0.944412                  \n",
       "MB-0120                                           0.599823                  \n",
       "MB-0121                                           0.056474                  \n",
       "MB-0122                                          -0.454606                  \n",
       "MB-0123                                           4.132531                  \n",
       "MB-0126                                          -2.216522                  \n",
       "MB-0131                                          -0.530690                  \n",
       "MB-0133                                          -2.443490                  \n",
       "MB-0135                                           2.118295                  \n",
       "MB-0136                                          -0.551310                  \n",
       "MB-0138                                          -2.162349                  \n",
       "MB-0139                                          -4.028101                  \n",
       "MB-0140                                           1.821616                  \n",
       "MB-0142                                          -3.524379                  \n",
       "MB-0143                                          -0.765246                  \n",
       "...                                                    ...                  \n",
       "MB-7264                                          -0.953181                  \n",
       "MB-7266                                           1.092236                  \n",
       "MB-7267                                           5.070132                  \n",
       "MB-7268                                           0.690283                  \n",
       "MB-7269                                           2.197838                  \n",
       "MB-7270                                           6.593430                  \n",
       "MB-7273                                           0.197701                  \n",
       "MB-7275                                           0.338423                  \n",
       "MB-7276                                          -0.607397                  \n",
       "MB-7277                                           0.260013                  \n",
       "MB-7278                                          -1.703582                  \n",
       "MB-7279                                          -1.698399                  \n",
       "MB-7280                                           0.736326                  \n",
       "MB-7281                                          -0.799303                  \n",
       "MB-7283                                           0.841771                  \n",
       "MB-7284                                          -0.059567                  \n",
       "MB-7285                                          -0.545414                  \n",
       "MB-7286                                          -0.506717                  \n",
       "MB-7287                                           0.465353                  \n",
       "MB-7288                                           0.688354                  \n",
       "MB-7289                                          -1.184310                  \n",
       "MB-7291                                           1.998022                  \n",
       "MB-7292                                           1.763887                  \n",
       "MB-7293                                           0.307119                  \n",
       "MB-7294                                           0.616089                  \n",
       "MB-7295                                           0.321434                  \n",
       "MB-7296                                           1.738995                  \n",
       "MB-7297                                           2.779270                  \n",
       "MB-7298                                          -0.057154                  \n",
       "MB-7299                                           0.926047                  \n",
       "\n",
       "                                        ...                                   \\\n",
       "MB-0002                                 ...                                    \n",
       "MB-0008                                 ...                                    \n",
       "MB-0010                                 ...                                    \n",
       "MB-0035                                 ...                                    \n",
       "MB-0036                                 ...                                    \n",
       "MB-0050                                 ...                                    \n",
       "MB-0059                                 ...                                    \n",
       "MB-0060                                 ...                                    \n",
       "MB-0066                                 ...                                    \n",
       "MB-0101                                 ...                                    \n",
       "MB-0102                                 ...                                    \n",
       "MB-0106                                 ...                                    \n",
       "MB-0112                                 ...                                    \n",
       "MB-0115                                 ...                                    \n",
       "MB-0117                                 ...                                    \n",
       "MB-0119                                 ...                                    \n",
       "MB-0120                                 ...                                    \n",
       "MB-0121                                 ...                                    \n",
       "MB-0122                                 ...                                    \n",
       "MB-0123                                 ...                                    \n",
       "MB-0126                                 ...                                    \n",
       "MB-0131                                 ...                                    \n",
       "MB-0133                                 ...                                    \n",
       "MB-0135                                 ...                                    \n",
       "MB-0136                                 ...                                    \n",
       "MB-0138                                 ...                                    \n",
       "MB-0139                                 ...                                    \n",
       "MB-0140                                 ...                                    \n",
       "MB-0142                                 ...                                    \n",
       "MB-0143                                 ...                                    \n",
       "...                                     ...                                    \n",
       "MB-7264                                 ...                                    \n",
       "MB-7266                                 ...                                    \n",
       "MB-7267                                 ...                                    \n",
       "MB-7268                                 ...                                    \n",
       "MB-7269                                 ...                                    \n",
       "MB-7270                                 ...                                    \n",
       "MB-7273                                 ...                                    \n",
       "MB-7275                                 ...                                    \n",
       "MB-7276                                 ...                                    \n",
       "MB-7277                                 ...                                    \n",
       "MB-7278                                 ...                                    \n",
       "MB-7279                                 ...                                    \n",
       "MB-7280                                 ...                                    \n",
       "MB-7281                                 ...                                    \n",
       "MB-7283                                 ...                                    \n",
       "MB-7284                                 ...                                    \n",
       "MB-7285                                 ...                                    \n",
       "MB-7286                                 ...                                    \n",
       "MB-7287                                 ...                                    \n",
       "MB-7288                                 ...                                    \n",
       "MB-7289                                 ...                                    \n",
       "MB-7291                                 ...                                    \n",
       "MB-7292                                 ...                                    \n",
       "MB-7293                                 ...                                    \n",
       "MB-7294                                 ...                                    \n",
       "MB-7295                                 ...                                    \n",
       "MB-7296                                 ...                                    \n",
       "MB-7297                                 ...                                    \n",
       "MB-7298                                 ...                                    \n",
       "MB-7299                                 ...                                    \n",
       "\n",
       "         Antigen processing: Ubiquitination & Proteasome degradation  \\\n",
       "MB-0002                                          -0.181212             \n",
       "MB-0008                                          -6.103682             \n",
       "MB-0010                                          -4.429469             \n",
       "MB-0035                                           2.307189             \n",
       "MB-0036                                           0.652174             \n",
       "MB-0050                                          -1.234766             \n",
       "MB-0059                                          -4.292349             \n",
       "MB-0060                                          -7.273609             \n",
       "MB-0066                                          -7.044344             \n",
       "MB-0101                                          -1.349735             \n",
       "MB-0102                                          -3.961124             \n",
       "MB-0106                                           2.509509             \n",
       "MB-0112                                          -3.463129             \n",
       "MB-0115                                          -5.246096             \n",
       "MB-0117                                           0.818814             \n",
       "MB-0119                                          -2.911408             \n",
       "MB-0120                                          -1.535755             \n",
       "MB-0121                                          -6.407047             \n",
       "MB-0122                                          -4.043551             \n",
       "MB-0123                                          -9.242506             \n",
       "MB-0126                                          -0.852170             \n",
       "MB-0131                                          -1.802845             \n",
       "MB-0133                                           2.981783             \n",
       "MB-0135                                          -3.796854             \n",
       "MB-0136                                          -5.329268             \n",
       "MB-0138                                           2.631016             \n",
       "MB-0139                                          -0.110126             \n",
       "MB-0140                                          -1.463628             \n",
       "MB-0142                                           1.139717             \n",
       "MB-0143                                          -1.103454             \n",
       "...                                                    ...             \n",
       "MB-7264                                           2.978944             \n",
       "MB-7266                                           0.236276             \n",
       "MB-7267                                           1.234845             \n",
       "MB-7268                                          -2.645635             \n",
       "MB-7269                                          -2.246201             \n",
       "MB-7270                                           0.104617             \n",
       "MB-7273                                          -6.583868             \n",
       "MB-7275                                          -6.027479             \n",
       "MB-7276                                          -0.924477             \n",
       "MB-7277                                          -2.903476             \n",
       "MB-7278                                           2.780067             \n",
       "MB-7279                                           7.453155             \n",
       "MB-7280                                           0.306340             \n",
       "MB-7281                                           0.093180             \n",
       "MB-7283                                           5.662923             \n",
       "MB-7284                                          -0.075522             \n",
       "MB-7285                                          -5.234862             \n",
       "MB-7286                                          -2.372422             \n",
       "MB-7287                                          -7.005773             \n",
       "MB-7288                                          -9.528711             \n",
       "MB-7289                                          -5.259995             \n",
       "MB-7291                                         -10.244129             \n",
       "MB-7292                                          -7.732811             \n",
       "MB-7293                                          -8.022697             \n",
       "MB-7294                                          -6.888719             \n",
       "MB-7295                                          -6.578778             \n",
       "MB-7296                                          -6.110505             \n",
       "MB-7297                                         -10.178550             \n",
       "MB-7298                                          -7.366374             \n",
       "MB-7299                                          -6.908489             \n",
       "\n",
       "         Class I MHC mediated antigen processing & presentation  \\\n",
       "MB-0002                                          10.501022        \n",
       "MB-0008                                          -4.968357        \n",
       "MB-0010                                           2.602108        \n",
       "MB-0035                                          11.536765        \n",
       "MB-0036                                           2.737150        \n",
       "MB-0050                                           5.523601        \n",
       "MB-0059                                           7.735078        \n",
       "MB-0060                                         -19.867712        \n",
       "MB-0066                                         -10.432402        \n",
       "MB-0101                                          -5.605123        \n",
       "MB-0102                                          -7.064365        \n",
       "MB-0106                                           6.347529        \n",
       "MB-0112                                          -2.173827        \n",
       "MB-0115                                         -12.025943        \n",
       "MB-0117                                           0.280332        \n",
       "MB-0119                                           6.341346        \n",
       "MB-0120                                          -3.482248        \n",
       "MB-0121                                           0.354208        \n",
       "MB-0122                                           7.237955        \n",
       "MB-0123                                           4.165809        \n",
       "MB-0126                                           3.772588        \n",
       "MB-0131                                           2.986812        \n",
       "MB-0133                                          10.042189        \n",
       "MB-0135                                           5.749571        \n",
       "MB-0136                                          -6.115133        \n",
       "MB-0138                                           6.741145        \n",
       "MB-0139                                           7.031245        \n",
       "MB-0140                                           9.266462        \n",
       "MB-0142                                          11.023046        \n",
       "MB-0143                                           6.532816        \n",
       "...                                                    ...        \n",
       "MB-7264                                          -8.202431        \n",
       "MB-7266                                           2.667299        \n",
       "MB-7267                                          -0.908210        \n",
       "MB-7268                                           8.161488        \n",
       "MB-7269                                         -20.507946        \n",
       "MB-7270                                          -3.871092        \n",
       "MB-7273                                          -9.537011        \n",
       "MB-7275                                          -1.047748        \n",
       "MB-7276                                          -2.676211        \n",
       "MB-7277                                           0.343151        \n",
       "MB-7278                                           9.351828        \n",
       "MB-7279                                          -6.069090        \n",
       "MB-7280                                           9.265659        \n",
       "MB-7281                                          -4.191171        \n",
       "MB-7283                                          17.127012        \n",
       "MB-7284                                           9.258899        \n",
       "MB-7285                                          -1.385978        \n",
       "MB-7286                                           2.396867        \n",
       "MB-7287                                         -10.989277        \n",
       "MB-7288                                          -4.563002        \n",
       "MB-7289                                          -1.504258        \n",
       "MB-7291                                         -13.150474        \n",
       "MB-7292                                          -2.052564        \n",
       "MB-7293                                          -5.319050        \n",
       "MB-7294                                          -3.931945        \n",
       "MB-7295                                          -0.410079        \n",
       "MB-7296                                           0.600571        \n",
       "MB-7297                                          -6.728893        \n",
       "MB-7298                                          -5.495021        \n",
       "MB-7299                                         -11.564774        \n",
       "\n",
       "         Antigen Presentation: Folding, assembly and peptide loading of class I MHC  \\\n",
       "MB-0002                                         -10.621735                            \n",
       "MB-0008                                           3.579570                            \n",
       "MB-0010                                          -4.274843                            \n",
       "MB-0035                                          -9.387837                            \n",
       "MB-0036                                          -1.630176                            \n",
       "MB-0050                                          -4.244558                            \n",
       "MB-0059                                          -7.386242                            \n",
       "MB-0060                                          14.711741                            \n",
       "MB-0066                                           7.673403                            \n",
       "MB-0101                                           3.450237                            \n",
       "MB-0102                                           3.386175                            \n",
       "MB-0106                                          -3.986429                            \n",
       "MB-0112                                           0.766776                            \n",
       "MB-0115                                           9.564923                            \n",
       "MB-0117                                           0.812773                            \n",
       "MB-0119                                          -6.942924                            \n",
       "MB-0120                                           1.582937                            \n",
       "MB-0121                                          -3.060088                            \n",
       "MB-0122                                          -7.595239                            \n",
       "MB-0123                                          -7.355088                            \n",
       "MB-0126                                          -3.266582                            \n",
       "MB-0131                                          -3.750671                            \n",
       "MB-0133                                          -6.609117                            \n",
       "MB-0135                                          -4.539532                            \n",
       "MB-0136                                           3.480316                            \n",
       "MB-0138                                          -4.268781                            \n",
       "MB-0139                                          -3.852609                            \n",
       "MB-0140                                          -8.249615                            \n",
       "MB-0142                                          -8.919237                            \n",
       "MB-0143                                          -6.297841                            \n",
       "...                                                    ...                            \n",
       "MB-7264                                           7.960864                            \n",
       "MB-7266                                           1.511926                            \n",
       "MB-7267                                          -1.150582                            \n",
       "MB-7268                                          -7.698986                            \n",
       "MB-7269                                          18.701986                            \n",
       "MB-7270                                           0.700887                            \n",
       "MB-7273                                           4.357210                            \n",
       "MB-7275                                          -1.119716                            \n",
       "MB-7276                                           3.357074                            \n",
       "MB-7277                                          -1.629507                            \n",
       "MB-7278                                          -7.465249                            \n",
       "MB-7279                                           8.397773                            \n",
       "MB-7280                                          -6.396199                            \n",
       "MB-7281                                           2.704735                            \n",
       "MB-7283                                         -12.113644                            \n",
       "MB-7284                                          -6.996106                            \n",
       "MB-7285                                          -1.994269                            \n",
       "MB-7286                                          -1.212736                            \n",
       "MB-7287                                           6.587463                            \n",
       "MB-7288                                           1.488169                            \n",
       "MB-7289                                          -0.598862                            \n",
       "MB-7291                                           6.470609                            \n",
       "MB-7292                                           0.100131                            \n",
       "MB-7293                                           1.760966                            \n",
       "MB-7294                                           0.534228                            \n",
       "MB-7295                                          -2.211857                            \n",
       "MB-7296                                          -2.990283                            \n",
       "MB-7297                                           2.466814                            \n",
       "MB-7298                                           3.069672                            \n",
       "MB-7299                                           7.531423                            \n",
       "\n",
       "         Kinesins  \\\n",
       "MB-0002 -2.783924   \n",
       "MB-0008  2.017033   \n",
       "MB-0010  5.476014   \n",
       "MB-0035 -1.037284   \n",
       "MB-0036 -1.113311   \n",
       "MB-0050 -5.265519   \n",
       "MB-0059 -0.915784   \n",
       "MB-0060  5.430391   \n",
       "MB-0066 -1.260229   \n",
       "MB-0101 -2.927546   \n",
       "MB-0102  1.580122   \n",
       "MB-0106 -5.207336   \n",
       "MB-0112 -3.123365   \n",
       "MB-0115  6.066910   \n",
       "MB-0117 -2.006572   \n",
       "MB-0119 -0.424924   \n",
       "MB-0120  0.281499   \n",
       "MB-0121 -3.040142   \n",
       "MB-0122 -5.352230   \n",
       "MB-0123  6.224326   \n",
       "MB-0126 -3.031947   \n",
       "MB-0131 -1.548971   \n",
       "MB-0133 -6.219803   \n",
       "MB-0135  3.203228   \n",
       "MB-0136 -1.196467   \n",
       "MB-0138 -2.985508   \n",
       "MB-0139 -6.474811   \n",
       "MB-0140  0.315797   \n",
       "MB-0142 -5.648750   \n",
       "MB-0143 -2.127102   \n",
       "...           ...   \n",
       "MB-7264 -0.747237   \n",
       "MB-7266 -1.543994   \n",
       "MB-7267  8.357569   \n",
       "MB-7268 -0.410765   \n",
       "MB-7269  5.781100   \n",
       "MB-7270  6.023480   \n",
       "MB-7273  1.776304   \n",
       "MB-7275 -1.453237   \n",
       "MB-7276 -2.800376   \n",
       "MB-7277 -3.523731   \n",
       "MB-7278 -5.574749   \n",
       "MB-7279  4.981525   \n",
       "MB-7280 -1.326755   \n",
       "MB-7281  0.990196   \n",
       "MB-7283  2.423513   \n",
       "MB-7284 -2.956348   \n",
       "MB-7285 -0.925810   \n",
       "MB-7286  0.224308   \n",
       "MB-7287  1.458673   \n",
       "MB-7288 -1.655888   \n",
       "MB-7289 -4.488339   \n",
       "MB-7291  2.560807   \n",
       "MB-7292  1.853204   \n",
       "MB-7293 -4.034656   \n",
       "MB-7294 -0.663199   \n",
       "MB-7295 -2.745921   \n",
       "MB-7296  5.587256   \n",
       "MB-7297  2.097607   \n",
       "MB-7298  0.566302   \n",
       "MB-7299 -0.303380   \n",
       "\n",
       "         Factors involved in megakaryocyte development and platelet production  \\\n",
       "MB-0002                                          -1.614593                       \n",
       "MB-0008                                           3.156719                       \n",
       "MB-0010                                           5.792390                       \n",
       "MB-0035                                          -1.172208                       \n",
       "MB-0036                                          -1.828239                       \n",
       "MB-0050                                          -5.665442                       \n",
       "MB-0059                                          -0.762069                       \n",
       "MB-0060                                           6.268124                       \n",
       "MB-0066                                          -0.731123                       \n",
       "MB-0101                                          -1.869052                       \n",
       "MB-0102                                           1.705319                       \n",
       "MB-0106                                          -5.675126                       \n",
       "MB-0112                                          -3.398603                       \n",
       "MB-0115                                           6.558084                       \n",
       "MB-0117                                          -2.226116                       \n",
       "MB-0119                                           0.076581                       \n",
       "MB-0120                                           1.953857                       \n",
       "MB-0121                                          -3.220359                       \n",
       "MB-0122                                          -5.548123                       \n",
       "MB-0123                                           6.990339                       \n",
       "MB-0126                                          -3.440848                       \n",
       "MB-0131                                          -1.714889                       \n",
       "MB-0133                                          -6.761085                       \n",
       "MB-0135                                           2.322894                       \n",
       "MB-0136                                          -2.514433                       \n",
       "MB-0138                                          -3.976150                       \n",
       "MB-0139                                          -7.965247                       \n",
       "MB-0140                                          -0.339202                       \n",
       "MB-0142                                          -6.126320                       \n",
       "MB-0143                                          -2.455823                       \n",
       "...                                                    ...                       \n",
       "MB-7264                                          -0.958069                       \n",
       "MB-7266                                          -1.962177                       \n",
       "MB-7267                                           9.896810                       \n",
       "MB-7268                                          -0.940649                       \n",
       "MB-7269                                           6.876094                       \n",
       "MB-7270                                           9.270829                       \n",
       "MB-7273                                           3.194313                       \n",
       "MB-7275                                          -1.435250                       \n",
       "MB-7276                                          -3.255994                       \n",
       "MB-7277                                          -3.699770                       \n",
       "MB-7278                                          -5.874127                       \n",
       "MB-7279                                           5.700316                       \n",
       "MB-7280                                          -0.831227                       \n",
       "MB-7281                                           2.182252                       \n",
       "MB-7283                                           2.340146                       \n",
       "MB-7284                                          -3.568803                       \n",
       "MB-7285                                          -0.647075                       \n",
       "MB-7286                                          -0.593532                       \n",
       "MB-7287                                           2.337528                       \n",
       "MB-7288                                          -1.267448                       \n",
       "MB-7289                                          -4.522991                       \n",
       "MB-7291                                           3.581800                       \n",
       "MB-7292                                           2.265941                       \n",
       "MB-7293                                          -4.194694                       \n",
       "MB-7294                                          -0.068830                       \n",
       "MB-7295                                          -2.813467                       \n",
       "MB-7296                                           5.737139                       \n",
       "MB-7297                                           3.400695                       \n",
       "MB-7298                                           1.095581                       \n",
       "MB-7299                                          -0.488631                       \n",
       "\n",
       "         Antigen activates B Cell Receptor (BCR) leading to generation of second messengers  \\\n",
       "MB-0002                                          -3.780976                                    \n",
       "MB-0008                                           3.019890                                    \n",
       "MB-0010                                          -5.347735                                    \n",
       "MB-0035                                          -6.415124                                    \n",
       "MB-0036                                          -4.086297                                    \n",
       "MB-0050                                           2.556065                                    \n",
       "MB-0059                                          -6.173665                                    \n",
       "MB-0060                                          -1.857444                                    \n",
       "MB-0066                                           0.353554                                    \n",
       "MB-0101                                           5.157788                                    \n",
       "MB-0102                                           2.999873                                    \n",
       "MB-0106                                          -3.342922                                    \n",
       "MB-0112                                           0.281370                                    \n",
       "MB-0115                                           3.361716                                    \n",
       "MB-0117                                           3.127701                                    \n",
       "MB-0119                                          -0.182723                                    \n",
       "MB-0120                                           5.230402                                    \n",
       "MB-0121                                          -2.223592                                    \n",
       "MB-0122                                          -2.932324                                    \n",
       "MB-0123                                          -1.275137                                    \n",
       "MB-0126                                          -3.510166                                    \n",
       "MB-0131                                          -5.144326                                    \n",
       "MB-0133                                          -2.707678                                    \n",
       "MB-0135                                           0.924889                                    \n",
       "MB-0136                                           2.689187                                    \n",
       "MB-0138                                           0.794619                                    \n",
       "MB-0139                                          -0.300409                                    \n",
       "MB-0140                                          -0.292043                                    \n",
       "MB-0142                                          -3.622275                                    \n",
       "MB-0143                                          -2.132147                                    \n",
       "...                                                    ...                                    \n",
       "MB-7264                                           8.858024                                    \n",
       "MB-7266                                          -6.226096                                    \n",
       "MB-7267                                           6.050182                                    \n",
       "MB-7268                                          -3.436144                                    \n",
       "MB-7269                                           5.948816                                    \n",
       "MB-7270                                           1.519696                                    \n",
       "MB-7273                                           5.475915                                    \n",
       "MB-7275                                          -0.190809                                    \n",
       "MB-7276                                           0.752929                                    \n",
       "MB-7277                                          -2.449184                                    \n",
       "MB-7278                                          -0.789296                                    \n",
       "MB-7279                                          -1.411486                                    \n",
       "MB-7280                                          -0.360274                                    \n",
       "MB-7281                                           2.212883                                    \n",
       "MB-7283                                          -3.977006                                    \n",
       "MB-7284                                          -1.943100                                    \n",
       "MB-7285                                           2.705253                                    \n",
       "MB-7286                                          -2.600160                                    \n",
       "MB-7287                                           5.110461                                    \n",
       "MB-7288                                           2.466028                                    \n",
       "MB-7289                                           5.110595                                    \n",
       "MB-7291                                           5.677145                                    \n",
       "MB-7292                                          -3.248151                                    \n",
       "MB-7293                                           0.838636                                    \n",
       "MB-7294                                          -2.938476                                    \n",
       "MB-7295                                          -1.638444                                    \n",
       "MB-7296                                          -2.236260                                    \n",
       "MB-7297                                           6.833561                                    \n",
       "MB-7298                                           3.134051                                    \n",
       "MB-7299                                          10.309915                                    \n",
       "\n",
       "         Signaling by the B Cell Receptor (BCR)  Ion channel transport  \\\n",
       "MB-0002                               -5.213696               2.340642   \n",
       "MB-0008                                4.179122               2.678819   \n",
       "MB-0010                               -3.749546              -0.974028   \n",
       "MB-0035                               -7.129510               0.346868   \n",
       "MB-0036                               -3.703407               3.655877   \n",
       "MB-0050                               -0.543039              -0.737432   \n",
       "MB-0059                               -6.540617               0.229692   \n",
       "MB-0060                                7.910823              -2.220703   \n",
       "MB-0066                                4.875847               1.274467   \n",
       "MB-0101                                5.911775               0.584062   \n",
       "MB-0102                                5.668929              -2.046671   \n",
       "MB-0106                               -4.616443               1.865447   \n",
       "MB-0112                                1.766547              -1.689046   \n",
       "MB-0115                                6.288507              -8.519237   \n",
       "MB-0117                                2.317191               0.827094   \n",
       "MB-0119                               -1.025801               0.215425   \n",
       "MB-0120                                3.683432               0.937613   \n",
       "MB-0121                               -0.487749               0.361180   \n",
       "MB-0122                               -2.600730               1.526380   \n",
       "MB-0123                               -0.749794               1.262404   \n",
       "MB-0126                               -4.230219               0.113761   \n",
       "MB-0131                               -4.683748               1.047786   \n",
       "MB-0133                               -5.525304               0.238325   \n",
       "MB-0135                               -0.893908               2.643863   \n",
       "MB-0136                                4.386436              -0.859905   \n",
       "MB-0138                               -2.320587               2.239991   \n",
       "MB-0139                               -4.190813               0.154110   \n",
       "MB-0140                               -3.042215               3.212023   \n",
       "MB-0142                               -4.970809               4.395123   \n",
       "MB-0143                               -2.518095               2.648164   \n",
       "...                                         ...                    ...   \n",
       "MB-7264                                8.301377              -1.376696   \n",
       "MB-7266                               -5.982386               3.705963   \n",
       "MB-7267                                5.022543              -4.776596   \n",
       "MB-7268                               -4.130540               0.890925   \n",
       "MB-7269                               10.695359              -5.042941   \n",
       "MB-7270                                3.669500              -5.082238   \n",
       "MB-7273                                8.468404              -3.994619   \n",
       "MB-7275                                0.521049              -1.946833   \n",
       "MB-7276                                2.364048               0.047615   \n",
       "MB-7277                                0.086364               0.993515   \n",
       "MB-7278                               -4.013051              -0.210591   \n",
       "MB-7279                               -0.457442               3.062864   \n",
       "MB-7280                               -4.104418               0.139250   \n",
       "MB-7281                                3.244051              -3.805825   \n",
       "MB-7283                               -8.119009               3.242792   \n",
       "MB-7284                               -4.104070               2.396170   \n",
       "MB-7285                                3.287334              -2.418825   \n",
       "MB-7286                               -4.352315               2.006475   \n",
       "MB-7287                                8.699111              -1.883775   \n",
       "MB-7288                                3.618482              -0.904307   \n",
       "MB-7289                                4.190233              -2.456881   \n",
       "MB-7291                               10.225146              -5.255381   \n",
       "MB-7292                               -0.804346              -0.104385   \n",
       "MB-7293                                2.630240              -3.092360   \n",
       "MB-7294                                1.727928               2.497594   \n",
       "MB-7295                               -0.490903               0.070499   \n",
       "MB-7296                               -1.231161               0.606136   \n",
       "MB-7297                                8.375035              -1.153599   \n",
       "MB-7298                                5.135721               1.004459   \n",
       "MB-7299                               10.764680              -1.806220   \n",
       "\n",
       "         Activation of GABAB receptors  \\\n",
       "MB-0002                      -2.081063   \n",
       "MB-0008                       0.760122   \n",
       "MB-0010                      -2.141831   \n",
       "MB-0035                       2.015445   \n",
       "MB-0036                      -0.986929   \n",
       "MB-0050                       1.518420   \n",
       "MB-0059                       0.725576   \n",
       "MB-0060                      -0.937799   \n",
       "MB-0066                      -0.689864   \n",
       "MB-0101                      -1.963235   \n",
       "MB-0102                      -3.694572   \n",
       "MB-0106                       1.400919   \n",
       "MB-0112                      -2.135168   \n",
       "MB-0115                      -0.397631   \n",
       "MB-0117                       1.706968   \n",
       "MB-0119                      -2.736801   \n",
       "MB-0120                      -2.077385   \n",
       "MB-0121                      -2.343898   \n",
       "MB-0122                      -0.434253   \n",
       "MB-0123                      -3.915522   \n",
       "MB-0126                      -0.226449   \n",
       "MB-0131                       0.745869   \n",
       "MB-0133                       3.263350   \n",
       "MB-0135                      -3.307961   \n",
       "MB-0136                      -1.844360   \n",
       "MB-0138                       1.720395   \n",
       "MB-0139                       2.372650   \n",
       "MB-0140                      -4.014152   \n",
       "MB-0142                      -0.061439   \n",
       "MB-0143                      -0.377887   \n",
       "...                                ...   \n",
       "MB-7264                      -1.475901   \n",
       "MB-7266                      -0.049891   \n",
       "MB-7267                      -4.174924   \n",
       "MB-7268                       0.940230   \n",
       "MB-7269                      -0.128779   \n",
       "MB-7270                      -2.763787   \n",
       "MB-7273                      -1.469257   \n",
       "MB-7275                      -2.628785   \n",
       "MB-7276                      -1.582996   \n",
       "MB-7277                      -1.396922   \n",
       "MB-7278                      -1.246493   \n",
       "MB-7279                       2.922412   \n",
       "MB-7280                      -2.458190   \n",
       "MB-7281                       0.245758   \n",
       "MB-7283                       2.700290   \n",
       "MB-7284                       3.613272   \n",
       "MB-7285                      -3.060629   \n",
       "MB-7286                       1.551420   \n",
       "MB-7287                      -2.270477   \n",
       "MB-7288                      -3.733653   \n",
       "MB-7289                      -2.388495   \n",
       "MB-7291                      -4.037444   \n",
       "MB-7292                      -4.864537   \n",
       "MB-7293                      -4.634184   \n",
       "MB-7294                      -3.496975   \n",
       "MB-7295                      -3.442825   \n",
       "MB-7296                      -0.894533   \n",
       "MB-7297                      -5.412340   \n",
       "MB-7298                      -2.584929   \n",
       "MB-7299                      -2.178345   \n",
       "\n",
       "         Inhibition  of voltage gated Ca2+ channels via Gbeta/gamma subunits  \n",
       "MB-0002                                          -1.718509                    \n",
       "MB-0008                                           0.911403                    \n",
       "MB-0010                                          -1.831642                    \n",
       "MB-0035                                           2.369224                    \n",
       "MB-0036                                          -1.141993                    \n",
       "MB-0050                                           1.390619                    \n",
       "MB-0059                                           1.175208                    \n",
       "MB-0060                                           0.211784                    \n",
       "MB-0066                                          -0.615685                    \n",
       "MB-0101                                          -2.617657                    \n",
       "MB-0102                                          -3.004808                    \n",
       "MB-0106                                           1.725544                    \n",
       "MB-0112                                          -1.900310                    \n",
       "MB-0115                                           1.017734                    \n",
       "MB-0117                                           1.396368                    \n",
       "MB-0119                                          -2.474468                    \n",
       "MB-0120                                          -1.412313                    \n",
       "MB-0121                                          -2.393808                    \n",
       "MB-0122                                          -0.132890                    \n",
       "MB-0123                                          -3.345704                    \n",
       "MB-0126                                           0.015675                    \n",
       "MB-0131                                          -0.520462                    \n",
       "MB-0133                                           2.628679                    \n",
       "MB-0135                                          -2.965665                    \n",
       "MB-0136                                          -2.120232                    \n",
       "MB-0138                                           1.111475                    \n",
       "MB-0139                                           1.756206                    \n",
       "MB-0140                                          -4.762921                    \n",
       "MB-0142                                          -0.582873                    \n",
       "MB-0143                                          -0.254645                    \n",
       "...                                                    ...                    \n",
       "MB-7264                                          -1.209309                    \n",
       "MB-7266                                           0.316525                    \n",
       "MB-7267                                          -3.890250                    \n",
       "MB-7268                                           0.841507                    \n",
       "MB-7269                                           0.578354                    \n",
       "MB-7270                                          -2.605367                    \n",
       "MB-7273                                          -2.158524                    \n",
       "MB-7275                                          -2.593172                    \n",
       "MB-7276                                          -1.464334                    \n",
       "MB-7277                                          -1.670098                    \n",
       "MB-7278                                          -1.164350                    \n",
       "MB-7279                                           2.362305                    \n",
       "MB-7280                                          -2.486319                    \n",
       "MB-7281                                           0.122075                    \n",
       "MB-7283                                           2.586261                    \n",
       "MB-7284                                           3.582648                    \n",
       "MB-7285                                          -3.616641                    \n",
       "MB-7286                                           1.824794                    \n",
       "MB-7287                                          -2.428680                    \n",
       "MB-7288                                          -3.187384                    \n",
       "MB-7289                                          -2.264113                    \n",
       "MB-7291                                          -3.730545                    \n",
       "MB-7292                                          -4.864987                    \n",
       "MB-7293                                          -4.191624                    \n",
       "MB-7294                                          -3.691912                    \n",
       "MB-7295                                          -3.079476                    \n",
       "MB-7296                                          -0.717148                    \n",
       "MB-7297                                          -4.411184                    \n",
       "MB-7298                                          -2.664394                    \n",
       "MB-7299                                          -1.081335                    \n",
       "\n",
       "[1992 rows x 2074 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pca_per_pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4.275524576934102e-14\n"
     ]
    }
   ],
   "source": [
    "from statsmodels.formula.api import ols\n",
    "import statsmodels.api as sm\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "results = ols(\"Q('Cell Cycle') ~ C(Q('Intclust 1 or not'))\", data=full_df).fit()\n",
    "table = sm.stats.anova_lm(results, typ=2)\n",
    "print(table['PR(>F)'][0])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4ER+\n",
      "9\n",
      "7\n",
      "3\n",
      "8\n",
      "10\n",
      "1\n",
      "2\n",
      "5\n",
      "4ER-\n",
      "6\n"
     ]
    }
   ],
   "source": [
    "from scipy.stats import ttest_ind, mannwhitneyu\n",
    "import qvalue as qv\n",
    "\n",
    "clusterframes = {}\n",
    "\n",
    "\n",
    "for i in full_df['Integrative Cluster'].unique():\n",
    "    if type(i) == str:\n",
    "        df_cluster = pd.DataFrame(index=pca_per_pathway.columns) \n",
    "        pvaluelist = []\n",
    "        for pathway in df_cluster.index:\n",
    "            results = ols(\"Q(pathway) ~ C(Q(f'Intclust {i} or not'))\", data=full_df).fit()\n",
    "            table = sm.stats.anova_lm(results, typ=2)\n",
    "            pvalue = table['PR(>F)'][0]\n",
    "            pvaluelist.append(pvalue)\n",
    "    \n",
    "        df_cluster['p-values'] = pvaluelist\n",
    "        qv.qvalues(df_cluster, 'p-values', 'qvalues')\n",
    "        print(i)\n",
    "        clusterframes[i] = df_cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'4ER+':                                                         p-values  \\\n",
       " cGMP effects                                        1.927328e-72   \n",
       " Cell-cell junction organization                     3.934492e-63   \n",
       " Gamma-carboxylation of protein precursors           2.921318e-61   \n",
       " Transport of gamma-carboxylated protein precurs...  8.144980e-60   \n",
       " Regulation of commissural axon pathfinding by S...  1.301501e-59   \n",
       " Tight junction interactions                         1.315182e-59   \n",
       " Hyaluronan uptake and degradation                   8.799324e-59   \n",
       " Resolution of D-Loop Structures                     3.636943e-58   \n",
       " Resolution of D-loop Structures through Hollida...  4.092748e-58   \n",
       " Intrinsic Pathway of Fibrin Clot Formation          6.531631e-57   \n",
       " Formation of Fibrin Clot (Clotting Cascade)         3.388969e-56   \n",
       " Nitric oxide stimulates guanylate cyclase           1.028844e-55   \n",
       " Gamma-carboxylation, transport, and amino-termi...  2.547621e-55   \n",
       " Removal of aminoterminal propeptides from gamma...  8.142074e-54   \n",
       " KSRP (KHSRP) binds and destabilizes mRNA            1.333336e-52   \n",
       " HDR through Homologous Recombination (HRR)          1.443566e-52   \n",
       " Resolution of D-loop Structures through Synthes...  5.177097e-52   \n",
       " HDR through Single Strand Annealing (SSA)           8.239589e-52   \n",
       " Polo-like kinase mediated events                    1.061252e-50   \n",
       " G2/M DNA replication checkpoint                     4.885853e-50   \n",
       " Cam-PDE 1 activation                                5.167461e-50   \n",
       " Homologous DNA Pairing and Strand Exchange          6.801105e-50   \n",
       " Extension of Telomeres                              2.753468e-49   \n",
       " E2F mediated regulation of DNA replication          5.450636e-49   \n",
       " Presynaptic phase of homologous DNA pairing and...  7.731126e-49   \n",
       " Depolymerisation of the Nuclear Lamina              3.939777e-48   \n",
       " Sema3A PAK dependent Axon repulsion                 7.320351e-48   \n",
       " EPHA-mediated growth cone collapse                  8.730207e-48   \n",
       " HDR through MMEJ (alt-NHEJ)                         1.227746e-47   \n",
       " Extrinsic Pathway of Fibrin Clot Formation          3.323719e-47   \n",
       " ...                                                          ...   \n",
       " Biosynthesis of DHA-derived sulfido conjugates      8.859045e-01   \n",
       " Biosynthesis of maresin conjugates in tissue re...  8.859055e-01   \n",
       " Phenylalanine and tyrosine metabolism               8.914937e-01   \n",
       " CLEC7A (Dectin-1) signaling                         8.959763e-01   \n",
       " Relaxin receptors                                   8.974814e-01   \n",
       " Josephin domain DUBs                                9.081831e-01   \n",
       " Cargo trafficking to the periciliary membrane       9.122261e-01   \n",
       " Peroxisomal lipid metabolism                        9.134744e-01   \n",
       " Orexin and neuropeptides FF and QRFP bind to th...  9.144818e-01   \n",
       " Toxicity of botulinum toxin type B (BoNT/B)         9.146443e-01   \n",
       " Serotonin and melatonin biosynthesis                9.229703e-01   \n",
       " Platelet Adhesion to exposed collagen               9.287677e-01   \n",
       " Activation of AMPK downstream of NMDARs             9.330878e-01   \n",
       " Signaling by NOTCH1                                 9.344067e-01   \n",
       " Biosynthesis of protectin and resolvin conjugat...  9.457690e-01   \n",
       " Signaling by Insulin receptor                       9.497160e-01   \n",
       " Olfactory Signaling Pathway                         9.503542e-01   \n",
       " Formation of the active cofactor, UDP-glucuronate   9.540020e-01   \n",
       " Ras activation upon Ca2+ influx through NMDA re...  9.655158e-01   \n",
       " DCC mediated attractive signaling                   9.677950e-01   \n",
       " Interleukin-1 family signaling                      9.708129e-01   \n",
       " Defective GCLC causes Hemolytic anemia due to g...  9.744829e-01   \n",
       " TICAM1, RIP1-mediated IKK complex recruitment       9.748136e-01   \n",
       " Gap junction trafficking                            9.867587e-01   \n",
       " Bicarbonate transporters                            9.898671e-01   \n",
       " Citric acid cycle (TCA cycle)                       9.917332e-01   \n",
       " SDK interactions                                    9.966741e-01   \n",
       " Defective CUBN causes hereditary megaloblastic ...  9.969287e-01   \n",
       " Defective AMN causes hereditary megaloblastic a...  9.969315e-01   \n",
       " FGFR1 ligand binding and activation                 9.989961e-01   \n",
       " \n",
       "                                                     cluster 4ER+ qvalues  \n",
       " cGMP effects                                                3.997278e-69  \n",
       " Cell-cell junction organization                             4.080068e-60  \n",
       " Gamma-carboxylation of protein precursors                   2.019604e-58  \n",
       " Transport of gamma-carboxylated protein precurs...          4.223172e-57  \n",
       " Regulation of commissural axon pathfinding by S...          4.546147e-57  \n",
       " Tight junction interactions                                 4.546147e-57  \n",
       " Hyaluronan uptake and degradation                           2.607114e-56  \n",
       " Resolution of D-Loop Structures                             9.428774e-56  \n",
       " Resolution of D-loop Structures through Hollida...          9.431510e-56  \n",
       " Intrinsic Pathway of Fibrin Clot Formation                  1.354660e-54  \n",
       " Formation of Fibrin Clot (Clotting Cascade)                 6.389747e-54  \n",
       " Nitric oxide stimulates guanylate cyclase                   1.778185e-53  \n",
       " Gamma-carboxylation, transport, and amino-termi...          4.064435e-53  \n",
       " Removal of aminoterminal propeptides from gamma...          1.206190e-51  \n",
       " KSRP (KHSRP) binds and destabilizes mRNA                    1.843559e-50  \n",
       " HDR through Homologous Recombination (HRR)                  1.871222e-50  \n",
       " Resolution of D-loop Structures through Synthes...          6.316059e-50  \n",
       " HDR through Single Strand Annealing (SSA)                   9.493837e-50  \n",
       " Polo-like kinase mediated events                            1.158441e-48  \n",
       " G2/M DNA replication checkpoint                             5.066629e-48  \n",
       " Cam-PDE 1 activation                                        5.103483e-48  \n",
       " Homologous DNA Pairing and Strand Exchange                  6.411587e-48  \n",
       " Extension of Telomeres                                      2.482910e-47  \n",
       " E2F mediated regulation of DNA replication                  4.710258e-47  \n",
       " Presynaptic phase of homologous DNA pairing and...          6.413742e-47  \n",
       " Depolymerisation of the Nuclear Lamina                      3.142730e-46  \n",
       " Sema3A PAK dependent Axon repulsion                         5.623114e-46  \n",
       " EPHA-mediated growth cone collapse                          6.466589e-46  \n",
       " HDR through MMEJ (alt-NHEJ)                                 8.780503e-46  \n",
       " Extrinsic Pathway of Fibrin Clot Formation                  2.297798e-45  \n",
       " ...                                                                  ...  \n",
       " Biosynthesis of DHA-derived sulfido conjugates              8.980294e-01  \n",
       " Biosynthesis of maresin conjugates in tissue re...          8.980294e-01  \n",
       " Phenylalanine and tyrosine metabolism                       9.032525e-01  \n",
       " CLEC7A (Dectin-1) signaling                                 9.073510e-01  \n",
       " Relaxin receptors                                           9.084317e-01  \n",
       " Josephin domain DUBs                                        9.188155e-01  \n",
       " Cargo trafficking to the periciliary membrane               9.224559e-01  \n",
       " Peroxisomal lipid metabolism                                9.232679e-01  \n",
       " Orexin and neuropeptides FF and QRFP bind to th...          9.235503e-01  \n",
       " Toxicity of botulinum toxin type B (BoNT/B)                 9.235503e-01  \n",
       " Serotonin and melatonin biosynthesis                        9.315039e-01  \n",
       " Platelet Adhesion to exposed collagen                       9.368990e-01  \n",
       " Activation of AMPK downstream of NMDARs                     9.407993e-01  \n",
       " Signaling by NOTCH1                                         9.416713e-01  \n",
       " Biosynthesis of protectin and resolvin conjugat...          9.526591e-01  \n",
       " Signaling by Insulin receptor                               9.561703e-01  \n",
       " Olfactory Signaling Pathway                                 9.563487e-01  \n",
       " Formation of the active cofactor, UDP-glucuronate           9.595539e-01  \n",
       " Ras activation upon Ca2+ influx through NMDA re...          9.706639e-01  \n",
       " DCC mediated attractive signaling                           9.724839e-01  \n",
       " Interleukin-1 family signaling                              9.750441e-01  \n",
       " Defective GCLC causes Hemolytic anemia due to g...          9.781149e-01  \n",
       " TICAM1, RIP1-mediated IKK complex recruitment               9.781149e-01  \n",
       " Gap junction trafficking                                    9.896216e-01  \n",
       " Bicarbonate transporters                                    9.922592e-01  \n",
       " Citric acid cycle (TCA cycle)                               9.936496e-01  \n",
       " SDK interactions                                            9.974124e-01  \n",
       " Defective CUBN causes hereditary megaloblastic ...          9.974124e-01  \n",
       " Defective AMN causes hereditary megaloblastic a...          9.974124e-01  \n",
       " FGFR1 ligand binding and activation                         9.989961e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '9':                                                         p-values  \\\n",
       " Vitamin B2 (riboflavin) metabolism                  3.948892e-68   \n",
       " Membrane binding and targetting of GAG proteins     3.991672e-66   \n",
       " Synthesis And Processing Of GAG, GAGPOL Polypro...  3.994236e-66   \n",
       " Glutamate and glutamine metabolism                  1.304903e-51   \n",
       " Synthesis of PS                                     1.517347e-49   \n",
       " KSRP (KHSRP) binds and destabilizes mRNA            4.268898e-46   \n",
       " Assembly Of The HIV Virion                          4.627320e-45   \n",
       " Tristetraprolin (TTP, ZFP36) binds and destabil...  4.946865e-39   \n",
       " Butyrate Response Factor 1 (BRF1) binds and des...  3.651237e-38   \n",
       " Protein repair                                      2.988241e-23   \n",
       " GDP-fucose biosynthesis                             6.627830e-23   \n",
       " Sema3A PAK dependent Axon repulsion                 2.510037e-20   \n",
       " Processing of Intronless Pre-mRNAs                  4.906374e-20   \n",
       " Telomere Extension By Telomerase                    8.258695e-20   \n",
       " Mitotic Telophase/Cytokinesis                       7.572306e-18   \n",
       " HDR through MMEJ (alt-NHEJ)                         1.477254e-16   \n",
       " RNA Polymerase III Chain Elongation                 7.498774e-16   \n",
       " Phosphorylation of proteins involved in G1/S tr...  1.215504e-15   \n",
       " Cooperation of Prefoldin and TriC/CCT  in actin...  1.512143e-15   \n",
       " Prefoldin mediated transfer of substrate  to CC...  1.725364e-15   \n",
       " APC/C:Cdc20 mediated degradation of Cyclin B        2.139594e-15   \n",
       " Retinoid metabolism and transport                   3.184889e-15   \n",
       " Metabolism of fat-soluble vitamins                  3.445194e-15   \n",
       " CRMPs in Sema3A signaling                           3.736259e-15   \n",
       " Cyclin A/B1/B2 associated events during G2/M tr...  7.450524e-15   \n",
       " Negative regulation of FGFR1 signaling              2.241511e-14   \n",
       " Visual phototransduction                            4.485357e-14   \n",
       " Negative regulation of FGFR3 signaling              4.774234e-14   \n",
       " Synthesis of active ubiquitin: roles of E1 and ...  6.919105e-14   \n",
       " Type I hemidesmosome assembly                       7.206938e-14   \n",
       " ...                                                          ...   \n",
       " Events associated with phagocytolytic activity ...  9.422749e-01   \n",
       " Antigen processing-Cross presentation               9.429062e-01   \n",
       " Regulated Necrosis                                  9.446099e-01   \n",
       " RIPK1-mediated regulated necrosis                   9.446259e-01   \n",
       " Post NMDA receptor activation events                9.448925e-01   \n",
       " Unfolded Protein Response (UPR)                     9.449588e-01   \n",
       " IKBKG deficiency causes anhidrotic ectodermal d...  9.465572e-01   \n",
       " IKBKB deficiency causes SCID                        9.465577e-01   \n",
       " CTLA4 inhibitory signaling                          9.479387e-01   \n",
       " Intrinsic Pathway for Apoptosis                     9.482367e-01   \n",
       " Defective ABCA1 causes Tangier disease              9.562203e-01   \n",
       " Intestinal hexose absorption                        9.572481e-01   \n",
       " HS-GAG biosynthesis                                 9.595522e-01   \n",
       " Phospholipase C-mediated cascade: FGFR1             9.598675e-01   \n",
       " FCGR activation                                     9.632201e-01   \n",
       " RNF mutants show enhanced WNT signaling and pro...  9.691918e-01   \n",
       " NOD1/2 Signaling Pathway                            9.755379e-01   \n",
       " Creation of C4 and C2 activators                    9.807417e-01   \n",
       " The citric acid (TCA) cycle and respiratory ele...  9.812124e-01   \n",
       " Intestinal saccharidase deficiencies                9.831065e-01   \n",
       " Sphingolipid metabolism                             9.843183e-01   \n",
       " Type II Na+/Pi cotransporters                       9.877599e-01   \n",
       " Synthesis of glycosylphosphatidylinositol (GPI)     9.882160e-01   \n",
       " TNF signaling                                       9.890175e-01   \n",
       " Formation of Senescence-Associated Heterochroma...  9.894104e-01   \n",
       " Tachykinin receptors bind tachykinins               9.963387e-01   \n",
       " Melanin biosynthesis                                9.981244e-01   \n",
       " Josephin domain DUBs                                9.983286e-01   \n",
       " ER-Phagosome pathway                                9.986797e-01   \n",
       " CD22 mediated BCR regulation                        9.993704e-01   \n",
       " \n",
       "                                                     cluster 9 qvalues  \n",
       " Vitamin B2 (riboflavin) metabolism                       8.190002e-65  \n",
       " Membrane binding and targetting of GAG proteins          2.761348e-63  \n",
       " Synthesis And Processing Of GAG, GAGPOL Polypro...       2.761348e-63  \n",
       " Glutamate and glutamine metabolism                       6.765921e-49  \n",
       " Synthesis of PS                                          6.293957e-47  \n",
       " KSRP (KHSRP) binds and destabilizes mRNA                 1.475616e-43  \n",
       " Assembly Of The HIV Virion                               1.371009e-42  \n",
       " Tristetraprolin (TTP, ZFP36) binds and destabil...       1.282475e-36  \n",
       " Butyrate Response Factor 1 (BRF1) binds and des...       8.414073e-36  \n",
       " Protein repair                                           6.197613e-21  \n",
       " GDP-fucose biosynthesis                                  1.249647e-20  \n",
       " Sema3A PAK dependent Axon repulsion                      4.338181e-18  \n",
       " Processing of Intronless Pre-mRNAs                       7.827554e-18  \n",
       " Telomere Extension By Telomerase                         1.223467e-17  \n",
       " Mitotic Telophase/Cytokinesis                            1.046997e-15  \n",
       " HDR through MMEJ (alt-NHEJ)                              1.914890e-14  \n",
       " RNA Polymerase III Chain Elongation                      9.148505e-14  \n",
       " Phosphorylation of proteins involved in G1/S tr...       1.400531e-13  \n",
       " Cooperation of Prefoldin and TriC/CCT  in actin...       1.650623e-13  \n",
       " Prefoldin mediated transfer of substrate  to CC...       1.789203e-13  \n",
       " APC/C:Cdc20 mediated degradation of Cyclin B             2.113104e-13  \n",
       " Retinoid metabolism and transport                        3.002482e-13  \n",
       " Metabolism of fat-soluble vitamins                       3.106666e-13  \n",
       " CRMPs in Sema3A signaling                                3.228751e-13  \n",
       " Cyclin A/B1/B2 associated events during G2/M tr...       6.180955e-13  \n",
       " Negative regulation of FGFR1 signaling                   1.788036e-12  \n",
       " Visual phototransduction                                 3.445418e-12  \n",
       " Negative regulation of FGFR3 signaling                   3.536343e-12  \n",
       " Synthesis of active ubiquitin: roles of E1 and ...       4.948353e-12  \n",
       " Type I hemidesmosome assembly                            4.982396e-12  \n",
       " ...                                                               ...  \n",
       " Events associated with phagocytolytic activity ...       9.556372e-01  \n",
       " Antigen processing-Cross presentation                    9.558101e-01  \n",
       " Regulated Necrosis                                       9.560217e-01  \n",
       " RIPK1-mediated regulated necrosis                        9.560217e-01  \n",
       " Post NMDA receptor activation events                     9.560217e-01  \n",
       " Unfolded Protein Response (UPR)                          9.560217e-01  \n",
       " IKBKG deficiency causes anhidrotic ectodermal d...       9.567060e-01  \n",
       " IKBKB deficiency causes SCID                             9.567060e-01  \n",
       " CTLA4 inhibitory signaling                               9.574698e-01  \n",
       " Intrinsic Pathway for Apoptosis                          9.574698e-01  \n",
       " Defective ABCA1 causes Tangier disease                   9.650612e-01  \n",
       " Intestinal hexose absorption                             9.656287e-01  \n",
       " HS-GAG biosynthesis                                      9.673300e-01  \n",
       " Phospholipase C-mediated cascade: FGFR1                  9.673300e-01  \n",
       " FCGR activation                                          9.702372e-01  \n",
       " RNF mutants show enhanced WNT signaling and pro...       9.757785e-01  \n",
       " NOD1/2 Signaling Pathway                                 9.816912e-01  \n",
       " Creation of C4 and C2 activators                         9.864442e-01  \n",
       " The citric acid (TCA) cycle and respiratory ele...       9.864442e-01  \n",
       " Intestinal saccharidase deficiencies                     9.878696e-01  \n",
       " Sphingolipid metabolism                                  9.886083e-01  \n",
       " Type II Na+/Pi cotransporters                            9.915627e-01  \n",
       " Synthesis of glycosylphosphatidylinositol (GPI)          9.915627e-01  \n",
       " TNF signaling                                            9.918014e-01  \n",
       " Formation of Senescence-Associated Heterochroma...       9.918014e-01  \n",
       " Tachykinin receptors bind tachykinins                    9.982639e-01  \n",
       " Melanin biosynthesis                                     9.991615e-01  \n",
       " Josephin domain DUBs                                     9.991615e-01  \n",
       " ER-Phagosome pathway                                     9.991615e-01  \n",
       " CD22 mediated BCR regulation                             9.993704e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '7':                                                         p-values  \\\n",
       " Ion channel transport                               1.640604e-38   \n",
       " Cytosolic iron-sulfur cluster assembly              1.824016e-37   \n",
       " Activation of NMDA receptors and postsynaptic e...  8.506606e-32   \n",
       " Ubiquinol biosynthesis                              9.629594e-32   \n",
       " Post NMDA receptor activation events                1.584346e-31   \n",
       " Cargo trafficking to the periciliary membrane       1.821808e-31   \n",
       " DAG and IP3 signaling                               4.097831e-31   \n",
       " RUNX1 regulates estrogen receptor mediated tran...  5.704728e-31   \n",
       " RAB geranylgeranylation                             4.967613e-30   \n",
       " Calmodulin induced events                           7.465837e-30   \n",
       " CaM pathway                                         7.486654e-30   \n",
       " Ca-dependent events                                 1.374186e-29   \n",
       " Intraflagellar transport                            2.137824e-29   \n",
       " BBSome-mediated cargo-targeting to cilium           2.157409e-29   \n",
       " Amino acids regulate mTORC1                         6.445892e-29   \n",
       " PLC beta mediated events                            1.203616e-28   \n",
       " RUNX1 regulates transcription of genes involved...  1.433393e-28   \n",
       " G-protein mediated events                           5.502056e-28   \n",
       " Acyl chain remodelling of PE                        6.932745e-28   \n",
       " Carboxyterminal post-translational modification...  2.632075e-26   \n",
       " Metabolism of vitamins and cofactors                3.769045e-26   \n",
       " Metabolism of nucleotides                           4.895072e-26   \n",
       " Signaling by Insulin receptor                       5.439869e-26   \n",
       " Cytosolic sulfonation of small molecules            9.713389e-26   \n",
       " Metabolism of cofactors                             1.158535e-25   \n",
       " Stimuli-sensing channels                            2.039599e-25   \n",
       " Lipid particle organization                         2.897295e-25   \n",
       " PKA-mediated phosphorylation of CREB                3.405077e-25   \n",
       " ROS and RNS production in phagocytes                3.882821e-25   \n",
       " Iron uptake and transport                           2.367555e-24   \n",
       " ...                                                          ...   \n",
       " PLCG1 events in ERBB2 signaling                     9.400478e-01   \n",
       " Conjugation of benzoate with glycine                9.453368e-01   \n",
       " RHO GTPases regulate CFTR trafficking               9.479479e-01   \n",
       " Transcription of the HIV genome                     9.480759e-01   \n",
       " NF-kB is activated and signals survival             9.499303e-01   \n",
       " Amine ligand-binding receptors                      9.525666e-01   \n",
       " Formation of RNA Pol II elongation complex          9.556048e-01   \n",
       " RNA Polymerase II Transcription Elongation          9.556053e-01   \n",
       " IRAK2 mediated activation of TAK1 complex           9.564849e-01   \n",
       " Butyrate Response Factor 1 (BRF1) binds and des...  9.578173e-01   \n",
       " AXIN mutants destabilize the destruction comple...  9.584161e-01   \n",
       " APC truncation mutants have impaired AXIN binding   9.584234e-01   \n",
       " truncated APC mutants destabilize the destructi...  9.584282e-01   \n",
       " AMER1 mutants destabilize the destruction complex   9.584303e-01   \n",
       " Truncations of AMER1 destabilize the destructio...  9.584336e-01   \n",
       " AXIN missense mutants destabilize the destructi...  9.584421e-01   \n",
       " Mitochondrial iron-sulfur cluster biogenesis        9.587529e-01   \n",
       " TP53 regulates transcription of additional cell...  9.606462e-01   \n",
       " TP53 regulates transcription of several additio...  9.614567e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...  9.672792e-01   \n",
       " Signalling to STAT3                                 9.686440e-01   \n",
       " CYP2E1 reactions                                    9.707636e-01   \n",
       " Transcriptional Regulation by VENTX                 9.767974e-01   \n",
       " Zinc influx into cells by the SLC39 gene family     9.885994e-01   \n",
       " PKA activation in glucagon signalling               9.901310e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...  9.939880e-01   \n",
       " Diseases associated with visual transduction        9.970615e-01   \n",
       " Defective Mismatch Repair Associated With MSH6      9.979216e-01   \n",
       " Retinoid cycle disease events                       9.983375e-01   \n",
       " Nuclear Receptor transcription pathway              9.994275e-01   \n",
       " \n",
       "                                                     cluster 7 qvalues  \n",
       " Ion channel transport                                    3.402613e-35  \n",
       " Cytosolic iron-sulfur cluster assembly                   1.891505e-34  \n",
       " Activation of NMDA receptors and postsynaptic e...       4.992945e-29  \n",
       " Ubiquinol biosynthesis                                   4.992945e-29  \n",
       " Post NMDA receptor activation events                     6.297383e-29  \n",
       " Cargo trafficking to the periciliary membrane            6.297383e-29  \n",
       " DAG and IP3 signaling                                    1.214129e-28  \n",
       " RUNX1 regulates estrogen receptor mediated tran...       1.478951e-28  \n",
       " RAB geranylgeranylation                                  1.144759e-27  \n",
       " Calmodulin induced events                                1.411575e-27  \n",
       " CaM pathway                                              1.411575e-27  \n",
       " Ca-dependent events                                      2.375051e-27  \n",
       " Intraflagellar transport                                 3.196047e-27  \n",
       " BBSome-mediated cargo-targeting to cilium                3.196047e-27  \n",
       " Amino acids regulate mTORC1                              8.912520e-27  \n",
       " PLC beta mediated events                                 1.560188e-26  \n",
       " RUNX1 regulates transcription of genes involved...       1.748740e-26  \n",
       " G-protein mediated events                                6.339591e-26  \n",
       " Acyl chain remodelling of PE                             7.567639e-26  \n",
       " Carboxyterminal post-translational modification...       2.729462e-24  \n",
       " Metabolism of vitamins and cofactors                     3.722380e-24  \n",
       " Metabolism of nucleotides                                4.614718e-24  \n",
       " Signaling by Insulin receptor                            4.905342e-24  \n",
       " Cytosolic sulfonation of small molecules                 8.393987e-24  \n",
       " Metabolism of cofactors                                  9.611204e-24  \n",
       " Stimuli-sensing channels                                 1.626972e-23  \n",
       " Lipid particle organization                              2.225552e-23  \n",
       " PKA-mediated phosphorylation of CREB                     2.522190e-23  \n",
       " ROS and RNS production in phagocytes                     2.776886e-23  \n",
       " Iron uptake and transport                                1.636770e-22  \n",
       " ...                                                               ...  \n",
       " PLCG1 events in ERBB2 signaling                          9.533785e-01  \n",
       " Conjugation of benzoate with glycine                     9.582739e-01  \n",
       " RHO GTPases regulate CFTR trafficking                    9.601120e-01  \n",
       " Transcription of the HIV genome                          9.601120e-01  \n",
       " NF-kB is activated and signals survival                  9.615205e-01  \n",
       " Amine ligand-binding receptors                           9.637186e-01  \n",
       " Formation of RNA Pol II elongation complex               9.648003e-01  \n",
       " RNA Polymerase II Transcription Elongation               9.648003e-01  \n",
       " IRAK2 mediated activation of TAK1 complex                9.648003e-01  \n",
       " Butyrate Response Factor 1 (BRF1) binds and des...       9.648003e-01  \n",
       " AXIN mutants destabilize the destruction comple...       9.648003e-01  \n",
       " APC truncation mutants have impaired AXIN binding        9.648003e-01  \n",
       " truncated APC mutants destabilize the destructi...       9.648003e-01  \n",
       " AMER1 mutants destabilize the destruction complex        9.648003e-01  \n",
       " Truncations of AMER1 destabilize the destructio...       9.648003e-01  \n",
       " AXIN missense mutants destabilize the destructi...       9.648003e-01  \n",
       " Mitochondrial iron-sulfur cluster biogenesis             9.648003e-01  \n",
       " TP53 regulates transcription of additional cell...       9.662368e-01  \n",
       " TP53 regulates transcription of several additio...       9.665832e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...       9.719657e-01  \n",
       " Signalling to STAT3                                      9.728657e-01  \n",
       " CYP2E1 reactions                                         9.745226e-01  \n",
       " Transcriptional Regulation by VENTX                      9.801053e-01  \n",
       " Zinc influx into cells by the SLC39 gene family          9.914676e-01  \n",
       " PKA activation in glucagon signalling                    9.925237e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...       9.959087e-01  \n",
       " Diseases associated with visual transduction             9.985059e-01  \n",
       " Defective Mismatch Repair Associated With MSH6           9.988191e-01  \n",
       " Retinoid cycle disease events                            9.988191e-01  \n",
       " Nuclear Receptor transcription pathway                   9.994275e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '3':                                                         p-values  \\\n",
       " Factors involved in megakaryocyte development a...  4.135886e-50   \n",
       " Cyclin A/B1/B2 associated events during G2/M tr...  1.957121e-49   \n",
       " KSRP (KHSRP) binds and destabilizes mRNA            4.755711e-49   \n",
       " COPI-dependent Golgi-to-ER retrograde traffic       2.893104e-47   \n",
       " Kinesins                                            2.993898e-47   \n",
       " Mitotic Telophase/Cytokinesis                       8.032517e-47   \n",
       " Unwinding of DNA                                    1.495898e-44   \n",
       " alpha-linolenic acid (ALA) metabolism               3.213107e-44   \n",
       " alpha-linolenic (omega3) and linoleic (omega6) ...  3.213185e-44   \n",
       " Cell Cycle, Mitotic                                 4.159790e-43   \n",
       " RHO GTPases Activate Formins                        4.676564e-43   \n",
       " Cell Cycle                                          4.715948e-43   \n",
       " Nucleotide salvage                                  6.432860e-43   \n",
       " Mitotic Prometaphase                                7.534139e-43   \n",
       " Polo-like kinase mediated events                    1.031463e-42   \n",
       " Mitotic Spindle Checkpoint                          1.237966e-42   \n",
       " Vitamin B2 (riboflavin) metabolism                  2.710062e-42   \n",
       " EML4 and NUDC in mitotic spindle formation          2.883703e-42   \n",
       " Resolution of Sister Chromatid Cohesion             3.012956e-42   \n",
       " Amplification of signal from the kinetochores       9.527618e-42   \n",
       " Amplification  of signal from unattached  kinet...  9.527771e-42   \n",
       " Golgi-to-ER retrograde transport                    1.379958e-41   \n",
       " DNA strand elongation                               4.934838e-41   \n",
       " G1/S-Specific Transcription                         9.032238e-41   \n",
       " Activation of ATR in response to replication st...  5.293577e-40   \n",
       " Resolution of D-Loop Structures                     6.661439e-40   \n",
       " Resolution of D-loop Structures through Hollida...  1.017519e-39   \n",
       " Phosphorylation of Emi1                             1.880399e-39   \n",
       " Regulation of TP53 Activity through Phosphoryla...  2.206834e-39   \n",
       " Activation of the pre-replicative complex           2.922633e-39   \n",
       " ...                                                          ...   \n",
       " Downregulation of ERBB2:ERBB3 signaling             9.509402e-01   \n",
       " Xenobiotics                                         9.519978e-01   \n",
       " Activated NTRK3 signals through RAS                 9.523156e-01   \n",
       " Synthesis of IP3 and IP4 in the cytosol             9.535863e-01   \n",
       " Signaling by NOTCH3                                 9.550203e-01   \n",
       " mRNA 3'-end processing                              9.558731e-01   \n",
       " Transcriptional activation of mitochondrial bio...  9.585961e-01   \n",
       " Regulation of lipid metabolism by PPARalpha         9.590037e-01   \n",
       " Diseases of carbohydrate metabolism                 9.609743e-01   \n",
       " PPARA activates gene expression                     9.612673e-01   \n",
       " Orexin and neuropeptides FF and QRFP bind to th...  9.640987e-01   \n",
       " Activation of SMO                                   9.702354e-01   \n",
       " Vasopressin-like receptors                          9.739473e-01   \n",
       " Glutamate Neurotransmitter Release Cycle            9.751637e-01   \n",
       " Defective C1GALT1C1 causes Tn polyagglutination...  9.770339e-01   \n",
       " Defective GALNT3 causes familial hyperphosphate...  9.793373e-01   \n",
       " ROS and RNS production in phagocytes                9.800587e-01   \n",
       " Defective ACTH causes Obesity and Pro-opiomelan...  9.806185e-01   \n",
       " Signaling by NTRK3 (TRKC)                           9.822851e-01   \n",
       " SHC-related events triggered by IGF1R               9.833145e-01   \n",
       " Signaling by NTRK2 (TRKB)                           9.838959e-01   \n",
       " Diseases associated with surfactant metabolism      9.848415e-01   \n",
       " POU5F1 (OCT4), SOX2, NANOG activate genes relat...  9.857551e-01   \n",
       " SUMO is transferred from E1 to E2 (UBE2I, UBC9)     9.875113e-01   \n",
       " MTF1 activates gene expression                      9.895293e-01   \n",
       " E3 ubiquitin ligases ubiquitinate target proteins   9.929483e-01   \n",
       " Myogenesis                                          9.942931e-01   \n",
       " GRB2 events in ERBB2 signaling                      9.948973e-01   \n",
       " Defective SLC3A1 causes cystinuria (CSNU)           9.952553e-01   \n",
       " Defective SLC7A9 causes cystinuria (CSNU)           9.952817e-01   \n",
       " \n",
       "                                                     cluster 3 qvalues  \n",
       " Factors involved in megakaryocyte development a...       8.577828e-47  \n",
       " Cyclin A/B1/B2 associated events during G2/M tr...       2.029535e-46  \n",
       " KSRP (KHSRP) binds and destabilizes mRNA                 3.287781e-46  \n",
       " COPI-dependent Golgi-to-ER retrograde traffic            1.241869e-44  \n",
       " Kinesins                                                 1.241869e-44  \n",
       " Mitotic Telophase/Cytokinesis                            2.776574e-44  \n",
       " Unwinding of DNA                                         4.432131e-42  \n",
       " alpha-linolenic acid (ALA) metabolism                    7.404605e-42  \n",
       " alpha-linolenic (omega3) and linoleic (omega6) ...       7.404605e-42  \n",
       " Cell Cycle, Mitotic                                      8.150731e-41  \n",
       " RHO GTPases Activate Formins                             8.150731e-41  \n",
       " Cell Cycle                                               8.150731e-41  \n",
       " Nucleotide salvage                                       1.026289e-40  \n",
       " Mitotic Prometaphase                                     1.116129e-40  \n",
       " Polo-like kinase mediated events                         1.426169e-40  \n",
       " Mitotic Spindle Checkpoint                               1.604714e-40  \n",
       " Vitamin B2 (riboflavin) metabolism                       3.288879e-40  \n",
       " EML4 and NUDC in mitotic spindle formation               3.288879e-40  \n",
       " Resolution of Sister Chromatid Cohesion                  3.288879e-40  \n",
       " Amplification of signal from the kinetochores            9.409808e-40  \n",
       " Amplification  of signal from unattached  kinet...       9.409808e-40  \n",
       " Golgi-to-ER retrograde transport                         1.300924e-39  \n",
       " DNA strand elongation                                    4.449936e-39  \n",
       " G1/S-Specific Transcription                              7.805359e-39  \n",
       " Activation of ATR in response to replication st...       4.391551e-38  \n",
       " Resolution of D-Loop Structures                          5.313778e-38  \n",
       " Resolution of D-loop Structures through Hollida...       7.816052e-38  \n",
       " Phosphorylation of Emi1                                  1.392838e-37  \n",
       " Regulation of TP53 Activity through Phosphoryla...       1.578267e-37  \n",
       " Activation of the pre-replicative complex                2.020514e-37  \n",
       " ...                                                               ...  \n",
       " Downregulation of ERBB2:ERBB3 signaling                  9.644254e-01  \n",
       " Xenobiotics                                              9.648767e-01  \n",
       " Activated NTRK3 signals through RAS                      9.648767e-01  \n",
       " Synthesis of IP3 and IP4 in the cytosol                  9.656924e-01  \n",
       " Signaling by NOTCH3                                      9.666726e-01  \n",
       " mRNA 3'-end processing                                   9.670638e-01  \n",
       " Transcriptional activation of mitochondrial bio...       9.692854e-01  \n",
       " Regulation of lipid metabolism by PPARalpha              9.692854e-01  \n",
       " Diseases of carbohydrate metabolism                      9.706272e-01  \n",
       " PPARA activates gene expression                          9.706272e-01  \n",
       " Orexin and neuropeptides FF and QRFP bind to th...       9.730125e-01  \n",
       " Activation of SMO                                        9.787297e-01  \n",
       " Vasopressin-like receptors                               9.819965e-01  \n",
       " Glutamate Neurotransmitter Release Cycle                 9.827452e-01  \n",
       " Defective C1GALT1C1 causes Tn polyagglutination...       9.841517e-01  \n",
       " Defective GALNT3 causes familial hyperphosphate...       9.859930e-01  \n",
       " ROS and RNS production in phagocytes                     9.862405e-01  \n",
       " Defective ACTH causes Obesity and Pro-opiomelan...       9.863253e-01  \n",
       " Signaling by NTRK3 (TRKC)                                9.875227e-01  \n",
       " SHC-related events triggered by IGF1R                    9.880787e-01  \n",
       " Signaling by NTRK2 (TRKB)                                9.881840e-01  \n",
       " Diseases associated with surfactant metabolism           9.886551e-01  \n",
       " POU5F1 (OCT4), SOX2, NANOG activate genes relat...       9.890934e-01  \n",
       " SUMO is transferred from E1 to E2 (UBE2I, UBC9)          9.903764e-01  \n",
       " MTF1 activates gene expression                           9.919207e-01  \n",
       " E3 ubiquitin ligases ubiquitinate target proteins        9.948670e-01  \n",
       " Myogenesis                                               9.952817e-01  \n",
       " GRB2 events in ERBB2 signaling                           9.952817e-01  \n",
       " Defective SLC3A1 causes cystinuria (CSNU)                9.952817e-01  \n",
       " Defective SLC7A9 causes cystinuria (CSNU)                9.952817e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '8':                                                          p-values  \\\n",
       " Noncanonical activation of NOTCH3                   2.705516e-132   \n",
       " HHAT G278V abrogates palmitoylation of Hh-Np         6.439906e-89   \n",
       " Intraflagellar transport                             7.810098e-70   \n",
       " Synthesis of epoxy (EET) and dihydroxyeicosatri...   7.967127e-65   \n",
       " Sphingolipid metabolism                              7.632915e-63   \n",
       " Biosynthesis of the N-glycan precursor (dolicho...   5.679008e-60   \n",
       " GAB1 signalosome                                     6.749583e-60   \n",
       " Synthesis of substrates in N-glycan biosythesis      4.167203e-58   \n",
       " Metabolism of vitamins and cofactors                 2.677592e-57   \n",
       " Complement cascade                                   4.526283e-56   \n",
       " Regulation of Complement cascade                     5.470481e-56   \n",
       " Nucleobase catabolism                                6.816699e-56   \n",
       " Phase I - Functionalization of compounds             1.804039e-55   \n",
       " Glycosphingolipid metabolism                         1.482786e-54   \n",
       " Transport of bile salts and organic acids, meta...   1.625934e-53   \n",
       " SLC-mediated transmembrane transport                 2.403111e-53   \n",
       " Initial triggering of complement                     1.131702e-52   \n",
       " Sialic acid metabolism                               6.773265e-52   \n",
       " Nuclear signaling by ERBB4                           1.128254e-51   \n",
       " RUNX1 regulates transcription of genes involved...   2.216459e-51   \n",
       " Synthesis of PC                                      8.455407e-51   \n",
       " Activation of AMPK downstream of NMDARs              9.107337e-51   \n",
       " Metallothioneins bind metals                         1.242298e-50   \n",
       " Ubiquinol biosynthesis                               1.665840e-50   \n",
       " Dectin-2 family                                      7.919848e-50   \n",
       " Interleukin-12 signaling                             2.368283e-49   \n",
       " Response to metal ions                               6.907736e-49   \n",
       " Fcgamma receptor (FCGR) dependent phagocytosis       9.390850e-49   \n",
       " Growth hormone receptor signaling                    3.071657e-48   \n",
       " Carboxyterminal post-translational modification...   5.396248e-48   \n",
       " ...                                                           ...   \n",
       " Acrosome Reaction and Sperm:Oocyte Membrane Bin...   8.739063e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...   8.765359e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression to limit...   8.768751e-01   \n",
       " Signaling by NODAL                                   8.815603e-01   \n",
       " Proton-coupled monocarboxylate transport             8.880708e-01   \n",
       " Defective MMADHC causes methylmalonic aciduria ...   8.963631e-01   \n",
       " Positive epigenetic regulation of rRNA expression    9.077293e-01   \n",
       " Phospholipase C-mediated cascade; FGFR2              9.105103e-01   \n",
       " Beta defensins                                       9.120495e-01   \n",
       " Citric acid cycle (TCA cycle)                        9.145863e-01   \n",
       " Transcriptional activation of mitochondrial bio...   9.225251e-01   \n",
       " Axonal growth stimulation                            9.235720e-01   \n",
       " Regulation of cholesterol biosynthesis by SREBP...   9.241249e-01   \n",
       " Listeria monocytogenes entry into host cells         9.273376e-01   \n",
       " Reuptake of GABA                                     9.274893e-01   \n",
       " Inhibition of Host mRNA Processing and RNA Sile...   9.433511e-01   \n",
       " Dual Incision in GG-NER                              9.469723e-01   \n",
       " rRNA modification in the nucleus and cytosol         9.493211e-01   \n",
       " Synthesis of PIPs in the nucleus                     9.605797e-01   \n",
       " Deadenylation of mRNA                                9.643467e-01   \n",
       " Formation of the active cofactor, UDP-glucuronate    9.678308e-01   \n",
       " PI-3K cascade:FGFR2                                  9.710387e-01   \n",
       " Telomere Extension By Telomerase                     9.765480e-01   \n",
       " Asparagine N-linked glycosylation                    9.785510e-01   \n",
       " FGFR2c ligand binding and activation                 9.848400e-01   \n",
       " Nonhomologous End-Joining (NHEJ)                     9.904836e-01   \n",
       " Activated point mutants of FGFR2                     9.935883e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression to contr...   9.959878e-01   \n",
       " Defective ABCG5 causes sitosterolemia                9.995878e-01   \n",
       " Defective ABCG8 causes gallbladder disease 4 an...   9.995904e-01   \n",
       " \n",
       "                                                     cluster 8 qvalues  \n",
       " Noncanonical activation of NOTCH3                       5.611239e-129  \n",
       " HHAT G278V abrogates palmitoylation of Hh-Np             6.678182e-86  \n",
       " Intraflagellar transport                                 5.399381e-67  \n",
       " Synthesis of epoxy (EET) and dihydroxyeicosatri...       4.130955e-62  \n",
       " Sphingolipid metabolism                                  3.166133e-60  \n",
       " Biosynthesis of the N-glycan precursor (dolicho...       1.963044e-57  \n",
       " GAB1 signalosome                                         1.999805e-57  \n",
       " Synthesis of substrates in N-glycan biosythesis          1.080347e-55  \n",
       " Metabolism of vitamins and cofactors                     6.170362e-55  \n",
       " Complement cascade                                       9.387510e-54  \n",
       " Regulation of Complement cascade                         1.031434e-53  \n",
       " Nucleobase catabolism                                    1.178153e-53  \n",
       " Phase I - Functionalization of compounds                 2.878137e-53  \n",
       " Glycosphingolipid metabolism                             2.196641e-52  \n",
       " Transport of bile salts and organic acids, meta...       2.248125e-51  \n",
       " SLC-mediated transmembrane transport                     3.115032e-51  \n",
       " Initial triggering of complement                         1.380677e-50  \n",
       " Sialic acid metabolism                                   7.804307e-50  \n",
       " Nuclear signaling by ERBB4                               1.231578e-49  \n",
       " RUNX1 regulates transcription of genes involved...       2.298468e-49  \n",
       " Synthesis of PC                                          8.350721e-49  \n",
       " Activation of AMPK downstream of NMDARs                  8.585735e-49  \n",
       " Metallothioneins bind metals                             1.120229e-48  \n",
       " Ubiquinol biosynthesis                                   1.439563e-48  \n",
       " Dectin-2 family                                          6.570306e-48  \n",
       " Interleukin-12 signaling                                 1.889161e-47  \n",
       " Response to metal ions                                   5.306165e-47  \n",
       " Fcgamma receptor (FCGR) dependent phagocytosis           6.955937e-47  \n",
       " Growth hormone receptor signaling                        2.196764e-46  \n",
       " Carboxyterminal post-translational modification...       3.730606e-46  \n",
       " ...                                                               ...  \n",
       " Acrosome Reaction and Sperm:Oocyte Membrane Bin...       8.862991e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression linked t...       8.884411e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression to limit...       8.884411e-01  \n",
       " Signaling by NODAL                                       8.927520e-01  \n",
       " Proton-coupled monocarboxylate transport                 8.989062e-01  \n",
       " Defective MMADHC causes methylmalonic aciduria ...       9.068571e-01  \n",
       " Positive epigenetic regulation of rRNA expression        9.179086e-01  \n",
       " Phospholipase C-mediated cascade; FGFR2                  9.202721e-01  \n",
       " Beta defensins                                           9.213788e-01  \n",
       " Citric acid cycle (TCA cycle)                            9.234917e-01  \n",
       " Transcriptional activation of mitochondrial bio...       9.310546e-01  \n",
       " Axonal growth stimulation                                9.316578e-01  \n",
       " Regulation of cholesterol biosynthesis by SREBP...       9.317623e-01  \n",
       " Listeria monocytogenes entry into host cells             9.342462e-01  \n",
       " Reuptake of GABA                                         9.342462e-01  \n",
       " Inhibition of Host mRNA Processing and RNA Sile...       9.497623e-01  \n",
       " Dual Incision in GG-NER                                  9.529454e-01  \n",
       " rRNA modification in the nucleus and cytosol             9.548457e-01  \n",
       " Synthesis of PIPs in the nucleus                         9.657015e-01  \n",
       " Deadenylation of mRNA                                    9.690190e-01  \n",
       " Formation of the active cofactor, UDP-glucuronate        9.720490e-01  \n",
       " PI-3K cascade:FGFR2                                      9.747988e-01  \n",
       " Telomere Extension By Telomerase                         9.798551e-01  \n",
       " Asparagine N-linked glycosylation                        9.813901e-01  \n",
       " FGFR2c ligand binding and activation                     9.872200e-01  \n",
       " Nonhomologous End-Joining (NHEJ)                         9.923975e-01  \n",
       " Activated point mutants of FGFR2                         9.950276e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression to contr...       9.969492e-01  \n",
       " Defective ABCG5 causes sitosterolemia                    9.995904e-01  \n",
       " Defective ABCG8 causes gallbladder disease 4 an...       9.995904e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '10':                                                          p-values  \\\n",
       " Cyclin D associated events in G1                    1.427101e-263   \n",
       " G1 Phase                                            1.427198e-263   \n",
       " Diseases of Cellular Senescence                     1.376071e-243   \n",
       " Evasion of Oxidative Stress Induced Senescence ...  1.376196e-243   \n",
       " Evasion of Oncogene Induced Senescence Due to p...  1.376234e-243   \n",
       " Evasion of Oncogene Induced Senescence Due to D...  1.376276e-243   \n",
       " Evasion of Oxidative Stress Induced Senescence ...  1.376337e-243   \n",
       " Transcriptional Regulation by MECP2                 2.905553e-240   \n",
       " SLC-mediated transmembrane transport                5.027414e-218   \n",
       " Oncogene Induced Senescence                         2.611509e-216   \n",
       " Transport of bile salts and organic acids, meta...  1.455934e-209   \n",
       " SUMOylation of DNA replication proteins             1.056100e-205   \n",
       " Metabolism of nucleotides                           3.313264e-203   \n",
       " RHO GTPases Activate Rhotekin and Rhophilins        2.495662e-202   \n",
       " G2 Phase                                            1.316584e-200   \n",
       " SLC transporter disorders                           1.260639e-194   \n",
       " mRNA Splicing - Minor Pathway                       1.982378e-194   \n",
       " Serine biosynthesis                                 1.434324e-191   \n",
       " SUMO E3 ligases SUMOylate target proteins           1.755944e-190   \n",
       " Sphingolipid metabolism                             7.898832e-189   \n",
       " Synthesis of PC                                     9.179694e-189   \n",
       " RHO GTPases Activate Formins                        1.355443e-187   \n",
       " Glycosphingolipid metabolism                        2.111430e-185   \n",
       " Regulation of TP53 Activity through Phosphoryla...  2.292009e-184   \n",
       " Purine catabolism                                   5.535426e-184   \n",
       " Insulin-like Growth Factor-2 mRNA Binding Prote...  1.646116e-183   \n",
       " Transport of Mature mRNA Derived from an Intron...  5.751717e-183   \n",
       " Vpr-mediated nuclear import of PICs                 1.848215e-182   \n",
       " RHO GTPase Effectors                                4.304460e-182   \n",
       " Miscellaneous transport and binding events          6.789150e-182   \n",
       " ...                                                           ...   \n",
       " Defective CYP11A1 causes Adrenal insufficiency,...   8.705560e-01   \n",
       " HCN channels                                         8.751906e-01   \n",
       " RNA Polymerase I Promoter Escape                     8.818945e-01   \n",
       " Protein localization                                 8.892508e-01   \n",
       " Nephrin family interactions                          9.044421e-01   \n",
       " Senescence-Associated Secretory Phenotype (SASP)     9.120354e-01   \n",
       " RMTs methylate histone arginines                     9.138475e-01   \n",
       " HSP90 chaperone cycle for steroid hormone recep...   9.188744e-01   \n",
       " FGFR4 ligand binding and activation                  9.198857e-01   \n",
       " Retinoid cycle disease events                        9.215073e-01   \n",
       " Oxidative Stress Induced Senescence                  9.219946e-01   \n",
       " Activated NTRK2 signals through CDK5                 9.220643e-01   \n",
       " Diseases associated with visual transduction         9.221253e-01   \n",
       " Phospholipase C-mediated cascade; FGFR4              9.321284e-01   \n",
       " Regulation of glycolysis by fructose 2,6-bispho...   9.335735e-01   \n",
       " Degradation of cysteine and homocysteine             9.336197e-01   \n",
       " Synthesis of PIPs at the ER membrane                 9.391002e-01   \n",
       " Synthesis of PIPs at the late endosome membrane      9.429334e-01   \n",
       " Relaxin receptors                                    9.571103e-01   \n",
       " NoRC negatively regulates rRNA expression            9.588862e-01   \n",
       " The canonical retinoid cycle in rods (twilight ...   9.610922e-01   \n",
       " Negative epigenetic regulation of rRNA expression    9.632637e-01   \n",
       " Rhesus glycoproteins mediate ammonium transport.     9.651657e-01   \n",
       " Cell death signalling via NRAGE, NRIF and NADE       9.702176e-01   \n",
       " Regulation of gene expression in early pancreat...   9.758999e-01   \n",
       " RSK activation                                       9.788090e-01   \n",
       " RUNX2 regulates genes involved in differentiati...   9.863815e-01   \n",
       " Rap1 signalling                                      9.888607e-01   \n",
       " Cohesin Loading onto Chromatin                       9.959711e-01   \n",
       " Hyaluronan biosynthesis and export                   9.965646e-01   \n",
       " \n",
       "                                                     cluster 10 qvalues  \n",
       " Cyclin D associated events in G1                         1.480004e-260  \n",
       " G1 Phase                                                 1.480004e-260  \n",
       " Diseases of Cellular Senescence                          4.077889e-241  \n",
       " Evasion of Oxidative Stress Induced Senescence ...       4.077889e-241  \n",
       " Evasion of Oncogene Induced Senescence Due to p...       4.077889e-241  \n",
       " Evasion of Oncogene Induced Senescence Due to D...       4.077889e-241  \n",
       " Evasion of Oxidative Stress Induced Senescence ...       4.077889e-241  \n",
       " Transcriptional Regulation by MECP2                      7.532646e-238  \n",
       " SLC-mediated transmembrane transport                     1.158540e-215  \n",
       " Oncogene Induced Senescence                              5.416269e-214  \n",
       " Transport of bile salts and organic acids, meta...       2.745097e-207  \n",
       " SUMOylation of DNA replication proteins                  1.825292e-203  \n",
       " Metabolism of nucleotides                                5.285930e-201  \n",
       " RHO GTPases Activate Rhotekin and Rhophilins             3.697145e-200  \n",
       " G2 Phase                                                 1.820397e-198  \n",
       " SLC transporter disorders                                1.634103e-192  \n",
       " mRNA Splicing - Minor Pathway                            2.418501e-192  \n",
       " Serine biosynthesis                                      1.652660e-189  \n",
       " SUMO E3 ligases SUMOylate target proteins                1.916752e-188  \n",
       " Sphingolipid metabolism                                  8.191089e-187  \n",
       " Synthesis of PC                                          9.066041e-187  \n",
       " RHO GTPases Activate Formins                             1.277813e-185  \n",
       " Glycosphingolipid metabolism                             1.903959e-183  \n",
       " Regulation of TP53 Activity through Phosphoryla...       1.980678e-182  \n",
       " Purine catabolism                                        4.592189e-182  \n",
       " Insulin-like Growth Factor-2 mRNA Binding Prote...       1.313094e-181  \n",
       " Transport of Mature mRNA Derived from an Intron...       4.418171e-181  \n",
       " Vpr-mediated nuclear import of PICs                      1.368999e-180  \n",
       " RHO GTPase Effectors                                     3.078431e-180  \n",
       " Miscellaneous transport and binding events               4.693566e-180  \n",
       " ...                                                                ...  \n",
       " Defective CYP11A1 causes Adrenal insufficiency,...        8.829012e-01  \n",
       " HCN channels                                              8.871678e-01  \n",
       " RNA Polymerase I Promoter Escape                          8.935267e-01  \n",
       " Protein localization                                      9.005401e-01  \n",
       " Nephrin family interactions                               9.154773e-01  \n",
       " Senescence-Associated Secretory Phenotype (SASP)          9.227129e-01  \n",
       " RMTs methylate histone arginines                          9.240955e-01  \n",
       " HSP90 chaperone cycle for steroid hormone recep...        9.287259e-01  \n",
       " FGFR4 ligand binding and activation                       9.292952e-01  \n",
       " Retinoid cycle disease events                             9.297461e-01  \n",
       " Oxidative Stress Induced Senescence                       9.297461e-01  \n",
       " Activated NTRK2 signals through CDK5                      9.297461e-01  \n",
       " Diseases associated with visual transduction              9.297461e-01  \n",
       " Phospholipase C-mediated cascade; FGFR4                   9.393753e-01  \n",
       " Regulation of glycolysis by fructose 2,6-bispho...        9.399646e-01  \n",
       " Degradation of cysteine and homocysteine                  9.399646e-01  \n",
       " Synthesis of PIPs at the ER membrane                      9.450237e-01  \n",
       " Synthesis of PIPs at the late endosome membrane           9.484209e-01  \n",
       " Relaxin receptors                                         9.622136e-01  \n",
       " NoRC negatively regulates rRNA expression                 9.635320e-01  \n",
       " The canonical retinoid cycle in rods (twilight ...        9.652809e-01  \n",
       " Negative epigenetic regulation of rRNA expression         9.669936e-01  \n",
       " Rhesus glycoproteins mediate ammonium transport.          9.684343e-01  \n",
       " Cell death signalling via NRAGE, NRIF and NADE            9.730325e-01  \n",
       " Regulation of gene expression in early pancreat...        9.782583e-01  \n",
       " RSK activation                                            9.807004e-01  \n",
       " RUNX2 regulates genes involved in differentiati...        9.878104e-01  \n",
       " Rap1 signalling                                           9.898152e-01  \n",
       " Cohesin Loading onto Chromatin                            9.964515e-01  \n",
       " Hyaluronan biosynthesis and export                        9.965646e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '1':                                                         p-values  \\\n",
       " RMTs methylate histone arginines                    2.678034e-44   \n",
       " HDMs demethylate histones                           1.726380e-40   \n",
       " Senescence-Associated Secretory Phenotype (SASP)    6.854062e-40   \n",
       " Cellular Senescence                                 8.041793e-40   \n",
       " Nucleosome assembly                                 4.833426e-39   \n",
       " Deposition of new CENPA-containing nucleosomes ...  4.833462e-39   \n",
       " Chromosome Maintenance                              1.275674e-38   \n",
       " Estrogen-dependent gene expression                  2.784706e-38   \n",
       " Meiosis                                             7.138403e-38   \n",
       " Oxidative Stress Induced Senescence                 9.169580e-38   \n",
       " Reproduction                                        1.678011e-37   \n",
       " HCMV Late Events                                    3.518082e-37   \n",
       " Telomere Maintenance                                3.605096e-37   \n",
       " Meiotic recombination                               3.834539e-37   \n",
       " RHO GTPases activate PKNs                           4.352758e-37   \n",
       " Transcriptional regulation by small RNAs            4.790038e-37   \n",
       " HCMV Early Events                                   1.153335e-36   \n",
       " Activation of HOX genes during differentiation      1.157267e-36   \n",
       " Activation of anterior HOX genes in hindbrain d...  1.157274e-36   \n",
       " Condensation of Prophase Chromosomes                2.111715e-36   \n",
       " Transcriptional regulation of granulopoiesis        2.593317e-36   \n",
       " Metalloprotease DUBs                                4.207767e-36   \n",
       " Base Excision Repair                                5.440705e-36   \n",
       " Gene Silencing by RNA                               1.324682e-35   \n",
       " DNA methylation                                     2.041734e-35   \n",
       " Formation of the beta-catenin:TCF transactivati...  2.284391e-35   \n",
       " PRC2 methylates histones and DNA                    2.414402e-35   \n",
       " HDACs deacetylate histones                          2.878176e-35   \n",
       " Activated PKN1 stimulates transcription of AR (...  3.943624e-35   \n",
       " RNA Polymerase I Promoter Opening                   4.066430e-35   \n",
       " ...                                                          ...   \n",
       " Activation of Ca-permeable Kainate Receptor         9.673156e-01   \n",
       " Classical Kir channels                              9.684032e-01   \n",
       " Eicosanoids                                         9.689333e-01   \n",
       " Zinc transporters                                   9.700864e-01   \n",
       " Signaling by BRAF and RAF fusions                   9.701674e-01   \n",
       " Synthesis of PIPs at the early endosome membrane    9.728586e-01   \n",
       " Zinc influx into cells by the SLC39 gene family     9.749692e-01   \n",
       " CREB1 phosphorylation through NMDA receptor-med...  9.764244e-01   \n",
       " CYP2E1 reactions                                    9.766004e-01   \n",
       " Biosynthesis of maresin conjugates in tissue re...  9.767414e-01   \n",
       " Biosynthesis of DHA-derived sulfido conjugates      9.767422e-01   \n",
       " Nef mediated downregulation of MHC class I comp...  9.780690e-01   \n",
       " Creation of C4 and C2 activators                    9.784340e-01   \n",
       " Serotonin Neurotransmitter Release Cycle            9.792587e-01   \n",
       " HuR (ELAVL1) binds and stabilizes mRNA              9.809400e-01   \n",
       " Miscellaneous substrates                            9.821285e-01   \n",
       " Activated NTRK3 signals through RAS                 9.830366e-01   \n",
       " Classical antibody-mediated complement activation   9.849891e-01   \n",
       " Defective Mismatch Repair Associated With MSH2      9.852698e-01   \n",
       " IRAK2 mediated activation of TAK1 complex           9.854209e-01   \n",
       " SHC-mediated cascade:FGFR4                          9.870537e-01   \n",
       " Role of phospholipids in phagocytosis               9.883636e-01   \n",
       " Regulation of IFNA signaling                        9.890569e-01   \n",
       " Xenobiotics                                         9.890907e-01   \n",
       " Major pathway of rRNA processing in the nucleol...  9.922244e-01   \n",
       " 5-Phosphoribose 1-diphosphate biosynthesis          9.933281e-01   \n",
       " Trafficking and processing of endosomal TLR         9.973872e-01   \n",
       " CD28 dependent PI3K/Akt signaling                   9.980622e-01   \n",
       " Constitutive Signaling by AKT1 E17K in Cancer       9.988501e-01   \n",
       " Dectin-2 family                                     9.989896e-01   \n",
       " \n",
       "                                                     cluster 1 qvalues  \n",
       " RMTs methylate histone arginines                         5.554242e-41  \n",
       " HDMs demethylate histones                                1.790256e-37  \n",
       " Senescence-Associated Secretory Phenotype (SASP)         4.169670e-37  \n",
       " Cellular Senescence                                      4.169670e-37  \n",
       " Nucleosome assembly                                      1.670767e-36  \n",
       " Deposition of new CENPA-containing nucleosomes ...       1.670767e-36  \n",
       " Chromosome Maintenance                                   3.779639e-36  \n",
       " Estrogen-dependent gene expression                       7.219350e-36  \n",
       " Meiosis                                                  1.645005e-35  \n",
       " Oxidative Stress Induced Senescence                      1.901771e-35  \n",
       " Reproduction                                             3.163813e-35  \n",
       " HCMV Late Events                                         5.680596e-35  \n",
       " Telomere Maintenance                                     5.680596e-35  \n",
       " Meiotic recombination                                    5.680596e-35  \n",
       " RHO GTPases activate PKNs                                6.018413e-35  \n",
       " Transcriptional regulation by small RNAs                 6.209087e-35  \n",
       " HCMV Early Events                                        1.263256e-34  \n",
       " Activation of HOX genes during differentiation           1.263256e-34  \n",
       " Activation of anterior HOX genes in hindbrain d...       1.263256e-34  \n",
       " Condensation of Prophase Chromosomes                     2.189849e-34  \n",
       " Transcriptional regulation of granulopoiesis             2.561209e-34  \n",
       " Metalloprotease DUBs                                     3.966777e-34  \n",
       " Base Excision Repair                                     4.906097e-34  \n",
       " Gene Silencing by RNA                                    1.144746e-33  \n",
       " DNA methylation                                          1.693822e-33  \n",
       " Formation of the beta-catenin:TCF transactivati...       1.822241e-33  \n",
       " PRC2 methylates histones and DNA                         1.854619e-33  \n",
       " HDACs deacetylate histones                               2.131906e-33  \n",
       " Activated PKN1 stimulates transcription of AR (...       2.787751e-33  \n",
       " RNA Polymerase I Promoter Opening                        2.787751e-33  \n",
       " ...                                                               ...  \n",
       " Activation of Ca-permeable Kainate Receptor              9.810330e-01  \n",
       " Classical Kir channels                                   9.816560e-01  \n",
       " Eicosanoids                                              9.817135e-01  \n",
       " Zinc transporters                                        9.820045e-01  \n",
       " Signaling by BRAF and RAF fusions                        9.820045e-01  \n",
       " Synthesis of PIPs at the early endosome membrane         9.842482e-01  \n",
       " Zinc influx into cells by the SLC39 gene family          9.857729e-01  \n",
       " CREB1 phosphorylation through NMDA receptor-med...       9.857729e-01  \n",
       " CYP2E1 reactions                                         9.857729e-01  \n",
       " Biosynthesis of maresin conjugates in tissue re...       9.857729e-01  \n",
       " Biosynthesis of DHA-derived sulfido conjugates           9.857729e-01  \n",
       " Nef mediated downregulation of MHC class I comp...       9.865203e-01  \n",
       " Creation of C4 and C2 activators                         9.865203e-01  \n",
       " Serotonin Neurotransmitter Release Cycle                 9.868720e-01  \n",
       " HuR (ELAVL1) binds and stabilizes mRNA                   9.880862e-01  \n",
       " Miscellaneous substrates                                 9.888032e-01  \n",
       " Activated NTRK3 signals through RAS                      9.892372e-01  \n",
       " Classical antibody-mediated complement activation        9.901952e-01  \n",
       " Defective Mismatch Repair Associated With MSH2           9.901952e-01  \n",
       " IRAK2 mediated activation of TAK1 complex                9.901952e-01  \n",
       " SHC-mediated cascade:FGFR4                               9.913556e-01  \n",
       " Role of phospholipids in phagocytosis                    9.919604e-01  \n",
       " Regulation of IFNA signaling                             9.919604e-01  \n",
       " Xenobiotics                                              9.919604e-01  \n",
       " Major pathway of rRNA processing in the nucleol...       9.946222e-01  \n",
       " 5-Phosphoribose 1-diphosphate biosynthesis               9.952476e-01  \n",
       " Trafficking and processing of endosomal TLR              9.988320e-01  \n",
       " CD28 dependent PI3K/Akt signaling                        9.989896e-01  \n",
       " Constitutive Signaling by AKT1 E17K in Cancer            9.989896e-01  \n",
       " Dectin-2 family                                          9.989896e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '2':                                                         p-values  \\\n",
       " Signaling by Insulin receptor                       3.788784e-12   \n",
       " Iron uptake and transport                           2.132196e-11   \n",
       " Transferrin endocytosis and recycling               2.429206e-11   \n",
       " Insulin receptor recycling                          2.846298e-11   \n",
       " Amino acids regulate mTORC1                         5.325704e-11   \n",
       " ROS and RNS production in phagocytes                1.264879e-10   \n",
       " RUNX3 regulates WNT signaling                       2.533284e-10   \n",
       " Ephrin signaling                                    4.560452e-10   \n",
       " TRAIL  signaling                                    2.159461e-09   \n",
       " Ca2+ activated K+ channels                          3.365146e-09   \n",
       " Estrogen-dependent nuclear events downstream of...  2.218629e-08   \n",
       " Ion channel transport                               3.403472e-08   \n",
       " Repression of WNT target genes                      3.538041e-08   \n",
       " Activation of SMO                                   3.723712e-08   \n",
       " Enzymatic degradation of dopamine by COMT           1.300325e-07   \n",
       " Dopamine clearance from the synaptic cleft          1.300327e-07   \n",
       " NOSIP mediated eNOS trafficking                     3.139949e-07   \n",
       " TGFBR1 KD Mutants in Cancer                         7.112294e-07   \n",
       " SMAD2/3 Phosphorylation Motif Mutants in Cancer     7.112335e-07   \n",
       " RUNX3 regulates CDKN1A transcription                8.469400e-07   \n",
       " Amine Oxidase reactions                             1.729631e-06   \n",
       " Phase 4 - resting membrane potential                1.856042e-06   \n",
       " Organic cation transport                            3.624653e-06   \n",
       " Defective ABCC8 can cause hypoglycemias and hyp...  5.006503e-06   \n",
       " Sodium-coupled phosphate cotransporters             5.183871e-06   \n",
       " PAOs oxidise polyamines to amines                   5.430314e-06   \n",
       " Cell junction organization                          5.879203e-06   \n",
       " Cell-Cell communication                             6.272364e-06   \n",
       " Organic cation/anion/zwitterion transport           6.813788e-06   \n",
       " RUNX2 regulates osteoblast differentiation          8.126514e-06   \n",
       " ...                                                          ...   \n",
       " Autodegradation of Cdh1 by Cdh1:APC/C               9.781982e-01   \n",
       " Orc1 removal from chromatin                         9.782737e-01   \n",
       " RAF-independent MAPK1/3 activation                  9.787485e-01   \n",
       " GPVI-mediated activation cascade                    9.788075e-01   \n",
       " Aggrephagy                                          9.817235e-01   \n",
       " Metabolism of ingested H2SeO4 and H2SeO3 into H2Se  9.819096e-01   \n",
       " Cytochrome c-mediated apoptotic response            9.825656e-01   \n",
       " Digestion                                           9.827568e-01   \n",
       " Downstream signal transduction                      9.844796e-01   \n",
       " Interactions of Vpr with host cellular proteins     9.844939e-01   \n",
       " Toxicity of botulinum toxin type E (BoNT/E)         9.855745e-01   \n",
       " Pyruvate metabolism and Citric Acid (TCA) cycle     9.867105e-01   \n",
       " The retinoid cycle in cones (daylight vision)       9.868557e-01   \n",
       " IRS-mediated signalling                             9.882559e-01   \n",
       " SMAC, XIAP-regulated apoptotic response             9.888819e-01   \n",
       " Pyrimidine salvage                                  9.890483e-01   \n",
       " Formation of editosomes by ADAR proteins            9.890489e-01   \n",
       " C6 deamination of adenosine                         9.890498e-01   \n",
       " mRNA Editing: A to I Conversion                     9.890499e-01   \n",
       " Triglyceride biosynthesis                           9.911754e-01   \n",
       " IKBKG deficiency causes anhidrotic ectodermal d...  9.914715e-01   \n",
       " IKBKB deficiency causes SCID                        9.914746e-01   \n",
       " Cellular responses to external stimuli              9.916512e-01   \n",
       " Cellular responses to stress                        9.922531e-01   \n",
       " Host Interactions with Influenza Factors            9.932679e-01   \n",
       " NS1 Mediated Effects on Host Pathways               9.933151e-01   \n",
       " Biosynthesis of maresin-like SPMs                   9.936552e-01   \n",
       " Defective AVP does not bind AVPR2 and causes ne...  9.950403e-01   \n",
       " Chylomicron remodeling                              9.963634e-01   \n",
       " Assembly of the pre-replicative complex             9.978066e-01   \n",
       " \n",
       "                                                     cluster 2 qvalues  \n",
       " Signaling by Insulin receptor                            7.857937e-09  \n",
       " Iron uptake and transport                                1.475806e-08  \n",
       " Transferrin endocytosis and recycling                    1.475806e-08  \n",
       " Insulin receptor recycling                               1.475806e-08  \n",
       " Amino acids regulate mTORC1                              2.209102e-08  \n",
       " ROS and RNS production in phagocytes                     4.372264e-08  \n",
       " RUNX3 regulates WNT signaling                            7.505758e-08  \n",
       " Ephrin signaling                                         1.182297e-07  \n",
       " TRAIL  signaling                                         4.976358e-07  \n",
       " Ca2+ activated K+ channels                               6.979312e-07  \n",
       " Estrogen-dependent nuclear events downstream of...       4.183125e-06  \n",
       " Ion channel transport                                    5.516413e-06  \n",
       " Repression of WNT target genes                           5.516413e-06  \n",
       " Activation of SMO                                        5.516413e-06  \n",
       " Enzymatic degradation of dopamine by COMT                1.685549e-05  \n",
       " Dopamine clearance from the synaptic cleft               1.685549e-05  \n",
       " NOSIP mediated eNOS trafficking                          3.830737e-05  \n",
       " TGFBR1 KD Mutants in Cancer                              7.763675e-05  \n",
       " SMAD2/3 Phosphorylation Motif Mutants in Cancer          7.763675e-05  \n",
       " RUNX3 regulates CDKN1A transcription                     8.782768e-05  \n",
       " Amine Oxidase reactions                                  1.708216e-04  \n",
       " Phase 4 - resting membrane potential                     1.749741e-04  \n",
       " Organic cation transport                                 3.268491e-04  \n",
       " Defective ABCC8 can cause hypoglycemias and hyp...       4.300540e-04  \n",
       " Sodium-coupled phosphate cotransporters                  4.300540e-04  \n",
       " PAOs oxidise polyamines to amines                        4.331720e-04  \n",
       " Cell junction organization                               4.516099e-04  \n",
       " Cell-Cell communication                                  4.646029e-04  \n",
       " Organic cation/anion/zwitterion transport                4.873033e-04  \n",
       " RUNX2 regulates osteoblast differentiation               5.618130e-04  \n",
       " ...                                                               ...  \n",
       " Autodegradation of Cdh1 by Cdh1:APC/C                    9.912338e-01  \n",
       " Orc1 removal from chromatin                              9.912338e-01  \n",
       " RAF-independent MAPK1/3 activation                       9.912338e-01  \n",
       " GPVI-mediated activation cascade                         9.912338e-01  \n",
       " Aggrephagy                                               9.932932e-01  \n",
       " Metabolism of ingested H2SeO4 and H2SeO3 into H2Se       9.932932e-01  \n",
       " Cytochrome c-mediated apoptotic response                 9.932932e-01  \n",
       " Digestion                                                9.932932e-01  \n",
       " Downstream signal transduction                           9.940800e-01  \n",
       " Interactions of Vpr with host cellular proteins          9.940800e-01  \n",
       " Toxicity of botulinum toxin type E (BoNT/E)              9.943235e-01  \n",
       " Pyruvate metabolism and Citric Acid (TCA) cycle          9.943235e-01  \n",
       " The retinoid cycle in cones (daylight vision)            9.943235e-01  \n",
       " IRS-mediated signalling                                  9.943235e-01  \n",
       " SMAC, XIAP-regulated apoptotic response                  9.943235e-01  \n",
       " Pyrimidine salvage                                       9.943235e-01  \n",
       " Formation of editosomes by ADAR proteins                 9.943235e-01  \n",
       " C6 deamination of adenosine                              9.943235e-01  \n",
       " mRNA Editing: A to I Conversion                          9.943235e-01  \n",
       " Triglyceride biosynthesis                                9.950095e-01  \n",
       " IKBKG deficiency causes anhidrotic ectodermal d...       9.950095e-01  \n",
       " IKBKB deficiency causes SCID                             9.950095e-01  \n",
       " Cellular responses to external stimuli                   9.950095e-01  \n",
       " Cellular responses to stress                             9.950945e-01  \n",
       " Host Interactions with Influenza Factors                 9.950945e-01  \n",
       " NS1 Mediated Effects on Host Pathways                    9.950945e-01  \n",
       " Biosynthesis of maresin-like SPMs                        9.950945e-01  \n",
       " Defective AVP does not bind AVPR2 and causes ne...       9.960007e-01  \n",
       " Chylomicron remodeling                                   9.968441e-01  \n",
       " Assembly of the pre-replicative complex                  9.978066e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '5':                                                         p-values  \\\n",
       " Metabolism of amine-derived hormones                3.032964e-82   \n",
       " Catecholamine biosynthesis                          1.499834e-80   \n",
       " Synthesis of PIPs in the nucleus                    1.037797e-74   \n",
       " GRB7 events in ERBB2 signaling                      2.392231e-68   \n",
       " PI5P Regulates TP53 Acetylation                     3.651662e-68   \n",
       " Fertilization                                       5.587300e-64   \n",
       " Sperm Motility And Taxes                            1.281686e-58   \n",
       " Tryptophan catabolism                               2.305909e-50   \n",
       " Metal sequestration by antimicrobial proteins       1.651867e-46   \n",
       " Activation of gene expression by SREBF (SREBP)      3.274298e-41   \n",
       " Proline catabolism                                  8.778566e-39   \n",
       " p130Cas linkage to MAPK signaling for integrins     2.574007e-38   \n",
       " BH3-only proteins associate with and inactivate...  5.216641e-38   \n",
       " Cholesterol biosynthesis                            1.272247e-35   \n",
       " GRB2:SOS provides linkage to MAPK signaling for...  1.903499e-35   \n",
       " Manipulation of host energy metabolism              5.916988e-33   \n",
       " Caspase-mediated cleavage of cytoskeletal proteins  5.839867e-32   \n",
       " Signaling by Leptin                                 2.521929e-31   \n",
       " RUNX1 regulates estrogen receptor mediated tran...  3.693794e-29   \n",
       " Linoleic acid (LA) metabolism                       7.052584e-28   \n",
       " Activation of AMPK downstream of NMDARs             1.090605e-27   \n",
       " alpha-linolenic acid (ALA) metabolism               3.962613e-27   \n",
       " alpha-linolenic (omega3) and linoleic (omega6) ...  3.963159e-27   \n",
       " Visual phototransduction                            4.172611e-27   \n",
       " N-glycan antennae elongation in the medial/tran...  8.537272e-27   \n",
       " IRS activation                                      6.581233e-26   \n",
       " Activation of caspases through apoptosome-media...  1.054786e-25   \n",
       " PI3K Cascade                                        1.144614e-25   \n",
       " Metabolism of folate and pterines                   3.272367e-25   \n",
       " Reactions specific to the complex N-glycan synt...  3.448863e-25   \n",
       " ...                                                          ...   \n",
       " FRS-mediated FGFR4 signaling                        9.390876e-01   \n",
       " Regulated proteolysis of p75NTR                     9.412384e-01   \n",
       " RUNX2 regulates bone development                    9.422876e-01   \n",
       " Negative epigenetic regulation of rRNA expression   9.430113e-01   \n",
       " Diseases associated with glycosylation precurso...  9.470275e-01   \n",
       " HuR (ELAVL1) binds and stabilizes mRNA              9.495924e-01   \n",
       " Digestion                                           9.502517e-01   \n",
       " STING mediated induction of host immune responses   9.515773e-01   \n",
       " Tachykinin receptors bind tachykinins               9.587869e-01   \n",
       " Mitophagy                                           9.604139e-01   \n",
       " Vitamins                                            9.636805e-01   \n",
       " RUNX1 interacts with co-factors whose precise e...  9.649128e-01   \n",
       " Defective LFNG causes SCDO3                         9.657203e-01   \n",
       " Sodium-coupled sulphate, di- and tri-carboxylat...  9.751795e-01   \n",
       " Electron transport from NADPH to Ferredoxin         9.775834e-01   \n",
       " ABO blood group biosynthesis                        9.821868e-01   \n",
       " CD209 (DC-SIGN) signaling                           9.827903e-01   \n",
       " ATF6 (ATF6-alpha) activates chaperone genes         9.832202e-01   \n",
       " NOTCH3 Activation and Transmission of Signal to...  9.835990e-01   \n",
       " RNA Polymerase I Transcription                      9.869608e-01   \n",
       " Clathrin-mediated endocytosis                       9.877379e-01   \n",
       " Intracellular metabolism of fatty acids regulat...  9.885944e-01   \n",
       " TGFBR1 LBD Mutants in Cancer                        9.893856e-01   \n",
       " Events associated with phagocytolytic activity ...  9.905370e-01   \n",
       " Constitutive Signaling by EGFRvIII                  9.909202e-01   \n",
       " Signaling by EGFRvIII in Cancer                     9.909927e-01   \n",
       " Breakdown of the nuclear lamina                     9.965854e-01   \n",
       " FGFRL1 modulation of FGFR1 signaling                9.973365e-01   \n",
       " RNA Polymerase I Promoter Clearance                 9.981905e-01   \n",
       " RAS signaling downstream of NF1 loss-of-functio...  9.989674e-01   \n",
       " \n",
       "                                                     cluster 5 qvalues  \n",
       " Metabolism of amine-derived hormones                     6.290367e-79  \n",
       " Catecholamine biosynthesis                               1.555328e-77  \n",
       " Synthesis of PIPs in the nucleus                         7.174636e-72  \n",
       " GRB7 events in ERBB2 signaling                           1.240372e-65  \n",
       " PI5P Regulates TP53 Acetylation                          1.514709e-65  \n",
       " Fertilization                                            1.931343e-61  \n",
       " Sperm Motility And Taxes                                 3.797451e-56  \n",
       " Tryptophan catabolism                                    5.978068e-48  \n",
       " Metal sequestration by antimicrobial proteins            3.806636e-44  \n",
       " Activation of gene expression by SREBF (SREBP)           6.790894e-39  \n",
       " Proline catabolism                                       1.655159e-36  \n",
       " p130Cas linkage to MAPK signaling for integrins          4.448743e-36  \n",
       " BH3-only proteins associate with and inactivate...       8.322549e-36  \n",
       " Cholesterol biosynthesis                                 1.884743e-33  \n",
       " GRB2:SOS provides linkage to MAPK signaling for...       2.631904e-33  \n",
       " Manipulation of host energy metabolism                   7.669896e-31  \n",
       " Caspase-mediated cleavage of cytoskeletal proteins       7.124638e-30  \n",
       " Signaling by Leptin                                      2.905823e-29  \n",
       " RUNX1 regulates estrogen receptor mediated tran...       4.032068e-27  \n",
       " Linoleic acid (LA) metabolism                            7.313530e-26  \n",
       " Activation of AMPK downstream of NMDARs                  1.077102e-25  \n",
       " alpha-linolenic acid (ALA) metabolism                    3.573735e-25  \n",
       " alpha-linolenic (omega3) and linoleic (omega6) ...       3.573735e-25  \n",
       " Visual phototransduction                                 3.605831e-25  \n",
       " N-glycan antennae elongation in the medial/tran...       7.082521e-25  \n",
       " IRS activation                                           5.249799e-24  \n",
       " Activation of caspases through apoptosome-media...       8.102320e-24  \n",
       " PI3K Cascade                                             8.478319e-24  \n",
       " Metabolism of folate and pterines                        2.340307e-23  \n",
       " Reactions specific to the complex N-glycan synt...       2.384314e-23  \n",
       " ...                                                               ...  \n",
       " FRS-mediated FGFR4 signaling                             9.524047e-01  \n",
       " Regulated proteolysis of p75NTR                          9.541194e-01  \n",
       " RUNX2 regulates bone development                         9.547164e-01  \n",
       " Negative epigenetic regulation of rRNA expression        9.549831e-01  \n",
       " Diseases associated with glycosylation precurso...       9.585823e-01  \n",
       " HuR (ELAVL1) binds and stabilizes mRNA                   9.607096e-01  \n",
       " Digestion                                                9.609079e-01  \n",
       " STING mediated induction of host immune responses        9.617794e-01  \n",
       " Tachykinin receptors bind tachykinins                    9.685943e-01  \n",
       " Mitophagy                                                9.697656e-01  \n",
       " Vitamins                                                 9.725904e-01  \n",
       " RUNX1 interacts with co-factors whose precise e...       9.733605e-01  \n",
       " Defective LFNG causes SCDO3                              9.737014e-01  \n",
       " Sodium-coupled sulphate, di- and tri-carboxylat...       9.827611e-01  \n",
       " Electron transport from NADPH to Ferredoxin              9.847051e-01  \n",
       " ABO blood group biosynthesis                             9.888436e-01  \n",
       " CD209 (DC-SIGN) signaling                                9.888436e-01  \n",
       " ATF6 (ATF6-alpha) activates chaperone genes              9.888436e-01  \n",
       " NOTCH3 Activation and Transmission of Signal to...       9.888436e-01  \n",
       " RNA Polymerase I Transcription                           9.917426e-01  \n",
       " Clathrin-mediated endocytosis                            9.920428e-01  \n",
       " Intracellular metabolism of fatty acids regulat...       9.924225e-01  \n",
       " TGFBR1 LBD Mutants in Cancer                             9.927362e-01  \n",
       " Events associated with phagocytolytic activity ...       9.929077e-01  \n",
       " Constitutive Signaling by EGFRvIII                       9.929077e-01  \n",
       " Signaling by EGFRvIII in Cancer                          9.929077e-01  \n",
       " Breakdown of the nuclear lamina                          9.980291e-01  \n",
       " FGFRL1 modulation of FGFR1 signaling                     9.982992e-01  \n",
       " RNA Polymerase I Promoter Clearance                      9.986720e-01  \n",
       " RAS signaling downstream of NF1 loss-of-functio...       9.989674e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '4ER-':                                                         p-values  \\\n",
       " Rho GTPase cycle                                    2.056710e-44   \n",
       " Cell-Cell communication                             2.692864e-35   \n",
       " Constitutive Signaling by Aberrant PI3K in Cancer   6.001913e-35   \n",
       " PI3K/AKT Signaling in Cancer                        1.023620e-34   \n",
       " G-protein mediated events                           1.553954e-34   \n",
       " Reelin signalling pathway                           1.583128e-34   \n",
       " Regulation of signaling by CBL                      5.406275e-34   \n",
       " Acetylcholine regulates insulin secretion           5.868915e-34   \n",
       " PLC beta mediated events                            1.733770e-33   \n",
       " RUNX1 regulates estrogen receptor mediated tran...  2.352336e-33   \n",
       " Transfer of LPS from LBP carrier to CD14            2.549549e-33   \n",
       " Regulation of KIT signaling                         3.258910e-33   \n",
       " Interleukin-3, Interleukin-5 and GM-CSF signaling   5.632264e-33   \n",
       " PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling     6.496485e-33   \n",
       " RET signaling                                       1.261225e-32   \n",
       " Negative regulation of the PI3K/AKT network         2.899708e-32   \n",
       " Platelet Aggregation (Plug Formation)               8.084091e-32   \n",
       " Synthesis of IP3 and IP4 in the cytosol             9.763577e-32   \n",
       " PECAM1 interactions                                 1.105236e-31   \n",
       " GAB1 signalosome                                    3.073702e-31   \n",
       " ABC transporters in lipid homeostasis               3.155560e-31   \n",
       " Inositol phosphate metabolism                       9.304153e-31   \n",
       " Interleukin receptor SHC signaling                  1.734594e-30   \n",
       " RUNX1 regulates transcription of genes involved...  1.392708e-29   \n",
       " Synthesis of PI                                     1.439539e-29   \n",
       " Transport of inorganic cations/anions and amino...  2.637903e-29   \n",
       " Calmodulin induced events                           3.819340e-29   \n",
       " CaM pathway                                         3.828558e-29   \n",
       " Interleukin-4 and Interleukin-13 signaling          7.439253e-29   \n",
       " RHO GTPases Activate NADPH Oxidases                 1.282226e-28   \n",
       " ...                                                          ...   \n",
       " Signal transduction by L1                           9.512491e-01   \n",
       " phospho-PLA2 pathway                                9.518211e-01   \n",
       " SMAC (DIABLO) binds to IAPs                         9.531349e-01   \n",
       " SMAC(DIABLO)-mediated dissociation of IAP:caspa...  9.531386e-01   \n",
       " NR1H2 & NR1H3 regulate gene expression to limit...  9.532104e-01   \n",
       " Negative regulation of NOTCH4 signaling             9.543599e-01   \n",
       " Loss of proteins required for interphase microt...  9.543724e-01   \n",
       " Loss of Nlp from mitotic centrosomes                9.543727e-01   \n",
       " TRAF6 mediated NF-kB activation                     9.546345e-01   \n",
       " Glycogen synthesis                                  9.551008e-01   \n",
       " Toxicity of botulinum toxin type C (BoNT/C)         9.585835e-01   \n",
       " Switching of origins to a post-replicative state    9.587367e-01   \n",
       " Intra-Golgi and retrograde Golgi-to-ER traffic      9.645465e-01   \n",
       " Proline catabolism                                  9.683300e-01   \n",
       " MyD88-independent TLR4 cascade                      9.686892e-01   \n",
       " TRIF(TICAM1)-mediated TLR4 signaling                9.686896e-01   \n",
       " Regulation of APC/C activators between G1/S and...  9.687294e-01   \n",
       " Cdc20:Phospho-APC/C mediated degradation of Cyc...  9.749326e-01   \n",
       " p75 NTR receptor-mediated signalling                9.755873e-01   \n",
       " RUNX1 regulates expression of components of tig...  9.807361e-01   \n",
       " RNA Polymerase II Transcription                     9.827828e-01   \n",
       " FGFR1 ligand binding and activation                 9.830755e-01   \n",
       " Dual Incision in GG-NER                             9.834114e-01   \n",
       " Digestion of dietary carbohydrate                   9.834668e-01   \n",
       " VLDLR internalisation and degradation               9.848543e-01   \n",
       " Defective GCLC causes Hemolytic anemia due to g...  9.854067e-01   \n",
       " Cyclin A:Cdk2-associated events at S phase entry    9.873545e-01   \n",
       " Highly calcium permeable postsynaptic nicotinic...  9.909011e-01   \n",
       " Aggrephagy                                          9.926341e-01   \n",
       " MET receptor recycling                              9.961448e-01   \n",
       " \n",
       "                                                     cluster 4ER- qvalues  \n",
       " Rho GTPase cycle                                            4.265616e-41  \n",
       " Cell-Cell communication                                     2.792500e-32  \n",
       " Constitutive Signaling by Aberrant PI3K in Cancer           4.149323e-32  \n",
       " PI3K/AKT Signaling in Cancer                                5.307472e-32  \n",
       " G-protein mediated events                                   5.472345e-32  \n",
       " Reelin signalling pathway                                   5.472345e-32  \n",
       " Regulation of signaling by CBL                              1.521516e-31  \n",
       " Acetylcholine regulates insulin secretion                   1.521516e-31  \n",
       " PLC beta mediated events                                    3.995376e-31  \n",
       " RUNX1 regulates estrogen receptor mediated tran...          4.807058e-31  \n",
       " Transfer of LPS from LBP carrier to CD14                    4.807058e-31  \n",
       " Regulation of KIT signaling                                 5.632483e-31  \n",
       " Interleukin-3, Interleukin-5 and GM-CSF signaling           8.985628e-31  \n",
       " PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling             9.624079e-31  \n",
       " RET signaling                                               1.743854e-30  \n",
       " Negative regulation of the PI3K/AKT network                 3.758746e-30  \n",
       " Platelet Aggregation (Plug Formation)                       9.862591e-30  \n",
       " Synthesis of IP3 and IP4 in the cytosol                     1.124981e-29  \n",
       " PECAM1 interactions                                         1.206452e-29  \n",
       " GAB1 signalosome                                            3.116491e-29  \n",
       " ABC transporters in lipid homeostasis                       3.116491e-29  \n",
       " Inositol phosphate metabolism                               8.771279e-29  \n",
       " Interleukin receptor SHC signaling                          1.564152e-28  \n",
       " RUNX1 regulates transcription of genes involved...          1.194242e-27  \n",
       " Synthesis of PI                                             1.194242e-27  \n",
       " Transport of inorganic cations/anions and amino...          2.104235e-27  \n",
       " Calmodulin induced events                                   2.835868e-27  \n",
       " CaM pathway                                                 2.835868e-27  \n",
       " Interleukin-4 and Interleukin-13 signaling                  5.320348e-27  \n",
       " RHO GTPases Activate NADPH Oxidases                         8.864459e-27  \n",
       " ...                                                                  ...  \n",
       " Signal transduction by L1                                   9.643994e-01  \n",
       " phospho-PLA2 pathway                                        9.643994e-01  \n",
       " SMAC (DIABLO) binds to IAPs                                 9.643994e-01  \n",
       " SMAC(DIABLO)-mediated dissociation of IAP:caspa...          9.643994e-01  \n",
       " NR1H2 & NR1H3 regulate gene expression to limit...          9.643994e-01  \n",
       " Negative regulation of NOTCH4 signaling                     9.643994e-01  \n",
       " Loss of proteins required for interphase microt...          9.643994e-01  \n",
       " Loss of Nlp from mitotic centrosomes                        9.643994e-01  \n",
       " TRAF6 mediated NF-kB activation                             9.643994e-01  \n",
       " Glycogen synthesis                                          9.644007e-01  \n",
       " Toxicity of botulinum toxin type C (BoNT/C)                 9.671303e-01  \n",
       " Switching of origins to a post-replicative state            9.671303e-01  \n",
       " Intra-Golgi and retrograde Golgi-to-ER traffic              9.725179e-01  \n",
       " Proline catabolism                                          9.748398e-01  \n",
       " MyD88-independent TLR4 cascade                              9.748398e-01  \n",
       " TRIF(TICAM1)-mediated TLR4 signaling                        9.748398e-01  \n",
       " Regulation of APC/C activators between G1/S and...          9.748398e-01  \n",
       " Cdc20:Phospho-APC/C mediated degradation of Cyc...          9.806063e-01  \n",
       " p75 NTR receptor-mediated signalling                        9.807891e-01  \n",
       " RUNX1 regulates expression of components of tig...          9.854878e-01  \n",
       " RNA Polymerase II Transcription                             9.863202e-01  \n",
       " FGFR1 ligand binding and activation                         9.863202e-01  \n",
       " Dual Incision in GG-NER                                     9.863202e-01  \n",
       " Digestion of dietary carbohydrate                           9.863202e-01  \n",
       " VLDLR internalisation and degradation                       9.872343e-01  \n",
       " Defective GCLC causes Hemolytic anemia due to g...          9.873109e-01  \n",
       " Cyclin A:Cdk2-associated events at S phase entry            9.887848e-01  \n",
       " Highly calcium permeable postsynaptic nicotinic...          9.918576e-01  \n",
       " Aggrephagy                                                  9.931129e-01  \n",
       " MET receptor recycling                                      9.961448e-01  \n",
       " \n",
       " [2074 rows x 2 columns],\n",
       " '6':                                                          p-values  \\\n",
       " Abacavir transport and metabolism                   3.937531e-248   \n",
       " Formation of Fibrin Clot (Clotting Cascade)          7.911799e-61   \n",
       " NGF-independant TRKA activation                      7.399481e-23   \n",
       " Vitamin B2 (riboflavin) metabolism                   1.092961e-21   \n",
       " Stimuli-sensing channels                             8.048739e-19   \n",
       " Constitutive Signaling by Overexpressed ERBB2        2.829107e-14   \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...   5.744285e-14   \n",
       " Ketone body metabolism                               2.100085e-12   \n",
       " Interleukin-1 family signaling                       2.359154e-12   \n",
       " Muscarinic acetylcholine receptors                   3.015843e-12   \n",
       " Passive transport by Aquaporins                      2.987857e-11   \n",
       " Meiosis                                              6.041974e-10   \n",
       " Cam-PDE 1 activation                                 6.554718e-10   \n",
       " IGF1R signaling cascade                              2.416179e-09   \n",
       " Toll Like Receptor 2 (TLR2) Cascade                  2.900491e-09   \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...   3.139896e-09   \n",
       " Activation of RAC1 downstream of NMDARs              3.518012e-09   \n",
       " Protein localization                                 4.089408e-09   \n",
       " Synthesis of PI                                      4.976171e-09   \n",
       " HSF1 activation                                      9.306833e-09   \n",
       " Detoxification of Reactive Oxygen Species            1.222043e-08   \n",
       " Signaling by moderate kinase activity BRAF mutants   1.351493e-08   \n",
       " Sodium-coupled phosphate cotransporters              1.486434e-08   \n",
       " Toxicity of botulinum toxin type G (BoNT/G)          1.854682e-08   \n",
       " IRE1alpha activates chaperones                       1.992904e-08   \n",
       " Biosynthesis of maresin-like SPMs                    2.296534e-08   \n",
       " Utilization of Ketone Bodies                         2.848385e-08   \n",
       " Asymmetric localization of PCP proteins              2.968894e-08   \n",
       " HIV Infection                                        2.980989e-08   \n",
       " Processing of Intronless Pre-mRNAs                   3.032209e-08   \n",
       " ...                                                           ...   \n",
       " MAPK6/MAPK4 signaling                                9.644683e-01   \n",
       " Nuclear import of Rev protein                        9.649291e-01   \n",
       " Interactions of Vpr with host cellular proteins      9.651402e-01   \n",
       " Regulation of TP53 Expression                        9.669865e-01   \n",
       " Phospholipase C-mediated cascade; FGFR3              9.669877e-01   \n",
       " Cohesin Loading onto Chromatin                       9.677787e-01   \n",
       " Cytosolic sensors of pathogen-associated DNA         9.728216e-01   \n",
       " TNF receptor superfamily (TNFSF) members mediat...   9.731401e-01   \n",
       " ERKs are inactivated                                 9.735971e-01   \n",
       " Interleukin-33 signaling                             9.752904e-01   \n",
       " Retinoid metabolism and transport                    9.783102e-01   \n",
       " Common Pathway of Fibrin Clot Formation              9.783576e-01   \n",
       " Adenylate cyclase inhibitory pathway                 9.790545e-01   \n",
       " Vif-mediated degradation of APOBEC3G                 9.790620e-01   \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...   9.859063e-01   \n",
       " EML4 and NUDC in mitotic spindle formation           9.876845e-01   \n",
       " Hyaluronan biosynthesis and export                   9.897185e-01   \n",
       " Regulation of TP53 Degradation                       9.925059e-01   \n",
       " CD28 dependent Vav1 pathway                          9.927261e-01   \n",
       " Synthesis of PG                                      9.929512e-01   \n",
       " Hemostasis                                           9.950324e-01   \n",
       " Transcriptional Regulation by MECP2                  9.967100e-01   \n",
       " ABC transporters in lipid homeostasis                9.967103e-01   \n",
       " Oligomerization of connexins into connexons          9.992339e-01   \n",
       " Bicarbonate transporters                             9.993454e-01   \n",
       " Nuclear Pore Complex (NPC) Disassembly               9.993502e-01   \n",
       " Cholesterol biosynthesis via desmosterol             9.993513e-01   \n",
       " Cholesterol biosynthesis via lathosterol             9.993523e-01   \n",
       " Nuclear Envelope Breakdown                           9.993533e-01   \n",
       " Selective autophagy                                  9.993537e-01   \n",
       " \n",
       "                                                     cluster 6 qvalues  \\\n",
       " Abacavir transport and metabolism                            0.721052   \n",
       " Formation of Fibrin Clot (Clotting Cascade)                  0.010930   \n",
       " NGF-independant TRKA activation                              0.672498   \n",
       " Vitamin B2 (riboflavin) metabolism                           0.012640   \n",
       " Stimuli-sensing channels                                     0.013170   \n",
       " Constitutive Signaling by Overexpressed ERBB2                0.235623   \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...           0.425309   \n",
       " Ketone body metabolism                                       0.774677   \n",
       " Interleukin-1 family signaling                               0.166103   \n",
       " Muscarinic acetylcholine receptors                           0.893613   \n",
       " Passive transport by Aquaporins                              0.767503   \n",
       " Meiosis                                                      0.011648   \n",
       " Cam-PDE 1 activation                                         0.725417   \n",
       " IGF1R signaling cascade                                      0.866957   \n",
       " Toll Like Receptor 2 (TLR2) Cascade                          0.015923   \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...           0.695167   \n",
       " Activation of RAC1 downstream of NMDARs                      0.037309   \n",
       " Protein localization                                         0.150830   \n",
       " Synthesis of PI                                              0.000007   \n",
       " HSF1 activation                                              0.000081   \n",
       " Detoxification of Reactive Oxygen Species                    0.000475   \n",
       " Signaling by moderate kinase activity BRAF mutants           0.190970   \n",
       " Sodium-coupled phosphate cotransporters                      0.000532   \n",
       " Toxicity of botulinum toxin type G (BoNT/G)                  0.854307   \n",
       " IRE1alpha activates chaperones                               0.885288   \n",
       " Biosynthesis of maresin-like SPMs                            0.673541   \n",
       " Utilization of Ketone Bodies                                 0.921605   \n",
       " Asymmetric localization of PCP proteins                      0.865087   \n",
       " HIV Infection                                                0.474888   \n",
       " Processing of Intronless Pre-mRNAs                           0.869225   \n",
       " ...                                                               ...   \n",
       " MAPK6/MAPK4 signaling                                        0.929732   \n",
       " Nuclear import of Rev protein                                0.332920   \n",
       " Interactions of Vpr with host cellular proteins              0.727640   \n",
       " Regulation of TP53 Expression                                0.175044   \n",
       " Phospholipase C-mediated cascade; FGFR3                      0.043519   \n",
       " Cohesin Loading onto Chromatin                               0.540713   \n",
       " Cytosolic sensors of pathogen-associated DNA                 0.482458   \n",
       " TNF receptor superfamily (TNFSF) members mediat...           0.000754   \n",
       " ERKs are inactivated                                         0.001009   \n",
       " Interleukin-33 signaling                                     0.602062   \n",
       " Retinoid metabolism and transport                            0.007721   \n",
       " Common Pathway of Fibrin Clot Formation                      0.011850   \n",
       " Adenylate cyclase inhibitory pathway                         0.000785   \n",
       " Vif-mediated degradation of APOBEC3G                         0.770731   \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...           0.206649   \n",
       " EML4 and NUDC in mitotic spindle formation                   0.077326   \n",
       " Hyaluronan biosynthesis and export                           0.662456   \n",
       " Regulation of TP53 Degradation                               0.410493   \n",
       " CD28 dependent Vav1 pathway                                  0.046330   \n",
       " Synthesis of PG                                              0.000001   \n",
       " Hemostasis                                                   0.000471   \n",
       " Transcriptional Regulation by MECP2                          0.000052   \n",
       " ABC transporters in lipid homeostasis                        0.000418   \n",
       " Oligomerization of connexins into connexons                  0.755363   \n",
       " Bicarbonate transporters                                     0.247779   \n",
       " Nuclear Pore Complex (NPC) Disassembly                       0.301383   \n",
       " Cholesterol biosynthesis via desmosterol                     0.248585   \n",
       " Cholesterol biosynthesis via lathosterol                     0.248585   \n",
       " Nuclear Envelope Breakdown                                   0.301032   \n",
       " Selective autophagy                                          0.301032   \n",
       " \n",
       "                                                     cluster nan qvalues  \n",
       " Abacavir transport and metabolism                         8.166440e-245  \n",
       " Formation of Fibrin Clot (Clotting Cascade)                8.204535e-58  \n",
       " NGF-independant TRKA activation                            5.115508e-20  \n",
       " Vitamin B2 (riboflavin) metabolism                         5.667005e-19  \n",
       " Stimuli-sensing channels                                   3.338617e-16  \n",
       " Constitutive Signaling by Overexpressed ERBB2              9.779280e-12  \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...         1.701950e-11  \n",
       " Ketone body metabolism                                     5.436539e-10  \n",
       " Interleukin-1 family signaling                             5.436539e-10  \n",
       " Muscarinic acetylcholine receptors                         6.254858e-10  \n",
       " Passive transport by Aquaporins                            5.633469e-09  \n",
       " Meiosis                                                    1.044255e-07  \n",
       " Cam-PDE 1 activation                                       1.045730e-07  \n",
       " IGF1R signaling cascade                                    3.579396e-07  \n",
       " Toll Like Receptor 2 (TLR2) Cascade                        4.010412e-07  \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...         4.070090e-07  \n",
       " Activation of RAC1 downstream of NMDARs                    4.291974e-07  \n",
       " Protein localization                                       4.711907e-07  \n",
       " Synthesis of PI                                            5.431883e-07  \n",
       " HSF1 activation                                            9.651185e-07  \n",
       " Detoxification of Reactive Oxygen Species                  1.206913e-06  \n",
       " Signaling by moderate kinase activity BRAF mutants         1.274089e-06  \n",
       " Sodium-coupled phosphate cotransporters                    1.340376e-06  \n",
       " Toxicity of botulinum toxin type G (BoNT/G)                1.602755e-06  \n",
       " IRE1alpha activates chaperones                             1.653313e-06  \n",
       " Biosynthesis of maresin-like SPMs                          1.831927e-06  \n",
       " Utilization of Ketone Bodies                               2.028650e-06  \n",
       " Asymmetric localization of PCP proteins                    2.028650e-06  \n",
       " HIV Infection                                              2.028650e-06  \n",
       " Processing of Intronless Pre-mRNAs                         2.028650e-06  \n",
       " ...                                                                 ...  \n",
       " MAPK6/MAPK4 signaling                                      9.778704e-01  \n",
       " Nuclear import of Rev protein                              9.778704e-01  \n",
       " Interactions of Vpr with host cellular proteins            9.778704e-01  \n",
       " Regulation of TP53 Expression                              9.787860e-01  \n",
       " Phospholipase C-mediated cascade; FGFR3                    9.787860e-01  \n",
       " Cohesin Loading onto Chromatin                             9.791088e-01  \n",
       " Cytosolic sensors of pathogen-associated DNA               9.835560e-01  \n",
       " TNF receptor superfamily (TNFSF) members mediat...         9.835560e-01  \n",
       " ERKs are inactivated                                       9.835560e-01  \n",
       " Interleukin-33 signaling                                   9.847869e-01  \n",
       " Retinoid metabolism and transport                          9.866738e-01  \n",
       " Common Pathway of Fibrin Clot Formation                    9.866738e-01  \n",
       " Adenylate cyclase inhibitory pathway                       9.866738e-01  \n",
       " Vif-mediated degradation of APOBEC3G                       9.866738e-01  \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...         9.930887e-01  \n",
       " EML4 and NUDC in mitotic spindle formation                 9.943970e-01  \n",
       " Hyaluronan biosynthesis and export                         9.959613e-01  \n",
       " Regulation of TP53 Degradation                             9.977620e-01  \n",
       " CD28 dependent Vav1 pathway                                9.977620e-01  \n",
       " Synthesis of PG                                            9.977620e-01  \n",
       " Hemostasis                                                 9.993537e-01  \n",
       " Transcriptional Regulation by MECP2                        9.993537e-01  \n",
       " ABC transporters in lipid homeostasis                      9.993537e-01  \n",
       " Oligomerization of connexins into connexons                9.993537e-01  \n",
       " Bicarbonate transporters                                   9.993537e-01  \n",
       " Nuclear Pore Complex (NPC) Disassembly                     9.993537e-01  \n",
       " Cholesterol biosynthesis via desmosterol                   9.993537e-01  \n",
       " Cholesterol biosynthesis via lathosterol                   9.993537e-01  \n",
       " Nuclear Envelope Breakdown                                 9.993537e-01  \n",
       " Selective autophagy                                        9.993537e-01  \n",
       " \n",
       " [2074 rows x 3 columns],\n",
       " nan:                                                          p-values  \\\n",
       " Abacavir transport and metabolism                   3.937531e-248   \n",
       " Formation of Fibrin Clot (Clotting Cascade)          7.911799e-61   \n",
       " NGF-independant TRKA activation                      7.399481e-23   \n",
       " Vitamin B2 (riboflavin) metabolism                   1.092961e-21   \n",
       " Stimuli-sensing channels                             8.048739e-19   \n",
       " Constitutive Signaling by Overexpressed ERBB2        2.829107e-14   \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...   5.744285e-14   \n",
       " Ketone body metabolism                               2.100085e-12   \n",
       " Interleukin-1 family signaling                       2.359154e-12   \n",
       " Muscarinic acetylcholine receptors                   3.015843e-12   \n",
       " Passive transport by Aquaporins                      2.987857e-11   \n",
       " Meiosis                                              6.041974e-10   \n",
       " Cam-PDE 1 activation                                 6.554718e-10   \n",
       " IGF1R signaling cascade                              2.416179e-09   \n",
       " Toll Like Receptor 2 (TLR2) Cascade                  2.900491e-09   \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...   3.139896e-09   \n",
       " Activation of RAC1 downstream of NMDARs              3.518012e-09   \n",
       " Protein localization                                 4.089408e-09   \n",
       " Synthesis of PI                                      4.976171e-09   \n",
       " HSF1 activation                                      9.306833e-09   \n",
       " Detoxification of Reactive Oxygen Species            1.222043e-08   \n",
       " Signaling by moderate kinase activity BRAF mutants   1.351493e-08   \n",
       " Sodium-coupled phosphate cotransporters              1.486434e-08   \n",
       " Toxicity of botulinum toxin type G (BoNT/G)          1.854682e-08   \n",
       " IRE1alpha activates chaperones                       1.992904e-08   \n",
       " Biosynthesis of maresin-like SPMs                    2.296534e-08   \n",
       " Utilization of Ketone Bodies                         2.848385e-08   \n",
       " Asymmetric localization of PCP proteins              2.968894e-08   \n",
       " HIV Infection                                        2.980989e-08   \n",
       " Processing of Intronless Pre-mRNAs                   3.032209e-08   \n",
       " ...                                                           ...   \n",
       " MAPK6/MAPK4 signaling                                9.644683e-01   \n",
       " Nuclear import of Rev protein                        9.649291e-01   \n",
       " Interactions of Vpr with host cellular proteins      9.651402e-01   \n",
       " Regulation of TP53 Expression                        9.669865e-01   \n",
       " Phospholipase C-mediated cascade; FGFR3              9.669877e-01   \n",
       " Cohesin Loading onto Chromatin                       9.677787e-01   \n",
       " Cytosolic sensors of pathogen-associated DNA         9.728216e-01   \n",
       " TNF receptor superfamily (TNFSF) members mediat...   9.731401e-01   \n",
       " ERKs are inactivated                                 9.735971e-01   \n",
       " Interleukin-33 signaling                             9.752904e-01   \n",
       " Retinoid metabolism and transport                    9.783102e-01   \n",
       " Common Pathway of Fibrin Clot Formation              9.783576e-01   \n",
       " Adenylate cyclase inhibitory pathway                 9.790545e-01   \n",
       " Vif-mediated degradation of APOBEC3G                 9.790620e-01   \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...   9.859063e-01   \n",
       " EML4 and NUDC in mitotic spindle formation           9.876845e-01   \n",
       " Hyaluronan biosynthesis and export                   9.897185e-01   \n",
       " Regulation of TP53 Degradation                       9.925059e-01   \n",
       " CD28 dependent Vav1 pathway                          9.927261e-01   \n",
       " Synthesis of PG                                      9.929512e-01   \n",
       " Hemostasis                                           9.950324e-01   \n",
       " Transcriptional Regulation by MECP2                  9.967100e-01   \n",
       " ABC transporters in lipid homeostasis                9.967103e-01   \n",
       " Oligomerization of connexins into connexons          9.992339e-01   \n",
       " Bicarbonate transporters                             9.993454e-01   \n",
       " Nuclear Pore Complex (NPC) Disassembly               9.993502e-01   \n",
       " Cholesterol biosynthesis via desmosterol             9.993513e-01   \n",
       " Cholesterol biosynthesis via lathosterol             9.993523e-01   \n",
       " Nuclear Envelope Breakdown                           9.993533e-01   \n",
       " Selective autophagy                                  9.993537e-01   \n",
       " \n",
       "                                                     cluster 6 qvalues  \\\n",
       " Abacavir transport and metabolism                            0.721052   \n",
       " Formation of Fibrin Clot (Clotting Cascade)                  0.010930   \n",
       " NGF-independant TRKA activation                              0.672498   \n",
       " Vitamin B2 (riboflavin) metabolism                           0.012640   \n",
       " Stimuli-sensing channels                                     0.013170   \n",
       " Constitutive Signaling by Overexpressed ERBB2                0.235623   \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...           0.425309   \n",
       " Ketone body metabolism                                       0.774677   \n",
       " Interleukin-1 family signaling                               0.166103   \n",
       " Muscarinic acetylcholine receptors                           0.893613   \n",
       " Passive transport by Aquaporins                              0.767503   \n",
       " Meiosis                                                      0.011648   \n",
       " Cam-PDE 1 activation                                         0.725417   \n",
       " IGF1R signaling cascade                                      0.866957   \n",
       " Toll Like Receptor 2 (TLR2) Cascade                          0.015923   \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...           0.695167   \n",
       " Activation of RAC1 downstream of NMDARs                      0.037309   \n",
       " Protein localization                                         0.150830   \n",
       " Synthesis of PI                                              0.000007   \n",
       " HSF1 activation                                              0.000081   \n",
       " Detoxification of Reactive Oxygen Species                    0.000475   \n",
       " Signaling by moderate kinase activity BRAF mutants           0.190970   \n",
       " Sodium-coupled phosphate cotransporters                      0.000532   \n",
       " Toxicity of botulinum toxin type G (BoNT/G)                  0.854307   \n",
       " IRE1alpha activates chaperones                               0.885288   \n",
       " Biosynthesis of maresin-like SPMs                            0.673541   \n",
       " Utilization of Ketone Bodies                                 0.921605   \n",
       " Asymmetric localization of PCP proteins                      0.865087   \n",
       " HIV Infection                                                0.474888   \n",
       " Processing of Intronless Pre-mRNAs                           0.869225   \n",
       " ...                                                               ...   \n",
       " MAPK6/MAPK4 signaling                                        0.929732   \n",
       " Nuclear import of Rev protein                                0.332920   \n",
       " Interactions of Vpr with host cellular proteins              0.727640   \n",
       " Regulation of TP53 Expression                                0.175044   \n",
       " Phospholipase C-mediated cascade; FGFR3                      0.043519   \n",
       " Cohesin Loading onto Chromatin                               0.540713   \n",
       " Cytosolic sensors of pathogen-associated DNA                 0.482458   \n",
       " TNF receptor superfamily (TNFSF) members mediat...           0.000754   \n",
       " ERKs are inactivated                                         0.001009   \n",
       " Interleukin-33 signaling                                     0.602062   \n",
       " Retinoid metabolism and transport                            0.007721   \n",
       " Common Pathway of Fibrin Clot Formation                      0.011850   \n",
       " Adenylate cyclase inhibitory pathway                         0.000785   \n",
       " Vif-mediated degradation of APOBEC3G                         0.770731   \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...           0.206649   \n",
       " EML4 and NUDC in mitotic spindle formation                   0.077326   \n",
       " Hyaluronan biosynthesis and export                           0.662456   \n",
       " Regulation of TP53 Degradation                               0.410493   \n",
       " CD28 dependent Vav1 pathway                                  0.046330   \n",
       " Synthesis of PG                                              0.000001   \n",
       " Hemostasis                                                   0.000471   \n",
       " Transcriptional Regulation by MECP2                          0.000052   \n",
       " ABC transporters in lipid homeostasis                        0.000418   \n",
       " Oligomerization of connexins into connexons                  0.755363   \n",
       " Bicarbonate transporters                                     0.247779   \n",
       " Nuclear Pore Complex (NPC) Disassembly                       0.301383   \n",
       " Cholesterol biosynthesis via desmosterol                     0.248585   \n",
       " Cholesterol biosynthesis via lathosterol                     0.248585   \n",
       " Nuclear Envelope Breakdown                                   0.301032   \n",
       " Selective autophagy                                          0.301032   \n",
       " \n",
       "                                                     cluster nan qvalues  \n",
       " Abacavir transport and metabolism                         8.166440e-245  \n",
       " Formation of Fibrin Clot (Clotting Cascade)                8.204535e-58  \n",
       " NGF-independant TRKA activation                            5.115508e-20  \n",
       " Vitamin B2 (riboflavin) metabolism                         5.667005e-19  \n",
       " Stimuli-sensing channels                                   3.338617e-16  \n",
       " Constitutive Signaling by Overexpressed ERBB2              9.779280e-12  \n",
       " NEP/NS2 Interacts with the Cellular Export Mach...         1.701950e-11  \n",
       " Ketone body metabolism                                     5.436539e-10  \n",
       " Interleukin-1 family signaling                             5.436539e-10  \n",
       " Muscarinic acetylcholine receptors                         6.254858e-10  \n",
       " Passive transport by Aquaporins                            5.633469e-09  \n",
       " Meiosis                                                    1.044255e-07  \n",
       " Cam-PDE 1 activation                                       1.045730e-07  \n",
       " IGF1R signaling cascade                                    3.579396e-07  \n",
       " Toll Like Receptor 2 (TLR2) Cascade                        4.010412e-07  \n",
       " Constitutive Signaling by NOTCH1 HD+PEST Domain...         4.070090e-07  \n",
       " Activation of RAC1 downstream of NMDARs                    4.291974e-07  \n",
       " Protein localization                                       4.711907e-07  \n",
       " Synthesis of PI                                            5.431883e-07  \n",
       " HSF1 activation                                            9.651185e-07  \n",
       " Detoxification of Reactive Oxygen Species                  1.206913e-06  \n",
       " Signaling by moderate kinase activity BRAF mutants         1.274089e-06  \n",
       " Sodium-coupled phosphate cotransporters                    1.340376e-06  \n",
       " Toxicity of botulinum toxin type G (BoNT/G)                1.602755e-06  \n",
       " IRE1alpha activates chaperones                             1.653313e-06  \n",
       " Biosynthesis of maresin-like SPMs                          1.831927e-06  \n",
       " Utilization of Ketone Bodies                               2.028650e-06  \n",
       " Asymmetric localization of PCP proteins                    2.028650e-06  \n",
       " HIV Infection                                              2.028650e-06  \n",
       " Processing of Intronless Pre-mRNAs                         2.028650e-06  \n",
       " ...                                                                 ...  \n",
       " MAPK6/MAPK4 signaling                                      9.778704e-01  \n",
       " Nuclear import of Rev protein                              9.778704e-01  \n",
       " Interactions of Vpr with host cellular proteins            9.778704e-01  \n",
       " Regulation of TP53 Expression                              9.787860e-01  \n",
       " Phospholipase C-mediated cascade; FGFR3                    9.787860e-01  \n",
       " Cohesin Loading onto Chromatin                             9.791088e-01  \n",
       " Cytosolic sensors of pathogen-associated DNA               9.835560e-01  \n",
       " TNF receptor superfamily (TNFSF) members mediat...         9.835560e-01  \n",
       " ERKs are inactivated                                       9.835560e-01  \n",
       " Interleukin-33 signaling                                   9.847869e-01  \n",
       " Retinoid metabolism and transport                          9.866738e-01  \n",
       " Common Pathway of Fibrin Clot Formation                    9.866738e-01  \n",
       " Adenylate cyclase inhibitory pathway                       9.866738e-01  \n",
       " Vif-mediated degradation of APOBEC3G                       9.866738e-01  \n",
       " Mismatch repair (MMR) directed by MSH2:MSH3 (Mu...         9.930887e-01  \n",
       " EML4 and NUDC in mitotic spindle formation                 9.943970e-01  \n",
       " Hyaluronan biosynthesis and export                         9.959613e-01  \n",
       " Regulation of TP53 Degradation                             9.977620e-01  \n",
       " CD28 dependent Vav1 pathway                                9.977620e-01  \n",
       " Synthesis of PG                                            9.977620e-01  \n",
       " Hemostasis                                                 9.993537e-01  \n",
       " Transcriptional Regulation by MECP2                        9.993537e-01  \n",
       " ABC transporters in lipid homeostasis                      9.993537e-01  \n",
       " Oligomerization of connexins into connexons                9.993537e-01  \n",
       " Bicarbonate transporters                                   9.993537e-01  \n",
       " Nuclear Pore Complex (NPC) Disassembly                     9.993537e-01  \n",
       " Cholesterol biosynthesis via desmosterol                   9.993537e-01  \n",
       " Cholesterol biosynthesis via lathosterol                   9.993537e-01  \n",
       " Nuclear Envelope Breakdown                                 9.993537e-01  \n",
       " Selective autophagy                                        9.993537e-01  \n",
       " \n",
       " [2074 rows x 3 columns]}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clusterframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4ER+\n",
      "(1196, 2)\n",
      "9\n",
      "(581, 2)\n",
      "7\n",
      "(1150, 2)\n",
      "3\n",
      "(959, 2)\n",
      "8\n",
      "(1313, 2)\n",
      "10\n",
      "(1571, 2)\n",
      "1\n",
      "(725, 2)\n",
      "2\n",
      "(39, 2)\n",
      "5\n",
      "(650, 2)\n",
      "4ER-\n",
      "(961, 2)\n",
      "6\n",
      "(218, 2)\n"
     ]
    }
   ],
   "source": [
    "for i in clusterframes:\n",
    "    print(i)\n",
    "    df = clusterframes[i]\n",
    "    print(df[df['qvalues'] <= 0.001].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
