{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import networkx as nx\n",
    "from networkx.readwrite import json_graph\n",
    "import json\n",
    "from IPython.core.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "relation_file = \"../data/ReactomePathwaysRelation.txt\"\n",
    "pathway_name = \"../data/ReactomePathways.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def generate_tree(relation_file = relation_file):\n",
    "    rel_df = pd.read_csv(relation_file, sep = \"\\t\", header = None, index_col = 0, names=['id'])\n",
    "    real_names = pd.read_csv(pathway_name, sep='\\t', index_col=0, names=['pathway', 'name', 'organism'])\n",
    "    \n",
    "    \n",
    "    cut = rel_df.index.str.contains('MMU') & rel_df['id'].str.contains('MMU')\n",
    "    rel_df = rel_df.loc[cut]\n",
    "    \n",
    "    namelist = []\n",
    "    for i in rel_df.index:\n",
    "        conversion = real_names.loc[i, 'name']\n",
    "        namelist.append(conversion)\n",
    "        \n",
    "    childlist = []   \n",
    "    for i in rel_df['id']:\n",
    "        conversion = real_names.loc[i, 'name']\n",
    "        childlist.append(conversion)\n",
    "    names = pd.DataFrame()\n",
    "    names['parent'] = namelist\n",
    "    names['child'] = childlist\n",
    "    \n",
    "    \n",
    "    G = nx.DiGraph()\n",
    "    G.add_edges_from(names.values)\n",
    "    roots = [n for n,d in G.in_degree() if d==0]\n",
    "    \n",
    "    roots_df = pd.DataFrame(columns = [['parentId', 'id']])\n",
    "    roots_df['id'] = roots\n",
    "    roots_df['parentId'] = 'Human'\n",
    "    \n",
    "    roots_df = pd.DataFrame(roots_df.values, columns = ['parentId', 'id'])\n",
    "    rel_df = pd.DataFrame(names.values, columns = ['parentId', 'id'])\n",
    "    \n",
    "    tree = roots_df.append(rel_df)\n",
    "    return tree\n",
    "\n",
    "rel_df = generate_tree()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "def default(o):\n",
    "     if isinstance(o, np.integer): return int(o)\n",
    "     raise TypeError\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                        value  ngenes  \\\n",
      "Chromosome Maintenance                              28.690694     112   \n",
      "RMTs methylate histone arginines                    27.805190      81   \n",
      "Cellular Senescence                                 27.248127     221   \n",
      "Nucleosome assembly                                 27.248127      75   \n",
      "Deposition of new CENPA-containing nucleosomes ...  27.248127      75   \n",
      "HDMs demethylate histones                           26.953585      50   \n",
      "Senescence-Associated Secretory Phenotype (SASP)    26.833820     123   \n",
      "Telomere Maintenance                                25.978580      85   \n",
      "Estrogen-dependent gene expression                  25.929925     151   \n",
      "Oxidative Stress Induced Senescence                 25.555362     135   \n",
      "Meiosis                                             25.506739     127   \n",
      "Reproduction                                        25.441606     157   \n",
      "Base Excision Repair                                25.441606     148   \n",
      "Activation of anterior HOX genes in hindbrain d...  25.369519     125   \n",
      "Activation of HOX genes during differentiation      25.369519     125   \n",
      "Transcriptional regulation by small RNAs            25.369519     109   \n",
      "RHO GTPases activate PKNs                           25.324784     100   \n",
      "Meiotic recombination                               25.214335      94   \n",
      "ESR-mediated signaling                              25.103617     231   \n",
      "HCMV Late Events                                    24.953541     121   \n",
      "Condensation of Prophase Chromosomes                24.830450      77   \n",
      "Transcriptional regulation of granulopoiesis        24.713803     106   \n",
      "Gene Silencing by RNA                               24.713803     153   \n",
      "Formation of the beta-catenin:TCF transactivati...  24.593877     152   \n",
      "DNA Double Strand Break Response                    24.588573      86   \n",
      "HCMV Early Events                                   24.557416     145   \n",
      "Pre-NOTCH Expression and Processing                 24.249882     150   \n",
      "Metalloprotease DUBs                                24.191437      38   \n",
      "DNA methylation                                     24.175055      66   \n",
      "Pre-NOTCH Transcription and Translation             24.175055     128   \n",
      "...                                                       ...     ...   \n",
      "Transcriptional regulation of white adipocyte d...   0.320208     145   \n",
      "Activated NTRK3 signals through RAS                  0.320208       9   \n",
      "Autophagy                                            0.320208     217   \n",
      "Transcriptional Regulation by MECP2                  0.320208      78   \n",
      "Signaling by ROBO receptors                          0.320208     268   \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...   0.319917       2   \n",
      "Constitutive Signaling by Overexpressed ERBB2        0.318890      12   \n",
      "Organic anion transport                              0.318890       5   \n",
      "Myoclonic epilepsy of Lafora                         0.318601       9   \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...   0.318535      58   \n",
      "Heme degradation                                     0.316860      16   \n",
      "FOXO-mediated transcription of oxidative stress...   0.316628      35   \n",
      "SHC-mediated cascade:FGFR4                           0.315403      21   \n",
      "Creation of C4 and C2 activators                     0.315284     103   \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...   0.312310      20   \n",
      "Nef mediated downregulation of CD28 cell surfac...   0.309792       2   \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)       0.308856       2   \n",
      "EPH-ephrin mediated repulsion of cells               0.308856      67   \n",
      "Release of Hh-Np from the secreting cell             0.306687      12   \n",
      "Regulation of expression of SLITs and ROBOs          0.306412     212   \n",
      "Butyrophilin (BTN) family interactions               0.306030      19   \n",
      "IKK complex recruitment mediated by RIP1             0.305917      24   \n",
      "Digestion and absorption                             0.305697      29   \n",
      "Conjugation of salicylate with glycine               0.305049       8   \n",
      "Biosynthesis of protectin and resolvin conjugat...   0.304080       2   \n",
      "CTLA4 inhibitory signaling                           0.303435      22   \n",
      "Norepinephrine Neurotransmitter Release Cycle        0.302632      18   \n",
      "Regulation of PTEN localization                      0.302360       9   \n",
      "Clathrin-mediated endocytosis                        0.302360     146   \n",
      "Adaptive Immune System                               0.301401    1413   \n",
      "\n",
      "                                                   Organism    max_val     C  \n",
      "Chromosome Maintenance                                Human  28.690694     1  \n",
      "RMTs methylate histone arginines                      Human  28.690694     2  \n",
      "Cellular Senescence                                   Human  28.690694     3  \n",
      "Nucleosome assembly                                   Human  28.690694     4  \n",
      "Deposition of new CENPA-containing nucleosomes ...    Human  28.690694     5  \n",
      "HDMs demethylate histones                             Human  28.690694     6  \n",
      "Senescence-Associated Secretory Phenotype (SASP)      Human  28.690694     7  \n",
      "Telomere Maintenance                                  Human  28.690694     8  \n",
      "Estrogen-dependent gene expression                    Human  28.690694     9  \n",
      "Oxidative Stress Induced Senescence                   Human  28.690694    10  \n",
      "Meiosis                                               Human  28.690694    11  \n",
      "Reproduction                                          Human  28.690694    12  \n",
      "Base Excision Repair                                  Human  28.690694    13  \n",
      "Activation of anterior HOX genes in hindbrain d...    Human  28.690694    14  \n",
      "Activation of HOX genes during differentiation        Human  28.690694    15  \n",
      "Transcriptional regulation by small RNAs              Human  28.690694    16  \n",
      "RHO GTPases activate PKNs                             Human  28.690694    17  \n",
      "Meiotic recombination                                 Human  28.690694    18  \n",
      "ESR-mediated signaling                                Human  28.690694    19  \n",
      "HCMV Late Events                                      Human  28.690694    20  \n",
      "Condensation of Prophase Chromosomes                  Human  28.690694    21  \n",
      "Transcriptional regulation of granulopoiesis          Human  28.690694    22  \n",
      "Gene Silencing by RNA                                 Human  28.690694    23  \n",
      "Formation of the beta-catenin:TCF transactivati...    Human  28.690694    24  \n",
      "DNA Double Strand Break Response                      Human  28.690694    25  \n",
      "HCMV Early Events                                     Human  28.690694    26  \n",
      "Pre-NOTCH Expression and Processing                   Human  28.690694    27  \n",
      "Metalloprotease DUBs                                  Human  28.690694    28  \n",
      "DNA methylation                                       Human  28.690694    29  \n",
      "Pre-NOTCH Transcription and Translation               Human  28.690694    30  \n",
      "...                                                     ...        ...   ...  \n",
      "Transcriptional regulation of white adipocyte d...    Human  28.690694  2045  \n",
      "Activated NTRK3 signals through RAS                   Human  28.690694  2046  \n",
      "Autophagy                                             Human  28.690694  2047  \n",
      "Transcriptional Regulation by MECP2                   Human  28.690694  2048  \n",
      "Signaling by ROBO receptors                           Human  28.690694  2049  \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...    Human  28.690694  2050  \n",
      "Constitutive Signaling by Overexpressed ERBB2         Human  28.690694  2051  \n",
      "Organic anion transport                               Human  28.690694  2052  \n",
      "Myoclonic epilepsy of Lafora                          Human  28.690694  2053  \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...    Human  28.690694  2054  \n",
      "Heme degradation                                      Human  28.690694  2055  \n",
      "FOXO-mediated transcription of oxidative stress...    Human  28.690694  2056  \n",
      "SHC-mediated cascade:FGFR4                            Human  28.690694  2057  \n",
      "Creation of C4 and C2 activators                      Human  28.690694  2058  \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...    Human  28.690694  2059  \n",
      "Nef mediated downregulation of CD28 cell surfac...    Human  28.690694  2060  \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)        Human  28.690694  2061  \n",
      "EPH-ephrin mediated repulsion of cells                Human  28.690694  2062  \n",
      "Release of Hh-Np from the secreting cell              Human  28.690694  2063  \n",
      "Regulation of expression of SLITs and ROBOs           Human  28.690694  2064  \n",
      "Butyrophilin (BTN) family interactions                Human  28.690694  2065  \n",
      "IKK complex recruitment mediated by RIP1              Human  28.690694  2066  \n",
      "Digestion and absorption                              Human  28.690694  2067  \n",
      "Conjugation of salicylate with glycine                Human  28.690694  2068  \n",
      "Biosynthesis of protectin and resolvin conjugat...    Human  28.690694  2069  \n",
      "CTLA4 inhibitory signaling                            Human  28.690694  2070  \n",
      "Norepinephrine Neurotransmitter Release Cycle         Human  28.690694  2071  \n",
      "Regulation of PTEN localization                       Human  28.690694  2072  \n",
      "Clathrin-mediated endocytosis                         Human  28.690694  2073  \n",
      "Adaptive Immune System                                Human  28.690694  2074  \n",
      "\n",
      "[2074 rows x 5 columns]\n",
      "                                                        value  ngenes  \\\n",
      "Chromosome Maintenance                              28.690694     112   \n",
      "RMTs methylate histone arginines                    27.805190      81   \n",
      "Cellular Senescence                                 27.248127     221   \n",
      "Nucleosome assembly                                 27.248127      75   \n",
      "Deposition of new CENPA-containing nucleosomes ...  27.248127      75   \n",
      "HDMs demethylate histones                           26.953585      50   \n",
      "Senescence-Associated Secretory Phenotype (SASP)    26.833820     123   \n",
      "Telomere Maintenance                                25.978580      85   \n",
      "Estrogen-dependent gene expression                  25.929925     151   \n",
      "Oxidative Stress Induced Senescence                 25.555362     135   \n",
      "Meiosis                                             25.506739     127   \n",
      "Reproduction                                        25.441606     157   \n",
      "Base Excision Repair                                25.441606     148   \n",
      "Activation of anterior HOX genes in hindbrain d...  25.369519     125   \n",
      "Activation of HOX genes during differentiation      25.369519     125   \n",
      "Transcriptional regulation by small RNAs            25.369519     109   \n",
      "RHO GTPases activate PKNs                           25.324784     100   \n",
      "Meiotic recombination                               25.214335      94   \n",
      "ESR-mediated signaling                              25.103617     231   \n",
      "HCMV Late Events                                    24.953541     121   \n",
      "Condensation of Prophase Chromosomes                24.830450      77   \n",
      "Transcriptional regulation of granulopoiesis        24.713803     106   \n",
      "Gene Silencing by RNA                               24.713803     153   \n",
      "Formation of the beta-catenin:TCF transactivati...  24.593877     152   \n",
      "DNA Double Strand Break Response                    24.588573      86   \n",
      "HCMV Early Events                                   24.557416     145   \n",
      "Pre-NOTCH Expression and Processing                 24.249882     150   \n",
      "Metalloprotease DUBs                                24.191437      38   \n",
      "DNA methylation                                     24.175055      66   \n",
      "Pre-NOTCH Transcription and Translation             24.175055     128   \n",
      "...                                                       ...     ...   \n",
      "Activated NTRK3 signals through RAS                  0.320208       9   \n",
      "Autophagy                                            0.320208     217   \n",
      "Transcriptional Regulation by MECP2                  0.320208      78   \n",
      "Signaling by ROBO receptors                          0.320208     268   \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...   0.319917       2   \n",
      "Constitutive Signaling by Overexpressed ERBB2        0.318890      12   \n",
      "Organic anion transport                              0.318890       5   \n",
      "Myoclonic epilepsy of Lafora                         0.318601       9   \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...   0.318535      58   \n",
      "Heme degradation                                     0.316860      16   \n",
      "FOXO-mediated transcription of oxidative stress...   0.316628      35   \n",
      "SHC-mediated cascade:FGFR4                           0.315403      21   \n",
      "Creation of C4 and C2 activators                     0.315284     103   \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...   0.312310      20   \n",
      "Nef mediated downregulation of CD28 cell surfac...   0.309792       2   \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)       0.308856       2   \n",
      "EPH-ephrin mediated repulsion of cells               0.308856      67   \n",
      "Release of Hh-Np from the secreting cell             0.306687      12   \n",
      "Regulation of expression of SLITs and ROBOs          0.306412     212   \n",
      "Butyrophilin (BTN) family interactions               0.306030      19   \n",
      "IKK complex recruitment mediated by RIP1             0.305917      24   \n",
      "Digestion and absorption                             0.305697      29   \n",
      "Conjugation of salicylate with glycine               0.305049       8   \n",
      "Biosynthesis of protectin and resolvin conjugat...   0.304080       2   \n",
      "CTLA4 inhibitory signaling                           0.303435      22   \n",
      "Norepinephrine Neurotransmitter Release Cycle        0.302632      18   \n",
      "Regulation of PTEN localization                      0.302360       9   \n",
      "Clathrin-mediated endocytosis                        0.302360     146   \n",
      "Adaptive Immune System                               0.301401    1413   \n",
      "Human                                                0.000000   23888   \n",
      "\n",
      "                                                   Organism    max_val       C  \n",
      "Chromosome Maintenance                                Human  28.690694     1.0  \n",
      "RMTs methylate histone arginines                      Human  28.690694     2.0  \n",
      "Cellular Senescence                                   Human  28.690694     3.0  \n",
      "Nucleosome assembly                                   Human  28.690694     4.0  \n",
      "Deposition of new CENPA-containing nucleosomes ...    Human  28.690694     5.0  \n",
      "HDMs demethylate histones                             Human  28.690694     6.0  \n",
      "Senescence-Associated Secretory Phenotype (SASP)      Human  28.690694     7.0  \n",
      "Telomere Maintenance                                  Human  28.690694     8.0  \n",
      "Estrogen-dependent gene expression                    Human  28.690694     9.0  \n",
      "Oxidative Stress Induced Senescence                   Human  28.690694    10.0  \n",
      "Meiosis                                               Human  28.690694    11.0  \n",
      "Reproduction                                          Human  28.690694    12.0  \n",
      "Base Excision Repair                                  Human  28.690694    13.0  \n",
      "Activation of anterior HOX genes in hindbrain d...    Human  28.690694    14.0  \n",
      "Activation of HOX genes during differentiation        Human  28.690694    15.0  \n",
      "Transcriptional regulation by small RNAs              Human  28.690694    16.0  \n",
      "RHO GTPases activate PKNs                             Human  28.690694    17.0  \n",
      "Meiotic recombination                                 Human  28.690694    18.0  \n",
      "ESR-mediated signaling                                Human  28.690694    19.0  \n",
      "HCMV Late Events                                      Human  28.690694    20.0  \n",
      "Condensation of Prophase Chromosomes                  Human  28.690694    21.0  \n",
      "Transcriptional regulation of granulopoiesis          Human  28.690694    22.0  \n",
      "Gene Silencing by RNA                                 Human  28.690694    23.0  \n",
      "Formation of the beta-catenin:TCF transactivati...    Human  28.690694    24.0  \n",
      "DNA Double Strand Break Response                      Human  28.690694    25.0  \n",
      "HCMV Early Events                                     Human  28.690694    26.0  \n",
      "Pre-NOTCH Expression and Processing                   Human  28.690694    27.0  \n",
      "Metalloprotease DUBs                                  Human  28.690694    28.0  \n",
      "DNA methylation                                       Human  28.690694    29.0  \n",
      "Pre-NOTCH Transcription and Translation               Human  28.690694    30.0  \n",
      "...                                                     ...        ...     ...  \n",
      "Activated NTRK3 signals through RAS                   Human  28.690694  2046.0  \n",
      "Autophagy                                             Human  28.690694  2047.0  \n",
      "Transcriptional Regulation by MECP2                   Human  28.690694  2048.0  \n",
      "Signaling by ROBO receptors                           Human  28.690694  2049.0  \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...    Human  28.690694  2050.0  \n",
      "Constitutive Signaling by Overexpressed ERBB2         Human  28.690694  2051.0  \n",
      "Organic anion transport                               Human  28.690694  2052.0  \n",
      "Myoclonic epilepsy of Lafora                          Human  28.690694  2053.0  \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...    Human  28.690694  2054.0  \n",
      "Heme degradation                                      Human  28.690694  2055.0  \n",
      "FOXO-mediated transcription of oxidative stress...    Human  28.690694  2056.0  \n",
      "SHC-mediated cascade:FGFR4                            Human  28.690694  2057.0  \n",
      "Creation of C4 and C2 activators                      Human  28.690694  2058.0  \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...    Human  28.690694  2059.0  \n",
      "Nef mediated downregulation of CD28 cell surfac...    Human  28.690694  2060.0  \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)        Human  28.690694  2061.0  \n",
      "EPH-ephrin mediated repulsion of cells                Human  28.690694  2062.0  \n",
      "Release of Hh-Np from the secreting cell              Human  28.690694  2063.0  \n",
      "Regulation of expression of SLITs and ROBOs           Human  28.690694  2064.0  \n",
      "Butyrophilin (BTN) family interactions                Human  28.690694  2065.0  \n",
      "IKK complex recruitment mediated by RIP1              Human  28.690694  2066.0  \n",
      "Digestion and absorption                              Human  28.690694  2067.0  \n",
      "Conjugation of salicylate with glycine                Human  28.690694  2068.0  \n",
      "Biosynthesis of protectin and resolvin conjugat...    Human  28.690694  2069.0  \n",
      "CTLA4 inhibitory signaling                            Human  28.690694  2070.0  \n",
      "Norepinephrine Neurotransmitter Release Cycle         Human  28.690694  2071.0  \n",
      "Regulation of PTEN localization                       Human  28.690694  2072.0  \n",
      "Clathrin-mediated endocytosis                         Human  28.690694  2073.0  \n",
      "Adaptive Immune System                                Human  28.690694  2074.0  \n",
      "Human                                                 Human  28.690694     NaN  \n",
      "\n",
      "[2075 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "def sunburst(in_df, outname = 'sun_tree.json'):\n",
    "    # in_df has as indexes the pathway reactome ID, 'value' as the colors to be plot, 'ngenes' as the width, and 'descr' as the description of the pathway\n",
    "    \n",
    "    #in_df = in_df.loc[[x for x in in_df.index if 'MMU' in x]]\n",
    "    max_val = in_df['value'].max()\n",
    "    in_df['max_val'] = max_val\n",
    "    sorted_in_df = in_df.sort_values(by='value', ascending = False)\n",
    "    sorted_in_df['C'] = np.arange(len(sorted_in_df))+1\n",
    "    in_df = sorted_in_df\n",
    "    print(in_df)\n",
    "    topPaths = rel_df.loc[(rel_df['parentId'] == 'Human'), 'id']\n",
    "    homoNgenes = np.sum(in_df.loc[[x in topPaths.tolist() for x in in_df.index],'ngenes'])\n",
    "    homoNode = pd.DataFrame([[0,homoNgenes,\"Human\", max_val]], columns = [\"value\", \"ngenes\", \"Organism\", 'max_val']).xs(0)\n",
    "    homoNode.name = 'Human'\n",
    "    \n",
    "    in_df = in_df.append(homoNode)\n",
    "    print(in_df)\n",
    "    \n",
    "    topDict = in_df.to_dict()\n",
    "    \n",
    "    \n",
    "    pathways = in_df.index\n",
    "    \n",
    "    n_path = len(pathways)\n",
    "    \n",
    "    subset_vec = [x in pathways for x in rel_df.iloc[:,0]] and [x in pathways for x in rel_df.iloc[:,1]] \n",
    "    sub_rel_df = rel_df[subset_vec] \n",
    "    \n",
    "    G = nx.DiGraph()\n",
    "    \n",
    "    G.add_nodes_from(pathways)\n",
    "    G.add_edges_from(sub_rel_df.values)\n",
    "    \n",
    "    tree = nx.algorithms.dag.dag_to_branching(G)\n",
    "    \n",
    "    secondDict = nx.get_node_attributes(tree,'source')\n",
    "    \n",
    "    thirdDict = {'value':{}, 'ngenes':{}, 'max_val': {}}\n",
    "    for key, value in secondDict.items():\n",
    "        thirdDict['value'].update({key : topDict['value'][value]})\n",
    "        thirdDict['ngenes'].update({key : topDict['ngenes'][value]})\n",
    "        thirdDict['max_val'].update({key : topDict['max_val'][value]})\n",
    "\n",
    "        \n",
    "        \n",
    "    nx.set_node_attributes(tree, thirdDict['value'], name = 'value')\n",
    "    nx.set_node_attributes(tree, thirdDict['ngenes'], name = 'ngenes')\n",
    "    nx.set_node_attributes(tree, thirdDict['max_val'], name = 'max_val')\n",
    "    \n",
    "    root = [v for v, d in tree.in_degree() if d == 0][0]\n",
    "    out_json = json_graph.tree_data(tree, root)\n",
    "    \n",
    "    with open(outname, 'w') as outfile:\n",
    "        json.dump(out_json, outfile, default=default)\n",
    "\n",
    "sunburst(df_dict['Cluster 1 q-values'], outname = 'sunburst/intclust__1.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_reactome(file_name, gene_name_start = \"ENSG0\"):\n",
    "    df = pd.read_csv(file_name, sep='\\t', header=None)\n",
    "    \n",
    "    if gene_name_start == None:\n",
    "        sub_df = df\n",
    "    else:\n",
    "        subset_vec = df[0].str.startswith(gene_name_start)\n",
    "        sub_df = df.loc[subset_vec]\n",
    "    \n",
    "    genes_df = sub_df.groupby(1)[0].apply(list)\n",
    "    names_df = sub_df.groupby(1)[3].max()\n",
    "    \n",
    "    out_df = pd.concat([genes_df,names_df], axis=1)\n",
    "    out_df.columns = ['genes', 'pathway_name']\n",
    "    out_df.index = out_df.pathway_name\n",
    "    \n",
    "    return out_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-138-d6b68f1e301c>, line 24)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-138-d6b68f1e301c>\"\u001b[0;36m, line \u001b[0;32m24\u001b[0m\n\u001b[0;31m    df_dict[i] = df\u001b[0m\n\u001b[0m          ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\"../data/pathway_qvalues.csv\", index_col = 0)\n",
    "\n",
    "clusterindex = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21]\n",
    "cluster_df = cluster_df.iloc[:,clusterindex]\n",
    "reactome_ngenes = read_reactome(\"../data/Ensembl2Reactome_All_Levels.txt\")\n",
    "\n",
    "length_dict = {}\n",
    "for i in cluster_df.index:\n",
    "    nr_genes = len(reactome_ngenes.loc[i, \"genes\"])\n",
    "    #print(nr_genes)\n",
    "    length_dict[i] = nr_genes\n",
    "\n",
    "cluster_df['ngenes'] = cluster_df.index.map(length_dict)\n",
    "\n",
    "\n",
    "df_dict = {}\n",
    "\n",
    "for i in cluster_df.iloc[:,:-1]:\n",
    "    df = pd.DataFrame(index = cluster_df.index)\n",
    "    df['value'] = cluster_df.loc[:,i]\n",
    "    df['ngenes'] = cluster_df.loc[:,'ngenes']\n",
    "    df['Organism'] = 'Human'\n",
    "    print(df['value'].max\n",
    "    df_dict[i] = df\n",
    "    \n",
    "for i in df_dict:\n",
    "    clust = i.strip('Cluster q-va')\n",
    "    sunburst(df_dict[i], outname = f'../exp/intclust_{clust}.json')\n",
    "    \n",
    "#sunburst(cluster_df, outname = 'pathways_per_intclusts.json')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
