{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from ast import literal_eval\n",
    "import numpy as np\n",
    "from IPython.core.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))\n",
    "from statistics import mean, median\n",
    "import scipy\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn import preprocessing\n",
    "from gprofiler import GProfiler\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.pyplot import figure\n",
    "figure(num=None, figsize=(8, 6), dpi=80, facecolor='w', edgecolor='k')\n",
    "import operator\n",
    "import qvalue as qv\n",
    "\n",
    "\n",
    "#Reactome file containing information on pathways, the genes they contain and pathway name, also including the illumina identifier for the genes.\n",
    "\n",
    "reactome = pd.read_csv('../data/reactome.csv', sep=',', index_col = 0)\n",
    "\n",
    "def read_reactome(file_name, gene_name_start = \"ENSG0\"):\n",
    "    df = pd.read_csv(file_name, sep='\\t', header=None)\n",
    "    \n",
    "    if gene_name_start == None:\n",
    "        sub_df = df\n",
    "    else:\n",
    "        subset_vec = df[0].str.startswith(gene_name_start)\n",
    "        sub_df = df.loc[subset_vec]\n",
    "    \n",
    "    genes_df = sub_df.groupby(1)[0].apply(list)\n",
    "    names_df = sub_df.groupby(1)[3].max()\n",
    "    \n",
    "    out_df = pd.concat([genes_df,names_df], axis=1)\n",
    "    out_df.columns = ['genes', 'pathway_name']\n",
    "    \n",
    "    return out_df\n",
    "\n",
    "low_level = read_reactome('../data/Ensembl2Reactome_All_Levels.txt.gz')\n",
    "\n",
    "def my_pca(df, n_pc=1, normalize=True):\n",
    "    df = df.dropna(axis = 0, how = 'all')#redundant, but keeping it just in case\n",
    "    X = df.values.T\n",
    "    if normalize:\n",
    "        X2 = preprocessing.scale(X)\n",
    "    else:\n",
    "        X2 = X\n",
    "    pca = PCA(n_components = n_pc)\n",
    "    pca.fit(X2)\n",
    "    my_pca.pca = pca  #needed for components\n",
    "    Xnew = pca.fit_transform(X2)\n",
    "    out_df = pd.DataFrame(Xnew.transpose(), index=list(range(1,n_pc+1)), columns=df.columns)\n",
    "    out_df = out_df.transpose()\n",
    "    \n",
    "    return out_df, my_pca.pca.components_, my_pca.pca.explained_variance_ratio_\n",
    "\n",
    "#Importing metabric dataset, dividing up what is clinical/expression data and changing the type of the expression columns to float\n",
    "metabric_data = pd.read_csv('../data/metabric.csv.gz')\n",
    "#clinical_data = metabric_data.iloc[:27, :]\n",
    "expression_data = metabric_data.iloc[27:,:]\n",
    "\n",
    "#print(expression_data.columns)\n",
    "dtypedict = {}\n",
    "for i in expression_data.columns[1:]:\n",
    "    dtypedict[i] = 'float32'\n",
    "expression_data = expression_data.astype(dtypedict)\n",
    "\n",
    "\n",
    "\n",
    "new_clinical_patient = pd.read_csv('../data/data_clinical_patient.txt', sep='\\t', index_col=0).iloc[4:]\n",
    "new_clinical_sample = pd.read_csv('../data/data_clinical_sample.txt', sep='\\t', index_col=0).iloc[4:]\n",
    "new_clinical = pd.concat([new_clinical_patient, new_clinical_sample.reindex(new_clinical_patient.index)], axis=1)\n",
    "new_clinical['Triple Neg'] = new_clinical.apply(lambda row: True if ((row['ER Status'] == 'Negative') \n",
    "                                                                     and (row['PR Status'] == 'Negative') \n",
    "                                                                     and (row['HER2 Status'] == 'Negative')) else False, axis = 1)\n",
    "\n",
    "new_clinical['ER-/PR-/HER2+'] = new_clinical.apply(lambda row: True if ((row['ER Status'] == 'Negative') \n",
    "                                                                     and (row['PR Status'] == 'Negative') \n",
    "                                                                     and (row['HER2 Status'] == 'Positive')) else False, axis = 1)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "genes = expression_data['Unnamed: 0'].values.tolist()\n",
    "\n",
    "gp = GProfiler(return_dataframe = True)\n",
    "gp = gp.convert(organism='hsapiens',\n",
    "          query=genes)\n",
    "\n",
    "gp = gp.loc[gp['n_converted'] == 1]\n",
    "gp = gp.loc[gp['name'] != 'None']\n",
    "gp = gp.set_index('incoming')\n",
    "gprofiler_names = gp\n",
    "gprofiler_names\n",
    "\n",
    "dataset = expression_data.set_index('Unnamed: 0') #gene_patient\n",
    "pca_per_pathway = pd.DataFrame(index=expression_data.columns)\n",
    "\n",
    "real_gene_names = pd.read_csv('../data/illumina2symbol.txt', sep=\"\\t\", index_col = 0)\n",
    "\n",
    "\n",
    "\n",
    "genes_components_per_pathway = {} #nested dictionary where the 'outer dictionary' is the pathway names as keys and values are \n",
    "                                  #another dictionary with genes as keys and components as values\n",
    "\n",
    "for pathway in reactome.index:\n",
    "    genes = reactome.loc[pathway, \"illumina\"]\n",
    "    genes = literal_eval(genes)\n",
    "    genes = list(filter(lambda a: a != 'NaN', genes))\n",
    "    pathwaydata = dataset.loc[genes]\n",
    "    if pathwaydata.index.empty == True:\n",
    "        pass\n",
    "    else:\n",
    "        pathwaydata = pathwaydata.dropna(axis = 0, how = 'any') #has to be done so the lists match, this makes the dropna in my_pca function obsolete\n",
    "        presentgenes = pathwaydata.index.values.tolist()\n",
    "        if len(presentgenes) <= 1:\n",
    "            pass\n",
    "        else:\n",
    "            res, components, explained_variance = my_pca(pathwaydata)\n",
    "            pathwayname = reactome.loc[pathway, 'pathway_name']\n",
    "            pca_per_pathway[pathwayname] = res\n",
    "\n",
    "            components = components.tolist()[0]\n",
    "            innerdict = {}\n",
    "            for i in range(0, len(presentgenes)):\n",
    "                component = components[i]\n",
    "                gene = genes[i]\n",
    "                if gene in real_gene_names.index:\n",
    "                    real_name = real_gene_names.loc[gene, \"symbol\"]\n",
    "                    innerdict[real_name] = component\n",
    "                elif gene in gprofiler_names.index:\n",
    "                    real_name = gprofiler_names.loc[gene, 'name']\n",
    "                    innerdict[real_name] = component\n",
    "                else:\n",
    "                    innerdict[gene] = component\n",
    "            sorted_innerdict = sorted(innerdict.items(), key = operator.itemgetter(1), reverse = True)\n",
    "            genes_components_per_pathway[pathwayname] = [sorted_innerdict, explained_variance.flat[0]]\n",
    "\n",
    "pca_per_pathway = pca_per_pathway.iloc[1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_df = pd.concat([pca_per_pathway, new_clinical.reindex(pca_per_pathway.index)], axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MB-0112    150.0\n",
       "MB-0406    180.0\n",
       "MB-0453    100.0\n",
       "MB-0656    100.0\n",
       "MB-0657    100.0\n",
       "MB-0660    160.0\n",
       "MB-3436    130.0\n",
       "MB-6063    120.0\n",
       "MB-7148    130.0\n",
       "MB-7165    182.0\n",
       "Name: Tumor Size, dtype: float64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "size = full_df['Tumor Size'].astype(float)\n",
    "size[size >= 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 1 p-values\n",
      "Interleukin-6 signaling                                   5.186702e-04\n",
      "Apoptosis                                                 3.172921e-05\n",
      "Hemostasis                                                1.654836e-03\n",
      "Intrinsic Pathway for Apoptosis                           9.744648e-02\n",
      "PKB-mediated events                                       2.635249e-03\n",
      "PI3K Cascade                                              1.102894e-01\n",
      "MAPK3 (ERK1) activation                                   3.697860e-02\n",
      "Translesion synthesis by REV1                             1.084238e-05\n",
      "Translesion synthesis by Y family DNA polymeras...        5.224905e-16\n",
      "Recognition of DNA damage by PCNA-containing re...        4.624878e-12\n",
      "Translesion Synthesis by POLH                             6.687313e-08\n",
      "Recognition and association of DNA glycosylase ...        2.122693e-26\n",
      "Cleavage of the damaged pyrimidine                        2.122693e-26\n",
      "Recognition and association of DNA glycosylase ...        4.773533e-26\n",
      "Cleavage of the damaged purine                            4.773533e-26\n",
      "Displacement of DNA glycosylase by APEX1                  4.353853e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair            4.005788e-16\n",
      "Resolution of AP sites via the multiple-nucleot...        6.134033e-15\n",
      "Resolution of AP sites via the single-nucleotid...        1.435967e-01\n",
      "SLBP independent Processing of Histone Pre-mRNAs          2.314352e-11\n",
      "Activation of BIM and translocation to mitochon...        4.218222e-01\n",
      "Activation of BAD and translocation to mitochon...        1.822113e-01\n",
      "Activation of NOXA and translocation to mitocho...        4.196699e-01\n",
      "Activation and oligomerization of BAK protein             1.041101e-01\n",
      "BH3-only proteins associate with and inactivate...        1.373968e-04\n",
      "Release of apoptotic factors from the mitochondria        1.538820e-11\n",
      "Formation of apoptosome                                   2.534480e-06\n",
      "Activation of caspases through apoptosome-media...        2.440473e-03\n",
      "Cytochrome c-mediated apoptotic response                  1.583060e-05\n",
      "SMAC (DIABLO) binds to IAPs                               2.519045e-10\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               2.696089e-05\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        7.477461e-03\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        1.912407e-01\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        4.175796e-01\n",
      "MyD88 dependent cascade initiated on endosome             1.535840e-01\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        7.339643e-03\n",
      "N-glycan antennae elongation in the medial/tran...        3.839682e-02\n",
      "N-Glycan antennae elongation                              1.292233e-07\n",
      "Reactions specific to the complex N-glycan synt...        8.736699e-03\n",
      "Retinoid metabolism and transport                         2.182472e-11\n",
      "MyD88 cascade initiated on plasma membrane                4.064523e-01\n",
      "Nonsense Mediated Decay (NMD) independent of th...        2.425522e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        2.623748e-01\n",
      "Termination of O-glycan biosynthesis                      1.954879e-02\n",
      "Amyloid fiber formation                                   1.194285e-26\n",
      "Serine biosynthesis                                       5.859981e-03\n",
      "GABA receptor activation                                  4.713226e-04\n",
      "GABA B receptor activation                                4.869945e-04\n",
      "Regulation of Complement cascade                          3.358472e-01\n",
      "Growth hormone receptor signaling                         1.071568e-02\n",
      "Antigen processing: Ubiquitination & Proteasome...        8.743192e-04\n",
      "Class I MHC mediated antigen processing & prese...        3.368259e-02\n",
      "Antigen Presentation: Folding, assembly and pep...        1.444640e-01\n",
      "Kinesins                                                  3.877829e-14\n",
      "Factors involved in megakaryocyte development a...        4.903330e-16\n",
      "Antigen activates B Cell Receptor (BCR) leading...        2.747768e-01\n",
      "Signaling by the B Cell Receptor (BCR)                    6.008986e-02\n",
      "Ion channel transport                                     1.949687e-03\n",
      "Activation of GABAB receptors                             4.869945e-04\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        2.476705e-03\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 1 p-values  \\\n",
      "Chromosome Maintenance                                       32.007503   \n",
      "RMTs methylate histone arginines                             30.820969   \n",
      "Cellular Senescence                                          29.984688   \n",
      "Nucleosome assembly                                          29.865966   \n",
      "Deposition of new CENPA-containing nucleosomes ...           29.865966   \n",
      "HDMs demethylate histones                                    29.492243   \n",
      "Senescence-Associated Secretory Phenotype (SASP)             29.305531   \n",
      "Telomere Maintenance                                         28.392299   \n",
      "Estrogen-dependent gene expression                           28.292491   \n",
      "Oxidative Stress Induced Senescence                          27.872171   \n",
      "Meiosis                                                      27.782155   \n",
      "Reproduction                                                 27.652561   \n",
      "Base Excision Repair                                         27.644471   \n",
      "Transcriptional regulation by small RNAs                     27.485140   \n",
      "Activation of HOX genes during differentiation               27.482208   \n",
      "Activation of anterior HOX genes in hindbrain d...           27.482208   \n",
      "RHO GTPases activate PKNs                                    27.411144   \n",
      "Meiotic recombination                                        27.275871   \n",
      "ESR-mediated signaling                                       27.141672   \n",
      "HCMV Late Events                                             26.969320   \n",
      "Condensation of Prophase Chromosomes                         26.825039   \n",
      "Transcriptional regulation of granulopoiesis                 26.674659   \n",
      "Gene Silencing by RNA                                        26.668884   \n",
      "Formation of the beta-catenin:TCF transactivati...           26.530474   \n",
      "DNA Double Strand Break Response                             26.507442   \n",
      "HCMV Early Events                                            26.459252   \n",
      "Pre-NOTCH Expression and Processing                          26.135327   \n",
      "Metalloprotease DUBs                                         26.061087   \n",
      "Pre-NOTCH Transcription and Translation                      26.024008   \n",
      "DNA methylation                                              26.014742   \n",
      "...                                                                ...   \n",
      "TNF receptor superfamily (TNFSF) members mediat...            0.326090   \n",
      "Competing endogenous RNAs (ceRNAs) regulate PTE...            0.325978   \n",
      "Axon guidance                                                 0.325699   \n",
      "Transcriptional Regulation by MECP2                           0.325587   \n",
      "Transcriptional regulation of white adipocyte d...            0.325587   \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...            0.324972   \n",
      "Organic anion transport                                       0.323633   \n",
      "Constitutive Signaling by Overexpressed ERBB2                 0.323522   \n",
      "Myoclonic epilepsy of Lafora                                  0.323021   \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...            0.322743   \n",
      "Heme degradation                                              0.321355   \n",
      "FOXO-mediated transcription of oxidative stress...            0.320524   \n",
      "SHC-mediated cascade:FGFR4                                    0.318978   \n",
      "Creation of C4 and C2 activators                              0.318647   \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...            0.315571   \n",
      "Nef mediated downregulation of CD28 cell surfac...            0.312734   \n",
      "EPH-ephrin mediated repulsion of cells                        0.311430   \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.311322   \n",
      "Release of Hh-Np from the secreting cell                      0.308996   \n",
      "Digestion and absorption                                      0.308726   \n",
      "Regulation of expression of SLITs and ROBOs                   0.308511   \n",
      "Butyrophilin (BTN) family interactions                        0.307918   \n",
      "IKK complex recruitment mediated by RIP1                      0.307596   \n",
      "Conjugation of salicylate with glycine                        0.306307   \n",
      "Biosynthesis of protectin and resolvin conjugat...            0.305129   \n",
      "CTLA4 inhibitory signaling                                    0.304274   \n",
      "Clathrin-mediated endocytosis                                 0.302675   \n",
      "Regulation of PTEN localization                               0.302569   \n",
      "Norepinephrine Neurotransmitter Release Cycle                 0.301613   \n",
      "Adaptive Immune System                                        0.301401   \n",
      "\n",
      "                                                    Cluster 1 q-values  \n",
      "Chromosome Maintenance                                       28.690694  \n",
      "RMTs methylate histone arginines                             27.805190  \n",
      "Cellular Senescence                                          27.248127  \n",
      "Nucleosome assembly                                          27.248127  \n",
      "Deposition of new CENPA-containing nucleosomes ...           27.248127  \n",
      "HDMs demethylate histones                                    26.953585  \n",
      "Senescence-Associated Secretory Phenotype (SASP)             26.833820  \n",
      "Telomere Maintenance                                         25.978580  \n",
      "Estrogen-dependent gene expression                           25.929925  \n",
      "Oxidative Stress Induced Senescence                          25.555362  \n",
      "Meiosis                                                      25.506739  \n",
      "Reproduction                                                 25.441606  \n",
      "Base Excision Repair                                         25.441606  \n",
      "Transcriptional regulation by small RNAs                     25.369519  \n",
      "Activation of HOX genes during differentiation               25.369519  \n",
      "Activation of anterior HOX genes in hindbrain d...           25.369519  \n",
      "RHO GTPases activate PKNs                                    25.324784  \n",
      "Meiotic recombination                                        25.214335  \n",
      "ESR-mediated signaling                                       25.103617  \n",
      "HCMV Late Events                                             24.953541  \n",
      "Condensation of Prophase Chromosomes                         24.830450  \n",
      "Transcriptional regulation of granulopoiesis                 24.713803  \n",
      "Gene Silencing by RNA                                        24.713803  \n",
      "Formation of the beta-catenin:TCF transactivati...           24.593877  \n",
      "DNA Double Strand Break Response                             24.588573  \n",
      "HCMV Early Events                                            24.557416  \n",
      "Pre-NOTCH Expression and Processing                          24.249882  \n",
      "Metalloprotease DUBs                                         24.191437  \n",
      "Pre-NOTCH Transcription and Translation                      24.175055  \n",
      "DNA methylation                                              24.175055  \n",
      "...                                                                ...  \n",
      "TNF receptor superfamily (TNFSF) members mediat...            0.320320  \n",
      "Competing endogenous RNAs (ceRNAs) regulate PTE...            0.320320  \n",
      "Axon guidance                                                 0.320320  \n",
      "Transcriptional Regulation by MECP2                           0.320320  \n",
      "Transcriptional regulation of white adipocyte d...            0.320320  \n",
      "Defective ABCC9 causes dilated cardiomyopathy 1...            0.319917  \n",
      "Organic anion transport                                       0.318890  \n",
      "Constitutive Signaling by Overexpressed ERBB2                 0.318890  \n",
      "Myoclonic epilepsy of Lafora                                  0.318601  \n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 h...            0.318535  \n",
      "Heme degradation                                              0.317358  \n",
      "FOXO-mediated transcription of oxidative stress...            0.316739  \n",
      "SHC-mediated cascade:FGFR4                                    0.315403  \n",
      "Creation of C4 and C2 activators                              0.315284  \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...            0.312419  \n",
      "Nef mediated downregulation of CD28 cell surfac...            0.309792  \n",
      "EPH-ephrin mediated repulsion of cells                        0.308802  \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.308802  \n",
      "Release of Hh-Np from the secreting cell                      0.306687  \n",
      "Digestion and absorption                                      0.306627  \n",
      "Regulation of expression of SLITs and ROBOs                   0.306622  \n",
      "Butyrophilin (BTN) family interactions                        0.306240  \n",
      "IKK complex recruitment mediated by RIP1                      0.306127  \n",
      "Conjugation of salicylate with glycine                        0.305049  \n",
      "Biosynthesis of protectin and resolvin conjugat...            0.304080  \n",
      "CTLA4 inhibitory signaling                                    0.303435  \n",
      "Clathrin-mediated endocytosis                                 0.302150  \n",
      "Regulation of PTEN localization                               0.302150  \n",
      "Norepinephrine Neurotransmitter Release Cycle                 0.301404  \n",
      "Adaptive Immune System                                        0.301401  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 10 p-values\n",
      "Interleukin-6 signaling                                    8.113491e-08\n",
      "Apoptosis                                                  9.733071e-60\n",
      "Hemostasis                                                 4.866084e-31\n",
      "Intrinsic Pathway for Apoptosis                            6.717934e-89\n",
      "PKB-mediated events                                        4.035898e-05\n",
      "PI3K Cascade                                               3.588494e-01\n",
      "MAPK3 (ERK1) activation                                    1.685368e-27\n",
      "Translesion synthesis by REV1                              4.862523e-62\n",
      "Translesion synthesis by Y family DNA polymeras...         2.988992e-80\n",
      "Recognition of DNA damage by PCNA-containing re...         6.726642e-79\n",
      "Translesion Synthesis by POLH                              1.052467e-56\n",
      "Recognition and association of DNA glycosylase ...         1.540760e-02\n",
      "Cleavage of the damaged pyrimidine                         1.540760e-02\n",
      "Recognition and association of DNA glycosylase ...         1.794004e-02\n",
      "Cleavage of the damaged purine                             1.794004e-02\n",
      "Displacement of DNA glycosylase by APEX1                   1.636136e-05\n",
      "POLB-Dependent Long Patch Base Excision Repair             2.603148e-41\n",
      "Resolution of AP sites via the multiple-nucleot...         8.022130e-81\n",
      "Resolution of AP sites via the single-nucleotid...         2.447003e-06\n",
      "SLBP independent Processing of Histone Pre-mRNAs           3.843756e-59\n",
      "Activation of BIM and translocation to mitochon...         2.750130e-12\n",
      "Activation of BAD and translocation to mitochon...         1.788741e-71\n",
      "Activation of NOXA and translocation to mitocho...         3.967036e-46\n",
      "Activation and oligomerization of BAK protein              5.101658e-15\n",
      "BH3-only proteins associate with and inactivate...         3.181037e-52\n",
      "Release of apoptotic factors from the mitochondria         1.875228e-02\n",
      "Formation of apoptosome                                    1.231837e-16\n",
      "Activation of caspases through apoptosome-media...         4.997623e-20\n",
      "Cytochrome c-mediated apoptotic response                   5.106461e-13\n",
      "SMAC (DIABLO) binds to IAPs                                1.855012e-01\n",
      "...                                                                 ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib                3.735455e-01\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...         3.630514e-14\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...         4.867215e-43\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...         4.105805e-16\n",
      "MyD88 dependent cascade initiated on endosome              8.827650e-42\n",
      "IRAK2 mediated activation of TAK1 complex upon ...         1.316297e-12\n",
      "N-glycan antennae elongation in the medial/tran...         5.880050e-62\n",
      "N-Glycan antennae elongation                               2.197468e-30\n",
      "Reactions specific to the complex N-glycan synt...         1.839690e-59\n",
      "Retinoid metabolism and transport                          7.753489e-26\n",
      "MyD88 cascade initiated on plasma membrane                 5.074291e-35\n",
      "Nonsense Mediated Decay (NMD) independent of th...         1.230198e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...         1.383414e-01\n",
      "Termination of O-glycan biosynthesis                       3.124746e-03\n",
      "Amyloid fiber formation                                    4.647414e-01\n",
      "Serine biosynthesis                                       3.419670e-104\n",
      "GABA receptor activation                                   1.535377e-16\n",
      "GABA B receptor activation                                 2.500667e-18\n",
      "Regulation of Complement cascade                           2.180431e-23\n",
      "Growth hormone receptor signaling                          3.723807e-78\n",
      "Antigen processing: Ubiquitination & Proteasome...         2.601348e-17\n",
      "Class I MHC mediated antigen processing & prese...         1.858449e-54\n",
      "Antigen Presentation: Folding, assembly and pep...         6.659953e-42\n",
      "Kinesins                                                   7.419908e-96\n",
      "Factors involved in megakaryocyte development a...        8.158073e-102\n",
      "Antigen activates B Cell Receptor (BCR) leading...         2.305780e-21\n",
      "Signaling by the B Cell Receptor (BCR)                     9.795044e-41\n",
      "Ion channel transport                                     1.690371e-100\n",
      "Activation of GABAB receptors                              2.500667e-18\n",
      "Inhibition  of voltage gated Ca2+ channels via ...         4.466268e-07\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 10 p-values  \\\n",
      "Transcriptional Regulation by MECP2                          117.073581   \n",
      "Metabolism of nucleotides                                    114.551727   \n",
      "G1 Phase                                                     112.833506   \n",
      "Cyclin D associated events in G1                             112.833506   \n",
      "SLC transporter disorders                                    111.215980   \n",
      "SLC-mediated transmembrane transport                         110.121892   \n",
      "SUMOylation of DNA replication proteins                      108.636816   \n",
      "RHO GTPase Effectors                                         108.588200   \n",
      "RHO GTPases Activate Formins                                 107.265588   \n",
      "SUMO E3 ligases SUMOylate target proteins                    106.897101   \n",
      "Oncogene Induced Senescence                                  106.891220   \n",
      "DAG and IP3 signaling                                        105.951164   \n",
      "Cardiac conduction                                           105.736984   \n",
      "Purine catabolism                                            105.455255   \n",
      "Cell Cycle, Mitotic                                          105.417876   \n",
      "Ubiquinol biosynthesis                                       105.309280   \n",
      "G2 Phase                                                     105.301108   \n",
      "Cell Cycle                                                   105.193740   \n",
      "Transport of bile salts and organic acids, meta...           104.892937   \n",
      "Signaling by Rho GTPases                                     104.788108   \n",
      "SUMOylation                                                  104.649582   \n",
      "Activation of AMPK downstream of NMDARs                      104.289151   \n",
      "Cell Cycle Checkpoints                                       104.261273   \n",
      "Amino acid transport across the plasma membrane              104.249658   \n",
      "Miscellaneous transport and binding events                   104.231076   \n",
      "M Phase                                                      104.041865   \n",
      "Serine biosynthesis                                          103.466016   \n",
      "BBSome-mediated cargo-targeting to cilium                    103.383884   \n",
      "Transcription of E2F targets under negative con...           103.353815   \n",
      "Mitotic G2-G2/M phases                                       103.263637   \n",
      "...                                                                 ...   \n",
      "Phospholipase C-mediated cascade; FGFR4                        0.370617   \n",
      "Cell death signalling via NRAGE, NRIF and NADE                 0.368461   \n",
      "Platelet Adhesion to exposed collagen                          0.365632   \n",
      "RHO GTPases activate PKNs                                      0.363060   \n",
      "Transport of gamma-carboxylated protein precurs...             0.360260   \n",
      "Synthesis, secretion, and inactivation of Gluco...             0.360163   \n",
      "DARPP-32 events                                                0.357666   \n",
      "Hyaluronan biosynthesis and export                             0.352943   \n",
      "Metabolism of RNA                                              0.347793   \n",
      "PLCG1 events in ERBB2 signaling                                0.347699   \n",
      "Calnexin/calreticulin cycle                                    0.344139   \n",
      "mRNA Splicing - Major Pathway                                  0.342788   \n",
      "Triglyceride metabolism                                        0.341486   \n",
      "Asparagine N-linked glycosylation                              0.336865   \n",
      "Amyloid fiber formation                                        0.332789   \n",
      "Retinoid cycle disease events                                  0.328972   \n",
      "Diseases associated with visual transduction                   0.327527   \n",
      "MET activates RAP1 and RAC1                                    0.327076   \n",
      "Nitric oxide stimulates guanylate cyclase                      0.326085   \n",
      "Defective MUT causes methylmalonic aciduria mut...             0.325276   \n",
      "Defective MMAA causes methylmalonic aciduria ty...             0.325142   \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...             0.324424   \n",
      "Relaxin receptors                                              0.323037   \n",
      "Methionine salvage pathway                                     0.322902   \n",
      "HSP90 chaperone cycle for steroid hormone recep...             0.314666   \n",
      "Listeria monocytogenes entry into host cells                   0.313965   \n",
      "ER Quality Control Compartment (ERQC)                          0.307440   \n",
      "N-glycan trimming in the ER and Calnexin/Calret...             0.306146   \n",
      "ESR-mediated signaling                                         0.304942   \n",
      "Regulation of gene expression in early pancreat...             0.301477   \n",
      "\n",
      "                                                    Cluster 10 q-values  \n",
      "Transcriptional Regulation by MECP2                          113.756772  \n",
      "Metabolism of nucleotides                                    111.535949  \n",
      "G1 Phase                                                     110.118757  \n",
      "Cyclin D associated events in G1                             110.118757  \n",
      "SLC transporter disorders                                    108.598141  \n",
      "SLC-mediated transmembrane transport                         107.583234  \n",
      "SUMOylation of DNA replication proteins                      106.174481  \n",
      "RHO GTPase Effectors                                         106.174481  \n",
      "RHO GTPases Activate Formins                                 104.903022  \n",
      "SUMO E3 ligases SUMOylate target proteins                    104.615804  \n",
      "Oncogene Induced Senescence                                  104.615804  \n",
      "DAG and IP3 signaling                                        103.713537  \n",
      "Cardiac conduction                                           103.534118  \n",
      "Purine catabolism                                            103.284574  \n",
      "Cell Cycle, Mitotic                                          103.277158  \n",
      "Ubiquinol biosynthesis                                       103.214748  \n",
      "G2 Phase                                                     103.214748  \n",
      "Cell Cycle                                                   103.132204  \n",
      "Transport of bile salts and organic acids, meta...           102.854882  \n",
      "Signaling by Rho GTPases                                     102.772329  \n",
      "SUMOylation                                                  102.654993  \n",
      "Activation of AMPK downstream of NMDARs                      102.314765  \n",
      "Cell Cycle Checkpoints                                       102.313061  \n",
      "Amino acid transport across the plasma membrane              102.313061  \n",
      "Miscellaneous transport and binding events                   102.312207  \n",
      "M Phase                                                      102.140029  \n",
      "Serine biosynthesis                                          101.580571  \n",
      "BBSome-mediated cargo-targeting to cilium                    101.514233  \n",
      "Transcription of E2F targets under negative con...           101.499404  \n",
      "Mitotic G2-G2/M phases                                       101.423949  \n",
      "...                                                                 ...  \n",
      "Phospholipase C-mediated cascade; FGFR4                        0.364501  \n",
      "Cell death signalling via NRAGE, NRIF and NADE                 0.362558  \n",
      "Platelet Adhesion to exposed collagen                          0.359941  \n",
      "RHO GTPases activate PKNs                                      0.357581  \n",
      "Transport of gamma-carboxylated protein precurs...             0.355108  \n",
      "Synthesis, secretion, and inactivation of Gluco...             0.355108  \n",
      "DARPP-32 events                                                0.352822  \n",
      "Hyaluronan biosynthesis and export                             0.348312  \n",
      "Metabolism of RNA                                              0.343491  \n",
      "PLCG1 events in ERBB2 signaling                                0.343491  \n",
      "Calnexin/calreticulin cycle                                    0.340142  \n",
      "mRNA Splicing - Major Pathway                                  0.339002  \n",
      "Triglyceride metabolism                                        0.337912  \n",
      "Asparagine N-linked glycosylation                              0.333502  \n",
      "Amyloid fiber formation                                        0.329636  \n",
      "Retinoid cycle disease events                                  0.326031  \n",
      "Diseases associated with visual transduction                   0.324796  \n",
      "MET activates RAP1 and RAC1                                    0.324556  \n",
      "Nitric oxide stimulates guanylate cyclase                      0.323776  \n",
      "Defective MUT causes methylmalonic aciduria mut...             0.323253  \n",
      "Defective MMAA causes methylmalonic aciduria ty...             0.323253  \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...             0.322745  \n",
      "Relaxin receptors                                              0.321644  \n",
      "Methionine salvage pathway                                     0.321644  \n",
      "HSP90 chaperone cycle for steroid hormone recep...             0.313618  \n",
      "Listeria monocytogenes entry into host cells                   0.313126  \n",
      "ER Quality Control Compartment (ERQC)                          0.306811  \n",
      "N-glycan trimming in the ER and Calnexin/Calret...             0.305727  \n",
      "ESR-mediated signaling                                         0.304732  \n",
      "Regulation of gene expression in early pancreat...             0.301477  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 2 p-values\n",
      "Interleukin-6 signaling                                   2.061650e-02\n",
      "Apoptosis                                                 4.259553e-01\n",
      "Hemostasis                                                4.387007e-03\n",
      "Intrinsic Pathway for Apoptosis                           1.230562e-01\n",
      "PKB-mediated events                                       3.317693e-01\n",
      "PI3K Cascade                                              5.224115e-03\n",
      "MAPK3 (ERK1) activation                                   2.308090e-01\n",
      "Translesion synthesis by REV1                             5.744746e-03\n",
      "Translesion synthesis by Y family DNA polymeras...        2.491368e-03\n",
      "Recognition of DNA damage by PCNA-containing re...        1.077563e-03\n",
      "Translesion Synthesis by POLH                             5.886253e-03\n",
      "Recognition and association of DNA glycosylase ...        5.473405e-04\n",
      "Cleavage of the damaged pyrimidine                        5.473405e-04\n",
      "Recognition and association of DNA glycosylase ...        4.744273e-04\n",
      "Cleavage of the damaged purine                            4.744273e-04\n",
      "Displacement of DNA glycosylase by APEX1                  4.330033e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair            1.725698e-02\n",
      "Resolution of AP sites via the multiple-nucleot...        6.173014e-04\n",
      "Resolution of AP sites via the single-nucleotid...        3.950033e-02\n",
      "SLBP independent Processing of Histone Pre-mRNAs          5.862874e-02\n",
      "Activation of BIM and translocation to mitochon...        1.366803e-01\n",
      "Activation of BAD and translocation to mitochon...        1.689707e-01\n",
      "Activation of NOXA and translocation to mitocho...        1.743375e-01\n",
      "Activation and oligomerization of BAK protein             4.115262e-01\n",
      "BH3-only proteins associate with and inactivate...        1.402237e-02\n",
      "Release of apoptotic factors from the mitochondria        3.944016e-01\n",
      "Formation of apoptosome                                   2.144507e-01\n",
      "Activation of caspases through apoptosome-media...        1.135379e-03\n",
      "Cytochrome c-mediated apoptotic response                  4.507120e-01\n",
      "SMAC (DIABLO) binds to IAPs                               4.770748e-01\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               2.470501e-01\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        1.051167e-01\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        1.041704e-01\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        1.657293e-01\n",
      "MyD88 dependent cascade initiated on endosome             9.081106e-02\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        4.254645e-01\n",
      "N-glycan antennae elongation in the medial/tran...        9.084518e-02\n",
      "N-Glycan antennae elongation                              1.994866e-01\n",
      "Reactions specific to the complex N-glycan synt...        5.935276e-03\n",
      "Retinoid metabolism and transport                         1.921735e-02\n",
      "MyD88 cascade initiated on plasma membrane                1.650548e-01\n",
      "Nonsense Mediated Decay (NMD) independent of th...        1.739085e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        1.961203e-01\n",
      "Termination of O-glycan biosynthesis                      1.275471e-02\n",
      "Amyloid fiber formation                                   4.292134e-04\n",
      "Serine biosynthesis                                       1.493588e-03\n",
      "GABA receptor activation                                  2.151211e-01\n",
      "GABA B receptor activation                                2.034772e-01\n",
      "Regulation of Complement cascade                          4.660825e-02\n",
      "Growth hormone receptor signaling                         9.496480e-04\n",
      "Antigen processing: Ubiquitination & Proteasome...        3.987456e-01\n",
      "Class I MHC mediated antigen processing & prese...        8.975799e-02\n",
      "Antigen Presentation: Folding, assembly and pep...        1.608921e-01\n",
      "Kinesins                                                  2.244814e-03\n",
      "Factors involved in megakaryocyte development a...        4.203851e-02\n",
      "Antigen activates B Cell Receptor (BCR) leading...        1.808444e-02\n",
      "Signaling by the B Cell Receptor (BCR)                    5.301814e-02\n",
      "Ion channel transport                                     1.618401e-08\n",
      "Activation of GABAB receptors                             2.034772e-01\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        2.164655e-01\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 2 p-values  \\\n",
      "RUNX3 regulates WNT signaling                                10.922527   \n",
      "Iron uptake and transport                                    10.627799   \n",
      "Signaling by Insulin receptor                                10.580309   \n",
      "Transferrin endocytosis and recycling                        10.573013   \n",
      "Insulin receptor recycling                                   10.483246   \n",
      "Amino acids regulate mTORC1                                  10.058843   \n",
      "ROS and RNS production in phagocytes                          9.675031   \n",
      "Ephrin signaling                                              9.631590   \n",
      "TRAIL  signaling                                              8.687381   \n",
      "Repression of WNT target genes                                8.633809   \n",
      "Ca2+ activated K+ channels                                    8.423447   \n",
      "Estrogen-dependent nuclear events downstream of...            8.420755   \n",
      "Ion channel transport                                         7.790914   \n",
      "NOSIP mediated eNOS trafficking                               7.289567   \n",
      "Activation of SMO                                             6.953467   \n",
      "RUNX3 regulates CDKN1A transcription                          6.839322   \n",
      "TGFBR1 KD Mutants in Cancer                                   6.487554   \n",
      "SMAD2/3 Phosphorylation Motif Mutants in Cancer               6.487554   \n",
      "Activation of RAC1                                            6.468384   \n",
      "Regulation of KIT signaling                                   6.263672   \n",
      "Phase 4 - resting membrane potential                          6.246239   \n",
      "Enzymatic degradation of dopamine by COMT                     6.223799   \n",
      "Dopamine clearance from the synaptic cleft                    6.223799   \n",
      "Amine Oxidase reactions                                       6.129484   \n",
      "Cell junction organization                                    6.086009   \n",
      "FasL/ CD95L signaling                                         6.043603   \n",
      "MAP3K8 (TPL2)-dependent MAPK1/3 activation                    5.985669   \n",
      "Organic cation transport                                      5.977610   \n",
      "Cell-Cell communication                                       5.966873   \n",
      "Glutamate Neurotransmitter Release Cycle                      5.918242   \n",
      "...                                                                ...   \n",
      "Cellular responses to stress                                  0.311088   \n",
      "Cellular responses to external stimuli                        0.310644   \n",
      "Tandem pore domain potassium channels                         0.310201   \n",
      "Defects in vitamin and cofactor metabolism                    0.310127   \n",
      "Gap junction assembly                                         0.310127   \n",
      "Downregulation of SMAD2/3:SMAD4 transcriptional...            0.310053   \n",
      "DDX58/IFIH1-mediated induction of interferon-al...            0.309611   \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      0.308800   \n",
      "Glucuronidation                                               0.308285   \n",
      "Synthesis of Leukotrienes (LT) and Eoxins (EX)                0.308212   \n",
      "Assembly of the pre-replicative complex                       0.307844   \n",
      "Xenobiotics                                                   0.307771   \n",
      "Lysosomal oligosaccharide catabolism                          0.307477   \n",
      "APC/C:Cdc20 mediated degradation of Securin                   0.307257   \n",
      "Pyrophosphate hydrolysis                                      0.306598   \n",
      "CDK-mediated phosphorylation and removal of Cdc6              0.306085   \n",
      "The activation of arylsulfatases                              0.306012   \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...            0.305866   \n",
      "MET activates STAT3                                           0.305866   \n",
      "alpha-linolenic acid (ALA) metabolism                         0.305793   \n",
      "Defects in biotin (Btn) metabolism                            0.305720   \n",
      "Aggrephagy                                                    0.304260   \n",
      "Toxicity of botulinum toxin type E (BoNT/E)                   0.304042   \n",
      "TFAP2A acts as a transcriptional repressor duri...            0.303241   \n",
      "Pyruvate metabolism and Citric Acid (TCA) cycle               0.302878   \n",
      "Peptide hormone biosynthesis                                  0.302225   \n",
      "Defective SLC16A1 causes symptomatic deficiency...            0.302152   \n",
      "PIP3 activates AKT signaling                                  0.302152   \n",
      "Response of EIF2AK1 (HRI) to heme deficiency                  0.301211   \n",
      "Defective AVP does not bind AVPR2 and causes ne...            0.301066   \n",
      "\n",
      "                                                    Cluster 2 q-values  \n",
      "RUNX3 regulates WNT signaling                                 7.865408  \n",
      "Iron uptake and transport                                     7.865408  \n",
      "Signaling by Insulin receptor                                 7.865408  \n",
      "Transferrin endocytosis and recycling                         7.865408  \n",
      "Insulin receptor recycling                                    7.865408  \n",
      "Amino acids regulate mTORC1                                   7.520185  \n",
      "ROS and RNS production in phagocytes                          7.217871  \n",
      "Ephrin signaling                                              7.217871  \n",
      "TRAIL  signaling                                              6.324815  \n",
      "Repression of WNT target genes                                6.317000  \n",
      "Ca2+ activated K+ channels                                    6.183127  \n",
      "Estrogen-dependent nuclear events downstream of...            6.183127  \n",
      "Ion channel transport                                         5.588048  \n",
      "NOSIP mediated eNOS trafficking                               5.118887  \n",
      "Activation of SMO                                             4.812750  \n",
      "RUNX3 regulates CDKN1A transcription                          4.726633  \n",
      "TGFBR1 KD Mutants in Cancer                                   4.430329  \n",
      "SMAD2/3 Phosphorylation Motif Mutants in Cancer               4.430329  \n",
      "Activation of RAC1                                            4.430329  \n",
      "Regulation of KIT signaling                                   4.268718  \n",
      "Phase 4 - resting membrane potential                          4.268718  \n",
      "Enzymatic degradation of dopamine by COMT                     4.268718  \n",
      "Dopamine clearance from the synaptic cleft                    4.268718  \n",
      "Amine Oxidase reactions                                       4.192886  \n",
      "Cell junction organization                                    4.167140  \n",
      "FasL/ CD95L signaling                                         4.141767  \n",
      "MAP3K8 (TPL2)-dependent MAPK1/3 activation                    4.112462  \n",
      "Organic cation transport                                      4.112462  \n",
      "Cell-Cell communication                                       4.112462  \n",
      "Glutamate Neurotransmitter Release Cycle                      4.078554  \n",
      "...                                                                ...  \n",
      "Cellular responses to stress                                  0.304998  \n",
      "Cellular responses to external stimuli                        0.304998  \n",
      "Tandem pore domain potassium channels                         0.304998  \n",
      "Defects in vitamin and cofactor metabolism                    0.304998  \n",
      "Gap junction assembly                                         0.304998  \n",
      "Downregulation of SMAD2/3:SMAD4 transcriptional...            0.304998  \n",
      "DDX58/IFIH1-mediated induction of interferon-al...            0.304768  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      0.304169  \n",
      "Glucuronidation                                               0.304004  \n",
      "Synthesis of Leukotrienes (LT) and Eoxins (EX)                0.304004  \n",
      "Assembly of the pre-replicative complex                       0.303985  \n",
      "Xenobiotics                                                   0.303985  \n",
      "Lysosomal oligosaccharide catabolism                          0.303903  \n",
      "APC/C:Cdc20 mediated degradation of Securin                   0.303894  \n",
      "Pyrophosphate hydrolysis                                      0.303831  \n",
      "CDK-mediated phosphorylation and removal of Cdc6              0.303831  \n",
      "The activation of arylsulfatases                              0.303831  \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...            0.303831  \n",
      "MET activates STAT3                                           0.303831  \n",
      "alpha-linolenic acid (ALA) metabolism                         0.303831  \n",
      "Defects in biotin (Btn) metabolism                            0.303831  \n",
      "Aggrephagy                                                    0.302582  \n",
      "Toxicity of botulinum toxin type E (BoNT/E)                   0.302573  \n",
      "TFAP2A acts as a transcriptional repressor duri...            0.301983  \n",
      "Pyruvate metabolism and Citric Acid (TCA) cycle               0.301830  \n",
      "Peptide hormone biosynthesis                                  0.301733  \n",
      "Defective SLC16A1 causes symptomatic deficiency...            0.301733  \n",
      "PIP3 activates AKT signaling                                  0.301733  \n",
      "Response of EIF2AK1 (HRI) to heme deficiency                  0.301066  \n",
      "Defective AVP does not bind AVPR2 and causes ne...            0.301066  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 3 p-values\n",
      "Interleukin-6 signaling                                   6.732247e-05\n",
      "Apoptosis                                                 4.219411e-07\n",
      "Hemostasis                                                1.455990e-01\n",
      "Intrinsic Pathway for Apoptosis                           4.484992e-06\n",
      "PKB-mediated events                                       4.619212e-02\n",
      "PI3K Cascade                                              4.321703e-05\n",
      "MAPK3 (ERK1) activation                                   3.573999e-01\n",
      "Translesion synthesis by REV1                             2.857977e-10\n",
      "Translesion synthesis by Y family DNA polymeras...        6.404465e-25\n",
      "Recognition of DNA damage by PCNA-containing re...        3.891546e-18\n",
      "Translesion Synthesis by POLH                             6.496640e-11\n",
      "Recognition and association of DNA glycosylase ...        2.020645e-01\n",
      "Cleavage of the damaged pyrimidine                        2.020645e-01\n",
      "Recognition and association of DNA glycosylase ...        2.665029e-01\n",
      "Cleavage of the damaged purine                            2.665029e-01\n",
      "Displacement of DNA glycosylase by APEX1                  4.653007e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair            3.213458e-30\n",
      "Resolution of AP sites via the multiple-nucleot...        4.025466e-31\n",
      "Resolution of AP sites via the single-nucleotid...        2.644028e-01\n",
      "SLBP independent Processing of Histone Pre-mRNAs          5.142166e-18\n",
      "Activation of BIM and translocation to mitochon...        7.057053e-02\n",
      "Activation of BAD and translocation to mitochon...        3.153758e-11\n",
      "Activation of NOXA and translocation to mitocho...        4.610054e-05\n",
      "Activation and oligomerization of BAK protein             6.639234e-06\n",
      "BH3-only proteins associate with and inactivate...        2.111566e-14\n",
      "Release of apoptotic factors from the mitochondria        2.702963e-08\n",
      "Formation of apoptosome                                   1.059267e-02\n",
      "Activation of caspases through apoptosome-media...        1.562569e-08\n",
      "Cytochrome c-mediated apoptotic response                  1.163446e-02\n",
      "SMAC (DIABLO) binds to IAPs                               2.670500e-04\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               1.138164e-02\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        6.527342e-05\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        6.401118e-03\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        1.091616e-01\n",
      "MyD88 dependent cascade initiated on endosome             9.764944e-03\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        3.487817e-01\n",
      "N-glycan antennae elongation in the medial/tran...        1.926167e-32\n",
      "N-Glycan antennae elongation                              3.005734e-21\n",
      "Reactions specific to the complex N-glycan synt...        1.031059e-27\n",
      "Retinoid metabolism and transport                         8.382379e-25\n",
      "MyD88 cascade initiated on plasma membrane                1.449840e-03\n",
      "Nonsense Mediated Decay (NMD) independent of th...        3.342711e-06\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        3.923120e-06\n",
      "Termination of O-glycan biosynthesis                      1.972460e-01\n",
      "Amyloid fiber formation                                   4.585101e-02\n",
      "Serine biosynthesis                                       3.599619e-09\n",
      "GABA receptor activation                                  2.145056e-05\n",
      "GABA B receptor activation                                1.203976e-05\n",
      "Regulation of Complement cascade                          9.811170e-03\n",
      "Growth hormone receptor signaling                         3.347964e-02\n",
      "Antigen processing: Ubiquitination & Proteasome...        4.404179e-02\n",
      "Class I MHC mediated antigen processing & prese...        1.568928e-04\n",
      "Antigen Presentation: Folding, assembly and pep...        1.912474e-03\n",
      "Kinesins                                                  3.038449e-48\n",
      "Factors involved in megakaryocyte development a...        1.069592e-51\n",
      "Antigen activates B Cell Receptor (BCR) leading...        4.556986e-01\n",
      "Signaling by the B Cell Receptor (BCR)                    1.121143e-02\n",
      "Ion channel transport                                     1.597139e-01\n",
      "Activation of GABAB receptors                             1.203976e-05\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        2.777406e-03\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 3 p-values  \\\n",
      "KSRP (KHSRP) binds and destabilizes mRNA                     51.830346   \n",
      "Factors involved in megakaryocyte development a...           50.970782   \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...           50.554597   \n",
      "Mitotic Telophase/Cytokinesis                                49.688706   \n",
      "COPI-dependent Golgi-to-ER retrograde traffic                47.921327   \n",
      "Kinesins                                                     47.517348   \n",
      "Vitamin B2 (riboflavin) metabolism                           46.671541   \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...           46.614596   \n",
      "alpha-linolenic acid (ALA) metabolism                        46.613208   \n",
      "Unwinding of DNA                                             46.508440   \n",
      "Cell Cycle, Mitotic                                          45.943671   \n",
      "Cell Cycle                                                   45.929201   \n",
      "RHO GTPases Activate Formins                                 45.826597   \n",
      "Mitotic Prometaphase                                         45.596339   \n",
      "Mitotic Spindle Checkpoint                                   44.990209   \n",
      "EML4 and NUDC in mitotic spindle formation                   44.606521   \n",
      "Polo-like kinase mediated events                             44.332035   \n",
      "Resolution of Sister Chromatid Cohesion                      44.242760   \n",
      "Amplification  of signal from unattached  kinet...           44.147499   \n",
      "Amplification of signal from the kinetochores                44.147499   \n",
      "Nucleotide salvage                                           44.131295   \n",
      "DNA strand elongation                                        43.972790   \n",
      "G1/S-Specific Transcription                                  43.863692   \n",
      "Golgi-to-ER retrograde transport                             43.590881   \n",
      "Activation of ATR in response to replication st...           43.475568   \n",
      "Resolution of D-Loop Structures                              43.340342   \n",
      "Resolution of D-loop Structures through Hollida...           43.133235   \n",
      "Regulation of TP53 Activity through Phosphoryla...           42.560668   \n",
      "Mitotic Metaphase/Anaphase Transition                        42.511640   \n",
      "Activation of the pre-replicative complex                    42.342914   \n",
      "...                                                                ...   \n",
      "Biosynthesis of maresin-like SPMs                             0.329814   \n",
      "Toxicity of botulinum toxin type E (BoNT/E)                   0.329774   \n",
      "Transferrin endocytosis and recycling                         0.328959   \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...            0.328552   \n",
      "Gamma carboxylation, hypusine formation and ary...            0.328187   \n",
      "Calcineurin activates NFAT                                    0.325715   \n",
      "TNFR1-mediated ceramide production                            0.325230   \n",
      "Signaling by FGFR4 in disease                                 0.324343   \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...            0.324101   \n",
      "TRIF-mediated programmed cell death                           0.323900   \n",
      "Regulation of actin dynamics for phagocytic cup...            0.323578   \n",
      "SHC-mediated cascade:FGFR3                                    0.322573   \n",
      "Regulation of TP53 Expression                                 0.322012   \n",
      "Downstream signaling of activated FGFR2                       0.318974   \n",
      "Role of ABL in ROBO-SLIT signaling                            0.316075   \n",
      "RUNX1 regulates transcription of genes involved...            0.315284   \n",
      "The IPAF inflammasome                                         0.313431   \n",
      "Suppression of phagosomal maturation                          0.310841   \n",
      "POU5F1 (OCT4), SOX2, NANOG repress genes relate...            0.309785   \n",
      "Signalling to RAS                                             0.309629   \n",
      "RNA Polymerase III Transcription Initiation Fro...            0.309551   \n",
      "Signaling by Erythropoietin                                   0.309473   \n",
      "Unblocking of NMDA receptors, glutamate binding...            0.307799   \n",
      "Signal regulatory protein family interactions                 0.307682   \n",
      "Vasopressin-like receptors                                    0.306402   \n",
      "Cellular responses to stress                                  0.306015   \n",
      "FasL/ CD95L signaling                                         0.305937   \n",
      "STING mediated induction of host immune responses             0.305473   \n",
      "Tandem pore domain potassium channels                         0.303621   \n",
      "Signaling by Insulin receptor                                 0.301969   \n",
      "\n",
      "                                                    Cluster 3 q-values  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                     48.513537  \n",
      "Factors involved in megakaryocyte development a...           47.955003  \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...           47.714910  \n",
      "Mitotic Telophase/Cytokinesis                                46.973957  \n",
      "COPI-dependent Golgi-to-ER retrograde traffic                45.303489  \n",
      "Kinesins                                                     44.978691  \n",
      "Vitamin B2 (riboflavin) metabolism                           44.250641  \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...           44.250641  \n",
      "alpha-linolenic acid (ALA) metabolism                        44.250641  \n",
      "Unwinding of DNA                                             44.191632  \n",
      "Cell Cycle, Mitotic                                          43.691573  \n",
      "Cell Cycle                                                   43.691573  \n",
      "RHO GTPases Activate Formins                                 43.623732  \n",
      "Mitotic Prometaphase                                         43.425658  \n",
      "Mitotic Spindle Checkpoint                                   42.849491  \n",
      "EML4 and NUDC in mitotic spindle formation                   42.493832  \n",
      "Polo-like kinase mediated events                             42.245675  \n",
      "Resolution of Sister Chromatid Cohesion                      42.181224  \n",
      "Amplification  of signal from unattached  kinet...           42.136705  \n",
      "Amplification of signal from the kinetochores                42.136705  \n",
      "Nucleotide salvage                                           42.136705  \n",
      "DNA strand elongation                                        41.998404  \n",
      "G1/S-Specific Transcription                                  41.908611  \n",
      "Golgi-to-ER retrograde transport                             41.654284  \n",
      "Activation of ATR in response to replication st...           41.556700  \n",
      "Resolution of D-Loop Structures                              41.438507  \n",
      "Resolution of D-loop Structures through Hollida...           41.247790  \n",
      "Regulation of TP53 Activity through Phosphoryla...           40.691017  \n",
      "Mitotic Metaphase/Anaphase Transition                        40.657229  \n",
      "Activation of the pre-replicative complex                    40.503226  \n",
      "...                                                                ...  \n",
      "Biosynthesis of maresin-like SPMs                             0.323870  \n",
      "Toxicity of botulinum toxin type E (BoNT/E)                   0.323870  \n",
      "Transferrin endocytosis and recycling                         0.323268  \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...            0.323074  \n",
      "Gamma carboxylation, hypusine formation and ary...            0.322920  \n",
      "Calcineurin activates NFAT                                    0.320660  \n",
      "TNFR1-mediated ceramide production                            0.320387  \n",
      "Signaling by FGFR4 in disease                                 0.319712  \n",
      "POU5F1 (OCT4), SOX2, NANOG activate genes relat...            0.319692  \n",
      "TRIF-mediated programmed cell death                           0.319692  \n",
      "Regulation of actin dynamics for phagocytic cup...            0.319581  \n",
      "SHC-mediated cascade:FGFR3                                    0.318788  \n",
      "Regulation of TP53 Expression                                 0.318437  \n",
      "Downstream signaling of activated FGFR2                       0.315611  \n",
      "Role of ABL in ROBO-SLIT signaling                            0.312923  \n",
      "RUNX1 regulates transcription of genes involved...            0.312343  \n",
      "The IPAF inflammasome                                         0.310700  \n",
      "Suppression of phagosomal maturation                          0.308321  \n",
      "POU5F1 (OCT4), SOX2, NANOG repress genes relate...            0.307795  \n",
      "Signalling to RAS                                             0.307795  \n",
      "RNA Polymerase III Transcription Initiation Fro...            0.307795  \n",
      "Signaling by Erythropoietin                                   0.307795  \n",
      "Unblocking of NMDA receptors, glutamate binding...            0.306424  \n",
      "Signal regulatory protein family interactions                 0.306424  \n",
      "Vasopressin-like receptors                                    0.305354  \n",
      "Cellular responses to stress                                  0.305308  \n",
      "FasL/ CD95L signaling                                         0.305308  \n",
      "STING mediated induction of host immune responses             0.305054  \n",
      "Tandem pore domain potassium channels                         0.303412  \n",
      "Signaling by Insulin receptor                                 0.301969  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 4ER+ p-values\n",
      "Interleukin-6 signaling                                      2.382400e-11\n",
      "Apoptosis                                                    2.805844e-01\n",
      "Hemostasis                                                   7.433489e-22\n",
      "Intrinsic Pathway for Apoptosis                              3.523330e-07\n",
      "PKB-mediated events                                          4.237196e-09\n",
      "PI3K Cascade                                                 7.161279e-08\n",
      "MAPK3 (ERK1) activation                                      3.307940e-05\n",
      "Translesion synthesis by REV1                                3.739303e-18\n",
      "Translesion synthesis by Y family DNA polymeras...           1.123579e-37\n",
      "Recognition of DNA damage by PCNA-containing re...           4.537208e-39\n",
      "Translesion Synthesis by POLH                                1.275302e-28\n",
      "Recognition and association of DNA glycosylase ...           7.132936e-09\n",
      "Cleavage of the damaged pyrimidine                           7.128123e-09\n",
      "Recognition and association of DNA glycosylase ...           2.461750e-08\n",
      "Cleavage of the damaged purine                               2.461750e-08\n",
      "Displacement of DNA glycosylase by APEX1                     1.683245e-03\n",
      "POLB-Dependent Long Patch Base Excision Repair               1.414075e-44\n",
      "Resolution of AP sites via the multiple-nucleot...           2.068090e-49\n",
      "Resolution of AP sites via the single-nucleotid...           8.393030e-08\n",
      "SLBP independent Processing of Histone Pre-mRNAs             2.173305e-38\n",
      "Activation of BIM and translocation to mitochon...           3.744621e-03\n",
      "Activation of BAD and translocation to mitochon...           7.781686e-16\n",
      "Activation of NOXA and translocation to mitocho...           3.839424e-11\n",
      "Activation and oligomerization of BAK protein                2.225515e-06\n",
      "BH3-only proteins associate with and inactivate...           1.234808e-01\n",
      "Release of apoptotic factors from the mitochondria           4.835744e-13\n",
      "Formation of apoptosome                                      5.078594e-03\n",
      "Activation of caspases through apoptosome-media...           1.812930e-03\n",
      "Cytochrome c-mediated apoptotic response                     1.055258e-02\n",
      "SMAC (DIABLO) binds to IAPs                                  5.023895e-06\n",
      "...                                                                   ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib                  4.893025e-16\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...           2.080577e-01\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...           4.641029e-06\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...           4.353571e-01\n",
      "MyD88 dependent cascade initiated on endosome                1.882664e-06\n",
      "IRAK2 mediated activation of TAK1 complex upon ...           5.802548e-18\n",
      "N-glycan antennae elongation in the medial/tran...           8.175113e-17\n",
      "N-Glycan antennae elongation                                 2.712359e-17\n",
      "Reactions specific to the complex N-glycan synt...           1.885045e-07\n",
      "Retinoid metabolism and transport                            1.876914e-10\n",
      "MyD88 cascade initiated on plasma membrane                   1.716259e-02\n",
      "Nonsense Mediated Decay (NMD) independent of th...           2.141007e-02\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...           1.537972e-02\n",
      "Termination of O-glycan biosynthesis                         4.991761e-06\n",
      "Amyloid fiber formation                                      1.847814e-12\n",
      "Serine biosynthesis                                          1.576684e-01\n",
      "GABA receptor activation                                     8.304744e-02\n",
      "GABA B receptor activation                                   5.740636e-02\n",
      "Regulation of Complement cascade                             2.948989e-14\n",
      "Growth hormone receptor signaling                            6.570171e-07\n",
      "Antigen processing: Ubiquitination & Proteasome...           3.046233e-01\n",
      "Class I MHC mediated antigen processing & prese...           1.120377e-03\n",
      "Antigen Presentation: Folding, assembly and pep...           6.103877e-05\n",
      "Kinesins                                                     2.951797e-40\n",
      "Factors involved in megakaryocyte development a...           2.722131e-42\n",
      "Antigen activates B Cell Receptor (BCR) leading...           3.918021e-12\n",
      "Signaling by the B Cell Receptor (BCR)                       5.987518e-06\n",
      "Ion channel transport                                        1.146093e-04\n",
      "Activation of GABAB receptors                                5.740636e-02\n",
      "Inhibition  of voltage gated Ca2+ channels via ...           1.286219e-03\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 4ER+ p-values  \\\n",
      "Resolution of D-Loop Structures                                 55.079736   \n",
      "Resolution of D-loop Structures through Hollida...              55.037805   \n",
      "HDR through Homologous Recombination (HRR)                      52.330210   \n",
      "HDR through Single Strand Annealing (SSA)                       51.859480   \n",
      "Homologous DNA Pairing and Strand Exchange                      51.325244   \n",
      "Resolution of D-loop Structures through Synthes...              50.708818   \n",
      "Presynaptic phase of homologous DNA pairing and...              50.437581   \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                        50.111266   \n",
      "Extension of Telomeres                                          49.924424   \n",
      "Polo-like kinase mediated events                                49.631319   \n",
      "E2F mediated regulation of DNA replication                      49.305432   \n",
      "Resolution of AP sites via the multiple-nucleot...              48.684431   \n",
      "cGMP effects                                                    48.438784   \n",
      "Resolution of Abasic Sites (AP sites)                           47.437593   \n",
      "E2F-enabled inhibition of pre-replication compl...              47.293971   \n",
      "HDR through MMEJ (alt-NHEJ)                                     47.093566   \n",
      "Homology Directed Repair                                        46.868812   \n",
      "Cell-cell junction organization                                 46.339753   \n",
      "G2/M DNA replication checkpoint                                 46.302080   \n",
      "Telomere C-strand (Lagging Strand) Synthesis                    46.202178   \n",
      "Lagging Strand Synthesis                                        45.746064   \n",
      "Mitotic Spindle Checkpoint                                      45.735626   \n",
      "Amplification  of signal from unattached  kinet...              45.058361   \n",
      "Amplification of signal from the kinetochores                   45.058361   \n",
      "Tight junction interactions                                     45.002661   \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...              44.757427   \n",
      "Depolymerisation of the Nuclear Lamina                          44.659241   \n",
      "Cell Cycle                                                      44.653553   \n",
      "PCNA-Dependent Long Patch Base Excision Repair                  44.371721   \n",
      "Post-chaperonin tubulin folding pathway                         44.157940   \n",
      "...                                                                   ...   \n",
      "MET activates RAP1 and RAC1                                      0.352596   \n",
      "Serotonin and melatonin biosynthesis                             0.351926   \n",
      "Lipophagy                                                        0.350100   \n",
      "Ion influx/efflux at host-pathogen interface                     0.349212   \n",
      "Cargo trafficking to the periciliary membrane                    0.346247   \n",
      "Glycerophospholipid catabolism                                   0.344353   \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...               0.343738   \n",
      "Platelet Adhesion to exposed collagen                            0.343694   \n",
      "Toxicity of botulinum toxin type C (BoNT/C)                      0.342598   \n",
      "Bicarbonate transporters                                         0.339280   \n",
      "GABA synthesis, release, reuptake and degradation                0.338976   \n",
      "Physiological factors                                            0.336114   \n",
      "IRF3-mediated induction of type I IFN                            0.333957   \n",
      "Activation of NMDA receptors and postsynaptic e...               0.333441   \n",
      "Ras activation upon Ca2+ influx through NMDA re...               0.333054   \n",
      "Lewis blood group biosynthesis                                   0.330227   \n",
      "Blood group systems biosynthesis                                 0.326907   \n",
      "Phospholipase C-mediated cascade: FGFR1                          0.320043   \n",
      "Defective AMN causes hereditary megaloblastic a...               0.319500   \n",
      "Defective CUBN causes hereditary megaloblastic ...               0.319500   \n",
      "CLEC7A (Dectin-1) signaling                                      0.319208   \n",
      "Orexin and neuropeptides FF and QRFP bind to th...               0.318166   \n",
      "DAG and IP3 signaling                                            0.317874   \n",
      "Formation of the active cofactor, UDP-glucuronate                0.314599   \n",
      "TICAM1, RIP1-mediated IKK complex recruitment                    0.312127   \n",
      "Signaling by NOTCH1                                              0.311799   \n",
      "GABA synthesis                                                   0.308160   \n",
      "Signaling by Insulin receptor                                    0.305399   \n",
      "Hyaluronan biosynthesis and export                               0.304025   \n",
      "InlB-mediated entry of Listeria monocytogenes i...               0.301651   \n",
      "\n",
      "                                                    Cluster 4ER+ q-values  \n",
      "Resolution of D-Loop Structures                                 52.022026  \n",
      "Resolution of D-loop Structures through Hollida...              52.022026  \n",
      "HDR through Homologous Recombination (HRR)                      49.490523  \n",
      "HDR through Single Strand Annealing (SSA)                       49.144731  \n",
      "Homologous DNA Pairing and Strand Exchange                      48.707406  \n",
      "Resolution of D-loop Structures through Synthes...              48.170160  \n",
      "Presynaptic phase of homologous DNA pairing and...              47.965871  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                        47.697547  \n",
      "Extension of Telomeres                                          47.561858  \n",
      "Polo-like kinase mediated events                                47.314511  \n",
      "E2F mediated regulation of DNA replication                      47.030016  \n",
      "Resolution of AP sites via the multiple-nucleot...              46.446803  \n",
      "cGMP effects                                                    46.235918  \n",
      "Resolution of Abasic Sites (AP sites)                           45.266912  \n",
      "E2F-enabled inhibition of pre-replication compl...              45.153253  \n",
      "HDR through MMEJ (alt-NHEJ)                                     44.980877  \n",
      "Homology Directed Repair                                        44.782452  \n",
      "Cell-cell junction organization                                 44.278217  \n",
      "G2/M DNA replication checkpoint                                 44.264025  \n",
      "Telomere C-strand (Lagging Strand) Synthesis                    44.186400  \n",
      "Lagging Strand Synthesis                                        43.761240  \n",
      "Mitotic Spindle Checkpoint                                      43.761240  \n",
      "Amplification  of signal from unattached  kinet...              43.121764  \n",
      "Amplification of signal from the kinetochores                   43.121764  \n",
      "Tight junction interactions                                     43.083792  \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...              42.855592  \n",
      "Depolymerisation of the Nuclear Lamina                          42.783902  \n",
      "Cell Cycle                                                      42.783902  \n",
      "PCNA-Dependent Long Patch Base Excision Repair                  42.517310  \n",
      "Post-chaperonin tubulin folding pathway                         42.326130  \n",
      "...                                                                   ...  \n",
      "MET activates RAP1 and RAC1                                      0.346481  \n",
      "Serotonin and melatonin biosynthesis                             0.346023  \n",
      "Lipophagy                                                        0.344410  \n",
      "Ion influx/efflux at host-pathogen interface                     0.343733  \n",
      "Cargo trafficking to the periciliary membrane                    0.340980  \n",
      "Glycerophospholipid catabolism                                   0.339299  \n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activa...               0.339063  \n",
      "Platelet Adhesion to exposed collagen                            0.339063  \n",
      "Toxicity of botulinum toxin type C (BoNT/C)                      0.338178  \n",
      "Bicarbonate transporters                                         0.335072  \n",
      "GABA synthesis, release, reuptake and degradation                0.334979  \n",
      "Physiological factors                                            0.332328  \n",
      "IRF3-mediated induction of type I IFN                            0.330383  \n",
      "Activation of NMDA receptors and postsynaptic e...               0.330077  \n",
      "Ras activation upon Ca2+ influx through NMDA re...               0.329902  \n",
      "Lewis blood group biosynthesis                                   0.327285  \n",
      "Blood group systems biosynthesis                                 0.324176  \n",
      "Phospholipase C-mediated cascade: FGFR1                          0.317523  \n",
      "Defective AMN causes hereditary megaloblastic a...               0.317401  \n",
      "Defective CUBN causes hereditary megaloblastic ...               0.317401  \n",
      "CLEC7A (Dectin-1) signaling                                      0.317319  \n",
      "Orexin and neuropeptides FF and QRFP bind to th...               0.316487  \n",
      "DAG and IP3 signaling                                            0.316406  \n",
      "Formation of the active cofactor, UDP-glucuronate                0.313341  \n",
      "TICAM1, RIP1-mediated IKK complex recruitment                    0.311079  \n",
      "Signaling by NOTCH1                                              0.310960  \n",
      "GABA synthesis                                                   0.307531  \n",
      "Signaling by Insulin receptor                                    0.304980  \n",
      "Hyaluronan biosynthesis and export                               0.303816  \n",
      "InlB-mediated entry of Listeria monocytogenes i...               0.301651  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 4ER- p-values\n",
      "Interleukin-6 signaling                                      1.216226e-05\n",
      "Apoptosis                                                    5.517435e-03\n",
      "Hemostasis                                                   5.832565e-20\n",
      "Intrinsic Pathway for Apoptosis                              2.421401e-01\n",
      "PKB-mediated events                                          3.710898e-06\n",
      "PI3K Cascade                                                 2.686865e-06\n",
      "MAPK3 (ERK1) activation                                      4.068825e-06\n",
      "Translesion synthesis by REV1                                3.318629e-06\n",
      "Translesion synthesis by Y family DNA polymeras...           1.109458e-05\n",
      "Recognition of DNA damage by PCNA-containing re...           1.435355e-06\n",
      "Translesion Synthesis by POLH                                2.081855e-09\n",
      "Recognition and association of DNA glycosylase ...           3.665520e-07\n",
      "Cleavage of the damaged pyrimidine                           3.665520e-07\n",
      "Recognition and association of DNA glycosylase ...           6.153651e-07\n",
      "Cleavage of the damaged purine                               6.153651e-07\n",
      "Displacement of DNA glycosylase by APEX1                     3.285675e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair               4.141285e-04\n",
      "Resolution of AP sites via the multiple-nucleot...           1.098432e-04\n",
      "Resolution of AP sites via the single-nucleotid...           5.914595e-11\n",
      "SLBP independent Processing of Histone Pre-mRNAs             1.814275e-05\n",
      "Activation of BIM and translocation to mitochon...           2.363799e-03\n",
      "Activation of BAD and translocation to mitochon...           2.220394e-01\n",
      "Activation of NOXA and translocation to mitocho...           1.072003e-01\n",
      "Activation and oligomerization of BAK protein                1.914571e-01\n",
      "BH3-only proteins associate with and inactivate...           8.793816e-20\n",
      "Release of apoptotic factors from the mitochondria           1.975204e-01\n",
      "Formation of apoptosome                                      2.103273e-01\n",
      "Activation of caspases through apoptosome-media...           5.384610e-11\n",
      "Cytochrome c-mediated apoptotic response                     4.099449e-01\n",
      "SMAC (DIABLO) binds to IAPs                                  3.726793e-01\n",
      "...                                                                   ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib                  1.029741e-04\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...           7.316716e-11\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...           3.596738e-10\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...           2.510564e-01\n",
      "MyD88 dependent cascade initiated on endosome                6.389430e-10\n",
      "IRAK2 mediated activation of TAK1 complex upon ...           3.307508e-16\n",
      "N-glycan antennae elongation in the medial/tran...           1.839575e-04\n",
      "N-Glycan antennae elongation                                 1.476828e-01\n",
      "Reactions specific to the complex N-glycan synt...           3.131877e-05\n",
      "Retinoid metabolism and transport                            1.304931e-06\n",
      "MyD88 cascade initiated on plasma membrane                   1.439895e-05\n",
      "Nonsense Mediated Decay (NMD) independent of th...           3.042572e-03\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...           3.517286e-03\n",
      "Termination of O-glycan biosynthesis                         2.599594e-03\n",
      "Amyloid fiber formation                                      5.992412e-07\n",
      "Serine biosynthesis                                          7.056392e-17\n",
      "GABA receptor activation                                     9.813712e-02\n",
      "GABA B receptor activation                                   9.487091e-02\n",
      "Regulation of Complement cascade                             1.960588e-16\n",
      "Growth hormone receptor signaling                            9.805229e-19\n",
      "Antigen processing: Ubiquitination & Proteasome...           3.360714e-01\n",
      "Class I MHC mediated antigen processing & prese...           5.637381e-10\n",
      "Antigen Presentation: Folding, assembly and pep...           1.007633e-08\n",
      "Kinesins                                                     1.076327e-01\n",
      "Factors involved in megakaryocyte development a...           4.877922e-01\n",
      "Antigen activates B Cell Receptor (BCR) leading...           5.017463e-13\n",
      "Signaling by the B Cell Receptor (BCR)                       2.110930e-10\n",
      "Ion channel transport                                        2.168201e-22\n",
      "Activation of GABAB receptors                                9.487091e-02\n",
      "Inhibition  of voltage gated Ca2+ channels via ...           1.445123e-01\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 4ER- p-values  \\\n",
      "RUNX1 regulates estrogen receptor mediated tran...              29.004629   \n",
      "Cell-Cell communication                                         28.544281   \n",
      "G-protein mediated events                                       28.332141   \n",
      "ABC transporters in lipid homeostasis                           27.924439   \n",
      "PLC beta mediated events                                        27.829339   \n",
      "GAB1 signalosome                                                25.801994   \n",
      "CaM pathway                                                     24.903599   \n",
      "Calmodulin induced events                                       24.903599   \n",
      "RET signaling                                                   24.638697   \n",
      "PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling                 24.248905   \n",
      "Ca-dependent events                                             24.195483   \n",
      "Transport of inorganic cations/anions and amino...              23.934508   \n",
      "RUNX1 regulates transcription of genes involved...              23.911883   \n",
      "Negative regulation of the PI3K/AKT network                     23.886662   \n",
      "Rho GTPase cycle                                                23.624824   \n",
      "Class I peroxisomal membrane protein import                     23.369635   \n",
      "SLC-mediated transmembrane transport                            23.317223   \n",
      "Acetylcholine regulates insulin secretion                       22.828718   \n",
      "Regulation of KIT signaling                                     22.786270   \n",
      "PI3K/AKT Signaling in Cancer                                    22.538350   \n",
      "Metabolism of vitamins and cofactors                            22.382536   \n",
      "Acyl chain remodelling of PG                                    22.032540   \n",
      "TFAP2 (AP-2) family regulates transcription of ...              22.023368   \n",
      "Constitutive Signaling by Aberrant PI3K in Cancer               21.906808   \n",
      "Nuclear signaling by ERBB4                                      21.837855   \n",
      "Cell junction organization                                      21.784764   \n",
      "Vitamin C (ascorbate) metabolism                                21.782277   \n",
      "Loss of Function of FBXW7 in Cancer and NOTCH1 ...              21.764044   \n",
      "FBXW7 Mutants and NOTCH1 in Cancer                              21.764044   \n",
      "Ion channel transport                                           21.663900   \n",
      "...                                                                   ...   \n",
      "RNA Polymerase II Transcription Elongation                       0.320499   \n",
      "Formation of RNA Pol II elongation complex                       0.320499   \n",
      "Constitutive Signaling by AKT1 E17K in Cancer                    0.320288   \n",
      "Tandem pore domain potassium channels                            0.318107   \n",
      "FGFR1b ligand binding and activation                             0.316636   \n",
      "APC/C-mediated degradation of cell cycle proteins                0.316007   \n",
      "Regulation of mitotic cell cycle                                 0.316007   \n",
      "Recruitment of NuMA to mitotic centrosomes                       0.314333   \n",
      "Presynaptic nicotinic acetylcholine receptors                    0.313777   \n",
      "NR1H2 & NR1H3 regulate gene expression to limit...               0.313082   \n",
      "Factors involved in megakaryocyte development a...               0.311765   \n",
      "FGFR1 ligand binding and activation                              0.310728   \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...               0.308662   \n",
      "Cholesterol biosynthesis via desmosterol                         0.308455   \n",
      "Cholesterol biosynthesis via lathosterol                         0.308455   \n",
      "Vitamin B5 (pantothenate) metabolism                             0.307769   \n",
      "Transcriptional Regulation by TP53                               0.307495   \n",
      "Transcriptional regulation by RUNX1                              0.307289   \n",
      "Signaling by FGFR in disease                                     0.305648   \n",
      "DDX58/IFIH1-mediated induction of interferon-al...               0.304353   \n",
      "phosphorylation site mutants of CTNNB1 are not ...               0.303537   \n",
      "T41 mutants of beta-catenin aren't phosphorylated                0.303537   \n",
      "S45 mutants of beta-catenin aren't phosphorylated                0.303537   \n",
      "Misspliced GSK3beta mutants stabilize beta-catenin               0.303537   \n",
      "S37 mutants of beta-catenin aren't phosphorylated                0.303537   \n",
      "S33 mutants of beta-catenin aren't phosphorylated                0.303537   \n",
      "MAPK1 (ERK2) activation                                          0.302925   \n",
      "SUMOylation of DNA methylation proteins                          0.302180   \n",
      "Lysine catabolism                                                0.301909   \n",
      "Vitamin B2 (riboflavin) metabolism                               0.301030   \n",
      "\n",
      "                                                    Cluster 4ER- q-values  \n",
      "RUNX1 regulates estrogen receptor mediated tran...              25.687820  \n",
      "Cell-Cell communication                                         25.528502  \n",
      "G-protein mediated events                                       25.492454  \n",
      "ABC transporters in lipid homeostasis                           25.211500  \n",
      "PLC beta mediated events                                        25.211500  \n",
      "GAB1 signalosome                                                23.263336  \n",
      "CaM pathway                                                     22.489880  \n",
      "Calmodulin induced events                                       22.489880  \n",
      "RET signaling                                                   22.276130  \n",
      "PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling                 21.932096  \n",
      "Ca-dependent events                                             21.920067  \n",
      "Transport of inorganic cations/anions and amino...              21.715981  \n",
      "RUNX1 regulates transcription of genes involved...              21.715981  \n",
      "Negative regulation of the PI3K/AKT network                     21.715981  \n",
      "Rho GTPase cycle                                                21.484106  \n",
      "Class I peroxisomal membrane protein import                     21.256946  \n",
      "SLC-mediated transmembrane transport                            21.230864  \n",
      "Acetylcholine regulates insulin secretion                       20.767181  \n",
      "Regulation of KIT signaling                                     20.748215  \n",
      "PI3K/AKT Signaling in Cancer                                    20.522571  \n",
      "Metabolism of vitamins and cofactors                            20.387946  \n",
      "Acyl chain remodelling of PG                                    20.068287  \n",
      "TFAP2 (AP-2) family regulates transcription of ...              20.068287  \n",
      "Constitutive Signaling by Aberrant PI3K in Cancer               19.970210  \n",
      "Nuclear signaling by ERBB4                                      19.918987  \n",
      "Cell junction organization                                      19.909633  \n",
      "Vitamin C (ascorbate) metabolism                                19.909633  \n",
      "Loss of Function of FBXW7 in Cancer and NOTCH1 ...              19.909633  \n",
      "FBXW7 Mutants and NOTCH1 in Cancer                              19.909633  \n",
      "Ion channel transport                                           19.824213  \n",
      "...                                                                   ...  \n",
      "RNA Polymerase II Transcription Elongation                       0.314597  \n",
      "Formation of RNA Pol II elongation complex                       0.314597  \n",
      "Constitutive Signaling by AKT1 E17K in Cancer                    0.314597  \n",
      "Tandem pore domain potassium channels                            0.312628  \n",
      "FGFR1b ligand binding and activation                             0.311369  \n",
      "APC/C-mediated degradation of cell cycle proteins                0.311163  \n",
      "Regulation of mitotic cell cycle                                 0.311163  \n",
      "Recruitment of NuMA to mitotic centrosomes                       0.309702  \n",
      "Presynaptic nicotinic acetylcholine receptors                    0.309357  \n",
      "NR1H2 & NR1H3 regulate gene expression to limit...               0.308874  \n",
      "Factors involved in megakaryocyte development a...               0.307768  \n",
      "FGFR1 ligand binding and activation                              0.306943  \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...               0.305303  \n",
      "Cholesterol biosynthesis via desmosterol                         0.305303  \n",
      "Cholesterol biosynthesis via lathosterol                         0.305303  \n",
      "Vitamin B5 (pantothenate) metabolism                             0.304827  \n",
      "Transcriptional Regulation by TP53                               0.304769  \n",
      "Transcriptional regulation by RUNX1                              0.304769  \n",
      "Signaling by FGFR in disease                                     0.303338  \n",
      "DDX58/IFIH1-mediated induction of interferon-al...               0.302698  \n",
      "phosphorylation site mutants of CTNNB1 are not ...               0.302698  \n",
      "T41 mutants of beta-catenin aren't phosphorylated                0.302698  \n",
      "S45 mutants of beta-catenin aren't phosphorylated                0.302698  \n",
      "Misspliced GSK3beta mutants stabilize beta-catenin               0.302698  \n",
      "S37 mutants of beta-catenin aren't phosphorylated                0.302698  \n",
      "S33 mutants of beta-catenin aren't phosphorylated                0.302698  \n",
      "MAPK1 (ERK2) activation                                          0.302297  \n",
      "SUMOylation of DNA methylation proteins                          0.301761  \n",
      "Lysine catabolism                                                0.301700  \n",
      "Vitamin B2 (riboflavin) metabolism                               0.301030  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 5 p-values\n",
      "Interleukin-6 signaling                                   1.425766e-03\n",
      "Apoptosis                                                 1.427767e-04\n",
      "Hemostasis                                                2.010551e-05\n",
      "Intrinsic Pathway for Apoptosis                           1.145333e-07\n",
      "PKB-mediated events                                       2.561220e-02\n",
      "PI3K Cascade                                              1.278530e-28\n",
      "MAPK3 (ERK1) activation                                   3.487466e-02\n",
      "Translesion synthesis by REV1                             2.439997e-01\n",
      "Translesion synthesis by Y family DNA polymeras...        9.708217e-10\n",
      "Recognition of DNA damage by PCNA-containing re...        3.216622e-05\n",
      "Translesion Synthesis by POLH                             2.787454e-03\n",
      "Recognition and association of DNA glycosylase ...        2.771763e-01\n",
      "Cleavage of the damaged pyrimidine                        2.771763e-01\n",
      "Recognition and association of DNA glycosylase ...        3.032770e-01\n",
      "Cleavage of the damaged purine                            3.032770e-01\n",
      "Displacement of DNA glycosylase by APEX1                  1.311928e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair            3.621402e-08\n",
      "Resolution of AP sites via the multiple-nucleot...        3.112291e-08\n",
      "Resolution of AP sites via the single-nucleotid...        2.508484e-07\n",
      "SLBP independent Processing of Histone Pre-mRNAs          3.834886e-11\n",
      "Activation of BIM and translocation to mitochon...        3.269224e-01\n",
      "Activation of BAD and translocation to mitochon...        4.975259e-16\n",
      "Activation of NOXA and translocation to mitocho...        1.157524e-01\n",
      "Activation and oligomerization of BAK protein             3.646018e-03\n",
      "BH3-only proteins associate with and inactivate...        1.256223e-38\n",
      "Release of apoptotic factors from the mitochondria        7.875590e-06\n",
      "Formation of apoptosome                                   2.406842e-01\n",
      "Activation of caspases through apoptosome-media...        8.188220e-27\n",
      "Cytochrome c-mediated apoptotic response                  7.740755e-02\n",
      "SMAC (DIABLO) binds to IAPs                               1.610450e-06\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               5.436981e-05\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        8.260425e-04\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        3.045147e-03\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        4.661553e-01\n",
      "MyD88 dependent cascade initiated on endosome             5.193681e-03\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        3.147450e-02\n",
      "N-glycan antennae elongation in the medial/tran...        9.853647e-27\n",
      "N-Glycan antennae elongation                              9.315883e-15\n",
      "Reactions specific to the complex N-glycan synt...        6.483228e-24\n",
      "Retinoid metabolism and transport                         9.912446e-21\n",
      "MyD88 cascade initiated on plasma membrane                9.950681e-03\n",
      "Nonsense Mediated Decay (NMD) independent of th...        1.003270e-03\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        8.912277e-04\n",
      "Termination of O-glycan biosynthesis                      9.125245e-02\n",
      "Amyloid fiber formation                                   1.747283e-01\n",
      "Serine biosynthesis                                       1.324032e-22\n",
      "GABA receptor activation                                  2.743486e-03\n",
      "GABA B receptor activation                                5.486394e-03\n",
      "Regulation of Complement cascade                          2.051963e-09\n",
      "Growth hormone receptor signaling                         6.550425e-09\n",
      "Antigen processing: Ubiquitination & Proteasome...        3.962509e-02\n",
      "Class I MHC mediated antigen processing & prese...        6.262835e-05\n",
      "Antigen Presentation: Folding, assembly and pep...        3.673256e-02\n",
      "Kinesins                                                  6.135433e-18\n",
      "Factors involved in megakaryocyte development a...        2.157351e-21\n",
      "Antigen activates B Cell Receptor (BCR) leading...        3.341374e-07\n",
      "Signaling by the B Cell Receptor (BCR)                    3.774743e-06\n",
      "Ion channel transport                                     2.502968e-14\n",
      "Activation of GABAB receptors                             5.486394e-03\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        9.795394e-02\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 5 p-values  \\\n",
      "Metabolism of amine-derived hormones                         61.133293   \n",
      "Catecholamine biosynthesis                                   52.603890   \n",
      "GRB7 events in ERBB2 signaling                               45.011106   \n",
      "Tryptophan catabolism                                        40.794582   \n",
      "Fertilization                                                40.386047   \n",
      "BH3-only proteins associate with and inactivate...           37.900933   \n",
      "Activation of gene expression by SREBF (SREBP)               37.479928   \n",
      "Sperm Motility And Taxes                                     36.172752   \n",
      "Cholesterol biosynthesis                                     35.291599   \n",
      "Proline catabolism                                           31.609162   \n",
      "Activation of AMPK downstream of NMDARs                      31.547697   \n",
      "Caspase-mediated cleavage of cytoskeletal proteins           31.352100   \n",
      "Manipulation of host energy metabolism                       30.693629   \n",
      "Signaling by Leptin                                          30.184298   \n",
      "Metabolism of folate and pterines                            29.729644   \n",
      "Synthesis of PIPs in the nucleus                             29.396906   \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...           28.535223   \n",
      "alpha-linolenic acid (ALA) metabolism                        28.535223   \n",
      "Metal sequestration by antimicrobial proteins                28.142466   \n",
      "Linoleic acid (LA) metabolism                                28.112221   \n",
      "PI3K Cascade                                                 27.893289   \n",
      "Glutathione synthesis and recycling                          27.856139   \n",
      "PI5P Regulates TP53 Acetylation                              27.801741   \n",
      "IRS activation                                               27.418677   \n",
      "Visual phototransduction                                     27.301313   \n",
      "p130Cas linkage to MAPK signaling for integrins              27.113438   \n",
      "Purine catabolism                                            26.853924   \n",
      "Transcriptional Regulation by MECP2                          26.541331   \n",
      "Response of EIF2AK1 (HRI) to heme deficiency                 26.372980   \n",
      "Phosphate bond hydrolysis by NUDT proteins                   26.120248   \n",
      "...                                                                ...   \n",
      "STING mediated induction of host immune responses             0.324546   \n",
      "Signaling by VEGF                                             0.324062   \n",
      "eNOS activation                                               0.323289   \n",
      "Metabolism of nitric oxide: eNOS activation and...            0.321891   \n",
      "Breakdown of the nuclear lamina                               0.321314   \n",
      "FGFR1c ligand binding and activation                          0.319635   \n",
      "Regulation of Insulin-like Growth Factor (IGF) ...            0.318200   \n",
      "Phospholipase C-mediated cascade; FGFR4                       0.316960   \n",
      "GLI proteins bind promoters of Hh responsive ge...            0.316865   \n",
      "Protein methylation                                           0.315486   \n",
      "LGI-ADAM interactions                                         0.315391   \n",
      "Metabolism of ingested SeMet, Sec, MeSec into H2Se            0.314822   \n",
      "PTK6 Regulates RTKs and Their Effectors AKT1 an...            0.313591   \n",
      "Biosynthesis of DPAn-3-derived protectins and r...            0.312034   \n",
      "Biosynthesis of DPAn-3-derived 13-series resolvins            0.310482   \n",
      "InlA-mediated entry of Listeria monocytogenes i...            0.309310   \n",
      "Amine ligand-binding receptors                                0.308421   \n",
      "Phospholipase C-mediated cascade; FGFR3                       0.308328   \n",
      "Biosynthesis of DPAn-3-derived maresins                       0.308188   \n",
      "B-WICH complex positively regulates rRNA expres...            0.307628   \n",
      "Fructose biosynthesis                                         0.307581   \n",
      "Myoclonic epilepsy of Lafora                                  0.306789   \n",
      "NoRC negatively regulates rRNA expression                     0.306323   \n",
      "Ligand-receptor interactions                                  0.304606   \n",
      "Pre-NOTCH Processing in the Endoplasmic Reticulum             0.303727   \n",
      "Membrane binding and targetting of GAG proteins               0.303588   \n",
      "Synthesis And Processing Of GAG, GAGPOL Polypro...            0.303449   \n",
      "Digestion and absorption                                      0.303218   \n",
      "Hypusine synthesis from eIF5A-lysine                          0.302019   \n",
      "Cytosolic sensors of pathogen-associated DNA                  0.301559   \n",
      "\n",
      "                                                    Cluster 5 q-values  \n",
      "Metabolism of amine-derived hormones                         57.816484  \n",
      "Catecholamine biosynthesis                                   49.588111  \n",
      "GRB7 events in ERBB2 signaling                               42.171419  \n",
      "Tryptophan catabolism                                        38.079833  \n",
      "Fertilization                                                37.768208  \n",
      "BH3-only proteins associate with and inactivate...           35.362276  \n",
      "Activation of gene expression by SREBF (SREBP)               35.008218  \n",
      "Sperm Motility And Taxes                                     33.759033  \n",
      "Cholesterol biosynthesis                                     32.929033  \n",
      "Proline catabolism                                           29.292353  \n",
      "Activation of AMPK downstream of NMDARs                      29.272281  \n",
      "Caspase-mediated cleavage of cytoskeletal proteins           29.114473  \n",
      "Manipulation of host energy metabolism                       28.490764  \n",
      "Signaling by Leptin                                          28.013617  \n",
      "Metabolism of folate and pterines                            27.588927  \n",
      "Synthesis of PIPs in the nucleus                             27.284218  \n",
      "alpha-linolenic (omega3) and linoleic (omega6) ...           26.473687  \n",
      "alpha-linolenic acid (ALA) metabolism                        26.473687  \n",
      "Metal sequestration by antimicrobial proteins                26.104411  \n",
      "Linoleic acid (LA) metabolism                                26.096443  \n",
      "PI3K Cascade                                                 25.898700  \n",
      "Glutathione synthesis and recycling                          25.881753  \n",
      "PI5P Regulates TP53 Acetylation                              25.846660  \n",
      "IRS activation                                               25.482079  \n",
      "Visual phototransduction                                     25.382444  \n",
      "p130Cas linkage to MAPK signaling for integrins              25.211603  \n",
      "Purine catabolism                                            24.968479  \n",
      "Transcriptional Regulation by MECP2                          24.671681  \n",
      "Response of EIF2AK1 (HRI) to heme deficiency                 24.518569  \n",
      "Phosphate bond hydrolysis by NUDT proteins                   24.280561  \n",
      "...                                                                ...  \n",
      "STING mediated induction of host immune responses             0.318430  \n",
      "Signaling by VEGF                                             0.318159  \n",
      "eNOS activation                                               0.317598  \n",
      "Metabolism of nitric oxide: eNOS activation and...            0.316412  \n",
      "Breakdown of the nuclear lamina                               0.316047  \n",
      "FGFR1c ligand binding and activation                          0.314580  \n",
      "Regulation of Insulin-like Growth Factor (IGF) ...            0.313357  \n",
      "Phospholipase C-mediated cascade; FGFR4                       0.312445  \n",
      "GLI proteins bind promoters of Hh responsive ge...            0.312445  \n",
      "Protein methylation                                           0.311394  \n",
      "LGI-ADAM interactions                                         0.311394  \n",
      "Metabolism of ingested SeMet, Sec, MeSec into H2Se            0.311036  \n",
      "PTK6 Regulates RTKs and Their Effectors AKT1 an...            0.310017  \n",
      "Biosynthesis of DPAn-3-derived protectins and r...            0.308671  \n",
      "Biosynthesis of DPAn-3-derived 13-series resolvins            0.307330  \n",
      "InlA-mediated entry of Listeria monocytogenes i...            0.306369  \n",
      "Amine ligand-binding receptors                                0.305878  \n",
      "Phospholipase C-mediated cascade; FGFR3                       0.305878  \n",
      "Biosynthesis of DPAn-3-derived maresins                       0.305878  \n",
      "B-WICH complex positively regulates rRNA expres...            0.305692  \n",
      "Fructose biosynthesis                                         0.305692  \n",
      "Myoclonic epilepsy of Lafora                                  0.305110  \n",
      "NoRC negatively regulates rRNA expression                     0.304855  \n",
      "Ligand-receptor interactions                                  0.303348  \n",
      "Pre-NOTCH Processing in the Endoplasmic Reticulum             0.302821  \n",
      "Membrane binding and targetting of GAG proteins               0.302821  \n",
      "Synthesis And Processing Of GAG, GAGPOL Polypro...            0.302821  \n",
      "Digestion and absorption                                      0.302799  \n",
      "Hypusine synthesis from eIF5A-lysine                          0.301810  \n",
      "Cytosolic sensors of pathogen-associated DNA                  0.301559  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 6 p-values\n",
      "Interleukin-6 signaling                                       0.001069\n",
      "Apoptosis                                                     0.410647\n",
      "Hemostasis                                                    0.000010\n",
      "Intrinsic Pathway for Apoptosis                               0.405259\n",
      "PKB-mediated events                                           0.049219\n",
      "PI3K Cascade                                                  0.000027\n",
      "MAPK3 (ERK1) activation                                       0.110215\n",
      "Translesion synthesis by REV1                                 0.009114\n",
      "Translesion synthesis by Y family DNA polymeras...            0.001082\n",
      "Recognition of DNA damage by PCNA-containing re...            0.006773\n",
      "Translesion Synthesis by POLH                                 0.005669\n",
      "Recognition and association of DNA glycosylase ...            0.003256\n",
      "Cleavage of the damaged pyrimidine                            0.003256\n",
      "Recognition and association of DNA glycosylase ...            0.003879\n",
      "Cleavage of the damaged purine                                0.003879\n",
      "Displacement of DNA glycosylase by APEX1                      0.331783\n",
      "POLB-Dependent Long Patch Base Excision Repair                0.036227\n",
      "Resolution of AP sites via the multiple-nucleot...            0.007903\n",
      "Resolution of AP sites via the single-nucleotid...            0.000437\n",
      "SLBP independent Processing of Histone Pre-mRNAs              0.098267\n",
      "Activation of BIM and translocation to mitochon...            0.403542\n",
      "Activation of BAD and translocation to mitochon...            0.002120\n",
      "Activation of NOXA and translocation to mitocho...            0.059726\n",
      "Activation and oligomerization of BAK protein                 0.200692\n",
      "BH3-only proteins associate with and inactivate...            0.000013\n",
      "Release of apoptotic factors from the mitochondria            0.001535\n",
      "Formation of apoptosome                                       0.003400\n",
      "Activation of caspases through apoptosome-media...            0.433722\n",
      "Cytochrome c-mediated apoptotic response                      0.009844\n",
      "SMAC (DIABLO) binds to IAPs                                   0.012832\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib                   0.284698\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...            0.005675\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...            0.003164\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...            0.430161\n",
      "MyD88 dependent cascade initiated on endosome                 0.003684\n",
      "IRAK2 mediated activation of TAK1 complex upon ...            0.000188\n",
      "N-glycan antennae elongation in the medial/tran...            0.085294\n",
      "N-Glycan antennae elongation                                  0.114445\n",
      "Reactions specific to the complex N-glycan synt...            0.071520\n",
      "Retinoid metabolism and transport                             0.000634\n",
      "MyD88 cascade initiated on plasma membrane                    0.078954\n",
      "Nonsense Mediated Decay (NMD) independent of th...            0.038168\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...            0.041992\n",
      "Termination of O-glycan biosynthesis                          0.000013\n",
      "Amyloid fiber formation                                       0.005638\n",
      "Serine biosynthesis                                           0.000011\n",
      "GABA receptor activation                                      0.207409\n",
      "GABA B receptor activation                                    0.209843\n",
      "Regulation of Complement cascade                              0.001201\n",
      "Growth hormone receptor signaling                             0.000015\n",
      "Antigen processing: Ubiquitination & Proteasome...            0.292332\n",
      "Class I MHC mediated antigen processing & prese...            0.011275\n",
      "Antigen Presentation: Folding, assembly and pep...            0.016727\n",
      "Kinesins                                                      0.000522\n",
      "Factors involved in megakaryocyte development a...            0.001408\n",
      "Antigen activates B Cell Receptor (BCR) leading...            0.000127\n",
      "Signaling by the B Cell Receptor (BCR)                        0.001777\n",
      "Ion channel transport                                         0.000001\n",
      "Activation of GABAB receptors                                 0.209843\n",
      "Inhibition  of voltage gated Ca2+ channels via ...            0.378241\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 6 p-values  \\\n",
      "Signaling by plasma membrane FGFR1 fusions                   51.157171   \n",
      "FGFR1 mutant receptor activation                             26.423871   \n",
      "TRAIL  signaling                                             18.662574   \n",
      "Repression of WNT target genes                               15.768507   \n",
      "FGFR1b ligand binding and activation                         15.648199   \n",
      "FGFR1c and Klotho ligand binding and activation              14.267684   \n",
      "RUNX3 regulates WNT signaling                                13.642549   \n",
      "Protein repair                                               13.098648   \n",
      "RUNX3 regulates CDKN1A transcription                         11.722635   \n",
      "Activated NTRK2 signals through CDK5                         11.480291   \n",
      "Synthesis of IP3 and IP4 in the cytosol                       9.939449   \n",
      "Binding of TCF/LEF:CTNNB1 to target gene promoters            9.403004   \n",
      "The NLRP3 inflammasome                                        9.290485   \n",
      "Androgen biosynthesis                                         8.979888   \n",
      "Spry regulation of FGF signaling                              8.854774   \n",
      "Sodium/Calcium exchangers                                     8.650182   \n",
      "Inositol phosphate metabolism                                 8.615409   \n",
      "Estrogen biosynthesis                                         8.595285   \n",
      "Interaction between PHLDA1 and AURKA                          8.563640   \n",
      "Inflammasomes                                                 8.347341   \n",
      "Reduction of cytosolic Ca++ levels                            8.233912   \n",
      "Neurotransmitter release cycle                                8.128633   \n",
      "Synthesis of very long-chain fatty acyl-CoAs                  8.092565   \n",
      "Molybdenum cofactor biosynthesis                              7.961696   \n",
      "Rho GTPase cycle                                              7.945779   \n",
      "Biosynthesis of electrophilic -3 PUFA oxo-deri...            7.825275   \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      7.816671   \n",
      "Synthesis of PI                                               7.778970   \n",
      "Interleukin-18 signaling                                      7.769918   \n",
      "Synthesis of PG                                               7.740417   \n",
      "...                                                                ...   \n",
      "truncated APC mutants destabilize the destructi...            0.314112   \n",
      "AMER1 mutants destabilize the destruction complex             0.314112   \n",
      "APC truncation mutants have impaired AXIN binding             0.314043   \n",
      "Post-transcriptional silencing by small RNAs                  0.313768   \n",
      "Disease                                                       0.311642   \n",
      "Regulation of mRNA stability by proteins that b...            0.311505   \n",
      "p53-Dependent G1 DNA Damage Response                          0.311505   \n",
      "p53-Dependent G1/S DNA damage checkpoint                      0.311505   \n",
      "Serotonin and melatonin biosynthesis                          0.310959   \n",
      "Asymmetric localization of PCP proteins                       0.310754   \n",
      "Activation of the TFAP2 (AP-2) family of transc...            0.310140   \n",
      "IRAK2 mediated activation of TAK1 complex                     0.309867   \n",
      "Cytosolic sulfonation of small molecules                      0.309186   \n",
      "Adenosine P1 receptors                                        0.309050   \n",
      "Oxygen-dependent proline hydroxylation of Hypox...            0.308235   \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...            0.307218   \n",
      "Metal ion SLC transporters                                    0.307150   \n",
      "Highly sodium permeable acetylcholine nicotinic...            0.305595   \n",
      "Defective DHDDS causes retinitis pigmentosa 59                0.304853   \n",
      "Insulin receptor signalling cascade                           0.304584   \n",
      "Loss of Function of TGFBR2 in Cancer                          0.303374   \n",
      "TGFBR2 Kinase Domain Mutants in Cancer                        0.303374   \n",
      "Defects in vitamin and cofactor metabolism                    0.303239   \n",
      "Defective POMGNT1 causes MDDGA3, MDDGB3 and MDDGC3            0.303105   \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...            0.302837   \n",
      "Metabolism of steroid hormones                                0.302703   \n",
      "Host Interactions of HIV factors                              0.302234   \n",
      "Defective CHST14 causes EDS, musculocontractura...            0.301631   \n",
      "Amine Oxidase reactions                                       0.301498   \n",
      "Regulation of PTEN stability and activity                     0.301164   \n",
      "\n",
      "                                                    Cluster 6 q-values  \n",
      "Signaling by plasma membrane FGFR1 fusions                   47.840363  \n",
      "FGFR1 mutant receptor activation                             23.408093  \n",
      "TRAIL  signaling                                             15.822886  \n",
      "Repression of WNT target genes                               13.053758  \n",
      "FGFR1b ligand binding and activation                         13.030361  \n",
      "FGFR1c and Klotho ligand binding and activation              11.729027  \n",
      "RUNX3 regulates WNT signaling                                11.170838  \n",
      "Protein repair                                               10.684929  \n",
      "RUNX3 regulates CDKN1A transcription                          9.360069  \n",
      "Activated NTRK2 signals through CDK5                          9.163482  \n",
      "Synthesis of IP3 and IP4 in the cytosol                       7.664033  \n",
      "Binding of TCF/LEF:CTNNB1 to target gene promoters            7.165376  \n",
      "The NLRP3 inflammasome                                        7.087620  \n",
      "Androgen biosynthesis                                         6.809207  \n",
      "Spry regulation of FGF signaling                              6.714056  \n",
      "Sodium/Calcium exchangers                                     6.537493  \n",
      "Inositol phosphate metabolism                                 6.533748  \n",
      "Estrogen biosynthesis                                         6.533748  \n",
      "Interaction between PHLDA1 and AURKA                          6.525585  \n",
      "Inflammasomes                                                 6.331562  \n",
      "Reduction of cytosolic Ca++ levels                            6.239323  \n",
      "Neurotransmitter release cycle                                6.154247  \n",
      "Synthesis of very long-chain fatty acyl-CoAs                  6.137484  \n",
      "Molybdenum cofactor biosynthesis                              6.026910  \n",
      "Rho GTPase cycle                                              6.026910  \n",
      "Biosynthesis of electrophilic -3 PUFA oxo-deri...            5.931226  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      5.931226  \n",
      "Synthesis of PI                                               5.915507  \n",
      "Interleukin-18 signaling                                      5.915507  \n",
      "Synthesis of PG                                               5.900729  \n",
      "...                                                                ...  \n",
      "truncated APC mutants destabilize the destructi...            0.308352  \n",
      "AMER1 mutants destabilize the destruction complex             0.308352  \n",
      "APC truncation mutants have impaired AXIN binding             0.308352  \n",
      "Post-transcriptional silencing by small RNAs                  0.308289  \n",
      "Disease                                                       0.306874  \n",
      "Regulation of mRNA stability by proteins that b...            0.306874  \n",
      "p53-Dependent G1 DNA Damage Response                          0.306874  \n",
      "p53-Dependent G1/S DNA damage checkpoint                      0.306874  \n",
      "Serotonin and melatonin biosynthesis                          0.306545  \n",
      "Asymmetric localization of PCP proteins                       0.306545  \n",
      "Activation of the TFAP2 (AP-2) family of transc...            0.306143  \n",
      "IRAK2 mediated activation of TAK1 complex                     0.306082  \n",
      "Cytosolic sulfonation of small molecules                      0.305687  \n",
      "Adenosine P1 receptors                                        0.305687  \n",
      "Oxygen-dependent proline hydroxylation of Hypox...            0.305082  \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...            0.304419  \n",
      "Metal ion SLC transporters                                    0.304419  \n",
      "Highly sodium permeable acetylcholine nicotinic...            0.303075  \n",
      "Defective DHDDS causes retinitis pigmentosa 59                0.302544  \n",
      "Insulin receptor signalling cascade                           0.302485  \n",
      "Loss of Function of TGFBR2 in Cancer                          0.301864  \n",
      "TGFBR2 Kinase Domain Mutants in Cancer                        0.301864  \n",
      "Defects in vitamin and cofactor metabolism                    0.301864  \n",
      "Defective POMGNT1 causes MDDGA3, MDDGB3 and MDDGC3            0.301864  \n",
      "NR1H2 & NR1H3 regulate gene expression linked t...            0.301864  \n",
      "Metabolism of steroid hormones                                0.301864  \n",
      "Host Interactions of HIV factors                              0.301605  \n",
      "Defective CHST14 causes EDS, musculocontractura...            0.301288  \n",
      "Amine Oxidase reactions                                       0.301288  \n",
      "Regulation of PTEN stability and activity                     0.301164  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 7 p-values\n",
      "Interleukin-6 signaling                                   7.374500e-02\n",
      "Apoptosis                                                 1.092507e-12\n",
      "Hemostasis                                                1.755731e-13\n",
      "Intrinsic Pathway for Apoptosis                           4.892944e-21\n",
      "PKB-mediated events                                       1.953309e-06\n",
      "PI3K Cascade                                              8.072916e-02\n",
      "MAPK3 (ERK1) activation                                   6.037839e-11\n",
      "Translesion synthesis by REV1                             1.166875e-02\n",
      "Translesion synthesis by Y family DNA polymeras...        1.186329e-08\n",
      "Recognition of DNA damage by PCNA-containing re...        4.011272e-06\n",
      "Translesion Synthesis by POLH                             2.867555e-06\n",
      "Recognition and association of DNA glycosylase ...        2.079250e-02\n",
      "Cleavage of the damaged pyrimidine                        2.079250e-02\n",
      "Recognition and association of DNA glycosylase ...        1.328188e-02\n",
      "Cleavage of the damaged purine                            1.328188e-02\n",
      "Displacement of DNA glycosylase by APEX1                  2.408974e-04\n",
      "POLB-Dependent Long Patch Base Excision Repair            2.565209e-01\n",
      "Resolution of AP sites via the multiple-nucleot...        1.764364e-04\n",
      "Resolution of AP sites via the single-nucleotid...        1.759900e-04\n",
      "SLBP independent Processing of Histone Pre-mRNAs          3.115926e-07\n",
      "Activation of BIM and translocation to mitochon...        3.228096e-02\n",
      "Activation of BAD and translocation to mitochon...        8.674905e-15\n",
      "Activation of NOXA and translocation to mitocho...        5.890016e-04\n",
      "Activation and oligomerization of BAK protein             1.119952e-01\n",
      "BH3-only proteins associate with and inactivate...        4.828710e-20\n",
      "Release of apoptotic factors from the mitochondria        4.803482e-02\n",
      "Formation of apoptosome                                   1.991970e-11\n",
      "Activation of caspases through apoptosome-media...        9.514684e-12\n",
      "Cytochrome c-mediated apoptotic response                  4.814829e-09\n",
      "SMAC (DIABLO) binds to IAPs                               4.585045e-02\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               1.911436e-02\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        3.206722e-04\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        2.840591e-11\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        1.487369e-01\n",
      "MyD88 dependent cascade initiated on endosome             3.643449e-11\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        6.473129e-06\n",
      "N-glycan antennae elongation in the medial/tran...        9.494959e-07\n",
      "N-Glycan antennae elongation                              1.313336e-02\n",
      "Reactions specific to the complex N-glycan synt...        5.166672e-06\n",
      "Retinoid metabolism and transport                         5.426275e-09\n",
      "MyD88 cascade initiated on plasma membrane                2.886721e-09\n",
      "Nonsense Mediated Decay (NMD) independent of th...        4.700578e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        4.429269e-01\n",
      "Termination of O-glycan biosynthesis                      2.382404e-02\n",
      "Amyloid fiber formation                                   1.519549e-02\n",
      "Serine biosynthesis                                       7.594028e-24\n",
      "GABA receptor activation                                  3.726844e-05\n",
      "GABA B receptor activation                                2.452429e-05\n",
      "Regulation of Complement cascade                          2.573612e-11\n",
      "Growth hormone receptor signaling                         2.206511e-20\n",
      "Antigen processing: Ubiquitination & Proteasome...        4.110439e-07\n",
      "Class I MHC mediated antigen processing & prese...        8.718712e-12\n",
      "Antigen Presentation: Folding, assembly and pep...        1.886840e-07\n",
      "Kinesins                                                  1.389083e-06\n",
      "Factors involved in megakaryocyte development a...        6.947969e-08\n",
      "Antigen activates B Cell Receptor (BCR) leading...        7.227805e-09\n",
      "Signaling by the B Cell Receptor (BCR)                    1.634247e-10\n",
      "Ion channel transport                                     2.497604e-36\n",
      "Activation of GABAB receptors                             2.452429e-05\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        3.418431e-03\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 7 p-values  \\\n",
      "Cytosolic iron-sulfur cluster assembly                       37.119923   \n",
      "Ion channel transport                                        35.602476   \n",
      "Ubiquinol biosynthesis                                       32.449809   \n",
      "CaM pathway                                                  30.711804   \n",
      "RUNX1 regulates estrogen receptor mediated tran...           30.708438   \n",
      "Calmodulin induced events                                    30.702379   \n",
      "Ca-dependent events                                          30.567229   \n",
      "DAG and IP3 signaling                                        30.479981   \n",
      "Amino acids regulate mTORC1                                  30.227688   \n",
      "RUNX1 regulates transcription of genes involved...           30.017712   \n",
      "Activation of NMDA receptors and postsynaptic e...           29.545872   \n",
      "PLC beta mediated events                                     29.365975   \n",
      "Post NMDA receptor activation events                         29.332431   \n",
      "BBSome-mediated cargo-targeting to cilium                    29.319282   \n",
      "G-protein mediated events                                    28.936659   \n",
      "Intraflagellar transport                                     28.235234   \n",
      "Cargo trafficking to the periciliary membrane                27.838213   \n",
      "PKA-mediated phosphorylation of CREB                         27.580874   \n",
      "Signaling by Insulin receptor                                27.551578   \n",
      "RAB geranylgeranylation                                      27.163346   \n",
      "ROS and RNS production in phagocytes                         26.593195   \n",
      "Acyl chain remodelling of PE                                 26.532589   \n",
      "Iron uptake and transport                                    26.216349   \n",
      "Amino acid transport across the plasma membrane              25.823183   \n",
      "Metabolism of nucleotides                                    25.478923   \n",
      "Metabolism of vitamins and cofactors                         24.715610   \n",
      "Carboxyterminal post-translational modification...           24.686723   \n",
      "Cardiac conduction                                           24.192944   \n",
      "Metabolism of cofactors                                      24.029496   \n",
      "Activation of AMPK downstream of NMDARs                      23.913943   \n",
      "...                                                                ...   \n",
      "Nonsense Mediated Decay (NMD) independent of th...            0.327849   \n",
      "Formation of a pool of free 40S subunits                      0.325855   \n",
      "Tat-mediated HIV elongation arrest and recovery               0.325612   \n",
      "Pausing and recovery of Tat-mediated HIV elonga...            0.325612   \n",
      "Biosynthesis of maresin-like SPMs                             0.325418   \n",
      "Formation of the active cofactor, UDP-glucuronate             0.324933   \n",
      "RNA Polymerase I Transcription Termination                    0.324836   \n",
      "Ribosomal scanning and start codon recognition                0.324352   \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.324062   \n",
      "Intestinal hexose absorption                                  0.323820   \n",
      "L13a-mediated translational silencing of Cerulo...            0.323579   \n",
      "Metabolism of Angiotensinogen to Angiotensins                 0.322855   \n",
      "Cap-dependent Translation Initiation                          0.322638   \n",
      "Eukaryotic Translation Initiation                             0.322614   \n",
      "Orexin and neuropeptides FF and QRFP bind to th...            0.321795   \n",
      "mTOR signalling                                               0.319060   \n",
      "Mitochondrial tRNA aminoacylation                             0.317818   \n",
      "Disassembly of the destruction complex and recr...            0.317723   \n",
      "Opsins                                                        0.317437   \n",
      "Activation of the mRNA upon binding of the cap-...            0.316722   \n",
      "SRP-dependent cotranslational protein targeting...            0.316722   \n",
      "Selenoamino acid metabolism                                   0.314254   \n",
      "GTP hydrolysis and joining of the 60S ribosomal...            0.313686   \n",
      "Defective CHST14 causes EDS, musculocontractura...            0.312553   \n",
      "Regulation of gap junction activity                           0.311752   \n",
      "Translation initiation complex formation                      0.310341   \n",
      "Formation of the ternary complex, and subsequen...            0.310154   \n",
      "Defective Mismatch Repair Associated With MSH6                0.308983   \n",
      "Methionine salvage pathway                                    0.308048   \n",
      "NF-kB is activated and signals survival                       0.304143   \n",
      "\n",
      "                                                    Cluster 7 q-values  \n",
      "Cytosolic iron-sulfur cluster assembly                       33.803114  \n",
      "Ion channel transport                                        32.586698  \n",
      "Ubiquinol biosynthesis                                       29.610121  \n",
      "CaM pathway                                                  28.163722  \n",
      "RUNX1 regulates estrogen receptor mediated tran...           28.163722  \n",
      "Calmodulin induced events                                    28.163722  \n",
      "Ca-dependent events                                          28.095518  \n",
      "DAG and IP3 signaling                                        28.066262  \n",
      "Amino acids regulate mTORC1                                  27.865122  \n",
      "RUNX1 regulates transcription of genes involved...           27.700903  \n",
      "Activation of NMDA receptors and postsynaptic e...           27.270456  \n",
      "PLC beta mediated events                                     27.148601  \n",
      "Post NMDA receptor activation events                         27.148601  \n",
      "BBSome-mediated cargo-targeting to cilium                    27.148601  \n",
      "G-protein mediated events                                    26.795941  \n",
      "Intraflagellar transport                                     26.122545  \n",
      "Cargo trafficking to the periciliary membrane                25.751854  \n",
      "PKA-mediated phosphorylation of CREB                         25.519338  \n",
      "Signaling by Insulin receptor                                25.513523  \n",
      "RAB geranylgeranylation                                      25.147568  \n",
      "ROS and RNS production in phagocytes                         24.598606  \n",
      "Acyl chain remodelling of PE                                 24.558202  \n",
      "Iron uptake and transport                                    24.261268  \n",
      "Amino acid transport across the plasma membrane              23.886585  \n",
      "Metabolism of nucleotides                                    23.560055  \n",
      "Metabolism of vitamins and cofactors                         22.813775  \n",
      "Carboxyterminal post-translational modification...           22.801278  \n",
      "Cardiac conduction                                           22.323293  \n",
      "Metabolism of cofactors                                      22.175085  \n",
      "Activation of AMPK downstream of NMDARs                      22.075478  \n",
      "...                                                                ...  \n",
      "Nonsense Mediated Decay (NMD) independent of th...            0.321733  \n",
      "Formation of a pool of free 40S subunits                      0.320151  \n",
      "Tat-mediated HIV elongation arrest and recovery               0.320151  \n",
      "Pausing and recovery of Tat-mediated HIV elonga...            0.320151  \n",
      "Biosynthesis of maresin-like SPMs                             0.320151  \n",
      "Formation of the active cofactor, UDP-glucuronate             0.319993  \n",
      "RNA Polymerase I Transcription Termination                    0.319993  \n",
      "Ribosomal scanning and start codon recognition                0.319721  \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.319642  \n",
      "Intestinal hexose absorption                                  0.319612  \n",
      "L13a-mediated translational silencing of Cerulo...            0.319582  \n",
      "Metabolism of Angiotensinogen to Angiotensins                 0.319250  \n",
      "Cap-dependent Translation Initiation                          0.319250  \n",
      "Eukaryotic Translation Initiation                             0.319250  \n",
      "Orexin and neuropeptides FF and QRFP bind to th...            0.318643  \n",
      "mTOR signalling                                               0.316119  \n",
      "Mitochondrial tRNA aminoacylation                             0.315203  \n",
      "Disassembly of the destruction complex and recr...            0.315203  \n",
      "Opsins                                                        0.315127  \n",
      "Activation of the mRNA upon binding of the cap-...            0.314833  \n",
      "SRP-dependent cotranslational protein targeting...            0.314833  \n",
      "Selenoamino acid metabolism                                   0.312575  \n",
      "GTP hydrolysis and joining of the 60S ribosomal...            0.312218  \n",
      "Defective CHST14 causes EDS, musculocontractura...            0.311294  \n",
      "Regulation of gap junction activity                           0.310703  \n",
      "Translation initiation complex formation                      0.309525  \n",
      "Formation of the ternary complex, and subsequen...            0.309525  \n",
      "Defective Mismatch Repair Associated With MSH6                0.308564  \n",
      "Methionine salvage pathway                                    0.307838  \n",
      "NF-kB is activated and signals survival                       0.304143  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 8 p-values\n",
      "Interleukin-6 signaling                                   3.903899e-08\n",
      "Apoptosis                                                 3.324388e-25\n",
      "Hemostasis                                                1.811018e-47\n",
      "Intrinsic Pathway for Apoptosis                           3.263356e-10\n",
      "PKB-mediated events                                       7.049243e-10\n",
      "PI3K Cascade                                              1.362356e-04\n",
      "MAPK3 (ERK1) activation                                   5.297822e-16\n",
      "Translesion synthesis by REV1                             4.464854e-04\n",
      "Translesion synthesis by Y family DNA polymeras...        7.456967e-08\n",
      "Recognition of DNA damage by PCNA-containing re...        8.682188e-05\n",
      "Translesion Synthesis by POLH                             1.606987e-01\n",
      "Recognition and association of DNA glycosylase ...        9.871047e-04\n",
      "Cleavage of the damaged pyrimidine                        9.871047e-04\n",
      "Recognition and association of DNA glycosylase ...        1.244310e-03\n",
      "Cleavage of the damaged purine                            1.244310e-03\n",
      "Displacement of DNA glycosylase by APEX1                  2.295830e-02\n",
      "POLB-Dependent Long Patch Base Excision Repair            5.772357e-03\n",
      "Resolution of AP sites via the multiple-nucleot...        3.750345e-05\n",
      "Resolution of AP sites via the single-nucleotid...        3.873764e-27\n",
      "SLBP independent Processing of Histone Pre-mRNAs          1.400692e-03\n",
      "Activation of BIM and translocation to mitochon...        2.875426e-01\n",
      "Activation of BAD and translocation to mitochon...        5.360021e-10\n",
      "Activation of NOXA and translocation to mitocho...        3.050981e-03\n",
      "Activation and oligomerization of BAK protein             9.851380e-03\n",
      "BH3-only proteins associate with and inactivate...        8.356056e-16\n",
      "Release of apoptotic factors from the mitochondria        5.971711e-04\n",
      "Formation of apoptosome                                   2.166667e-01\n",
      "Activation of caspases through apoptosome-media...        4.575354e-19\n",
      "Cytochrome c-mediated apoptotic response                  3.487100e-01\n",
      "SMAC (DIABLO) binds to IAPs                               1.908059e-03\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               4.123271e-03\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        1.057923e-15\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        7.662534e-25\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        3.948674e-05\n",
      "MyD88 dependent cascade initiated on endosome             9.760520e-24\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        1.205386e-29\n",
      "N-glycan antennae elongation in the medial/tran...        7.480910e-13\n",
      "N-Glycan antennae elongation                              8.731347e-08\n",
      "Reactions specific to the complex N-glycan synt...        3.897574e-13\n",
      "Retinoid metabolism and transport                         8.852532e-23\n",
      "MyD88 cascade initiated on plasma membrane                3.740664e-14\n",
      "Nonsense Mediated Decay (NMD) independent of th...        1.318548e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        1.067171e-01\n",
      "Termination of O-glycan biosynthesis                      2.834453e-17\n",
      "Amyloid fiber formation                                   1.962416e-03\n",
      "Serine biosynthesis                                       3.493027e-33\n",
      "GABA receptor activation                                  2.968715e-14\n",
      "GABA B receptor activation                                6.021604e-15\n",
      "Regulation of Complement cascade                          3.522025e-57\n",
      "Growth hormone receptor signaling                         1.182194e-53\n",
      "Antigen processing: Ubiquitination & Proteasome...        3.578764e-11\n",
      "Class I MHC mediated antigen processing & prese...        4.716160e-42\n",
      "Antigen Presentation: Folding, assembly and pep...        2.882821e-36\n",
      "Kinesins                                                  7.939270e-16\n",
      "Factors involved in megakaryocyte development a...        3.280411e-17\n",
      "Antigen activates B Cell Receptor (BCR) leading...        3.053966e-47\n",
      "Signaling by the B Cell Receptor (BCR)                    9.058014e-46\n",
      "Ion channel transport                                     9.396050e-33\n",
      "Activation of GABAB receptors                             6.021604e-15\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        1.721601e-10\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 8 p-values  \\\n",
      "Noncanonical activation of NOTCH3                           105.638692   \n",
      "HHAT G278V abrogates palmitoylation of Hh-Np                 73.661161   \n",
      "Sphingolipid metabolism                                      71.344263   \n",
      "GAB1 signalosome                                             71.258322   \n",
      "Intraflagellar transport                                     70.124779   \n",
      "Biosynthesis of the N-glycan precursor (dolicho...           64.755681   \n",
      "Synthesis of substrates in N-glycan biosythesis              64.580772   \n",
      "Transport of bile salts and organic acids, meta...           64.570253   \n",
      "SLC-mediated transmembrane transport                         63.770118   \n",
      "Glycosphingolipid metabolism                                 62.713278   \n",
      "Synthesis of epoxy (EET) and dihydroxyeicosatri...           59.236665   \n",
      "Sialic acid metabolism                                       58.509400   \n",
      "Synthesis of PC                                              57.506377   \n",
      "Complement cascade                                           56.482686   \n",
      "Metabolism of vitamins and cofactors                         56.465300   \n",
      "Regulation of Complement cascade                             56.453208   \n",
      "EGFR interacts with phospholipase C-gamma                    55.142265   \n",
      "RUNX1 regulates transcription of genes involved...           54.227634   \n",
      "Nucleobase catabolism                                        53.837400   \n",
      "Initial triggering of complement                             53.229063   \n",
      "Growth hormone receptor signaling                            52.927311   \n",
      "Phase I - Functionalization of compounds                     52.483602   \n",
      "Carboxyterminal post-translational modification...           52.133924   \n",
      "Interleukin-12 family signaling                              51.636550   \n",
      "RUNX3 Regulates Immune Response and Cell Migration           51.497259   \n",
      "Activation of AMPK downstream of NMDARs                      51.071199   \n",
      "Ubiquinol biosynthesis                                       51.067608   \n",
      "Alternative complement activation                            50.983620   \n",
      "Immune System                                                50.927667   \n",
      "Interleukin-12 signaling                                     50.806543   \n",
      "...                                                                ...   \n",
      "HDR through MMEJ (alt-NHEJ)                                   0.361828   \n",
      "Axonal growth stimulation                                     0.361142   \n",
      "Citric acid cycle (TCA cycle)                                 0.355943   \n",
      "Rap1 signalling                                               0.353320   \n",
      "IRF3 mediated activation of type 1 IFN                        0.352772   \n",
      "Formation of the active cofactor, UDP-glucuronate             0.351046   \n",
      "Lysosomal oligosaccharide catabolism                          0.350878   \n",
      "HIV Life Cycle                                                0.344908   \n",
      "Protein methylation                                           0.343499   \n",
      "mitochondrial fatty acid beta-oxidation of unsa...            0.343126   \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.334679   \n",
      "Digestion of dietary carbohydrate                             0.333988   \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      0.330912   \n",
      "HDL clearance                                                 0.329703   \n",
      "STAT6-mediated induction of chemokines                        0.328899   \n",
      "Glucuronidation                                               0.327615   \n",
      "Mitochondrial iron-sulfur cluster biogenesis                  0.327375   \n",
      "Zinc influx into cells by the SLC39 gene family               0.327095   \n",
      "tRNA modification in the mitochondrion                        0.326855   \n",
      "FGFR1c ligand binding and activation                          0.325138   \n",
      "HCMV Infection                                                0.324739   \n",
      "DNA Double Strand Break Response                              0.324381   \n",
      "Defective SLC16A1 causes symptomatic deficiency...            0.322673   \n",
      "Notch-HLH transcription pathway                               0.320971   \n",
      "ERKs are inactivated                                          0.319315   \n",
      "Melanin biosynthesis                                          0.314420   \n",
      "FGFR1 ligand binding and activation                           0.313681   \n",
      "Dual Incision in GG-NER                                       0.313564   \n",
      "Resolution of D-loop Structures through Hollida...            0.307426   \n",
      "Resolution of D-Loop Structures                               0.304902   \n",
      "\n",
      "                                                    Cluster 8 q-values  \n",
      "Noncanonical activation of NOTCH3                           102.321884  \n",
      "HHAT G278V abrogates palmitoylation of Hh-Np                 70.645382  \n",
      "Sphingolipid metabolism                                      68.543574  \n",
      "GAB1 signalosome                                             68.543574  \n",
      "Intraflagellar transport                                     67.506940  \n",
      "Biosynthesis of the N-glycan precursor (dolicho...           62.217024  \n",
      "Synthesis of substrates in N-glycan biosythesis              62.156534  \n",
      "Transport of bile salts and organic acids, meta...           62.156534  \n",
      "SLC-mediated transmembrane transport                         61.407552  \n",
      "Glycosphingolipid metabolism                                 60.396469  \n",
      "Synthesis of epoxy (EET) and dihydroxyeicosatri...           56.961249  \n",
      "Sialic acid metabolism                                       56.271772  \n",
      "Synthesis of PC                                              55.303511  \n",
      "Complement cascade                                           54.340519  \n",
      "Metabolism of vitamins and cofactors                         54.340519  \n",
      "Regulation of Complement cascade                             54.340519  \n",
      "EGFR interacts with phospholipase C-gamma                    53.055905  \n",
      "RUNX1 regulates transcription of genes involved...           52.166098  \n",
      "Nucleobase catabolism                                        51.799345  \n",
      "Initial triggering of complement                             51.213284  \n",
      "Growth hormone receptor signaling                            50.932722  \n",
      "Phase I - Functionalization of compounds                     50.509216  \n",
      "Carboxyterminal post-translational modification...           50.178843  \n",
      "Interleukin-12 family signaling                              49.699953  \n",
      "RUNX3 Regulates Immune Response and Cell Migration           49.578390  \n",
      "Activation of AMPK downstream of NMDARs                      49.182163  \n",
      "Ubiquinol biosynthesis                                       49.182163  \n",
      "Alternative complement activation                            49.113969  \n",
      "Immune System                                                49.073256  \n",
      "Interleukin-12 signaling                                     48.966855  \n",
      "...                                                                ...  \n",
      "HDR through MMEJ (alt-NHEJ)                                   0.355712  \n",
      "Axonal growth stimulation                                     0.355239  \n",
      "Citric acid cycle (TCA cycle)                                 0.350252  \n",
      "Rap1 signalling                                               0.347841  \n",
      "IRF3 mediated activation of type 1 IFN                        0.347505  \n",
      "Formation of the active cofactor, UDP-glucuronate             0.346035  \n",
      "Lysosomal oligosaccharide catabolism                          0.346035  \n",
      "HIV Life Cycle                                                0.340276  \n",
      "Protein methylation                                           0.339079  \n",
      "mitochondrial fatty acid beta-oxidation of unsa...            0.338918  \n",
      "TWIK-releated acid-sensitive K+ channel (TASK)                0.330682  \n",
      "Digestion of dietary carbohydrate                             0.330202  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                      0.327338  \n",
      "HDL clearance                                                 0.326340  \n",
      "STAT6-mediated induction of chemokines                        0.325747  \n",
      "Glucuronidation                                               0.324674  \n",
      "Mitochondrial iron-sulfur cluster biogenesis                  0.324644  \n",
      "Zinc influx into cells by the SLC39 gene family               0.324575  \n",
      "tRNA modification in the mitochondrion                        0.324545  \n",
      "FGFR1c ligand binding and activation                          0.323039  \n",
      "HCMV Infection                                                0.322851  \n",
      "DNA Double Strand Break Response                              0.322703  \n",
      "Defective SLC16A1 causes symptomatic deficiency...            0.321205  \n",
      "Notch-HLH transcription pathway                               0.319713  \n",
      "ERKs are inactivated                                          0.318267  \n",
      "Melanin biosynthesis                                          0.313582  \n",
      "FGFR1 ligand binding and activation                           0.313145  \n",
      "Dual Incision in GG-NER                                       0.313145  \n",
      "Resolution of D-loop Structures through Hollida...            0.307216  \n",
      "Resolution of D-Loop Structures                               0.304902  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    Cluster 9 p-values\n",
      "Interleukin-6 signaling                                   1.903171e-05\n",
      "Apoptosis                                                 3.249663e-01\n",
      "Hemostasis                                                4.092132e-04\n",
      "Intrinsic Pathway for Apoptosis                           4.138321e-01\n",
      "PKB-mediated events                                       6.177639e-02\n",
      "PI3K Cascade                                              9.455896e-06\n",
      "MAPK3 (ERK1) activation                                   1.334850e-02\n",
      "Translesion synthesis by REV1                             9.190198e-02\n",
      "Translesion synthesis by Y family DNA polymeras...        2.830620e-07\n",
      "Recognition of DNA damage by PCNA-containing re...        1.188100e-05\n",
      "Translesion Synthesis by POLH                             8.378610e-03\n",
      "Recognition and association of DNA glycosylase ...        2.898588e-02\n",
      "Cleavage of the damaged pyrimidine                        2.897599e-02\n",
      "Recognition and association of DNA glycosylase ...        4.178814e-02\n",
      "Cleavage of the damaged purine                            4.178814e-02\n",
      "Displacement of DNA glycosylase by APEX1                  2.274536e-01\n",
      "POLB-Dependent Long Patch Base Excision Repair            1.326491e-11\n",
      "Resolution of AP sites via the multiple-nucleot...        1.367507e-09\n",
      "Resolution of AP sites via the single-nucleotid...        6.935304e-03\n",
      "SLBP independent Processing of Histone Pre-mRNAs          6.125713e-08\n",
      "Activation of BIM and translocation to mitochon...        6.916510e-02\n",
      "Activation of BAD and translocation to mitochon...        1.665008e-04\n",
      "Activation of NOXA and translocation to mitocho...        3.711715e-01\n",
      "Activation and oligomerization of BAK protein             6.861007e-02\n",
      "BH3-only proteins associate with and inactivate...        6.611645e-02\n",
      "Release of apoptotic factors from the mitochondria        3.775396e-04\n",
      "Formation of apoptosome                                   1.694821e-01\n",
      "Activation of caspases through apoptosome-media...        2.321397e-03\n",
      "Cytochrome c-mediated apoptotic response                  2.378670e-01\n",
      "SMAC (DIABLO) binds to IAPs                               1.348535e-01\n",
      "...                                                                ...\n",
      "Resistance of ERBB2 KD mutants to lapatinib               4.066071e-03\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 s...        1.520142e-02\n",
      "TRAF6 mediated induction of NFkB and MAP kinase...        1.230064e-02\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 sti...        3.905891e-01\n",
      "MyD88 dependent cascade initiated on endosome             7.848048e-03\n",
      "IRAK2 mediated activation of TAK1 complex upon ...        2.117392e-03\n",
      "N-glycan antennae elongation in the medial/tran...        1.923207e-08\n",
      "N-Glycan antennae elongation                              5.578617e-05\n",
      "Reactions specific to the complex N-glycan synt...        3.219893e-08\n",
      "Retinoid metabolism and transport                         2.551188e-15\n",
      "MyD88 cascade initiated on plasma membrane                2.248833e-02\n",
      "Nonsense Mediated Decay (NMD) independent of th...        2.219887e-01\n",
      "Nonsense Mediated Decay (NMD) enhanced by the E...        1.896765e-01\n",
      "Termination of O-glycan biosynthesis                      5.306762e-02\n",
      "Amyloid fiber formation                                   1.057904e-02\n",
      "Serine biosynthesis                                       1.331745e-01\n",
      "GABA receptor activation                                  1.883020e-01\n",
      "GABA B receptor activation                                1.754366e-01\n",
      "Regulation of Complement cascade                          1.562066e-01\n",
      "Growth hormone receptor signaling                         4.062783e-01\n",
      "Antigen processing: Ubiquitination & Proteasome...        2.534867e-01\n",
      "Class I MHC mediated antigen processing & prese...        4.875098e-01\n",
      "Antigen Presentation: Folding, assembly and pep...        3.715105e-01\n",
      "Kinesins                                                  2.167304e-14\n",
      "Factors involved in megakaryocyte development a...        1.501172e-14\n",
      "Antigen activates B Cell Receptor (BCR) leading...        4.346734e-01\n",
      "Signaling by the B Cell Receptor (BCR)                    3.575844e-01\n",
      "Ion channel transport                                     1.226540e-05\n",
      "Activation of GABAB receptors                             1.754366e-01\n",
      "Inhibition  of voltage gated Ca2+ channels via ...        3.001546e-01\n",
      "\n",
      "[2074 rows x 1 columns]\n",
      "                                                    Cluster 9 p-values  \\\n",
      "Vitamin B2 (riboflavin) metabolism                           52.594921   \n",
      "Synthesis And Processing Of GAG, GAGPOL Polypro...           49.226819   \n",
      "Membrane binding and targetting of GAG proteins              49.224887   \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                     41.394100   \n",
      "Assembly Of The HIV Virion                                   38.593533   \n",
      "Tristetraprolin (TTP, ZFP36) binds and destabil...           37.536366   \n",
      "Glutamate and glutamine metabolism                           37.401918   \n",
      "Synthesis of PS                                              35.525131   \n",
      "Butyrate Response Factor 1 (BRF1) binds and des...           34.315077   \n",
      "Protein repair                                               22.613583   \n",
      "Processing of Intronless Pre-mRNAs                           21.759531   \n",
      "Mitotic Telophase/Cytokinesis                                20.847865   \n",
      "RNA Polymerase III Chain Elongation                          20.318293   \n",
      "GDP-fucose biosynthesis                                      20.257643   \n",
      "Sema3A PAK dependent Axon repulsion                          19.653908   \n",
      "Telomere Extension By Telomerase                             19.440286   \n",
      "RNA Polymerase III Transcription Termination                 18.424724   \n",
      "tRNA processing                                              17.371009   \n",
      "Cooperation of Prefoldin and TriC/CCT  in actin...           17.046528   \n",
      "Type I hemidesmosome assembly                                16.975536   \n",
      "Prefoldin mediated transfer of substrate  to CC...           16.947629   \n",
      "Phosphorylation of proteins involved in G1/S tr...           16.913615   \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...           16.359445   \n",
      "G0 and Early G1                                              16.055312   \n",
      "CRMPs in Sema3A signaling                                    15.974605   \n",
      "APC/C:Cdc20 mediated degradation of Cyclin B                 15.938383   \n",
      "HDR through MMEJ (alt-NHEJ)                                  15.622339   \n",
      "PERK regulates gene expression                               15.347719   \n",
      "Negative regulation of FGFR1 signaling                       15.206619   \n",
      "Abortive elongation of HIV-1 transcript in the ...           15.145529   \n",
      "...                                                                ...   \n",
      "Unfolded Protein Response (UPR)                               0.313293   \n",
      "BMAL1:CLOCK,NPAS2 activates circadian gene expr...            0.313186   \n",
      "FGFR2c ligand binding and activation                          0.313133   \n",
      "Carnitine synthesis                                           0.312442   \n",
      "Degradation of DVL                                            0.312335   \n",
      "Class I MHC mediated antigen processing & prese...            0.312017   \n",
      "FCGR activation                                               0.310585   \n",
      "Autodegradation of the E3 ubiquitin ligase COP1               0.310427   \n",
      "Prolonged ERK activation events                               0.308895   \n",
      "Alpha-defensins                                               0.308368   \n",
      "RNF mutants show enhanced WNT signaling and pro...            0.308368   \n",
      "HDL clearance                                                 0.307999   \n",
      "CD22 mediated BCR regulation                                  0.307369   \n",
      "Vpu mediated degradation of CD4                               0.306633   \n",
      "Regulation of activated PAK-2p34 by proteasome ...            0.306529   \n",
      "Entry of Influenza Virion into Host Cell via En...            0.305743   \n",
      "Oncogenic MAPK signaling                                      0.305533   \n",
      "Defective ABCA1 causes Tangier disease                        0.304749   \n",
      "Regulation of Apoptosis                                       0.304749   \n",
      "Activation of NMDA receptors and postsynaptic e...            0.304645   \n",
      "Interleukin-12 family signaling                               0.304383   \n",
      "Ubiquitin-dependent degradation of Cyclin D                   0.303810   \n",
      "Metabolism of ingested H2SeO4 and H2SeO3 into H2Se            0.302821   \n",
      "Activation of Ca-permeable Kainate Receptor                   0.301989   \n",
      "Intestinal hexose absorption                                  0.301989   \n",
      "Classical antibody-mediated complement activation             0.301989   \n",
      "Ionotropic activity of kainate receptors                      0.301989   \n",
      "DNA Damage Recognition in GG-NER                              0.301937   \n",
      "Regulation of TP53 Expression                                 0.301678   \n",
      "Antigen processing-Cross presentation                         0.301419   \n",
      "\n",
      "                                                    Cluster 9 q-values  \n",
      "Vitamin B2 (riboflavin) metabolism                           49.278113  \n",
      "Synthesis And Processing Of GAG, GAGPOL Polypro...           46.385199  \n",
      "Membrane binding and targetting of GAG proteins              46.385199  \n",
      "KSRP (KHSRP) binds and destabilizes mRNA                     38.679351  \n",
      "Assembly Of The HIV Virion                                   35.975694  \n",
      "Tristetraprolin (TTP, ZFP36) binds and destabil...           34.997709  \n",
      "Glutamate and glutamine metabolism                           34.930208  \n",
      "Synthesis of PS                                              33.111412  \n",
      "Butyrate Response Factor 1 (BRF1) binds and des...           31.952511  \n",
      "Protein repair                                               20.296774  \n",
      "Processing of Intronless Pre-mRNAs                           19.484115  \n",
      "Mitotic Telophase/Cytokinesis                                18.610237  \n",
      "RNA Polymerase III Chain Elongation                          18.115427  \n",
      "GDP-fucose biosynthesis                                      18.086962  \n",
      "Sema3A PAK dependent Axon repulsion                          17.513190  \n",
      "Telomere Extension By Telomerase                             17.327597  \n",
      "RNA Polymerase III Transcription Termination                 16.338365  \n",
      "tRNA processing                                              15.309472  \n",
      "Cooperation of Prefoldin and TriC/CCT  in actin...           15.008473  \n",
      "Type I hemidesmosome assembly                                14.959758  \n",
      "Prefoldin mediated transfer of substrate  to CC...           14.953040  \n",
      "Phosphorylation of proteins involved in G1/S tr...           14.939229  \n",
      "Cyclin A/B1/B2 associated events during G2/M tr...           14.404364  \n",
      "G0 and Early G1                                              14.118714  \n",
      "CRMPs in Sema3A signaling                                    14.055736  \n",
      "APC/C:Cdc20 mediated degradation of Cyclin B                 14.036548  \n",
      "HDR through MMEJ (alt-NHEJ)                                  13.736894  \n",
      "PERK regulates gene expression                               13.478068  \n",
      "Negative regulation of FGFR1 signaling                       13.352208  \n",
      "Abortive elongation of HIV-1 transcript in the ...           13.305842  \n",
      "...                                                                ...  \n",
      "Unfolded Protein Response (UPR)                               0.307442  \n",
      "BMAL1:CLOCK,NPAS2 activates circadian gene expr...            0.307442  \n",
      "FGFR2c ligand binding and activation                          0.307442  \n",
      "Carnitine synthesis                                           0.307068  \n",
      "Degradation of DVL                                            0.307068  \n",
      "Class I MHC mediated antigen processing & prese...            0.306962  \n",
      "FCGR activation                                               0.305795  \n",
      "Autodegradation of the E3 ubiquitin ligase COP1               0.305795  \n",
      "Prolonged ERK activation events                               0.304475  \n",
      "Alpha-defensins                                               0.304371  \n",
      "RNF mutants show enhanced WNT signaling and pro...            0.304371  \n",
      "HDL clearance                                                 0.304214  \n",
      "CD22 mediated BCR regulation                                  0.303794  \n",
      "Vpu mediated degradation of CD4                               0.303376  \n",
      "Regulation of activated PAK-2p34 by proteasome ...            0.303376  \n",
      "Entry of Influenza Virion into Host Cell via En...            0.302802  \n",
      "Oncogenic MAPK signaling                                      0.302802  \n",
      "Defective ABCA1 causes Tangier disease                        0.302545  \n",
      "Regulation of Apoptosis                                       0.302545  \n",
      "Activation of NMDA receptors and postsynaptic e...            0.302545  \n",
      "Interleukin-12 family signaling                               0.302495  \n",
      "Ubiquitin-dependent degradation of Cyclin D                   0.302131  \n",
      "Metabolism of ingested H2SeO4 and H2SeO3 into H2Se            0.301518  \n",
      "Activation of Ca-permeable Kainate Receptor                   0.301518  \n",
      "Intestinal hexose absorption                                  0.301518  \n",
      "Classical antibody-mediated complement activation             0.301518  \n",
      "Ionotropic activity of kainate receptors                      0.301518  \n",
      "DNA Damage Recognition in GG-NER                              0.301518  \n",
      "Regulation of TP53 Expression                                 0.301468  \n",
      "Antigen processing-Cross presentation                         0.301419  \n",
      "\n",
      "[2074 rows x 2 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cluster 1 p-values</th>\n",
       "      <th>Cluster 1 q-values</th>\n",
       "      <th>Cluster 10 p-values</th>\n",
       "      <th>Cluster 10 q-values</th>\n",
       "      <th>Cluster 2 p-values</th>\n",
       "      <th>Cluster 2 q-values</th>\n",
       "      <th>Cluster 3 p-values</th>\n",
       "      <th>Cluster 3 q-values</th>\n",
       "      <th>Cluster 4ER+ p-values</th>\n",
       "      <th>Cluster 4ER+ q-values</th>\n",
       "      <th>...</th>\n",
       "      <th>Cluster 5 p-values</th>\n",
       "      <th>Cluster 5 q-values</th>\n",
       "      <th>Cluster 6 p-values</th>\n",
       "      <th>Cluster 6 q-values</th>\n",
       "      <th>Cluster 7 p-values</th>\n",
       "      <th>Cluster 7 q-values</th>\n",
       "      <th>Cluster 8 p-values</th>\n",
       "      <th>Cluster 8 q-values</th>\n",
       "      <th>Cluster 9 p-values</th>\n",
       "      <th>Cluster 9 q-values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2-LTR circle formation</th>\n",
       "      <td>3.760961</td>\n",
       "      <td>3.335132</td>\n",
       "      <td>18.711431</td>\n",
       "      <td>18.385849</td>\n",
       "      <td>0.325594</td>\n",
       "      <td>0.312018</td>\n",
       "      <td>1.881159</td>\n",
       "      <td>1.696610</td>\n",
       "      <td>8.625987</td>\n",
       "      <td>8.211725</td>\n",
       "      <td>...</td>\n",
       "      <td>5.889667</td>\n",
       "      <td>5.319493</td>\n",
       "      <td>0.336657</td>\n",
       "      <td>0.323548</td>\n",
       "      <td>1.601092</td>\n",
       "      <td>1.461820</td>\n",
       "      <td>8.159628</td>\n",
       "      <td>7.821457</td>\n",
       "      <td>2.886596</td>\n",
       "      <td>2.479343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5-Phosphoribose 1-diphosphate biosynthesis</th>\n",
       "      <td>0.345861</td>\n",
       "      <td>0.333758</td>\n",
       "      <td>15.974278</td>\n",
       "      <td>15.677000</td>\n",
       "      <td>1.045455</td>\n",
       "      <td>0.770827</td>\n",
       "      <td>0.902001</td>\n",
       "      <td>0.812664</td>\n",
       "      <td>6.189245</td>\n",
       "      <td>5.861441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.685590</td>\n",
       "      <td>0.609081</td>\n",
       "      <td>1.176011</td>\n",
       "      <td>0.938421</td>\n",
       "      <td>3.107078</td>\n",
       "      <td>2.852851</td>\n",
       "      <td>2.044017</td>\n",
       "      <td>1.911900</td>\n",
       "      <td>0.935861</td>\n",
       "      <td>0.791947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A tetrasaccharide linker sequence is required for GAG synthesis</th>\n",
       "      <td>8.375755</td>\n",
       "      <td>7.582692</td>\n",
       "      <td>7.269704</td>\n",
       "      <td>7.085795</td>\n",
       "      <td>2.155733</td>\n",
       "      <td>1.521358</td>\n",
       "      <td>13.725188</td>\n",
       "      <td>12.984721</td>\n",
       "      <td>8.374111</td>\n",
       "      <td>7.970056</td>\n",
       "      <td>...</td>\n",
       "      <td>2.658408</td>\n",
       "      <td>2.334594</td>\n",
       "      <td>0.350185</td>\n",
       "      <td>0.332666</td>\n",
       "      <td>2.683999</td>\n",
       "      <td>2.454971</td>\n",
       "      <td>1.929601</td>\n",
       "      <td>1.804243</td>\n",
       "      <td>5.047063</td>\n",
       "      <td>4.435262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABC transporter disorders</th>\n",
       "      <td>7.503048</td>\n",
       "      <td>6.786122</td>\n",
       "      <td>33.002068</td>\n",
       "      <td>32.520950</td>\n",
       "      <td>0.387921</td>\n",
       "      <td>0.349866</td>\n",
       "      <td>4.239394</td>\n",
       "      <td>3.874409</td>\n",
       "      <td>1.241745</td>\n",
       "      <td>1.148691</td>\n",
       "      <td>...</td>\n",
       "      <td>2.193869</td>\n",
       "      <td>1.918848</td>\n",
       "      <td>0.344877</td>\n",
       "      <td>0.329532</td>\n",
       "      <td>6.063705</td>\n",
       "      <td>5.638434</td>\n",
       "      <td>17.625445</td>\n",
       "      <td>17.020443</td>\n",
       "      <td>0.476183</td>\n",
       "      <td>0.425300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABC transporters in lipid homeostasis</th>\n",
       "      <td>2.180941</td>\n",
       "      <td>1.890257</td>\n",
       "      <td>70.425206</td>\n",
       "      <td>69.536532</td>\n",
       "      <td>4.768345</td>\n",
       "      <td>3.243928</td>\n",
       "      <td>9.805411</td>\n",
       "      <td>9.191894</td>\n",
       "      <td>3.275117</td>\n",
       "      <td>3.069571</td>\n",
       "      <td>...</td>\n",
       "      <td>24.876615</td>\n",
       "      <td>23.161867</td>\n",
       "      <td>4.569565</td>\n",
       "      <td>3.576682</td>\n",
       "      <td>17.352672</td>\n",
       "      <td>16.035863</td>\n",
       "      <td>36.164804</td>\n",
       "      <td>35.018257</td>\n",
       "      <td>1.424468</td>\n",
       "      <td>1.211121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABC-family proteins mediated transport</th>\n",
       "      <td>7.218621</td>\n",
       "      <td>6.531222</td>\n",
       "      <td>35.018455</td>\n",
       "      <td>34.507147</td>\n",
       "      <td>0.340716</td>\n",
       "      <td>0.322760</td>\n",
       "      <td>4.383301</td>\n",
       "      <td>4.004512</td>\n",
       "      <td>1.664934</td>\n",
       "      <td>1.544301</td>\n",
       "      <td>...</td>\n",
       "      <td>2.240816</td>\n",
       "      <td>1.962628</td>\n",
       "      <td>0.375231</td>\n",
       "      <td>0.348679</td>\n",
       "      <td>6.529664</td>\n",
       "      <td>6.075583</td>\n",
       "      <td>15.774023</td>\n",
       "      <td>15.239687</td>\n",
       "      <td>0.453226</td>\n",
       "      <td>0.408027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABO blood group biosynthesis</th>\n",
       "      <td>0.821769</td>\n",
       "      <td>0.701136</td>\n",
       "      <td>2.607131</td>\n",
       "      <td>2.508858</td>\n",
       "      <td>1.903703</td>\n",
       "      <td>1.358481</td>\n",
       "      <td>0.393219</td>\n",
       "      <td>0.376135</td>\n",
       "      <td>5.278316</td>\n",
       "      <td>4.988857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.720298</td>\n",
       "      <td>0.638765</td>\n",
       "      <td>4.837029</td>\n",
       "      <td>3.773074</td>\n",
       "      <td>2.530621</td>\n",
       "      <td>2.310722</td>\n",
       "      <td>3.941170</td>\n",
       "      <td>3.726109</td>\n",
       "      <td>0.410581</td>\n",
       "      <td>0.375487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADP signalling through P2Y purinoceptor 1</th>\n",
       "      <td>1.665815</td>\n",
       "      <td>1.423457</td>\n",
       "      <td>6.976290</td>\n",
       "      <td>6.796835</td>\n",
       "      <td>0.916661</td>\n",
       "      <td>0.681919</td>\n",
       "      <td>13.591149</td>\n",
       "      <td>12.852980</td>\n",
       "      <td>11.845372</td>\n",
       "      <td>11.291991</td>\n",
       "      <td>...</td>\n",
       "      <td>1.295572</td>\n",
       "      <td>1.127437</td>\n",
       "      <td>4.455846</td>\n",
       "      <td>3.483429</td>\n",
       "      <td>0.618850</td>\n",
       "      <td>0.566623</td>\n",
       "      <td>1.117781</td>\n",
       "      <td>1.040433</td>\n",
       "      <td>5.692188</td>\n",
       "      <td>5.035296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADP signalling through P2Y purinoceptor 12</th>\n",
       "      <td>3.115592</td>\n",
       "      <td>2.737803</td>\n",
       "      <td>4.658852</td>\n",
       "      <td>4.515230</td>\n",
       "      <td>1.022234</td>\n",
       "      <td>0.755483</td>\n",
       "      <td>15.881230</td>\n",
       "      <td>15.061351</td>\n",
       "      <td>16.676885</td>\n",
       "      <td>15.928278</td>\n",
       "      <td>...</td>\n",
       "      <td>1.954865</td>\n",
       "      <td>1.707354</td>\n",
       "      <td>5.777599</td>\n",
       "      <td>4.481980</td>\n",
       "      <td>0.919203</td>\n",
       "      <td>0.832588</td>\n",
       "      <td>2.683503</td>\n",
       "      <td>2.523849</td>\n",
       "      <td>6.344734</td>\n",
       "      <td>5.614512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT phosphorylates targets in the cytosol</th>\n",
       "      <td>3.216786</td>\n",
       "      <td>2.832958</td>\n",
       "      <td>11.108934</td>\n",
       "      <td>10.879551</td>\n",
       "      <td>1.160667</td>\n",
       "      <td>0.850752</td>\n",
       "      <td>1.609689</td>\n",
       "      <td>1.449732</td>\n",
       "      <td>3.927190</td>\n",
       "      <td>3.694600</td>\n",
       "      <td>...</td>\n",
       "      <td>3.167744</td>\n",
       "      <td>2.790954</td>\n",
       "      <td>0.824871</td>\n",
       "      <td>0.669130</td>\n",
       "      <td>2.161445</td>\n",
       "      <td>1.969467</td>\n",
       "      <td>4.162660</td>\n",
       "      <td>3.941370</td>\n",
       "      <td>0.924645</td>\n",
       "      <td>0.783348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT phosphorylates targets in the nucleus</th>\n",
       "      <td>7.864724</td>\n",
       "      <td>7.114942</td>\n",
       "      <td>11.723281</td>\n",
       "      <td>11.484929</td>\n",
       "      <td>4.513568</td>\n",
       "      <td>3.061005</td>\n",
       "      <td>1.192219</td>\n",
       "      <td>1.071587</td>\n",
       "      <td>7.273187</td>\n",
       "      <td>6.907230</td>\n",
       "      <td>...</td>\n",
       "      <td>1.462585</td>\n",
       "      <td>1.269627</td>\n",
       "      <td>0.534124</td>\n",
       "      <td>0.446741</td>\n",
       "      <td>2.001840</td>\n",
       "      <td>1.817931</td>\n",
       "      <td>2.765016</td>\n",
       "      <td>2.602022</td>\n",
       "      <td>1.414306</td>\n",
       "      <td>1.203407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT-mediated inactivation of FOXO1A</th>\n",
       "      <td>5.360685</td>\n",
       "      <td>4.822751</td>\n",
       "      <td>13.729136</td>\n",
       "      <td>13.461933</td>\n",
       "      <td>3.602443</td>\n",
       "      <td>2.443997</td>\n",
       "      <td>0.590346</td>\n",
       "      <td>0.543050</td>\n",
       "      <td>4.670407</td>\n",
       "      <td>4.404751</td>\n",
       "      <td>...</td>\n",
       "      <td>1.540024</td>\n",
       "      <td>1.338493</td>\n",
       "      <td>0.421307</td>\n",
       "      <td>0.371670</td>\n",
       "      <td>1.669095</td>\n",
       "      <td>1.522548</td>\n",
       "      <td>3.719748</td>\n",
       "      <td>3.514202</td>\n",
       "      <td>0.793230</td>\n",
       "      <td>0.675628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ALKBH3 mediated reversal of alkylation damage</th>\n",
       "      <td>0.548935</td>\n",
       "      <td>0.478625</td>\n",
       "      <td>2.348202</td>\n",
       "      <td>2.254369</td>\n",
       "      <td>1.581827</td>\n",
       "      <td>1.139500</td>\n",
       "      <td>5.078548</td>\n",
       "      <td>4.655501</td>\n",
       "      <td>3.608131</td>\n",
       "      <td>3.386840</td>\n",
       "      <td>...</td>\n",
       "      <td>9.320050</td>\n",
       "      <td>8.521755</td>\n",
       "      <td>0.978490</td>\n",
       "      <td>0.787487</td>\n",
       "      <td>2.515340</td>\n",
       "      <td>2.296135</td>\n",
       "      <td>0.499449</td>\n",
       "      <td>0.476445</td>\n",
       "      <td>3.176725</td>\n",
       "      <td>2.745843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMER1 mutants destabilize the destruction complex</th>\n",
       "      <td>0.936291</td>\n",
       "      <td>0.794994</td>\n",
       "      <td>28.217593</td>\n",
       "      <td>27.796759</td>\n",
       "      <td>2.172428</td>\n",
       "      <td>1.528800</td>\n",
       "      <td>15.850162</td>\n",
       "      <td>15.038503</td>\n",
       "      <td>14.395657</td>\n",
       "      <td>13.742549</td>\n",
       "      <td>...</td>\n",
       "      <td>11.145170</td>\n",
       "      <td>10.280147</td>\n",
       "      <td>0.314112</td>\n",
       "      <td>0.308352</td>\n",
       "      <td>0.762996</td>\n",
       "      <td>0.694406</td>\n",
       "      <td>1.707118</td>\n",
       "      <td>1.596865</td>\n",
       "      <td>6.056340</td>\n",
       "      <td>5.363813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMPK inhibits chREBP transcriptional activation activity</th>\n",
       "      <td>8.371428</td>\n",
       "      <td>7.580172</td>\n",
       "      <td>8.648882</td>\n",
       "      <td>8.447351</td>\n",
       "      <td>2.175979</td>\n",
       "      <td>1.528800</td>\n",
       "      <td>3.232553</td>\n",
       "      <td>2.940640</td>\n",
       "      <td>5.671590</td>\n",
       "      <td>5.359532</td>\n",
       "      <td>...</td>\n",
       "      <td>6.082065</td>\n",
       "      <td>5.498453</td>\n",
       "      <td>4.526513</td>\n",
       "      <td>3.542143</td>\n",
       "      <td>7.904280</td>\n",
       "      <td>7.349399</td>\n",
       "      <td>6.659763</td>\n",
       "      <td>6.354947</td>\n",
       "      <td>7.099057</td>\n",
       "      <td>6.284676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC truncation mutants have impaired AXIN binding</th>\n",
       "      <td>0.936291</td>\n",
       "      <td>0.794994</td>\n",
       "      <td>28.217593</td>\n",
       "      <td>27.796759</td>\n",
       "      <td>2.172428</td>\n",
       "      <td>1.528800</td>\n",
       "      <td>15.850162</td>\n",
       "      <td>15.038503</td>\n",
       "      <td>14.395657</td>\n",
       "      <td>13.742549</td>\n",
       "      <td>...</td>\n",
       "      <td>11.145964</td>\n",
       "      <td>10.280147</td>\n",
       "      <td>0.314043</td>\n",
       "      <td>0.308352</td>\n",
       "      <td>0.762996</td>\n",
       "      <td>0.694406</td>\n",
       "      <td>1.707233</td>\n",
       "      <td>1.596865</td>\n",
       "      <td>6.056340</td>\n",
       "      <td>5.363813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC-Cdc20 mediated degradation of Nek2A</th>\n",
       "      <td>21.945291</td>\n",
       "      <td>20.473581</td>\n",
       "      <td>67.124764</td>\n",
       "      <td>66.274823</td>\n",
       "      <td>2.009577</td>\n",
       "      <td>1.429961</td>\n",
       "      <td>31.310749</td>\n",
       "      <td>30.019246</td>\n",
       "      <td>39.735972</td>\n",
       "      <td>38.264261</td>\n",
       "      <td>...</td>\n",
       "      <td>11.843548</td>\n",
       "      <td>10.924680</td>\n",
       "      <td>3.876048</td>\n",
       "      <td>3.015606</td>\n",
       "      <td>6.165634</td>\n",
       "      <td>5.735880</td>\n",
       "      <td>8.950053</td>\n",
       "      <td>8.591809</td>\n",
       "      <td>14.380475</td>\n",
       "      <td>12.665727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC/C-mediated degradation of cell cycle proteins</th>\n",
       "      <td>13.011680</td>\n",
       "      <td>11.885203</td>\n",
       "      <td>71.914874</td>\n",
       "      <td>71.011365</td>\n",
       "      <td>0.665471</td>\n",
       "      <td>0.518962</td>\n",
       "      <td>17.306354</td>\n",
       "      <td>16.428878</td>\n",
       "      <td>12.148558</td>\n",
       "      <td>11.583798</td>\n",
       "      <td>...</td>\n",
       "      <td>9.833084</td>\n",
       "      <td>9.015962</td>\n",
       "      <td>0.918796</td>\n",
       "      <td>0.740606</td>\n",
       "      <td>9.757787</td>\n",
       "      <td>9.070388</td>\n",
       "      <td>22.318410</td>\n",
       "      <td>21.556696</td>\n",
       "      <td>3.388987</td>\n",
       "      <td>2.943751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC/C:Cdc20 mediated degradation of Cyclin B</th>\n",
       "      <td>22.040693</td>\n",
       "      <td>20.562734</td>\n",
       "      <td>69.952725</td>\n",
       "      <td>69.072079</td>\n",
       "      <td>2.354643</td>\n",
       "      <td>1.624421</td>\n",
       "      <td>34.554555</td>\n",
       "      <td>33.082844</td>\n",
       "      <td>41.539732</td>\n",
       "      <td>39.930493</td>\n",
       "      <td>...</td>\n",
       "      <td>14.911636</td>\n",
       "      <td>13.842801</td>\n",
       "      <td>3.664181</td>\n",
       "      <td>2.852522</td>\n",
       "      <td>4.706908</td>\n",
       "      <td>4.363689</td>\n",
       "      <td>14.351334</td>\n",
       "      <td>13.857347</td>\n",
       "      <td>15.938383</td>\n",
       "      <td>14.036548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC/C:Cdc20 mediated degradation of Securin</th>\n",
       "      <td>8.393073</td>\n",
       "      <td>7.597402</td>\n",
       "      <td>35.796564</td>\n",
       "      <td>35.273546</td>\n",
       "      <td>0.307257</td>\n",
       "      <td>0.303894</td>\n",
       "      <td>5.574839</td>\n",
       "      <td>5.114759</td>\n",
       "      <td>2.013528</td>\n",
       "      <td>1.871651</td>\n",
       "      <td>...</td>\n",
       "      <td>3.177552</td>\n",
       "      <td>2.798763</td>\n",
       "      <td>0.406612</td>\n",
       "      <td>0.361877</td>\n",
       "      <td>6.043087</td>\n",
       "      <td>5.620040</td>\n",
       "      <td>18.291677</td>\n",
       "      <td>17.667715</td>\n",
       "      <td>0.666720</td>\n",
       "      <td>0.577283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC/C:Cdc20 mediated degradation of mitotic proteins</th>\n",
       "      <td>10.889119</td>\n",
       "      <td>9.920615</td>\n",
       "      <td>48.948295</td>\n",
       "      <td>48.275925</td>\n",
       "      <td>0.447278</td>\n",
       "      <td>0.384586</td>\n",
       "      <td>9.475519</td>\n",
       "      <td>8.871056</td>\n",
       "      <td>5.867643</td>\n",
       "      <td>5.549094</td>\n",
       "      <td>...</td>\n",
       "      <td>5.665495</td>\n",
       "      <td>5.114355</td>\n",
       "      <td>0.641753</td>\n",
       "      <td>0.527160</td>\n",
       "      <td>7.517541</td>\n",
       "      <td>6.988900</td>\n",
       "      <td>19.137797</td>\n",
       "      <td>18.480904</td>\n",
       "      <td>1.550581</td>\n",
       "      <td>1.322319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1</th>\n",
       "      <td>9.718463</td>\n",
       "      <td>8.808194</td>\n",
       "      <td>49.381426</td>\n",
       "      <td>48.704104</td>\n",
       "      <td>0.347536</td>\n",
       "      <td>0.327133</td>\n",
       "      <td>9.298696</td>\n",
       "      <td>8.702046</td>\n",
       "      <td>4.286565</td>\n",
       "      <td>4.037942</td>\n",
       "      <td>...</td>\n",
       "      <td>5.184148</td>\n",
       "      <td>4.667369</td>\n",
       "      <td>0.540742</td>\n",
       "      <td>0.451820</td>\n",
       "      <td>7.374682</td>\n",
       "      <td>6.859326</td>\n",
       "      <td>20.751335</td>\n",
       "      <td>20.051527</td>\n",
       "      <td>1.331734</td>\n",
       "      <td>1.131201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC:Cdc20 mediated degradation of cell cycle proteins prior to satisfation of the cell cycle checkpoint</th>\n",
       "      <td>9.890892</td>\n",
       "      <td>8.975483</td>\n",
       "      <td>44.195642</td>\n",
       "      <td>43.570799</td>\n",
       "      <td>0.354125</td>\n",
       "      <td>0.329369</td>\n",
       "      <td>7.674052</td>\n",
       "      <td>7.131029</td>\n",
       "      <td>4.115386</td>\n",
       "      <td>3.872662</td>\n",
       "      <td>...</td>\n",
       "      <td>4.509152</td>\n",
       "      <td>4.024213</td>\n",
       "      <td>0.526877</td>\n",
       "      <td>0.443318</td>\n",
       "      <td>7.066893</td>\n",
       "      <td>6.576807</td>\n",
       "      <td>18.317949</td>\n",
       "      <td>17.693106</td>\n",
       "      <td>1.081649</td>\n",
       "      <td>0.917740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APEX1-Independent Resolution of AP Sites via the Single Nucleotide Replacement Pathway</th>\n",
       "      <td>0.842751</td>\n",
       "      <td>0.719067</td>\n",
       "      <td>23.678476</td>\n",
       "      <td>23.307628</td>\n",
       "      <td>1.344850</td>\n",
       "      <td>0.972030</td>\n",
       "      <td>4.229399</td>\n",
       "      <td>3.865867</td>\n",
       "      <td>6.499152</td>\n",
       "      <td>6.158234</td>\n",
       "      <td>...</td>\n",
       "      <td>9.648178</td>\n",
       "      <td>8.839225</td>\n",
       "      <td>1.377732</td>\n",
       "      <td>1.101675</td>\n",
       "      <td>4.271260</td>\n",
       "      <td>3.949208</td>\n",
       "      <td>45.525588</td>\n",
       "      <td>43.986930</td>\n",
       "      <td>0.483647</td>\n",
       "      <td>0.430947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APOBEC3G mediated resistance to HIV-1 infection</th>\n",
       "      <td>6.390606</td>\n",
       "      <td>5.781368</td>\n",
       "      <td>53.944371</td>\n",
       "      <td>53.228535</td>\n",
       "      <td>0.622397</td>\n",
       "      <td>0.492591</td>\n",
       "      <td>14.143824</td>\n",
       "      <td>13.384523</td>\n",
       "      <td>7.291126</td>\n",
       "      <td>6.924194</td>\n",
       "      <td>...</td>\n",
       "      <td>13.041273</td>\n",
       "      <td>12.066887</td>\n",
       "      <td>2.395101</td>\n",
       "      <td>1.906705</td>\n",
       "      <td>4.625958</td>\n",
       "      <td>4.286873</td>\n",
       "      <td>33.272007</td>\n",
       "      <td>32.231660</td>\n",
       "      <td>10.549802</td>\n",
       "      <td>9.337594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARL13B-mediated ciliary trafficking of INPP5E</th>\n",
       "      <td>0.789112</td>\n",
       "      <td>0.673154</td>\n",
       "      <td>1.546303</td>\n",
       "      <td>1.476977</td>\n",
       "      <td>0.868220</td>\n",
       "      <td>0.646232</td>\n",
       "      <td>0.371532</td>\n",
       "      <td>0.357487</td>\n",
       "      <td>0.501319</td>\n",
       "      <td>0.477209</td>\n",
       "      <td>...</td>\n",
       "      <td>1.370428</td>\n",
       "      <td>1.192553</td>\n",
       "      <td>0.345023</td>\n",
       "      <td>0.329532</td>\n",
       "      <td>1.529787</td>\n",
       "      <td>1.395108</td>\n",
       "      <td>0.981842</td>\n",
       "      <td>0.916185</td>\n",
       "      <td>1.741873</td>\n",
       "      <td>1.482350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARMS-mediated activation</th>\n",
       "      <td>6.105456</td>\n",
       "      <td>5.523447</td>\n",
       "      <td>4.898852</td>\n",
       "      <td>4.751129</td>\n",
       "      <td>1.107040</td>\n",
       "      <td>0.810592</td>\n",
       "      <td>0.472567</td>\n",
       "      <td>0.442439</td>\n",
       "      <td>4.448748</td>\n",
       "      <td>4.193769</td>\n",
       "      <td>...</td>\n",
       "      <td>3.099961</td>\n",
       "      <td>2.730096</td>\n",
       "      <td>2.280779</td>\n",
       "      <td>1.814003</td>\n",
       "      <td>0.451066</td>\n",
       "      <td>0.422283</td>\n",
       "      <td>1.512565</td>\n",
       "      <td>1.410070</td>\n",
       "      <td>5.592169</td>\n",
       "      <td>4.948381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF4 activates genes in response to endoplasmic reticulum  stress</th>\n",
       "      <td>2.928160</td>\n",
       "      <td>2.561229</td>\n",
       "      <td>10.093901</td>\n",
       "      <td>9.876082</td>\n",
       "      <td>0.408322</td>\n",
       "      <td>0.363587</td>\n",
       "      <td>19.651999</td>\n",
       "      <td>18.687373</td>\n",
       "      <td>26.842697</td>\n",
       "      <td>25.763934</td>\n",
       "      <td>...</td>\n",
       "      <td>0.442169</td>\n",
       "      <td>0.413162</td>\n",
       "      <td>4.946292</td>\n",
       "      <td>3.863894</td>\n",
       "      <td>5.928387</td>\n",
       "      <td>5.511399</td>\n",
       "      <td>3.715069</td>\n",
       "      <td>3.509858</td>\n",
       "      <td>13.446183</td>\n",
       "      <td>11.921766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF6 (ATF6-alpha) activates chaperone genes</th>\n",
       "      <td>0.445184</td>\n",
       "      <td>0.401839</td>\n",
       "      <td>4.522333</td>\n",
       "      <td>4.382194</td>\n",
       "      <td>1.332106</td>\n",
       "      <td>0.964606</td>\n",
       "      <td>1.750143</td>\n",
       "      <td>1.576349</td>\n",
       "      <td>22.126203</td>\n",
       "      <td>21.185971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.680348</td>\n",
       "      <td>0.604837</td>\n",
       "      <td>0.333206</td>\n",
       "      <td>0.320888</td>\n",
       "      <td>1.203328</td>\n",
       "      <td>1.092805</td>\n",
       "      <td>39.566115</td>\n",
       "      <td>38.310005</td>\n",
       "      <td>0.486008</td>\n",
       "      <td>0.432599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF6 (ATF6-alpha) activates chaperones</th>\n",
       "      <td>0.401413</td>\n",
       "      <td>0.371061</td>\n",
       "      <td>2.989754</td>\n",
       "      <td>2.883799</td>\n",
       "      <td>1.203695</td>\n",
       "      <td>0.880323</td>\n",
       "      <td>1.614979</td>\n",
       "      <td>1.454720</td>\n",
       "      <td>23.422288</td>\n",
       "      <td>22.437918</td>\n",
       "      <td>...</td>\n",
       "      <td>0.350750</td>\n",
       "      <td>0.338216</td>\n",
       "      <td>0.597005</td>\n",
       "      <td>0.493980</td>\n",
       "      <td>0.961810</td>\n",
       "      <td>0.870052</td>\n",
       "      <td>35.982597</td>\n",
       "      <td>34.844766</td>\n",
       "      <td>0.738903</td>\n",
       "      <td>0.631609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mitochondrial fatty acid beta-oxidation of saturated fatty acids</th>\n",
       "      <td>0.744543</td>\n",
       "      <td>0.637398</td>\n",
       "      <td>5.198804</td>\n",
       "      <td>5.047240</td>\n",
       "      <td>0.349295</td>\n",
       "      <td>0.327793</td>\n",
       "      <td>2.565066</td>\n",
       "      <td>2.315700</td>\n",
       "      <td>0.886980</td>\n",
       "      <td>0.823266</td>\n",
       "      <td>...</td>\n",
       "      <td>7.448810</td>\n",
       "      <td>6.769171</td>\n",
       "      <td>0.441623</td>\n",
       "      <td>0.384169</td>\n",
       "      <td>0.345973</td>\n",
       "      <td>0.337088</td>\n",
       "      <td>2.048751</td>\n",
       "      <td>1.916066</td>\n",
       "      <td>1.577808</td>\n",
       "      <td>1.344860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mitochondrial fatty acid beta-oxidation of unsaturated fatty acids</th>\n",
       "      <td>1.632177</td>\n",
       "      <td>1.392737</td>\n",
       "      <td>2.156582</td>\n",
       "      <td>2.067146</td>\n",
       "      <td>0.459124</td>\n",
       "      <td>0.391321</td>\n",
       "      <td>3.501793</td>\n",
       "      <td>3.188013</td>\n",
       "      <td>0.857368</td>\n",
       "      <td>0.796073</td>\n",
       "      <td>...</td>\n",
       "      <td>4.155106</td>\n",
       "      <td>3.702215</td>\n",
       "      <td>1.962413</td>\n",
       "      <td>1.557827</td>\n",
       "      <td>0.657697</td>\n",
       "      <td>0.600244</td>\n",
       "      <td>0.343126</td>\n",
       "      <td>0.338918</td>\n",
       "      <td>1.287907</td>\n",
       "      <td>1.091672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p130Cas linkage to MAPK signaling for integrins</th>\n",
       "      <td>2.588006</td>\n",
       "      <td>2.257969</td>\n",
       "      <td>1.037744</td>\n",
       "      <td>0.986460</td>\n",
       "      <td>0.958216</td>\n",
       "      <td>0.708479</td>\n",
       "      <td>2.298793</td>\n",
       "      <td>2.073651</td>\n",
       "      <td>5.586869</td>\n",
       "      <td>5.278661</td>\n",
       "      <td>...</td>\n",
       "      <td>27.113438</td>\n",
       "      <td>25.211603</td>\n",
       "      <td>2.824638</td>\n",
       "      <td>2.222997</td>\n",
       "      <td>4.526958</td>\n",
       "      <td>4.193738</td>\n",
       "      <td>14.850274</td>\n",
       "      <td>14.342351</td>\n",
       "      <td>0.416824</td>\n",
       "      <td>0.380236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p38MAPK events</th>\n",
       "      <td>0.459301</td>\n",
       "      <td>0.411772</td>\n",
       "      <td>11.137599</td>\n",
       "      <td>10.907862</td>\n",
       "      <td>0.898098</td>\n",
       "      <td>0.667649</td>\n",
       "      <td>0.884350</td>\n",
       "      <td>0.797735</td>\n",
       "      <td>3.517022</td>\n",
       "      <td>3.301961</td>\n",
       "      <td>...</td>\n",
       "      <td>2.522183</td>\n",
       "      <td>2.211412</td>\n",
       "      <td>1.760402</td>\n",
       "      <td>1.393751</td>\n",
       "      <td>7.862333</td>\n",
       "      <td>7.309700</td>\n",
       "      <td>7.388025</td>\n",
       "      <td>7.068603</td>\n",
       "      <td>2.961988</td>\n",
       "      <td>2.546638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p53-Dependent G1 DNA Damage Response</th>\n",
       "      <td>7.153408</td>\n",
       "      <td>6.473087</td>\n",
       "      <td>44.975996</td>\n",
       "      <td>44.343134</td>\n",
       "      <td>0.357696</td>\n",
       "      <td>0.330087</td>\n",
       "      <td>5.965132</td>\n",
       "      <td>5.490308</td>\n",
       "      <td>2.126589</td>\n",
       "      <td>1.978278</td>\n",
       "      <td>...</td>\n",
       "      <td>3.150571</td>\n",
       "      <td>2.776266</td>\n",
       "      <td>0.311505</td>\n",
       "      <td>0.306874</td>\n",
       "      <td>6.576002</td>\n",
       "      <td>6.117731</td>\n",
       "      <td>19.476739</td>\n",
       "      <td>18.814107</td>\n",
       "      <td>0.547632</td>\n",
       "      <td>0.481974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p53-Dependent G1/S DNA damage checkpoint</th>\n",
       "      <td>7.153408</td>\n",
       "      <td>6.473087</td>\n",
       "      <td>44.975996</td>\n",
       "      <td>44.343134</td>\n",
       "      <td>0.357696</td>\n",
       "      <td>0.330087</td>\n",
       "      <td>5.965132</td>\n",
       "      <td>5.490308</td>\n",
       "      <td>2.126589</td>\n",
       "      <td>1.978278</td>\n",
       "      <td>...</td>\n",
       "      <td>3.150571</td>\n",
       "      <td>2.776266</td>\n",
       "      <td>0.311505</td>\n",
       "      <td>0.306874</td>\n",
       "      <td>6.576002</td>\n",
       "      <td>6.117731</td>\n",
       "      <td>19.476739</td>\n",
       "      <td>18.814107</td>\n",
       "      <td>0.547632</td>\n",
       "      <td>0.481974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p53-Independent DNA Damage Response</th>\n",
       "      <td>6.665075</td>\n",
       "      <td>6.031313</td>\n",
       "      <td>34.369896</td>\n",
       "      <td>33.868665</td>\n",
       "      <td>0.470981</td>\n",
       "      <td>0.397609</td>\n",
       "      <td>4.257262</td>\n",
       "      <td>3.889355</td>\n",
       "      <td>1.279853</td>\n",
       "      <td>1.184980</td>\n",
       "      <td>...</td>\n",
       "      <td>2.499587</td>\n",
       "      <td>2.192230</td>\n",
       "      <td>0.383212</td>\n",
       "      <td>0.351960</td>\n",
       "      <td>5.742845</td>\n",
       "      <td>5.338259</td>\n",
       "      <td>17.009579</td>\n",
       "      <td>16.431551</td>\n",
       "      <td>0.399163</td>\n",
       "      <td>0.369036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p53-Independent G1/S DNA damage checkpoint</th>\n",
       "      <td>6.665075</td>\n",
       "      <td>6.031313</td>\n",
       "      <td>34.369896</td>\n",
       "      <td>33.868665</td>\n",
       "      <td>0.470981</td>\n",
       "      <td>0.397609</td>\n",
       "      <td>4.257262</td>\n",
       "      <td>3.889355</td>\n",
       "      <td>1.279853</td>\n",
       "      <td>1.184980</td>\n",
       "      <td>...</td>\n",
       "      <td>2.499587</td>\n",
       "      <td>2.192230</td>\n",
       "      <td>0.383212</td>\n",
       "      <td>0.351960</td>\n",
       "      <td>5.742845</td>\n",
       "      <td>5.338259</td>\n",
       "      <td>17.009579</td>\n",
       "      <td>16.431551</td>\n",
       "      <td>0.399163</td>\n",
       "      <td>0.369036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p75 NTR receptor-mediated signalling</th>\n",
       "      <td>9.760672</td>\n",
       "      <td>8.848697</td>\n",
       "      <td>48.027490</td>\n",
       "      <td>47.362927</td>\n",
       "      <td>0.452507</td>\n",
       "      <td>0.386849</td>\n",
       "      <td>10.986857</td>\n",
       "      <td>10.339365</td>\n",
       "      <td>8.984090</td>\n",
       "      <td>8.557143</td>\n",
       "      <td>...</td>\n",
       "      <td>12.335284</td>\n",
       "      <td>11.391387</td>\n",
       "      <td>0.997443</td>\n",
       "      <td>0.802703</td>\n",
       "      <td>7.191423</td>\n",
       "      <td>6.690856</td>\n",
       "      <td>22.014209</td>\n",
       "      <td>21.263248</td>\n",
       "      <td>2.036557</td>\n",
       "      <td>1.735527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p75NTR negatively regulates cell cycle via SC1</th>\n",
       "      <td>3.834756</td>\n",
       "      <td>3.401919</td>\n",
       "      <td>6.212856</td>\n",
       "      <td>6.045882</td>\n",
       "      <td>0.354691</td>\n",
       "      <td>0.329614</td>\n",
       "      <td>2.044479</td>\n",
       "      <td>1.839604</td>\n",
       "      <td>1.773138</td>\n",
       "      <td>1.643850</td>\n",
       "      <td>...</td>\n",
       "      <td>6.993605</td>\n",
       "      <td>6.340497</td>\n",
       "      <td>2.007499</td>\n",
       "      <td>1.597563</td>\n",
       "      <td>6.314398</td>\n",
       "      <td>5.876111</td>\n",
       "      <td>2.089701</td>\n",
       "      <td>1.955021</td>\n",
       "      <td>6.926040</td>\n",
       "      <td>6.138148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p75NTR recruits signalling complexes</th>\n",
       "      <td>2.321031</td>\n",
       "      <td>2.016638</td>\n",
       "      <td>85.793526</td>\n",
       "      <td>84.669842</td>\n",
       "      <td>1.934117</td>\n",
       "      <td>1.382231</td>\n",
       "      <td>13.825430</td>\n",
       "      <td>13.079164</td>\n",
       "      <td>4.369223</td>\n",
       "      <td>4.117620</td>\n",
       "      <td>...</td>\n",
       "      <td>5.524764</td>\n",
       "      <td>4.981741</td>\n",
       "      <td>1.637828</td>\n",
       "      <td>1.299199</td>\n",
       "      <td>6.094992</td>\n",
       "      <td>5.667485</td>\n",
       "      <td>22.692936</td>\n",
       "      <td>21.915204</td>\n",
       "      <td>4.456849</td>\n",
       "      <td>3.901216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p75NTR regulates axonogenesis</th>\n",
       "      <td>1.250670</td>\n",
       "      <td>1.062906</td>\n",
       "      <td>23.282628</td>\n",
       "      <td>22.914721</td>\n",
       "      <td>1.274652</td>\n",
       "      <td>0.926424</td>\n",
       "      <td>11.677282</td>\n",
       "      <td>11.002938</td>\n",
       "      <td>1.670706</td>\n",
       "      <td>1.549243</td>\n",
       "      <td>...</td>\n",
       "      <td>0.517298</td>\n",
       "      <td>0.473491</td>\n",
       "      <td>2.930041</td>\n",
       "      <td>2.295100</td>\n",
       "      <td>2.198895</td>\n",
       "      <td>2.003646</td>\n",
       "      <td>1.853008</td>\n",
       "      <td>1.731545</td>\n",
       "      <td>3.167927</td>\n",
       "      <td>2.738173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p75NTR signals via NF-kB</th>\n",
       "      <td>4.456845</td>\n",
       "      <td>3.983892</td>\n",
       "      <td>88.998536</td>\n",
       "      <td>87.812061</td>\n",
       "      <td>1.797095</td>\n",
       "      <td>1.280323</td>\n",
       "      <td>17.083295</td>\n",
       "      <td>16.215193</td>\n",
       "      <td>5.772325</td>\n",
       "      <td>5.458114</td>\n",
       "      <td>...</td>\n",
       "      <td>5.680145</td>\n",
       "      <td>5.127512</td>\n",
       "      <td>1.442989</td>\n",
       "      <td>1.150666</td>\n",
       "      <td>6.979677</td>\n",
       "      <td>6.492815</td>\n",
       "      <td>24.891425</td>\n",
       "      <td>24.060338</td>\n",
       "      <td>6.025390</td>\n",
       "      <td>5.334922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phospho-PLA2 pathway</th>\n",
       "      <td>1.476583</td>\n",
       "      <td>1.256684</td>\n",
       "      <td>2.245882</td>\n",
       "      <td>2.153865</td>\n",
       "      <td>0.312051</td>\n",
       "      <td>0.305510</td>\n",
       "      <td>1.158092</td>\n",
       "      <td>1.041587</td>\n",
       "      <td>0.774850</td>\n",
       "      <td>0.722386</td>\n",
       "      <td>...</td>\n",
       "      <td>1.930719</td>\n",
       "      <td>1.687544</td>\n",
       "      <td>1.014991</td>\n",
       "      <td>0.817438</td>\n",
       "      <td>0.713007</td>\n",
       "      <td>0.649294</td>\n",
       "      <td>0.459800</td>\n",
       "      <td>0.440971</td>\n",
       "      <td>1.653140</td>\n",
       "      <td>1.407846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phosphorylation site mutants of CTNNB1 are not targeted to the proteasome by the destruction complex</th>\n",
       "      <td>1.443835</td>\n",
       "      <td>1.232440</td>\n",
       "      <td>12.657323</td>\n",
       "      <td>12.403472</td>\n",
       "      <td>1.720946</td>\n",
       "      <td>1.231489</td>\n",
       "      <td>12.017237</td>\n",
       "      <td>11.334906</td>\n",
       "      <td>12.437981</td>\n",
       "      <td>11.858976</td>\n",
       "      <td>...</td>\n",
       "      <td>9.043403</td>\n",
       "      <td>8.261889</td>\n",
       "      <td>0.515455</td>\n",
       "      <td>0.436943</td>\n",
       "      <td>0.416272</td>\n",
       "      <td>0.395880</td>\n",
       "      <td>0.545972</td>\n",
       "      <td>0.518360</td>\n",
       "      <td>5.771768</td>\n",
       "      <td>5.109136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA modification in the mitochondrion</th>\n",
       "      <td>2.371133</td>\n",
       "      <td>2.062072</td>\n",
       "      <td>0.675652</td>\n",
       "      <td>0.646197</td>\n",
       "      <td>1.109413</td>\n",
       "      <td>0.811796</td>\n",
       "      <td>4.769068</td>\n",
       "      <td>4.371338</td>\n",
       "      <td>11.603468</td>\n",
       "      <td>11.058981</td>\n",
       "      <td>...</td>\n",
       "      <td>2.081064</td>\n",
       "      <td>1.818485</td>\n",
       "      <td>1.396577</td>\n",
       "      <td>1.116217</td>\n",
       "      <td>4.459597</td>\n",
       "      <td>4.130008</td>\n",
       "      <td>11.043109</td>\n",
       "      <td>10.628847</td>\n",
       "      <td>5.907865</td>\n",
       "      <td>5.228546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA modification in the nucleus and cytosol</th>\n",
       "      <td>2.436913</td>\n",
       "      <td>2.121838</td>\n",
       "      <td>9.245907</td>\n",
       "      <td>9.037663</td>\n",
       "      <td>0.521597</td>\n",
       "      <td>0.429839</td>\n",
       "      <td>12.294303</td>\n",
       "      <td>11.595543</td>\n",
       "      <td>11.131025</td>\n",
       "      <td>10.605907</td>\n",
       "      <td>...</td>\n",
       "      <td>8.027170</td>\n",
       "      <td>7.311334</td>\n",
       "      <td>0.779591</td>\n",
       "      <td>0.632750</td>\n",
       "      <td>1.631546</td>\n",
       "      <td>1.489088</td>\n",
       "      <td>0.416904</td>\n",
       "      <td>0.405661</td>\n",
       "      <td>3.526579</td>\n",
       "      <td>3.058575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA processing</th>\n",
       "      <td>0.350723</td>\n",
       "      <td>0.337542</td>\n",
       "      <td>3.224591</td>\n",
       "      <td>3.113528</td>\n",
       "      <td>0.698690</td>\n",
       "      <td>0.539338</td>\n",
       "      <td>3.990782</td>\n",
       "      <td>3.644785</td>\n",
       "      <td>1.020705</td>\n",
       "      <td>0.945443</td>\n",
       "      <td>...</td>\n",
       "      <td>2.725126</td>\n",
       "      <td>2.395537</td>\n",
       "      <td>1.018255</td>\n",
       "      <td>0.818901</td>\n",
       "      <td>0.674974</td>\n",
       "      <td>0.615844</td>\n",
       "      <td>1.350553</td>\n",
       "      <td>1.258537</td>\n",
       "      <td>0.597357</td>\n",
       "      <td>0.520184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA processing in the mitochondrion</th>\n",
       "      <td>2.558716</td>\n",
       "      <td>2.231357</td>\n",
       "      <td>0.963222</td>\n",
       "      <td>0.915225</td>\n",
       "      <td>0.570653</td>\n",
       "      <td>0.460940</td>\n",
       "      <td>6.861804</td>\n",
       "      <td>6.353881</td>\n",
       "      <td>16.264847</td>\n",
       "      <td>15.535750</td>\n",
       "      <td>...</td>\n",
       "      <td>1.333103</td>\n",
       "      <td>1.159934</td>\n",
       "      <td>1.627142</td>\n",
       "      <td>1.293508</td>\n",
       "      <td>3.736491</td>\n",
       "      <td>3.445807</td>\n",
       "      <td>11.916951</td>\n",
       "      <td>11.478664</td>\n",
       "      <td>5.255992</td>\n",
       "      <td>4.629379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA processing in the nucleus and cytosol</th>\n",
       "      <td>0.361006</td>\n",
       "      <td>0.345227</td>\n",
       "      <td>3.312260</td>\n",
       "      <td>3.199842</td>\n",
       "      <td>0.727344</td>\n",
       "      <td>0.556974</td>\n",
       "      <td>3.906893</td>\n",
       "      <td>3.568721</td>\n",
       "      <td>1.010827</td>\n",
       "      <td>0.936062</td>\n",
       "      <td>...</td>\n",
       "      <td>2.780706</td>\n",
       "      <td>2.446620</td>\n",
       "      <td>1.018850</td>\n",
       "      <td>0.818981</td>\n",
       "      <td>0.653134</td>\n",
       "      <td>0.596403</td>\n",
       "      <td>1.322648</td>\n",
       "      <td>1.232241</td>\n",
       "      <td>0.576181</td>\n",
       "      <td>0.502162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snRNP Assembly</th>\n",
       "      <td>10.409022</td>\n",
       "      <td>9.474230</td>\n",
       "      <td>87.993579</td>\n",
       "      <td>86.835132</td>\n",
       "      <td>2.194038</td>\n",
       "      <td>1.529476</td>\n",
       "      <td>21.346077</td>\n",
       "      <td>20.314825</td>\n",
       "      <td>35.918982</td>\n",
       "      <td>34.593400</td>\n",
       "      <td>...</td>\n",
       "      <td>3.524237</td>\n",
       "      <td>3.118479</td>\n",
       "      <td>1.871249</td>\n",
       "      <td>1.486914</td>\n",
       "      <td>8.966442</td>\n",
       "      <td>8.330874</td>\n",
       "      <td>2.833084</td>\n",
       "      <td>2.666417</td>\n",
       "      <td>7.478763</td>\n",
       "      <td>6.619836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA Aminoacylation</th>\n",
       "      <td>1.180890</td>\n",
       "      <td>1.007096</td>\n",
       "      <td>65.611203</td>\n",
       "      <td>64.778694</td>\n",
       "      <td>2.447590</td>\n",
       "      <td>1.687084</td>\n",
       "      <td>21.124507</td>\n",
       "      <td>20.102164</td>\n",
       "      <td>20.240820</td>\n",
       "      <td>19.363344</td>\n",
       "      <td>...</td>\n",
       "      <td>12.059921</td>\n",
       "      <td>11.130502</td>\n",
       "      <td>0.470952</td>\n",
       "      <td>0.404870</td>\n",
       "      <td>4.014184</td>\n",
       "      <td>3.705549</td>\n",
       "      <td>2.740870</td>\n",
       "      <td>2.578790</td>\n",
       "      <td>5.738798</td>\n",
       "      <td>5.078087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA modification in the mitochondrion</th>\n",
       "      <td>1.567963</td>\n",
       "      <td>1.335730</td>\n",
       "      <td>8.183507</td>\n",
       "      <td>7.985954</td>\n",
       "      <td>0.945721</td>\n",
       "      <td>0.700426</td>\n",
       "      <td>1.697206</td>\n",
       "      <td>1.528691</td>\n",
       "      <td>3.171743</td>\n",
       "      <td>2.971210</td>\n",
       "      <td>...</td>\n",
       "      <td>0.477717</td>\n",
       "      <td>0.443303</td>\n",
       "      <td>0.438509</td>\n",
       "      <td>0.382555</td>\n",
       "      <td>1.882957</td>\n",
       "      <td>1.710411</td>\n",
       "      <td>0.326855</td>\n",
       "      <td>0.324545</td>\n",
       "      <td>2.707477</td>\n",
       "      <td>2.320087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA modification in the nucleus and cytosol</th>\n",
       "      <td>0.704413</td>\n",
       "      <td>0.604298</td>\n",
       "      <td>45.634924</td>\n",
       "      <td>44.991136</td>\n",
       "      <td>4.107528</td>\n",
       "      <td>2.790719</td>\n",
       "      <td>16.387902</td>\n",
       "      <td>15.548215</td>\n",
       "      <td>5.430805</td>\n",
       "      <td>5.131447</td>\n",
       "      <td>...</td>\n",
       "      <td>2.644672</td>\n",
       "      <td>2.323060</td>\n",
       "      <td>0.988084</td>\n",
       "      <td>0.795779</td>\n",
       "      <td>1.655744</td>\n",
       "      <td>1.510369</td>\n",
       "      <td>2.451006</td>\n",
       "      <td>2.300034</td>\n",
       "      <td>7.593586</td>\n",
       "      <td>6.720822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA processing</th>\n",
       "      <td>19.248006</td>\n",
       "      <td>17.817688</td>\n",
       "      <td>95.268268</td>\n",
       "      <td>93.947094</td>\n",
       "      <td>1.135550</td>\n",
       "      <td>0.834101</td>\n",
       "      <td>37.836909</td>\n",
       "      <td>36.163553</td>\n",
       "      <td>40.029946</td>\n",
       "      <td>38.551986</td>\n",
       "      <td>...</td>\n",
       "      <td>7.331414</td>\n",
       "      <td>6.661009</td>\n",
       "      <td>5.027523</td>\n",
       "      <td>3.915660</td>\n",
       "      <td>10.855245</td>\n",
       "      <td>10.082504</td>\n",
       "      <td>7.876478</td>\n",
       "      <td>7.545545</td>\n",
       "      <td>17.371009</td>\n",
       "      <td>15.309472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA processing in the mitochondrion</th>\n",
       "      <td>8.825378</td>\n",
       "      <td>7.982786</td>\n",
       "      <td>5.491192</td>\n",
       "      <td>5.335152</td>\n",
       "      <td>1.198887</td>\n",
       "      <td>0.876479</td>\n",
       "      <td>1.836096</td>\n",
       "      <td>1.654738</td>\n",
       "      <td>10.780951</td>\n",
       "      <td>10.275047</td>\n",
       "      <td>...</td>\n",
       "      <td>2.216781</td>\n",
       "      <td>1.940179</td>\n",
       "      <td>0.965501</td>\n",
       "      <td>0.777772</td>\n",
       "      <td>1.694317</td>\n",
       "      <td>1.544825</td>\n",
       "      <td>0.852926</td>\n",
       "      <td>0.796904</td>\n",
       "      <td>0.658380</td>\n",
       "      <td>0.569715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA processing in the nucleus</th>\n",
       "      <td>18.974620</td>\n",
       "      <td>17.566296</td>\n",
       "      <td>70.963336</td>\n",
       "      <td>70.066483</td>\n",
       "      <td>1.859653</td>\n",
       "      <td>1.325317</td>\n",
       "      <td>29.350417</td>\n",
       "      <td>28.112790</td>\n",
       "      <td>33.064652</td>\n",
       "      <td>31.827025</td>\n",
       "      <td>...</td>\n",
       "      <td>1.930719</td>\n",
       "      <td>1.687544</td>\n",
       "      <td>6.113450</td>\n",
       "      <td>4.731140</td>\n",
       "      <td>5.170739</td>\n",
       "      <td>4.804295</td>\n",
       "      <td>5.944671</td>\n",
       "      <td>5.661688</td>\n",
       "      <td>14.587067</td>\n",
       "      <td>12.860291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trans-Golgi Network Vesicle Budding</th>\n",
       "      <td>0.975492</td>\n",
       "      <td>0.828064</td>\n",
       "      <td>0.610988</td>\n",
       "      <td>0.584659</td>\n",
       "      <td>0.940660</td>\n",
       "      <td>0.697203</td>\n",
       "      <td>2.688724</td>\n",
       "      <td>2.429961</td>\n",
       "      <td>5.478459</td>\n",
       "      <td>5.175750</td>\n",
       "      <td>...</td>\n",
       "      <td>0.536009</td>\n",
       "      <td>0.488012</td>\n",
       "      <td>1.201339</td>\n",
       "      <td>0.958249</td>\n",
       "      <td>5.059475</td>\n",
       "      <td>4.699315</td>\n",
       "      <td>3.995115</td>\n",
       "      <td>3.778677</td>\n",
       "      <td>2.790866</td>\n",
       "      <td>2.395223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>truncated APC mutants destabilize the destruction complex</th>\n",
       "      <td>0.936291</td>\n",
       "      <td>0.794994</td>\n",
       "      <td>28.217593</td>\n",
       "      <td>27.796759</td>\n",
       "      <td>2.172428</td>\n",
       "      <td>1.528800</td>\n",
       "      <td>15.850162</td>\n",
       "      <td>15.038503</td>\n",
       "      <td>14.395657</td>\n",
       "      <td>13.742549</td>\n",
       "      <td>...</td>\n",
       "      <td>11.145567</td>\n",
       "      <td>10.280147</td>\n",
       "      <td>0.314112</td>\n",
       "      <td>0.308352</td>\n",
       "      <td>0.762996</td>\n",
       "      <td>0.694406</td>\n",
       "      <td>1.707233</td>\n",
       "      <td>1.596865</td>\n",
       "      <td>6.056340</td>\n",
       "      <td>5.363813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vRNP Assembly</th>\n",
       "      <td>6.286006</td>\n",
       "      <td>5.690183</td>\n",
       "      <td>15.047276</td>\n",
       "      <td>14.762683</td>\n",
       "      <td>0.392500</td>\n",
       "      <td>0.353300</td>\n",
       "      <td>4.664218</td>\n",
       "      <td>4.273752</td>\n",
       "      <td>13.511153</td>\n",
       "      <td>12.893314</td>\n",
       "      <td>...</td>\n",
       "      <td>1.673513</td>\n",
       "      <td>1.455002</td>\n",
       "      <td>1.218187</td>\n",
       "      <td>0.971786</td>\n",
       "      <td>1.383214</td>\n",
       "      <td>1.258135</td>\n",
       "      <td>0.576505</td>\n",
       "      <td>0.544803</td>\n",
       "      <td>1.716624</td>\n",
       "      <td>1.461645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2074 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Cluster 1 p-values  \\\n",
       "2-LTR circle formation                                        3.760961   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.345861   \n",
       "A tetrasaccharide linker sequence is required f...            8.375755   \n",
       "ABC transporter disorders                                     7.503048   \n",
       "ABC transporters in lipid homeostasis                         2.180941   \n",
       "ABC-family proteins mediated transport                        7.218621   \n",
       "ABO blood group biosynthesis                                  0.821769   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.665815   \n",
       "ADP signalling through P2Y purinoceptor 12                    3.115592   \n",
       "AKT phosphorylates targets in the cytosol                     3.216786   \n",
       "AKT phosphorylates targets in the nucleus                     7.864724   \n",
       "AKT-mediated inactivation of FOXO1A                           5.360685   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.548935   \n",
       "AMER1 mutants destabilize the destruction complex             0.936291   \n",
       "AMPK inhibits chREBP transcriptional activation...            8.371428   \n",
       "APC truncation mutants have impaired AXIN binding             0.936291   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      21.945291   \n",
       "APC/C-mediated degradation of cell cycle proteins            13.011680   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 22.040693   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   8.393073   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...           10.889119   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            9.718463   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            9.890892   \n",
       "APEX1-Independent Resolution of AP Sites via th...            0.842751   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               6.390606   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.789112   \n",
       "ARMS-mediated activation                                      6.105456   \n",
       "ATF4 activates genes in response to endoplasmic...            2.928160   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.445184   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.401413   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.744543   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.632177   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.588006   \n",
       "p38MAPK events                                                0.459301   \n",
       "p53-Dependent G1 DNA Damage Response                          7.153408   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      7.153408   \n",
       "p53-Independent DNA Damage Response                           6.665075   \n",
       "p53-Independent G1/S DNA damage checkpoint                    6.665075   \n",
       "p75 NTR receptor-mediated signalling                          9.760672   \n",
       "p75NTR negatively regulates cell cycle via SC1                3.834756   \n",
       "p75NTR recruits signalling complexes                          2.321031   \n",
       "p75NTR regulates axonogenesis                                 1.250670   \n",
       "p75NTR signals via NF-kB                                      4.456845   \n",
       "phospho-PLA2 pathway                                          1.476583   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            1.443835   \n",
       "rRNA modification in the mitochondrion                        2.371133   \n",
       "rRNA modification in the nucleus and cytosol                  2.436913   \n",
       "rRNA processing                                               0.350723   \n",
       "rRNA processing in the mitochondrion                          2.558716   \n",
       "rRNA processing in the nucleus and cytosol                    0.361006   \n",
       "snRNP Assembly                                               10.409022   \n",
       "tRNA Aminoacylation                                           1.180890   \n",
       "tRNA modification in the mitochondrion                        1.567963   \n",
       "tRNA modification in the nucleus and cytosol                  0.704413   \n",
       "tRNA processing                                              19.248006   \n",
       "tRNA processing in the mitochondrion                          8.825378   \n",
       "tRNA processing in the nucleus                               18.974620   \n",
       "trans-Golgi Network Vesicle Budding                           0.975492   \n",
       "truncated APC mutants destabilize the destructi...            0.936291   \n",
       "vRNP Assembly                                                 6.286006   \n",
       "\n",
       "                                                    Cluster 1 q-values  \\\n",
       "2-LTR circle formation                                        3.335132   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.333758   \n",
       "A tetrasaccharide linker sequence is required f...            7.582692   \n",
       "ABC transporter disorders                                     6.786122   \n",
       "ABC transporters in lipid homeostasis                         1.890257   \n",
       "ABC-family proteins mediated transport                        6.531222   \n",
       "ABO blood group biosynthesis                                  0.701136   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.423457   \n",
       "ADP signalling through P2Y purinoceptor 12                    2.737803   \n",
       "AKT phosphorylates targets in the cytosol                     2.832958   \n",
       "AKT phosphorylates targets in the nucleus                     7.114942   \n",
       "AKT-mediated inactivation of FOXO1A                           4.822751   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.478625   \n",
       "AMER1 mutants destabilize the destruction complex             0.794994   \n",
       "AMPK inhibits chREBP transcriptional activation...            7.580172   \n",
       "APC truncation mutants have impaired AXIN binding             0.794994   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      20.473581   \n",
       "APC/C-mediated degradation of cell cycle proteins            11.885203   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 20.562734   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   7.597402   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            9.920615   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            8.808194   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            8.975483   \n",
       "APEX1-Independent Resolution of AP Sites via th...            0.719067   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               5.781368   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.673154   \n",
       "ARMS-mediated activation                                      5.523447   \n",
       "ATF4 activates genes in response to endoplasmic...            2.561229   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.401839   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.371061   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.637398   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.392737   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.257969   \n",
       "p38MAPK events                                                0.411772   \n",
       "p53-Dependent G1 DNA Damage Response                          6.473087   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      6.473087   \n",
       "p53-Independent DNA Damage Response                           6.031313   \n",
       "p53-Independent G1/S DNA damage checkpoint                    6.031313   \n",
       "p75 NTR receptor-mediated signalling                          8.848697   \n",
       "p75NTR negatively regulates cell cycle via SC1                3.401919   \n",
       "p75NTR recruits signalling complexes                          2.016638   \n",
       "p75NTR regulates axonogenesis                                 1.062906   \n",
       "p75NTR signals via NF-kB                                      3.983892   \n",
       "phospho-PLA2 pathway                                          1.256684   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            1.232440   \n",
       "rRNA modification in the mitochondrion                        2.062072   \n",
       "rRNA modification in the nucleus and cytosol                  2.121838   \n",
       "rRNA processing                                               0.337542   \n",
       "rRNA processing in the mitochondrion                          2.231357   \n",
       "rRNA processing in the nucleus and cytosol                    0.345227   \n",
       "snRNP Assembly                                                9.474230   \n",
       "tRNA Aminoacylation                                           1.007096   \n",
       "tRNA modification in the mitochondrion                        1.335730   \n",
       "tRNA modification in the nucleus and cytosol                  0.604298   \n",
       "tRNA processing                                              17.817688   \n",
       "tRNA processing in the mitochondrion                          7.982786   \n",
       "tRNA processing in the nucleus                               17.566296   \n",
       "trans-Golgi Network Vesicle Budding                           0.828064   \n",
       "truncated APC mutants destabilize the destructi...            0.794994   \n",
       "vRNP Assembly                                                 5.690183   \n",
       "\n",
       "                                                    Cluster 10 p-values  \\\n",
       "2-LTR circle formation                                        18.711431   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    15.974278   \n",
       "A tetrasaccharide linker sequence is required f...             7.269704   \n",
       "ABC transporter disorders                                     33.002068   \n",
       "ABC transporters in lipid homeostasis                         70.425206   \n",
       "ABC-family proteins mediated transport                        35.018455   \n",
       "ABO blood group biosynthesis                                   2.607131   \n",
       "ADP signalling through P2Y purinoceptor 1                      6.976290   \n",
       "ADP signalling through P2Y purinoceptor 12                     4.658852   \n",
       "AKT phosphorylates targets in the cytosol                     11.108934   \n",
       "AKT phosphorylates targets in the nucleus                     11.723281   \n",
       "AKT-mediated inactivation of FOXO1A                           13.729136   \n",
       "ALKBH3 mediated reversal of alkylation damage                  2.348202   \n",
       "AMER1 mutants destabilize the destruction complex             28.217593   \n",
       "AMPK inhibits chREBP transcriptional activation...             8.648882   \n",
       "APC truncation mutants have impaired AXIN binding             28.217593   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       67.124764   \n",
       "APC/C-mediated degradation of cell cycle proteins             71.914874   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  69.952725   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   35.796564   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            48.948295   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            49.381426   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            44.195642   \n",
       "APEX1-Independent Resolution of AP Sites via th...            23.678476   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               53.944371   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                  1.546303   \n",
       "ARMS-mediated activation                                       4.898852   \n",
       "ATF4 activates genes in response to endoplasmic...            10.093901   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                    4.522333   \n",
       "ATF6 (ATF6-alpha) activates chaperones                         2.989754   \n",
       "...                                                                 ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...             5.198804   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...             2.156582   \n",
       "p130Cas linkage to MAPK signaling for integrins                1.037744   \n",
       "p38MAPK events                                                11.137599   \n",
       "p53-Dependent G1 DNA Damage Response                          44.975996   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      44.975996   \n",
       "p53-Independent DNA Damage Response                           34.369896   \n",
       "p53-Independent G1/S DNA damage checkpoint                    34.369896   \n",
       "p75 NTR receptor-mediated signalling                          48.027490   \n",
       "p75NTR negatively regulates cell cycle via SC1                 6.212856   \n",
       "p75NTR recruits signalling complexes                          85.793526   \n",
       "p75NTR regulates axonogenesis                                 23.282628   \n",
       "p75NTR signals via NF-kB                                      88.998536   \n",
       "phospho-PLA2 pathway                                           2.245882   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            12.657323   \n",
       "rRNA modification in the mitochondrion                         0.675652   \n",
       "rRNA modification in the nucleus and cytosol                   9.245907   \n",
       "rRNA processing                                                3.224591   \n",
       "rRNA processing in the mitochondrion                           0.963222   \n",
       "rRNA processing in the nucleus and cytosol                     3.312260   \n",
       "snRNP Assembly                                                87.993579   \n",
       "tRNA Aminoacylation                                           65.611203   \n",
       "tRNA modification in the mitochondrion                         8.183507   \n",
       "tRNA modification in the nucleus and cytosol                  45.634924   \n",
       "tRNA processing                                               95.268268   \n",
       "tRNA processing in the mitochondrion                           5.491192   \n",
       "tRNA processing in the nucleus                                70.963336   \n",
       "trans-Golgi Network Vesicle Budding                            0.610988   \n",
       "truncated APC mutants destabilize the destructi...            28.217593   \n",
       "vRNP Assembly                                                 15.047276   \n",
       "\n",
       "                                                    Cluster 10 q-values  \\\n",
       "2-LTR circle formation                                        18.385849   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    15.677000   \n",
       "A tetrasaccharide linker sequence is required f...             7.085795   \n",
       "ABC transporter disorders                                     32.520950   \n",
       "ABC transporters in lipid homeostasis                         69.536532   \n",
       "ABC-family proteins mediated transport                        34.507147   \n",
       "ABO blood group biosynthesis                                   2.508858   \n",
       "ADP signalling through P2Y purinoceptor 1                      6.796835   \n",
       "ADP signalling through P2Y purinoceptor 12                     4.515230   \n",
       "AKT phosphorylates targets in the cytosol                     10.879551   \n",
       "AKT phosphorylates targets in the nucleus                     11.484929   \n",
       "AKT-mediated inactivation of FOXO1A                           13.461933   \n",
       "ALKBH3 mediated reversal of alkylation damage                  2.254369   \n",
       "AMER1 mutants destabilize the destruction complex             27.796759   \n",
       "AMPK inhibits chREBP transcriptional activation...             8.447351   \n",
       "APC truncation mutants have impaired AXIN binding             27.796759   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       66.274823   \n",
       "APC/C-mediated degradation of cell cycle proteins             71.011365   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  69.072079   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   35.273546   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            48.275925   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            48.704104   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            43.570799   \n",
       "APEX1-Independent Resolution of AP Sites via th...            23.307628   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               53.228535   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                  1.476977   \n",
       "ARMS-mediated activation                                       4.751129   \n",
       "ATF4 activates genes in response to endoplasmic...             9.876082   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                    4.382194   \n",
       "ATF6 (ATF6-alpha) activates chaperones                         2.883799   \n",
       "...                                                                 ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...             5.047240   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...             2.067146   \n",
       "p130Cas linkage to MAPK signaling for integrins                0.986460   \n",
       "p38MAPK events                                                10.907862   \n",
       "p53-Dependent G1 DNA Damage Response                          44.343134   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      44.343134   \n",
       "p53-Independent DNA Damage Response                           33.868665   \n",
       "p53-Independent G1/S DNA damage checkpoint                    33.868665   \n",
       "p75 NTR receptor-mediated signalling                          47.362927   \n",
       "p75NTR negatively regulates cell cycle via SC1                 6.045882   \n",
       "p75NTR recruits signalling complexes                          84.669842   \n",
       "p75NTR regulates axonogenesis                                 22.914721   \n",
       "p75NTR signals via NF-kB                                      87.812061   \n",
       "phospho-PLA2 pathway                                           2.153865   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            12.403472   \n",
       "rRNA modification in the mitochondrion                         0.646197   \n",
       "rRNA modification in the nucleus and cytosol                   9.037663   \n",
       "rRNA processing                                                3.113528   \n",
       "rRNA processing in the mitochondrion                           0.915225   \n",
       "rRNA processing in the nucleus and cytosol                     3.199842   \n",
       "snRNP Assembly                                                86.835132   \n",
       "tRNA Aminoacylation                                           64.778694   \n",
       "tRNA modification in the mitochondrion                         7.985954   \n",
       "tRNA modification in the nucleus and cytosol                  44.991136   \n",
       "tRNA processing                                               93.947094   \n",
       "tRNA processing in the mitochondrion                           5.335152   \n",
       "tRNA processing in the nucleus                                70.066483   \n",
       "trans-Golgi Network Vesicle Budding                            0.584659   \n",
       "truncated APC mutants destabilize the destructi...            27.796759   \n",
       "vRNP Assembly                                                 14.762683   \n",
       "\n",
       "                                                    Cluster 2 p-values  \\\n",
       "2-LTR circle formation                                        0.325594   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    1.045455   \n",
       "A tetrasaccharide linker sequence is required f...            2.155733   \n",
       "ABC transporter disorders                                     0.387921   \n",
       "ABC transporters in lipid homeostasis                         4.768345   \n",
       "ABC-family proteins mediated transport                        0.340716   \n",
       "ABO blood group biosynthesis                                  1.903703   \n",
       "ADP signalling through P2Y purinoceptor 1                     0.916661   \n",
       "ADP signalling through P2Y purinoceptor 12                    1.022234   \n",
       "AKT phosphorylates targets in the cytosol                     1.160667   \n",
       "AKT phosphorylates targets in the nucleus                     4.513568   \n",
       "AKT-mediated inactivation of FOXO1A                           3.602443   \n",
       "ALKBH3 mediated reversal of alkylation damage                 1.581827   \n",
       "AMER1 mutants destabilize the destruction complex             2.172428   \n",
       "AMPK inhibits chREBP transcriptional activation...            2.175979   \n",
       "APC truncation mutants have impaired AXIN binding             2.172428   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       2.009577   \n",
       "APC/C-mediated degradation of cell cycle proteins             0.665471   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  2.354643   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.307257   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            0.447278   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            0.347536   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            0.354125   \n",
       "APEX1-Independent Resolution of AP Sites via th...            1.344850   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               0.622397   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.868220   \n",
       "ARMS-mediated activation                                      1.107040   \n",
       "ATF4 activates genes in response to endoplasmic...            0.408322   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   1.332106   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        1.203695   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.349295   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.459124   \n",
       "p130Cas linkage to MAPK signaling for integrins               0.958216   \n",
       "p38MAPK events                                                0.898098   \n",
       "p53-Dependent G1 DNA Damage Response                          0.357696   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.357696   \n",
       "p53-Independent DNA Damage Response                           0.470981   \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.470981   \n",
       "p75 NTR receptor-mediated signalling                          0.452507   \n",
       "p75NTR negatively regulates cell cycle via SC1                0.354691   \n",
       "p75NTR recruits signalling complexes                          1.934117   \n",
       "p75NTR regulates axonogenesis                                 1.274652   \n",
       "p75NTR signals via NF-kB                                      1.797095   \n",
       "phospho-PLA2 pathway                                          0.312051   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            1.720946   \n",
       "rRNA modification in the mitochondrion                        1.109413   \n",
       "rRNA modification in the nucleus and cytosol                  0.521597   \n",
       "rRNA processing                                               0.698690   \n",
       "rRNA processing in the mitochondrion                          0.570653   \n",
       "rRNA processing in the nucleus and cytosol                    0.727344   \n",
       "snRNP Assembly                                                2.194038   \n",
       "tRNA Aminoacylation                                           2.447590   \n",
       "tRNA modification in the mitochondrion                        0.945721   \n",
       "tRNA modification in the nucleus and cytosol                  4.107528   \n",
       "tRNA processing                                               1.135550   \n",
       "tRNA processing in the mitochondrion                          1.198887   \n",
       "tRNA processing in the nucleus                                1.859653   \n",
       "trans-Golgi Network Vesicle Budding                           0.940660   \n",
       "truncated APC mutants destabilize the destructi...            2.172428   \n",
       "vRNP Assembly                                                 0.392500   \n",
       "\n",
       "                                                    Cluster 2 q-values  \\\n",
       "2-LTR circle formation                                        0.312018   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.770827   \n",
       "A tetrasaccharide linker sequence is required f...            1.521358   \n",
       "ABC transporter disorders                                     0.349866   \n",
       "ABC transporters in lipid homeostasis                         3.243928   \n",
       "ABC-family proteins mediated transport                        0.322760   \n",
       "ABO blood group biosynthesis                                  1.358481   \n",
       "ADP signalling through P2Y purinoceptor 1                     0.681919   \n",
       "ADP signalling through P2Y purinoceptor 12                    0.755483   \n",
       "AKT phosphorylates targets in the cytosol                     0.850752   \n",
       "AKT phosphorylates targets in the nucleus                     3.061005   \n",
       "AKT-mediated inactivation of FOXO1A                           2.443997   \n",
       "ALKBH3 mediated reversal of alkylation damage                 1.139500   \n",
       "AMER1 mutants destabilize the destruction complex             1.528800   \n",
       "AMPK inhibits chREBP transcriptional activation...            1.528800   \n",
       "APC truncation mutants have impaired AXIN binding             1.528800   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       1.429961   \n",
       "APC/C-mediated degradation of cell cycle proteins             0.518962   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  1.624421   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.303894   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            0.384586   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            0.327133   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            0.329369   \n",
       "APEX1-Independent Resolution of AP Sites via th...            0.972030   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               0.492591   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.646232   \n",
       "ARMS-mediated activation                                      0.810592   \n",
       "ATF4 activates genes in response to endoplasmic...            0.363587   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.964606   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.880323   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.327793   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.391321   \n",
       "p130Cas linkage to MAPK signaling for integrins               0.708479   \n",
       "p38MAPK events                                                0.667649   \n",
       "p53-Dependent G1 DNA Damage Response                          0.330087   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.330087   \n",
       "p53-Independent DNA Damage Response                           0.397609   \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.397609   \n",
       "p75 NTR receptor-mediated signalling                          0.386849   \n",
       "p75NTR negatively regulates cell cycle via SC1                0.329614   \n",
       "p75NTR recruits signalling complexes                          1.382231   \n",
       "p75NTR regulates axonogenesis                                 0.926424   \n",
       "p75NTR signals via NF-kB                                      1.280323   \n",
       "phospho-PLA2 pathway                                          0.305510   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            1.231489   \n",
       "rRNA modification in the mitochondrion                        0.811796   \n",
       "rRNA modification in the nucleus and cytosol                  0.429839   \n",
       "rRNA processing                                               0.539338   \n",
       "rRNA processing in the mitochondrion                          0.460940   \n",
       "rRNA processing in the nucleus and cytosol                    0.556974   \n",
       "snRNP Assembly                                                1.529476   \n",
       "tRNA Aminoacylation                                           1.687084   \n",
       "tRNA modification in the mitochondrion                        0.700426   \n",
       "tRNA modification in the nucleus and cytosol                  2.790719   \n",
       "tRNA processing                                               0.834101   \n",
       "tRNA processing in the mitochondrion                          0.876479   \n",
       "tRNA processing in the nucleus                                1.325317   \n",
       "trans-Golgi Network Vesicle Budding                           0.697203   \n",
       "truncated APC mutants destabilize the destructi...            1.528800   \n",
       "vRNP Assembly                                                 0.353300   \n",
       "\n",
       "                                                    Cluster 3 p-values  \\\n",
       "2-LTR circle formation                                        1.881159   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.902001   \n",
       "A tetrasaccharide linker sequence is required f...           13.725188   \n",
       "ABC transporter disorders                                     4.239394   \n",
       "ABC transporters in lipid homeostasis                         9.805411   \n",
       "ABC-family proteins mediated transport                        4.383301   \n",
       "ABO blood group biosynthesis                                  0.393219   \n",
       "ADP signalling through P2Y purinoceptor 1                    13.591149   \n",
       "ADP signalling through P2Y purinoceptor 12                   15.881230   \n",
       "AKT phosphorylates targets in the cytosol                     1.609689   \n",
       "AKT phosphorylates targets in the nucleus                     1.192219   \n",
       "AKT-mediated inactivation of FOXO1A                           0.590346   \n",
       "ALKBH3 mediated reversal of alkylation damage                 5.078548   \n",
       "AMER1 mutants destabilize the destruction complex            15.850162   \n",
       "AMPK inhibits chREBP transcriptional activation...            3.232553   \n",
       "APC truncation mutants have impaired AXIN binding            15.850162   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      31.310749   \n",
       "APC/C-mediated degradation of cell cycle proteins            17.306354   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 34.554555   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   5.574839   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            9.475519   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            9.298696   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            7.674052   \n",
       "APEX1-Independent Resolution of AP Sites via th...            4.229399   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              14.143824   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.371532   \n",
       "ARMS-mediated activation                                      0.472567   \n",
       "ATF4 activates genes in response to endoplasmic...           19.651999   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   1.750143   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        1.614979   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            2.565066   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            3.501793   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.298793   \n",
       "p38MAPK events                                                0.884350   \n",
       "p53-Dependent G1 DNA Damage Response                          5.965132   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      5.965132   \n",
       "p53-Independent DNA Damage Response                           4.257262   \n",
       "p53-Independent G1/S DNA damage checkpoint                    4.257262   \n",
       "p75 NTR receptor-mediated signalling                         10.986857   \n",
       "p75NTR negatively regulates cell cycle via SC1                2.044479   \n",
       "p75NTR recruits signalling complexes                         13.825430   \n",
       "p75NTR regulates axonogenesis                                11.677282   \n",
       "p75NTR signals via NF-kB                                     17.083295   \n",
       "phospho-PLA2 pathway                                          1.158092   \n",
       "phosphorylation site mutants of CTNNB1 are not ...           12.017237   \n",
       "rRNA modification in the mitochondrion                        4.769068   \n",
       "rRNA modification in the nucleus and cytosol                 12.294303   \n",
       "rRNA processing                                               3.990782   \n",
       "rRNA processing in the mitochondrion                          6.861804   \n",
       "rRNA processing in the nucleus and cytosol                    3.906893   \n",
       "snRNP Assembly                                               21.346077   \n",
       "tRNA Aminoacylation                                          21.124507   \n",
       "tRNA modification in the mitochondrion                        1.697206   \n",
       "tRNA modification in the nucleus and cytosol                 16.387902   \n",
       "tRNA processing                                              37.836909   \n",
       "tRNA processing in the mitochondrion                          1.836096   \n",
       "tRNA processing in the nucleus                               29.350417   \n",
       "trans-Golgi Network Vesicle Budding                           2.688724   \n",
       "truncated APC mutants destabilize the destructi...           15.850162   \n",
       "vRNP Assembly                                                 4.664218   \n",
       "\n",
       "                                                    Cluster 3 q-values  \\\n",
       "2-LTR circle formation                                        1.696610   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.812664   \n",
       "A tetrasaccharide linker sequence is required f...           12.984721   \n",
       "ABC transporter disorders                                     3.874409   \n",
       "ABC transporters in lipid homeostasis                         9.191894   \n",
       "ABC-family proteins mediated transport                        4.004512   \n",
       "ABO blood group biosynthesis                                  0.376135   \n",
       "ADP signalling through P2Y purinoceptor 1                    12.852980   \n",
       "ADP signalling through P2Y purinoceptor 12                   15.061351   \n",
       "AKT phosphorylates targets in the cytosol                     1.449732   \n",
       "AKT phosphorylates targets in the nucleus                     1.071587   \n",
       "AKT-mediated inactivation of FOXO1A                           0.543050   \n",
       "ALKBH3 mediated reversal of alkylation damage                 4.655501   \n",
       "AMER1 mutants destabilize the destruction complex            15.038503   \n",
       "AMPK inhibits chREBP transcriptional activation...            2.940640   \n",
       "APC truncation mutants have impaired AXIN binding            15.038503   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      30.019246   \n",
       "APC/C-mediated degradation of cell cycle proteins            16.428878   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 33.082844   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   5.114759   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            8.871056   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            8.702046   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            7.131029   \n",
       "APEX1-Independent Resolution of AP Sites via th...            3.865867   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              13.384523   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.357487   \n",
       "ARMS-mediated activation                                      0.442439   \n",
       "ATF4 activates genes in response to endoplasmic...           18.687373   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   1.576349   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        1.454720   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            2.315700   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            3.188013   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.073651   \n",
       "p38MAPK events                                                0.797735   \n",
       "p53-Dependent G1 DNA Damage Response                          5.490308   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      5.490308   \n",
       "p53-Independent DNA Damage Response                           3.889355   \n",
       "p53-Independent G1/S DNA damage checkpoint                    3.889355   \n",
       "p75 NTR receptor-mediated signalling                         10.339365   \n",
       "p75NTR negatively regulates cell cycle via SC1                1.839604   \n",
       "p75NTR recruits signalling complexes                         13.079164   \n",
       "p75NTR regulates axonogenesis                                11.002938   \n",
       "p75NTR signals via NF-kB                                     16.215193   \n",
       "phospho-PLA2 pathway                                          1.041587   \n",
       "phosphorylation site mutants of CTNNB1 are not ...           11.334906   \n",
       "rRNA modification in the mitochondrion                        4.371338   \n",
       "rRNA modification in the nucleus and cytosol                 11.595543   \n",
       "rRNA processing                                               3.644785   \n",
       "rRNA processing in the mitochondrion                          6.353881   \n",
       "rRNA processing in the nucleus and cytosol                    3.568721   \n",
       "snRNP Assembly                                               20.314825   \n",
       "tRNA Aminoacylation                                          20.102164   \n",
       "tRNA modification in the mitochondrion                        1.528691   \n",
       "tRNA modification in the nucleus and cytosol                 15.548215   \n",
       "tRNA processing                                              36.163553   \n",
       "tRNA processing in the mitochondrion                          1.654738   \n",
       "tRNA processing in the nucleus                               28.112790   \n",
       "trans-Golgi Network Vesicle Budding                           2.429961   \n",
       "truncated APC mutants destabilize the destructi...           15.038503   \n",
       "vRNP Assembly                                                 4.273752   \n",
       "\n",
       "                                                    Cluster 4ER+ p-values  \\\n",
       "2-LTR circle formation                                           8.625987   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                       6.189245   \n",
       "A tetrasaccharide linker sequence is required f...               8.374111   \n",
       "ABC transporter disorders                                        1.241745   \n",
       "ABC transporters in lipid homeostasis                            3.275117   \n",
       "ABC-family proteins mediated transport                           1.664934   \n",
       "ABO blood group biosynthesis                                     5.278316   \n",
       "ADP signalling through P2Y purinoceptor 1                       11.845372   \n",
       "ADP signalling through P2Y purinoceptor 12                      16.676885   \n",
       "AKT phosphorylates targets in the cytosol                        3.927190   \n",
       "AKT phosphorylates targets in the nucleus                        7.273187   \n",
       "AKT-mediated inactivation of FOXO1A                              4.670407   \n",
       "ALKBH3 mediated reversal of alkylation damage                    3.608131   \n",
       "AMER1 mutants destabilize the destruction complex               14.395657   \n",
       "AMPK inhibits chREBP transcriptional activation...               5.671590   \n",
       "APC truncation mutants have impaired AXIN binding               14.395657   \n",
       "APC-Cdc20 mediated degradation of Nek2A                         39.735972   \n",
       "APC/C-mediated degradation of cell cycle proteins               12.148558   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                    41.539732   \n",
       "APC/C:Cdc20 mediated degradation of Securin                      2.013528   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...               5.867643   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...               4.286565   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...               4.115386   \n",
       "APEX1-Independent Resolution of AP Sites via th...               6.499152   \n",
       "APOBEC3G mediated resistance to HIV-1 infection                  7.291126   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                    0.501319   \n",
       "ARMS-mediated activation                                         4.448748   \n",
       "ATF4 activates genes in response to endoplasmic...              26.842697   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                     22.126203   \n",
       "ATF6 (ATF6-alpha) activates chaperones                          23.422288   \n",
       "...                                                                   ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...               0.886980   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...               0.857368   \n",
       "p130Cas linkage to MAPK signaling for integrins                  5.586869   \n",
       "p38MAPK events                                                   3.517022   \n",
       "p53-Dependent G1 DNA Damage Response                             2.126589   \n",
       "p53-Dependent G1/S DNA damage checkpoint                         2.126589   \n",
       "p53-Independent DNA Damage Response                              1.279853   \n",
       "p53-Independent G1/S DNA damage checkpoint                       1.279853   \n",
       "p75 NTR receptor-mediated signalling                             8.984090   \n",
       "p75NTR negatively regulates cell cycle via SC1                   1.773138   \n",
       "p75NTR recruits signalling complexes                             4.369223   \n",
       "p75NTR regulates axonogenesis                                    1.670706   \n",
       "p75NTR signals via NF-kB                                         5.772325   \n",
       "phospho-PLA2 pathway                                             0.774850   \n",
       "phosphorylation site mutants of CTNNB1 are not ...              12.437981   \n",
       "rRNA modification in the mitochondrion                          11.603468   \n",
       "rRNA modification in the nucleus and cytosol                    11.131025   \n",
       "rRNA processing                                                  1.020705   \n",
       "rRNA processing in the mitochondrion                            16.264847   \n",
       "rRNA processing in the nucleus and cytosol                       1.010827   \n",
       "snRNP Assembly                                                  35.918982   \n",
       "tRNA Aminoacylation                                             20.240820   \n",
       "tRNA modification in the mitochondrion                           3.171743   \n",
       "tRNA modification in the nucleus and cytosol                     5.430805   \n",
       "tRNA processing                                                 40.029946   \n",
       "tRNA processing in the mitochondrion                            10.780951   \n",
       "tRNA processing in the nucleus                                  33.064652   \n",
       "trans-Golgi Network Vesicle Budding                              5.478459   \n",
       "truncated APC mutants destabilize the destructi...              14.395657   \n",
       "vRNP Assembly                                                   13.511153   \n",
       "\n",
       "                                                    Cluster 4ER+ q-values  \\\n",
       "2-LTR circle formation                                           8.211725   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                       5.861441   \n",
       "A tetrasaccharide linker sequence is required f...               7.970056   \n",
       "ABC transporter disorders                                        1.148691   \n",
       "ABC transporters in lipid homeostasis                            3.069571   \n",
       "ABC-family proteins mediated transport                           1.544301   \n",
       "ABO blood group biosynthesis                                     4.988857   \n",
       "ADP signalling through P2Y purinoceptor 1                       11.291991   \n",
       "ADP signalling through P2Y purinoceptor 12                      15.928278   \n",
       "AKT phosphorylates targets in the cytosol                        3.694600   \n",
       "AKT phosphorylates targets in the nucleus                        6.907230   \n",
       "AKT-mediated inactivation of FOXO1A                              4.404751   \n",
       "ALKBH3 mediated reversal of alkylation damage                    3.386840   \n",
       "AMER1 mutants destabilize the destruction complex               13.742549   \n",
       "AMPK inhibits chREBP transcriptional activation...               5.359532   \n",
       "APC truncation mutants have impaired AXIN binding               13.742549   \n",
       "APC-Cdc20 mediated degradation of Nek2A                         38.264261   \n",
       "APC/C-mediated degradation of cell cycle proteins               11.583798   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                    39.930493   \n",
       "APC/C:Cdc20 mediated degradation of Securin                      1.871651   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...               5.549094   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...               4.037942   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...               3.872662   \n",
       "APEX1-Independent Resolution of AP Sites via th...               6.158234   \n",
       "APOBEC3G mediated resistance to HIV-1 infection                  6.924194   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                    0.477209   \n",
       "ARMS-mediated activation                                         4.193769   \n",
       "ATF4 activates genes in response to endoplasmic...              25.763934   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                     21.185971   \n",
       "ATF6 (ATF6-alpha) activates chaperones                          22.437918   \n",
       "...                                                                   ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...               0.823266   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...               0.796073   \n",
       "p130Cas linkage to MAPK signaling for integrins                  5.278661   \n",
       "p38MAPK events                                                   3.301961   \n",
       "p53-Dependent G1 DNA Damage Response                             1.978278   \n",
       "p53-Dependent G1/S DNA damage checkpoint                         1.978278   \n",
       "p53-Independent DNA Damage Response                              1.184980   \n",
       "p53-Independent G1/S DNA damage checkpoint                       1.184980   \n",
       "p75 NTR receptor-mediated signalling                             8.557143   \n",
       "p75NTR negatively regulates cell cycle via SC1                   1.643850   \n",
       "p75NTR recruits signalling complexes                             4.117620   \n",
       "p75NTR regulates axonogenesis                                    1.549243   \n",
       "p75NTR signals via NF-kB                                         5.458114   \n",
       "phospho-PLA2 pathway                                             0.722386   \n",
       "phosphorylation site mutants of CTNNB1 are not ...              11.858976   \n",
       "rRNA modification in the mitochondrion                          11.058981   \n",
       "rRNA modification in the nucleus and cytosol                    10.605907   \n",
       "rRNA processing                                                  0.945443   \n",
       "rRNA processing in the mitochondrion                            15.535750   \n",
       "rRNA processing in the nucleus and cytosol                       0.936062   \n",
       "snRNP Assembly                                                  34.593400   \n",
       "tRNA Aminoacylation                                             19.363344   \n",
       "tRNA modification in the mitochondrion                           2.971210   \n",
       "tRNA modification in the nucleus and cytosol                     5.131447   \n",
       "tRNA processing                                                 38.551986   \n",
       "tRNA processing in the mitochondrion                            10.275047   \n",
       "tRNA processing in the nucleus                                  31.827025   \n",
       "trans-Golgi Network Vesicle Budding                              5.175750   \n",
       "truncated APC mutants destabilize the destructi...              13.742549   \n",
       "vRNP Assembly                                                   12.893314   \n",
       "\n",
       "                                                           ...          \\\n",
       "2-LTR circle formation                                     ...           \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                 ...           \n",
       "A tetrasaccharide linker sequence is required f...         ...           \n",
       "ABC transporter disorders                                  ...           \n",
       "ABC transporters in lipid homeostasis                      ...           \n",
       "ABC-family proteins mediated transport                     ...           \n",
       "ABO blood group biosynthesis                               ...           \n",
       "ADP signalling through P2Y purinoceptor 1                  ...           \n",
       "ADP signalling through P2Y purinoceptor 12                 ...           \n",
       "AKT phosphorylates targets in the cytosol                  ...           \n",
       "AKT phosphorylates targets in the nucleus                  ...           \n",
       "AKT-mediated inactivation of FOXO1A                        ...           \n",
       "ALKBH3 mediated reversal of alkylation damage              ...           \n",
       "AMER1 mutants destabilize the destruction complex          ...           \n",
       "AMPK inhibits chREBP transcriptional activation...         ...           \n",
       "APC truncation mutants have impaired AXIN binding          ...           \n",
       "APC-Cdc20 mediated degradation of Nek2A                    ...           \n",
       "APC/C-mediated degradation of cell cycle proteins          ...           \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B               ...           \n",
       "APC/C:Cdc20 mediated degradation of Securin                ...           \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...         ...           \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...         ...           \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...         ...           \n",
       "APEX1-Independent Resolution of AP Sites via th...         ...           \n",
       "APOBEC3G mediated resistance to HIV-1 infection            ...           \n",
       "ARL13B-mediated ciliary trafficking of INPP5E              ...           \n",
       "ARMS-mediated activation                                   ...           \n",
       "ATF4 activates genes in response to endoplasmic...         ...           \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                ...           \n",
       "ATF6 (ATF6-alpha) activates chaperones                     ...           \n",
       "...                                                        ...           \n",
       "mitochondrial fatty acid beta-oxidation of satu...         ...           \n",
       "mitochondrial fatty acid beta-oxidation of unsa...         ...           \n",
       "p130Cas linkage to MAPK signaling for integrins            ...           \n",
       "p38MAPK events                                             ...           \n",
       "p53-Dependent G1 DNA Damage Response                       ...           \n",
       "p53-Dependent G1/S DNA damage checkpoint                   ...           \n",
       "p53-Independent DNA Damage Response                        ...           \n",
       "p53-Independent G1/S DNA damage checkpoint                 ...           \n",
       "p75 NTR receptor-mediated signalling                       ...           \n",
       "p75NTR negatively regulates cell cycle via SC1             ...           \n",
       "p75NTR recruits signalling complexes                       ...           \n",
       "p75NTR regulates axonogenesis                              ...           \n",
       "p75NTR signals via NF-kB                                   ...           \n",
       "phospho-PLA2 pathway                                       ...           \n",
       "phosphorylation site mutants of CTNNB1 are not ...         ...           \n",
       "rRNA modification in the mitochondrion                     ...           \n",
       "rRNA modification in the nucleus and cytosol               ...           \n",
       "rRNA processing                                            ...           \n",
       "rRNA processing in the mitochondrion                       ...           \n",
       "rRNA processing in the nucleus and cytosol                 ...           \n",
       "snRNP Assembly                                             ...           \n",
       "tRNA Aminoacylation                                        ...           \n",
       "tRNA modification in the mitochondrion                     ...           \n",
       "tRNA modification in the nucleus and cytosol               ...           \n",
       "tRNA processing                                            ...           \n",
       "tRNA processing in the mitochondrion                       ...           \n",
       "tRNA processing in the nucleus                             ...           \n",
       "trans-Golgi Network Vesicle Budding                        ...           \n",
       "truncated APC mutants destabilize the destructi...         ...           \n",
       "vRNP Assembly                                              ...           \n",
       "\n",
       "                                                    Cluster 5 p-values  \\\n",
       "2-LTR circle formation                                        5.889667   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.685590   \n",
       "A tetrasaccharide linker sequence is required f...            2.658408   \n",
       "ABC transporter disorders                                     2.193869   \n",
       "ABC transporters in lipid homeostasis                        24.876615   \n",
       "ABC-family proteins mediated transport                        2.240816   \n",
       "ABO blood group biosynthesis                                  0.720298   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.295572   \n",
       "ADP signalling through P2Y purinoceptor 12                    1.954865   \n",
       "AKT phosphorylates targets in the cytosol                     3.167744   \n",
       "AKT phosphorylates targets in the nucleus                     1.462585   \n",
       "AKT-mediated inactivation of FOXO1A                           1.540024   \n",
       "ALKBH3 mediated reversal of alkylation damage                 9.320050   \n",
       "AMER1 mutants destabilize the destruction complex            11.145170   \n",
       "AMPK inhibits chREBP transcriptional activation...            6.082065   \n",
       "APC truncation mutants have impaired AXIN binding            11.145964   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      11.843548   \n",
       "APC/C-mediated degradation of cell cycle proteins             9.833084   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 14.911636   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   3.177552   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            5.665495   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            5.184148   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            4.509152   \n",
       "APEX1-Independent Resolution of AP Sites via th...            9.648178   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              13.041273   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.370428   \n",
       "ARMS-mediated activation                                      3.099961   \n",
       "ATF4 activates genes in response to endoplasmic...            0.442169   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.680348   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.350750   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            7.448810   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            4.155106   \n",
       "p130Cas linkage to MAPK signaling for integrins              27.113438   \n",
       "p38MAPK events                                                2.522183   \n",
       "p53-Dependent G1 DNA Damage Response                          3.150571   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      3.150571   \n",
       "p53-Independent DNA Damage Response                           2.499587   \n",
       "p53-Independent G1/S DNA damage checkpoint                    2.499587   \n",
       "p75 NTR receptor-mediated signalling                         12.335284   \n",
       "p75NTR negatively regulates cell cycle via SC1                6.993605   \n",
       "p75NTR recruits signalling complexes                          5.524764   \n",
       "p75NTR regulates axonogenesis                                 0.517298   \n",
       "p75NTR signals via NF-kB                                      5.680145   \n",
       "phospho-PLA2 pathway                                          1.930719   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            9.043403   \n",
       "rRNA modification in the mitochondrion                        2.081064   \n",
       "rRNA modification in the nucleus and cytosol                  8.027170   \n",
       "rRNA processing                                               2.725126   \n",
       "rRNA processing in the mitochondrion                          1.333103   \n",
       "rRNA processing in the nucleus and cytosol                    2.780706   \n",
       "snRNP Assembly                                                3.524237   \n",
       "tRNA Aminoacylation                                          12.059921   \n",
       "tRNA modification in the mitochondrion                        0.477717   \n",
       "tRNA modification in the nucleus and cytosol                  2.644672   \n",
       "tRNA processing                                               7.331414   \n",
       "tRNA processing in the mitochondrion                          2.216781   \n",
       "tRNA processing in the nucleus                                1.930719   \n",
       "trans-Golgi Network Vesicle Budding                           0.536009   \n",
       "truncated APC mutants destabilize the destructi...           11.145567   \n",
       "vRNP Assembly                                                 1.673513   \n",
       "\n",
       "                                                    Cluster 5 q-values  \\\n",
       "2-LTR circle formation                                        5.319493   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.609081   \n",
       "A tetrasaccharide linker sequence is required f...            2.334594   \n",
       "ABC transporter disorders                                     1.918848   \n",
       "ABC transporters in lipid homeostasis                        23.161867   \n",
       "ABC-family proteins mediated transport                        1.962628   \n",
       "ABO blood group biosynthesis                                  0.638765   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.127437   \n",
       "ADP signalling through P2Y purinoceptor 12                    1.707354   \n",
       "AKT phosphorylates targets in the cytosol                     2.790954   \n",
       "AKT phosphorylates targets in the nucleus                     1.269627   \n",
       "AKT-mediated inactivation of FOXO1A                           1.338493   \n",
       "ALKBH3 mediated reversal of alkylation damage                 8.521755   \n",
       "AMER1 mutants destabilize the destruction complex            10.280147   \n",
       "AMPK inhibits chREBP transcriptional activation...            5.498453   \n",
       "APC truncation mutants have impaired AXIN binding            10.280147   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      10.924680   \n",
       "APC/C-mediated degradation of cell cycle proteins             9.015962   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 13.842801   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   2.798763   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            5.114355   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            4.667369   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            4.024213   \n",
       "APEX1-Independent Resolution of AP Sites via th...            8.839225   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              12.066887   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.192553   \n",
       "ARMS-mediated activation                                      2.730096   \n",
       "ATF4 activates genes in response to endoplasmic...            0.413162   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.604837   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.338216   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            6.769171   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            3.702215   \n",
       "p130Cas linkage to MAPK signaling for integrins              25.211603   \n",
       "p38MAPK events                                                2.211412   \n",
       "p53-Dependent G1 DNA Damage Response                          2.776266   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      2.776266   \n",
       "p53-Independent DNA Damage Response                           2.192230   \n",
       "p53-Independent G1/S DNA damage checkpoint                    2.192230   \n",
       "p75 NTR receptor-mediated signalling                         11.391387   \n",
       "p75NTR negatively regulates cell cycle via SC1                6.340497   \n",
       "p75NTR recruits signalling complexes                          4.981741   \n",
       "p75NTR regulates axonogenesis                                 0.473491   \n",
       "p75NTR signals via NF-kB                                      5.127512   \n",
       "phospho-PLA2 pathway                                          1.687544   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            8.261889   \n",
       "rRNA modification in the mitochondrion                        1.818485   \n",
       "rRNA modification in the nucleus and cytosol                  7.311334   \n",
       "rRNA processing                                               2.395537   \n",
       "rRNA processing in the mitochondrion                          1.159934   \n",
       "rRNA processing in the nucleus and cytosol                    2.446620   \n",
       "snRNP Assembly                                                3.118479   \n",
       "tRNA Aminoacylation                                          11.130502   \n",
       "tRNA modification in the mitochondrion                        0.443303   \n",
       "tRNA modification in the nucleus and cytosol                  2.323060   \n",
       "tRNA processing                                               6.661009   \n",
       "tRNA processing in the mitochondrion                          1.940179   \n",
       "tRNA processing in the nucleus                                1.687544   \n",
       "trans-Golgi Network Vesicle Budding                           0.488012   \n",
       "truncated APC mutants destabilize the destructi...           10.280147   \n",
       "vRNP Assembly                                                 1.455002   \n",
       "\n",
       "                                                    Cluster 6 p-values  \\\n",
       "2-LTR circle formation                                        0.336657   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    1.176011   \n",
       "A tetrasaccharide linker sequence is required f...            0.350185   \n",
       "ABC transporter disorders                                     0.344877   \n",
       "ABC transporters in lipid homeostasis                         4.569565   \n",
       "ABC-family proteins mediated transport                        0.375231   \n",
       "ABO blood group biosynthesis                                  4.837029   \n",
       "ADP signalling through P2Y purinoceptor 1                     4.455846   \n",
       "ADP signalling through P2Y purinoceptor 12                    5.777599   \n",
       "AKT phosphorylates targets in the cytosol                     0.824871   \n",
       "AKT phosphorylates targets in the nucleus                     0.534124   \n",
       "AKT-mediated inactivation of FOXO1A                           0.421307   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.978490   \n",
       "AMER1 mutants destabilize the destruction complex             0.314112   \n",
       "AMPK inhibits chREBP transcriptional activation...            4.526513   \n",
       "APC truncation mutants have impaired AXIN binding             0.314043   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       3.876048   \n",
       "APC/C-mediated degradation of cell cycle proteins             0.918796   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  3.664181   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.406612   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            0.641753   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            0.540742   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            0.526877   \n",
       "APEX1-Independent Resolution of AP Sites via th...            1.377732   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               2.395101   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.345023   \n",
       "ARMS-mediated activation                                      2.280779   \n",
       "ATF4 activates genes in response to endoplasmic...            4.946292   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.333206   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.597005   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.441623   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.962413   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.824638   \n",
       "p38MAPK events                                                1.760402   \n",
       "p53-Dependent G1 DNA Damage Response                          0.311505   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.311505   \n",
       "p53-Independent DNA Damage Response                           0.383212   \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.383212   \n",
       "p75 NTR receptor-mediated signalling                          0.997443   \n",
       "p75NTR negatively regulates cell cycle via SC1                2.007499   \n",
       "p75NTR recruits signalling complexes                          1.637828   \n",
       "p75NTR regulates axonogenesis                                 2.930041   \n",
       "p75NTR signals via NF-kB                                      1.442989   \n",
       "phospho-PLA2 pathway                                          1.014991   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.515455   \n",
       "rRNA modification in the mitochondrion                        1.396577   \n",
       "rRNA modification in the nucleus and cytosol                  0.779591   \n",
       "rRNA processing                                               1.018255   \n",
       "rRNA processing in the mitochondrion                          1.627142   \n",
       "rRNA processing in the nucleus and cytosol                    1.018850   \n",
       "snRNP Assembly                                                1.871249   \n",
       "tRNA Aminoacylation                                           0.470952   \n",
       "tRNA modification in the mitochondrion                        0.438509   \n",
       "tRNA modification in the nucleus and cytosol                  0.988084   \n",
       "tRNA processing                                               5.027523   \n",
       "tRNA processing in the mitochondrion                          0.965501   \n",
       "tRNA processing in the nucleus                                6.113450   \n",
       "trans-Golgi Network Vesicle Budding                           1.201339   \n",
       "truncated APC mutants destabilize the destructi...            0.314112   \n",
       "vRNP Assembly                                                 1.218187   \n",
       "\n",
       "                                                    Cluster 6 q-values  \\\n",
       "2-LTR circle formation                                        0.323548   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.938421   \n",
       "A tetrasaccharide linker sequence is required f...            0.332666   \n",
       "ABC transporter disorders                                     0.329532   \n",
       "ABC transporters in lipid homeostasis                         3.576682   \n",
       "ABC-family proteins mediated transport                        0.348679   \n",
       "ABO blood group biosynthesis                                  3.773074   \n",
       "ADP signalling through P2Y purinoceptor 1                     3.483429   \n",
       "ADP signalling through P2Y purinoceptor 12                    4.481980   \n",
       "AKT phosphorylates targets in the cytosol                     0.669130   \n",
       "AKT phosphorylates targets in the nucleus                     0.446741   \n",
       "AKT-mediated inactivation of FOXO1A                           0.371670   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.787487   \n",
       "AMER1 mutants destabilize the destruction complex             0.308352   \n",
       "AMPK inhibits chREBP transcriptional activation...            3.542143   \n",
       "APC truncation mutants have impaired AXIN binding             0.308352   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       3.015606   \n",
       "APC/C-mediated degradation of cell cycle proteins             0.740606   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  2.852522   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.361877   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            0.527160   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            0.451820   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            0.443318   \n",
       "APEX1-Independent Resolution of AP Sites via th...            1.101675   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               1.906705   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.329532   \n",
       "ARMS-mediated activation                                      1.814003   \n",
       "ATF4 activates genes in response to endoplasmic...            3.863894   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.320888   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.493980   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.384169   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.557827   \n",
       "p130Cas linkage to MAPK signaling for integrins               2.222997   \n",
       "p38MAPK events                                                1.393751   \n",
       "p53-Dependent G1 DNA Damage Response                          0.306874   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.306874   \n",
       "p53-Independent DNA Damage Response                           0.351960   \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.351960   \n",
       "p75 NTR receptor-mediated signalling                          0.802703   \n",
       "p75NTR negatively regulates cell cycle via SC1                1.597563   \n",
       "p75NTR recruits signalling complexes                          1.299199   \n",
       "p75NTR regulates axonogenesis                                 2.295100   \n",
       "p75NTR signals via NF-kB                                      1.150666   \n",
       "phospho-PLA2 pathway                                          0.817438   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.436943   \n",
       "rRNA modification in the mitochondrion                        1.116217   \n",
       "rRNA modification in the nucleus and cytosol                  0.632750   \n",
       "rRNA processing                                               0.818901   \n",
       "rRNA processing in the mitochondrion                          1.293508   \n",
       "rRNA processing in the nucleus and cytosol                    0.818981   \n",
       "snRNP Assembly                                                1.486914   \n",
       "tRNA Aminoacylation                                           0.404870   \n",
       "tRNA modification in the mitochondrion                        0.382555   \n",
       "tRNA modification in the nucleus and cytosol                  0.795779   \n",
       "tRNA processing                                               3.915660   \n",
       "tRNA processing in the mitochondrion                          0.777772   \n",
       "tRNA processing in the nucleus                                4.731140   \n",
       "trans-Golgi Network Vesicle Budding                           0.958249   \n",
       "truncated APC mutants destabilize the destructi...            0.308352   \n",
       "vRNP Assembly                                                 0.971786   \n",
       "\n",
       "                                                    Cluster 7 p-values  \\\n",
       "2-LTR circle formation                                        1.601092   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    3.107078   \n",
       "A tetrasaccharide linker sequence is required f...            2.683999   \n",
       "ABC transporter disorders                                     6.063705   \n",
       "ABC transporters in lipid homeostasis                        17.352672   \n",
       "ABC-family proteins mediated transport                        6.529664   \n",
       "ABO blood group biosynthesis                                  2.530621   \n",
       "ADP signalling through P2Y purinoceptor 1                     0.618850   \n",
       "ADP signalling through P2Y purinoceptor 12                    0.919203   \n",
       "AKT phosphorylates targets in the cytosol                     2.161445   \n",
       "AKT phosphorylates targets in the nucleus                     2.001840   \n",
       "AKT-mediated inactivation of FOXO1A                           1.669095   \n",
       "ALKBH3 mediated reversal of alkylation damage                 2.515340   \n",
       "AMER1 mutants destabilize the destruction complex             0.762996   \n",
       "AMPK inhibits chREBP transcriptional activation...            7.904280   \n",
       "APC truncation mutants have impaired AXIN binding             0.762996   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       6.165634   \n",
       "APC/C-mediated degradation of cell cycle proteins             9.757787   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  4.706908   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   6.043087   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            7.517541   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            7.374682   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            7.066893   \n",
       "APEX1-Independent Resolution of AP Sites via th...            4.271260   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               4.625958   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.529787   \n",
       "ARMS-mediated activation                                      0.451066   \n",
       "ATF4 activates genes in response to endoplasmic...            5.928387   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   1.203328   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.961810   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.345973   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.657697   \n",
       "p130Cas linkage to MAPK signaling for integrins               4.526958   \n",
       "p38MAPK events                                                7.862333   \n",
       "p53-Dependent G1 DNA Damage Response                          6.576002   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      6.576002   \n",
       "p53-Independent DNA Damage Response                           5.742845   \n",
       "p53-Independent G1/S DNA damage checkpoint                    5.742845   \n",
       "p75 NTR receptor-mediated signalling                          7.191423   \n",
       "p75NTR negatively regulates cell cycle via SC1                6.314398   \n",
       "p75NTR recruits signalling complexes                          6.094992   \n",
       "p75NTR regulates axonogenesis                                 2.198895   \n",
       "p75NTR signals via NF-kB                                      6.979677   \n",
       "phospho-PLA2 pathway                                          0.713007   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.416272   \n",
       "rRNA modification in the mitochondrion                        4.459597   \n",
       "rRNA modification in the nucleus and cytosol                  1.631546   \n",
       "rRNA processing                                               0.674974   \n",
       "rRNA processing in the mitochondrion                          3.736491   \n",
       "rRNA processing in the nucleus and cytosol                    0.653134   \n",
       "snRNP Assembly                                                8.966442   \n",
       "tRNA Aminoacylation                                           4.014184   \n",
       "tRNA modification in the mitochondrion                        1.882957   \n",
       "tRNA modification in the nucleus and cytosol                  1.655744   \n",
       "tRNA processing                                              10.855245   \n",
       "tRNA processing in the mitochondrion                          1.694317   \n",
       "tRNA processing in the nucleus                                5.170739   \n",
       "trans-Golgi Network Vesicle Budding                           5.059475   \n",
       "truncated APC mutants destabilize the destructi...            0.762996   \n",
       "vRNP Assembly                                                 1.383214   \n",
       "\n",
       "                                                    Cluster 7 q-values  \\\n",
       "2-LTR circle formation                                        1.461820   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    2.852851   \n",
       "A tetrasaccharide linker sequence is required f...            2.454971   \n",
       "ABC transporter disorders                                     5.638434   \n",
       "ABC transporters in lipid homeostasis                        16.035863   \n",
       "ABC-family proteins mediated transport                        6.075583   \n",
       "ABO blood group biosynthesis                                  2.310722   \n",
       "ADP signalling through P2Y purinoceptor 1                     0.566623   \n",
       "ADP signalling through P2Y purinoceptor 12                    0.832588   \n",
       "AKT phosphorylates targets in the cytosol                     1.969467   \n",
       "AKT phosphorylates targets in the nucleus                     1.817931   \n",
       "AKT-mediated inactivation of FOXO1A                           1.522548   \n",
       "ALKBH3 mediated reversal of alkylation damage                 2.296135   \n",
       "AMER1 mutants destabilize the destruction complex             0.694406   \n",
       "AMPK inhibits chREBP transcriptional activation...            7.349399   \n",
       "APC truncation mutants have impaired AXIN binding             0.694406   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       5.735880   \n",
       "APC/C-mediated degradation of cell cycle proteins             9.070388   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                  4.363689   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   5.620040   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            6.988900   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            6.859326   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            6.576807   \n",
       "APEX1-Independent Resolution of AP Sites via th...            3.949208   \n",
       "APOBEC3G mediated resistance to HIV-1 infection               4.286873   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.395108   \n",
       "ARMS-mediated activation                                      0.422283   \n",
       "ATF4 activates genes in response to endoplasmic...            5.511399   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   1.092805   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.870052   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            0.337088   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.600244   \n",
       "p130Cas linkage to MAPK signaling for integrins               4.193738   \n",
       "p38MAPK events                                                7.309700   \n",
       "p53-Dependent G1 DNA Damage Response                          6.117731   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      6.117731   \n",
       "p53-Independent DNA Damage Response                           5.338259   \n",
       "p53-Independent G1/S DNA damage checkpoint                    5.338259   \n",
       "p75 NTR receptor-mediated signalling                          6.690856   \n",
       "p75NTR negatively regulates cell cycle via SC1                5.876111   \n",
       "p75NTR recruits signalling complexes                          5.667485   \n",
       "p75NTR regulates axonogenesis                                 2.003646   \n",
       "p75NTR signals via NF-kB                                      6.492815   \n",
       "phospho-PLA2 pathway                                          0.649294   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.395880   \n",
       "rRNA modification in the mitochondrion                        4.130008   \n",
       "rRNA modification in the nucleus and cytosol                  1.489088   \n",
       "rRNA processing                                               0.615844   \n",
       "rRNA processing in the mitochondrion                          3.445807   \n",
       "rRNA processing in the nucleus and cytosol                    0.596403   \n",
       "snRNP Assembly                                                8.330874   \n",
       "tRNA Aminoacylation                                           3.705549   \n",
       "tRNA modification in the mitochondrion                        1.710411   \n",
       "tRNA modification in the nucleus and cytosol                  1.510369   \n",
       "tRNA processing                                              10.082504   \n",
       "tRNA processing in the mitochondrion                          1.544825   \n",
       "tRNA processing in the nucleus                                4.804295   \n",
       "trans-Golgi Network Vesicle Budding                           4.699315   \n",
       "truncated APC mutants destabilize the destructi...            0.694406   \n",
       "vRNP Assembly                                                 1.258135   \n",
       "\n",
       "                                                    Cluster 8 p-values  \\\n",
       "2-LTR circle formation                                        8.159628   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    2.044017   \n",
       "A tetrasaccharide linker sequence is required f...            1.929601   \n",
       "ABC transporter disorders                                    17.625445   \n",
       "ABC transporters in lipid homeostasis                        36.164804   \n",
       "ABC-family proteins mediated transport                       15.774023   \n",
       "ABO blood group biosynthesis                                  3.941170   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.117781   \n",
       "ADP signalling through P2Y purinoceptor 12                    2.683503   \n",
       "AKT phosphorylates targets in the cytosol                     4.162660   \n",
       "AKT phosphorylates targets in the nucleus                     2.765016   \n",
       "AKT-mediated inactivation of FOXO1A                           3.719748   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.499449   \n",
       "AMER1 mutants destabilize the destruction complex             1.707118   \n",
       "AMPK inhibits chREBP transcriptional activation...            6.659763   \n",
       "APC truncation mutants have impaired AXIN binding             1.707233   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       8.950053   \n",
       "APC/C-mediated degradation of cell cycle proteins            22.318410   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 14.351334   \n",
       "APC/C:Cdc20 mediated degradation of Securin                  18.291677   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...           19.137797   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...           20.751335   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...           18.317949   \n",
       "APEX1-Independent Resolution of AP Sites via th...           45.525588   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              33.272007   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.981842   \n",
       "ARMS-mediated activation                                      1.512565   \n",
       "ATF4 activates genes in response to endoplasmic...            3.715069   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                  39.566115   \n",
       "ATF6 (ATF6-alpha) activates chaperones                       35.982597   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            2.048751   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.343126   \n",
       "p130Cas linkage to MAPK signaling for integrins              14.850274   \n",
       "p38MAPK events                                                7.388025   \n",
       "p53-Dependent G1 DNA Damage Response                         19.476739   \n",
       "p53-Dependent G1/S DNA damage checkpoint                     19.476739   \n",
       "p53-Independent DNA Damage Response                          17.009579   \n",
       "p53-Independent G1/S DNA damage checkpoint                   17.009579   \n",
       "p75 NTR receptor-mediated signalling                         22.014209   \n",
       "p75NTR negatively regulates cell cycle via SC1                2.089701   \n",
       "p75NTR recruits signalling complexes                         22.692936   \n",
       "p75NTR regulates axonogenesis                                 1.853008   \n",
       "p75NTR signals via NF-kB                                     24.891425   \n",
       "phospho-PLA2 pathway                                          0.459800   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.545972   \n",
       "rRNA modification in the mitochondrion                       11.043109   \n",
       "rRNA modification in the nucleus and cytosol                  0.416904   \n",
       "rRNA processing                                               1.350553   \n",
       "rRNA processing in the mitochondrion                         11.916951   \n",
       "rRNA processing in the nucleus and cytosol                    1.322648   \n",
       "snRNP Assembly                                                2.833084   \n",
       "tRNA Aminoacylation                                           2.740870   \n",
       "tRNA modification in the mitochondrion                        0.326855   \n",
       "tRNA modification in the nucleus and cytosol                  2.451006   \n",
       "tRNA processing                                               7.876478   \n",
       "tRNA processing in the mitochondrion                          0.852926   \n",
       "tRNA processing in the nucleus                                5.944671   \n",
       "trans-Golgi Network Vesicle Budding                           3.995115   \n",
       "truncated APC mutants destabilize the destructi...            1.707233   \n",
       "vRNP Assembly                                                 0.576505   \n",
       "\n",
       "                                                    Cluster 8 q-values  \\\n",
       "2-LTR circle formation                                        7.821457   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    1.911900   \n",
       "A tetrasaccharide linker sequence is required f...            1.804243   \n",
       "ABC transporter disorders                                    17.020443   \n",
       "ABC transporters in lipid homeostasis                        35.018257   \n",
       "ABC-family proteins mediated transport                       15.239687   \n",
       "ABO blood group biosynthesis                                  3.726109   \n",
       "ADP signalling through P2Y purinoceptor 1                     1.040433   \n",
       "ADP signalling through P2Y purinoceptor 12                    2.523849   \n",
       "AKT phosphorylates targets in the cytosol                     3.941370   \n",
       "AKT phosphorylates targets in the nucleus                     2.602022   \n",
       "AKT-mediated inactivation of FOXO1A                           3.514202   \n",
       "ALKBH3 mediated reversal of alkylation damage                 0.476445   \n",
       "AMER1 mutants destabilize the destruction complex             1.596865   \n",
       "AMPK inhibits chREBP transcriptional activation...            6.354947   \n",
       "APC truncation mutants have impaired AXIN binding             1.596865   \n",
       "APC-Cdc20 mediated degradation of Nek2A                       8.591809   \n",
       "APC/C-mediated degradation of cell cycle proteins            21.556696   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 13.857347   \n",
       "APC/C:Cdc20 mediated degradation of Securin                  17.667715   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...           18.480904   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...           20.051527   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...           17.693106   \n",
       "APEX1-Independent Resolution of AP Sites via th...           43.986930   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              32.231660   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 0.916185   \n",
       "ARMS-mediated activation                                      1.410070   \n",
       "ATF4 activates genes in response to endoplasmic...            3.509858   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                  38.310005   \n",
       "ATF6 (ATF6-alpha) activates chaperones                       34.844766   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            1.916066   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            0.338918   \n",
       "p130Cas linkage to MAPK signaling for integrins              14.342351   \n",
       "p38MAPK events                                                7.068603   \n",
       "p53-Dependent G1 DNA Damage Response                         18.814107   \n",
       "p53-Dependent G1/S DNA damage checkpoint                     18.814107   \n",
       "p53-Independent DNA Damage Response                          16.431551   \n",
       "p53-Independent G1/S DNA damage checkpoint                   16.431551   \n",
       "p75 NTR receptor-mediated signalling                         21.263248   \n",
       "p75NTR negatively regulates cell cycle via SC1                1.955021   \n",
       "p75NTR recruits signalling complexes                         21.915204   \n",
       "p75NTR regulates axonogenesis                                 1.731545   \n",
       "p75NTR signals via NF-kB                                     24.060338   \n",
       "phospho-PLA2 pathway                                          0.440971   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            0.518360   \n",
       "rRNA modification in the mitochondrion                       10.628847   \n",
       "rRNA modification in the nucleus and cytosol                  0.405661   \n",
       "rRNA processing                                               1.258537   \n",
       "rRNA processing in the mitochondrion                         11.478664   \n",
       "rRNA processing in the nucleus and cytosol                    1.232241   \n",
       "snRNP Assembly                                                2.666417   \n",
       "tRNA Aminoacylation                                           2.578790   \n",
       "tRNA modification in the mitochondrion                        0.324545   \n",
       "tRNA modification in the nucleus and cytosol                  2.300034   \n",
       "tRNA processing                                               7.545545   \n",
       "tRNA processing in the mitochondrion                          0.796904   \n",
       "tRNA processing in the nucleus                                5.661688   \n",
       "trans-Golgi Network Vesicle Budding                           3.778677   \n",
       "truncated APC mutants destabilize the destructi...            1.596865   \n",
       "vRNP Assembly                                                 0.544803   \n",
       "\n",
       "                                                    Cluster 9 p-values  \\\n",
       "2-LTR circle formation                                        2.886596   \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.935861   \n",
       "A tetrasaccharide linker sequence is required f...            5.047063   \n",
       "ABC transporter disorders                                     0.476183   \n",
       "ABC transporters in lipid homeostasis                         1.424468   \n",
       "ABC-family proteins mediated transport                        0.453226   \n",
       "ABO blood group biosynthesis                                  0.410581   \n",
       "ADP signalling through P2Y purinoceptor 1                     5.692188   \n",
       "ADP signalling through P2Y purinoceptor 12                    6.344734   \n",
       "AKT phosphorylates targets in the cytosol                     0.924645   \n",
       "AKT phosphorylates targets in the nucleus                     1.414306   \n",
       "AKT-mediated inactivation of FOXO1A                           0.793230   \n",
       "ALKBH3 mediated reversal of alkylation damage                 3.176725   \n",
       "AMER1 mutants destabilize the destruction complex             6.056340   \n",
       "AMPK inhibits chREBP transcriptional activation...            7.099057   \n",
       "APC truncation mutants have impaired AXIN binding             6.056340   \n",
       "APC-Cdc20 mediated degradation of Nek2A                      14.380475   \n",
       "APC/C-mediated degradation of cell cycle proteins             3.388987   \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 15.938383   \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.666720   \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            1.550581   \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            1.331734   \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            1.081649   \n",
       "APEX1-Independent Resolution of AP Sites via th...            0.483647   \n",
       "APOBEC3G mediated resistance to HIV-1 infection              10.549802   \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.741873   \n",
       "ARMS-mediated activation                                      5.592169   \n",
       "ATF4 activates genes in response to endoplasmic...           13.446183   \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.486008   \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.738903   \n",
       "...                                                                ...   \n",
       "mitochondrial fatty acid beta-oxidation of satu...            1.577808   \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.287907   \n",
       "p130Cas linkage to MAPK signaling for integrins               0.416824   \n",
       "p38MAPK events                                                2.961988   \n",
       "p53-Dependent G1 DNA Damage Response                          0.547632   \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.547632   \n",
       "p53-Independent DNA Damage Response                           0.399163   \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.399163   \n",
       "p75 NTR receptor-mediated signalling                          2.036557   \n",
       "p75NTR negatively regulates cell cycle via SC1                6.926040   \n",
       "p75NTR recruits signalling complexes                          4.456849   \n",
       "p75NTR regulates axonogenesis                                 3.167927   \n",
       "p75NTR signals via NF-kB                                      6.025390   \n",
       "phospho-PLA2 pathway                                          1.653140   \n",
       "phosphorylation site mutants of CTNNB1 are not ...            5.771768   \n",
       "rRNA modification in the mitochondrion                        5.907865   \n",
       "rRNA modification in the nucleus and cytosol                  3.526579   \n",
       "rRNA processing                                               0.597357   \n",
       "rRNA processing in the mitochondrion                          5.255992   \n",
       "rRNA processing in the nucleus and cytosol                    0.576181   \n",
       "snRNP Assembly                                                7.478763   \n",
       "tRNA Aminoacylation                                           5.738798   \n",
       "tRNA modification in the mitochondrion                        2.707477   \n",
       "tRNA modification in the nucleus and cytosol                  7.593586   \n",
       "tRNA processing                                              17.371009   \n",
       "tRNA processing in the mitochondrion                          0.658380   \n",
       "tRNA processing in the nucleus                               14.587067   \n",
       "trans-Golgi Network Vesicle Budding                           2.790866   \n",
       "truncated APC mutants destabilize the destructi...            6.056340   \n",
       "vRNP Assembly                                                 1.716624   \n",
       "\n",
       "                                                    Cluster 9 q-values  \n",
       "2-LTR circle formation                                        2.479343  \n",
       "5-Phosphoribose 1-diphosphate biosynthesis                    0.791947  \n",
       "A tetrasaccharide linker sequence is required f...            4.435262  \n",
       "ABC transporter disorders                                     0.425300  \n",
       "ABC transporters in lipid homeostasis                         1.211121  \n",
       "ABC-family proteins mediated transport                        0.408027  \n",
       "ABO blood group biosynthesis                                  0.375487  \n",
       "ADP signalling through P2Y purinoceptor 1                     5.035296  \n",
       "ADP signalling through P2Y purinoceptor 12                    5.614512  \n",
       "AKT phosphorylates targets in the cytosol                     0.783348  \n",
       "AKT phosphorylates targets in the nucleus                     1.203407  \n",
       "AKT-mediated inactivation of FOXO1A                           0.675628  \n",
       "ALKBH3 mediated reversal of alkylation damage                 2.745843  \n",
       "AMER1 mutants destabilize the destruction complex             5.363813  \n",
       "AMPK inhibits chREBP transcriptional activation...            6.284676  \n",
       "APC truncation mutants have impaired AXIN binding             5.363813  \n",
       "APC-Cdc20 mediated degradation of Nek2A                      12.665727  \n",
       "APC/C-mediated degradation of cell cycle proteins             2.943751  \n",
       "APC/C:Cdc20 mediated degradation of Cyclin B                 14.036548  \n",
       "APC/C:Cdc20 mediated degradation of Securin                   0.577283  \n",
       "APC/C:Cdc20 mediated degradation of mitotic pro...            1.322319  \n",
       "APC/C:Cdh1 mediated degradation of Cdc20 and ot...            1.131201  \n",
       "APC:Cdc20 mediated degradation of cell cycle pr...            0.917740  \n",
       "APEX1-Independent Resolution of AP Sites via th...            0.430947  \n",
       "APOBEC3G mediated resistance to HIV-1 infection               9.337594  \n",
       "ARL13B-mediated ciliary trafficking of INPP5E                 1.482350  \n",
       "ARMS-mediated activation                                      4.948381  \n",
       "ATF4 activates genes in response to endoplasmic...           11.921766  \n",
       "ATF6 (ATF6-alpha) activates chaperone genes                   0.432599  \n",
       "ATF6 (ATF6-alpha) activates chaperones                        0.631609  \n",
       "...                                                                ...  \n",
       "mitochondrial fatty acid beta-oxidation of satu...            1.344860  \n",
       "mitochondrial fatty acid beta-oxidation of unsa...            1.091672  \n",
       "p130Cas linkage to MAPK signaling for integrins               0.380236  \n",
       "p38MAPK events                                                2.546638  \n",
       "p53-Dependent G1 DNA Damage Response                          0.481974  \n",
       "p53-Dependent G1/S DNA damage checkpoint                      0.481974  \n",
       "p53-Independent DNA Damage Response                           0.369036  \n",
       "p53-Independent G1/S DNA damage checkpoint                    0.369036  \n",
       "p75 NTR receptor-mediated signalling                          1.735527  \n",
       "p75NTR negatively regulates cell cycle via SC1                6.138148  \n",
       "p75NTR recruits signalling complexes                          3.901216  \n",
       "p75NTR regulates axonogenesis                                 2.738173  \n",
       "p75NTR signals via NF-kB                                      5.334922  \n",
       "phospho-PLA2 pathway                                          1.407846  \n",
       "phosphorylation site mutants of CTNNB1 are not ...            5.109136  \n",
       "rRNA modification in the mitochondrion                        5.228546  \n",
       "rRNA modification in the nucleus and cytosol                  3.058575  \n",
       "rRNA processing                                               0.520184  \n",
       "rRNA processing in the mitochondrion                          4.629379  \n",
       "rRNA processing in the nucleus and cytosol                    0.502162  \n",
       "snRNP Assembly                                                6.619836  \n",
       "tRNA Aminoacylation                                           5.078087  \n",
       "tRNA modification in the mitochondrion                        2.320087  \n",
       "tRNA modification in the nucleus and cytosol                  6.720822  \n",
       "tRNA processing                                              15.309472  \n",
       "tRNA processing in the mitochondrion                          0.569715  \n",
       "tRNA processing in the nucleus                               12.860291  \n",
       "trans-Golgi Network Vesicle Budding                           2.395223  \n",
       "truncated APC mutants destabilize the destructi...            5.363813  \n",
       "vRNP Assembly                                                 1.461645  \n",
       "\n",
       "[2074 rows x 22 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from scipy.stats import ttest_ind, mannwhitneyu\n",
    "import qvalue as qv\n",
    "\n",
    "\n",
    "grouped_ttest = pd.DataFrame(index=full_df.iloc[:,:-33].columns)\n",
    "grouped_by_cluster = full_df.groupby('Integrative Cluster')\n",
    "\n",
    "for group in grouped_by_cluster:\n",
    "    df = group[1].iloc[:,:-33]\n",
    "    groupname = group[0]\n",
    "    group2_df = full_df[full_df['Integrative Cluster'] != groupname].iloc[:,:-33]\n",
    "    pvaluelist = []\n",
    "    for pathway in df:\n",
    "        group = df[pathway]\n",
    "        group2 = group2_df[pathway]\n",
    "        test = mannwhitneyu(group, group2)\n",
    "        pvaluelist.append(test[1])\n",
    "        \n",
    "    empty_df = pd.DataFrame(index=full_df.iloc[:,:-33].columns)\n",
    "    \n",
    "    empty_df[f'Cluster {groupname} p-values'] = pvaluelist\n",
    "    print(empty_df)\n",
    "    qv.qvalues(empty_df, f'Cluster {groupname} p-values', f'Cluster {groupname} q-values')\n",
    "    empty_df[f'Cluster {groupname} p-values'] = -np.log10(empty_df[f'Cluster {groupname} p-values'])\n",
    "    empty_df[f'Cluster {groupname} q-values'] = -np.log10(empty_df[f'Cluster {groupname} q-values'])\n",
    "    print(empty_df)\n",
    "    grouped_ttest = pd.concat([grouped_ttest, empty_df], axis = 1)    \n",
    "        \n",
    "grouped_ttest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cluster 1 p-values        17.792930\n",
       "Cluster 1 q-values        16.449249\n",
       "Cluster 10 p-values      105.193740\n",
       "Cluster 10 q-values      103.132204\n",
       "Cluster 2 p-values         1.675756\n",
       "Cluster 2 q-values         1.201556\n",
       "Cluster 3 p-values        45.929201\n",
       "Cluster 3 q-values        43.691573\n",
       "Cluster 4ER+ p-values     44.653553\n",
       "Cluster 4ER+ q-values     42.783902\n",
       "Cluster 4ER- p-values      1.033751\n",
       "Cluster 4ER- q-values      0.918066\n",
       "Cluster 5 p-values        19.062704\n",
       "Cluster 5 q-values        17.741685\n",
       "Cluster 6 p-values         3.170968\n",
       "Cluster 6 q-values         2.472208\n",
       "Cluster 7 p-values         8.998319\n",
       "Cluster 7 q-values         8.356786\n",
       "Cluster 8 p-values        15.567307\n",
       "Cluster 8 q-values        15.037958\n",
       "Cluster 9 p-values        13.357865\n",
       "Cluster 9 q-values        11.853969\n",
       "Name: Cell Cycle, dtype: float64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_ttest.loc['Cell Cycle']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster 1 qvalues    16.449249\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 10 qvalues    103.132204\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 2 qvalues    1.201556\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 3 qvalues    43.691573\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 4ER+ qvalues    42.783902\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 4ER- qvalues    0.918066\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 5 qvalues    17.741685\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 6 qvalues    2.472208\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 7 qvalues    8.356786\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 8 qvalues    15.037958\n",
      "Name: Cell Cycle, dtype: float64\n",
      "cluster 9 qvalues    11.853969\n",
      "Name: Cell Cycle, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "for i in clusterframes:\n",
    "    print(clusterframes[i].loc['Cell Cycle'].iloc[4:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_ttest.to_csv('../exp/pathway_qvalues.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_ttest.sort_values(by='Cluster 7 q-values', ascending=False).iloc[:30,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "10\n",
      "2\n",
      "3\n",
      "4ER+\n",
      "4ER-\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n"
     ]
    }
   ],
   "source": [
    "from scipy.stats import ttest_ind, mannwhitneyu\n",
    "import qvalue as qv\n",
    "\n",
    "clusterframes = {}\n",
    "\n",
    "grouped_by_cluster = full_df.groupby('Integrative Cluster')\n",
    "\n",
    "for group in grouped_by_cluster:\n",
    "    df_cluster = pd.DataFrame(index=full_df.iloc[:,:-33].columns)\n",
    "    groupname = group[0]\n",
    "    df = group[1].iloc[:,:-33]\n",
    "    group2_df = full_df[full_df['Integrative Cluster'] != groupname].iloc[:,:-33]\n",
    "    pvaluelist = []\n",
    "    group1_mean_list = []\n",
    "    group2_mean_list = []\n",
    "    for pathway in df:\n",
    "        group = df[pathway]\n",
    "        group2 = group2_df[pathway]\n",
    "        test = mannwhitneyu(group, group2)\n",
    "        pvaluelist.append(test[1])\n",
    "        group_mean = group.mean()\n",
    "        group1_mean_list.append(group_mean)\n",
    "        group2_mean = group2.mean()\n",
    "        group2_mean_list.append(group2_mean) \n",
    "        \n",
    "    \n",
    "    df_cluster[f'Cluster {groupname}'] = group1_mean_list\n",
    "    df_cluster['Other clusters'] = group2_mean_list\n",
    "    df_cluster['Fold Change'] = np.log2(abs(df_cluster[f'Cluster {groupname}'])) - np.log2(abs(df_cluster['Other clusters']))\n",
    "    \n",
    "    \n",
    "    df_cluster['p-values'] = pvaluelist\n",
    "    qv.qvalues(df_cluster, 'p-values', f'cluster {groupname} qvalues')\n",
    "    df_cluster['p-values'] = -np.log10(df_cluster['p-values'])\n",
    "    df_cluster[f'cluster {groupname} qvalues'] = -np.log10(df_cluster[f'cluster {groupname} qvalues'])\n",
    "    print(groupname)\n",
    "    clusterframes[groupname] = df_cluster\n",
    "        \n",
    "        \n",
    "#clusterframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "clusterframes['5'][clusterframes['5']['cluster 5 qvalues'] > 3].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in new_clinical.columns:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'full_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-851470bde2ab>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     20\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m \u001b[0mcluster_clin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Pam50 + Claudin-low subtype'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Integrative Cluster'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-1-851470bde2ab>\u001b[0m in \u001b[0;36mcluster_clin\u001b[0;34m(clinical_cat, clustergroups)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mclinical_cat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'ER-/PR-/HER2+'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mcluster_clin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclinical_cat\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mclustergroups\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0mER_pos_neg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfull_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustergroups\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'total'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mfull_grp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfull_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mclinical_cat\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue_counts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'full_df' is not defined"
     ]
    }
   ],
   "source": [
    "clinical_cat = 'ER-/PR-/HER2+'\n",
    "def cluster_clin(clinical_cat, clustergroups):\n",
    "    ER_pos_neg = full_df.groupby(clustergroups)\n",
    "    print('total')\n",
    "    full_grp = full_df[clinical_cat].value_counts()\n",
    "    full_series_len = full_df[clinical_cat].count()\n",
    "    full_grp_df = pd.DataFrame()\n",
    "    full_grp_df['counts'] = full_grp\n",
    "    full_grp_df['%'] = round(100*(full_grp/full_series_len),2)\n",
    "    print(full_grp_df)\n",
    "    print('\\n\\n\\n')\n",
    "    for group in ER_pos_neg:\n",
    "        grp = group[1]\n",
    "        grp2 = grp[clinical_cat].value_counts()\n",
    "        series_len = grp[clinical_cat].count()\n",
    "        df = pd.DataFrame()\n",
    "        df['counts'] = grp2\n",
    "        df['%'] = round(100*(grp2/series_len),2)\n",
    "        print(f'cluster {group[0]}')\n",
    "        print(df)\n",
    "        print()\n",
    "cluster_clin('Pam50 + Claudin-low subtype', 'Integrative Cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_clin(\"ER Status\", 'Integrative Cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clus5 = new_clinical[new_clinical['Integrative Cluster'] == '']\n",
    "clus5['Age at Diagnosis'].astype(float).median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "clusterframes['6'].iloc[:45,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_clusterframe = pd.DataFrame(index=clusterframes['1'].index)\n",
    "for i in clusterframes:\n",
    "    series = clusterframes[i][f'cluster {i} qvalues']\n",
    "    full_clusterframe = pd.concat([full_clusterframe, series.reindex(full_clusterframe.index)], axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_clusterframe.to_csv(\"../data/pathway_qvalues.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "full_clusterframe.sort_values(by='cluster 6 qvalues', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clus9 = clusterframes['5']\n",
    "clus9[clus9['cluster 5 qvalues'] > 2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "genes_components_per_pathway['Cell-Cell communication'][0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca_per_pathway['Calmodulin induced events'].hist(bins=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "grouped_ttest_vs_intclust3.sort_values(by='9 qvalues', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from statsmodels.formula.api import ols\n",
    "results = ols(\"Q('M Phase') ~C(Q('Integrative Cluster'))\", data=full_df).fit()\n",
    "results.summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import qvalue as qv\n",
    "\n",
    "df = qv.qvalues(grouped_ttest_vs_intclust3, '1', '1 qv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
